{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "### This note book analyze figure 5 - synonymous candidate result\n",
    "# Author: Yiyun\n",
    "\n",
    "import pandas as pd\n",
    "import os,pickle\n",
    "import numpy as np\n",
    "from collections import Counter\n",
    "from multiprocessing import Pool"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "***\n",
    "### Synonymous candidates in CGC/PCAWG - TSG, oncogenes..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Read synonymous candidate list and create pivot table for genes\n",
    "cohorts = ['histology','organ','origin','system','pancancer']\n",
    "dir_out = './figure4/'\n",
    "\n",
    "# Read the significant gene dataframe after FDR calculation\n",
    "feature_type = 'histology';syn_nsyn = 'syn';run = 'cohort_072221';threshold = 1;\n",
    "df_syn = pd.read_csv(os.path.join(dir_out,feature_type+'.syn.df_all_forheatmap.'+run+'.'+str(threshold)+'.csv'),index_col = 0)\n",
    "df_syn['FDR'] = -np.log10(df_syn['FDR']) # Transform into log scale\n",
    "\n",
    "### read the census gene file and get cancer census gene list\n",
    "dir_anno = '../anno_ref/anlyze-manuscript'\n",
    "df_census = pd.read_csv(os.path.join(dir_anno,'Census_all.csv'))\n",
    "df_census = df_census.set_index('Gene Symbol')\n",
    "df_pcawg = pd.read_csv(os.path.join(dir_anno, 'TableS1_compendium_mutational_drivers.csv'))\n",
    "df_pcawg = df_pcawg[df_pcawg['Element_type'] == 'cds']\n",
    "df_pcawg = df_pcawg[df_pcawg['Category'].isin(['both','discovery_unique'])]\n",
    "lpg = set(df_pcawg[\"Gene\"].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Find overlap between synonymous candidate and CGC genes\n",
    "df_incgc = pd.DataFrame() # Dataframe for gene that is in Cancer Gene Census\n",
    "for gene in df_syn['gene']:\n",
    "    if gene in df_census.index.tolist():\n",
    "        df_incgc = pd.concat([df_incgc,df_census.loc[gene]], axis = 1)\n",
    "    if gene in lpg:\n",
    "        print(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BCL2</th>\n",
       "      <th>CALR</th>\n",
       "      <th>NACA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <td>B-cell CLL/lymphoma 2</td>\n",
       "      <td>calreticulin</td>\n",
       "      <td>nascent-polypeptide-associated complex alpha p...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Entrez GeneId</th>\n",
       "      <td>596</td>\n",
       "      <td>811</td>\n",
       "      <td>4666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genome Location</th>\n",
       "      <td>18:63123346-63320128</td>\n",
       "      <td>19:12938607-12944489</td>\n",
       "      <td>12:56712433-56725299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Tier</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hallmark</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Chr Band</th>\n",
       "      <td>21.33</td>\n",
       "      <td>13.13</td>\n",
       "      <td>13.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Somatic</th>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Germline</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Tumour Types(Somatic)</th>\n",
       "      <td>NHL, CLL</td>\n",
       "      <td>MPN, MDS</td>\n",
       "      <td>NHL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Tumour Types(Germline)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cancer Syndrome</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Tissue Type</th>\n",
       "      <td>L</td>\n",
       "      <td>L</td>\n",
       "      <td>L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Molecular Genetics</th>\n",
       "      <td>Dom</td>\n",
       "      <td>Dom</td>\n",
       "      <td>Dom</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Role in Cancer</th>\n",
       "      <td>oncogene, fusion</td>\n",
       "      <td>oncogene</td>\n",
       "      <td>fusion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mutation Types</th>\n",
       "      <td>T</td>\n",
       "      <td>F, Mis</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Translocation Partner</th>\n",
       "      <td>IGH</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BCL6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other Germline Mut</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other Syndrome</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Synonyms</th>\n",
       "      <td>596,BCL2,Bcl-2,ENSG00000171791.12,P10415,PPP1R50</td>\n",
       "      <td>811,CALR,CRT,ENSG00000179218.13,FLJ26680,P2779...</td>\n",
       "      <td>4666,E9PAV3,ENSG00000196531.10,NACA,NACA1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                    BCL2  \\\n",
       "Name                                               B-cell CLL/lymphoma 2   \n",
       "Entrez GeneId                                                        596   \n",
       "Genome Location                                     18:63123346-63320128   \n",
       "Tier                                                                   1   \n",
       "Hallmark                                                             NaN   \n",
       "Chr Band                                                           21.33   \n",
       "Somatic                                                              yes   \n",
       "Germline                                                             NaN   \n",
       "Tumour Types(Somatic)                                           NHL, CLL   \n",
       "Tumour Types(Germline)                                               NaN   \n",
       "Cancer Syndrome                                                      NaN   \n",
       "Tissue Type                                                            L   \n",
       "Molecular Genetics                                                   Dom   \n",
       "Role in Cancer                                          oncogene, fusion   \n",
       "Mutation Types                                                         T   \n",
       "Translocation Partner                                                IGH   \n",
       "Other Germline Mut                                                   NaN   \n",
       "Other Syndrome                                                       NaN   \n",
       "Synonyms                596,BCL2,Bcl-2,ENSG00000171791.12,P10415,PPP1R50   \n",
       "\n",
       "                                                                     CALR  \\\n",
       "Name                                                         calreticulin   \n",
       "Entrez GeneId                                                         811   \n",
       "Genome Location                                      19:12938607-12944489   \n",
       "Tier                                                                    1   \n",
       "Hallmark                                                              Yes   \n",
       "Chr Band                                                            13.13   \n",
       "Somatic                                                               yes   \n",
       "Germline                                                              NaN   \n",
       "Tumour Types(Somatic)                                            MPN, MDS   \n",
       "Tumour Types(Germline)                                                NaN   \n",
       "Cancer Syndrome                                                       NaN   \n",
       "Tissue Type                                                             L   \n",
       "Molecular Genetics                                                    Dom   \n",
       "Role in Cancer                                                   oncogene   \n",
       "Mutation Types                                                     F, Mis   \n",
       "Translocation Partner                                                 NaN   \n",
       "Other Germline Mut                                                    NaN   \n",
       "Other Syndrome                                                        NaN   \n",
       "Synonyms                811,CALR,CRT,ENSG00000179218.13,FLJ26680,P2779...   \n",
       "\n",
       "                                                                     NACA  \n",
       "Name                    nascent-polypeptide-associated complex alpha p...  \n",
       "Entrez GeneId                                                        4666  \n",
       "Genome Location                                      12:56712433-56725299  \n",
       "Tier                                                                    2  \n",
       "Hallmark                                                              NaN  \n",
       "Chr Band                                                             13.3  \n",
       "Somatic                                                               yes  \n",
       "Germline                                                              NaN  \n",
       "Tumour Types(Somatic)                                                 NHL  \n",
       "Tumour Types(Germline)                                                NaN  \n",
       "Cancer Syndrome                                                       NaN  \n",
       "Tissue Type                                                             L  \n",
       "Molecular Genetics                                                    Dom  \n",
       "Role in Cancer                                                     fusion  \n",
       "Mutation Types                                                          T  \n",
       "Translocation Partner                                                BCL6  \n",
       "Other Germline Mut                                                    NaN  \n",
       "Other Syndrome                                                        NaN  \n",
       "Synonyms                        4666,E9PAV3,ENSG00000196531.10,NACA,NACA1  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_incgc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "***\n",
    "### Analyze synonymous mutations in the candidate gene list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "out_dir = './figure5'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Read synonymous candidate list and create pivot table for genes\n",
    "cohorts = ['histology','organ','origin','system','pancancer']\n",
    "dir_out = './figure4/'\n",
    "\n",
    "dir_maf = '../maf_out/maf_cohorts_060121'\n",
    "feature_type = 'histology'\n",
    "\n",
    "# Read the significant gene dataframe after FDR calculation\n",
    "feature_type = 'histology';syn_nsyn = 'syn';run = 'cohort_072221';threshold = 1;\n",
    "df_syn = pd.read_csv(os.path.join(dir_out,feature_type+'.syn.df_all_forheatmap.'+run+'.'+str(threshold)+'.csv'),index_col = 0)\n",
    "df_syn = df_syn.set_index('gene')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>X</th>\n",
       "      <th>p</th>\n",
       "      <th>q</th>\n",
       "      <th>feature</th>\n",
       "      <th>exp.nonexp</th>\n",
       "      <th>FDR</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCL2</th>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.518254e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ITLN1</th>\n",
       "      <td>54504</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>CNS-PiloAstro</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.518254e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>XIRP2</th>\n",
       "      <td>109020</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Skin-Melanoma</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.518254e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SOX18</th>\n",
       "      <td>272521</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Eso-AdenoCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.518254e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SIGLEC15</th>\n",
       "      <td>345195</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Panc-AdenoCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.518254e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53I3</th>\n",
       "      <td>345196</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Panc-AdenoCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.518254e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CALR</th>\n",
       "      <td>399696</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Uterus-AdenoCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.518254e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PPWD1</th>\n",
       "      <td>417864</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Head-SCC</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.518254e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PURA</th>\n",
       "      <td>563208</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Breast-AdenoCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.518254e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HIST1H2BK</th>\n",
       "      <td>1</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>0.014953</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>exp</td>\n",
       "      <td>5.211509e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AKAP2</th>\n",
       "      <td>417865</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.036627</td>\n",
       "      <td>Head-SCC</td>\n",
       "      <td>exp</td>\n",
       "      <td>3.059533e-17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NOL9</th>\n",
       "      <td>2</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>0.051304</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>exp</td>\n",
       "      <td>7.729580e-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NACA</th>\n",
       "      <td>109027</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>0.016218</td>\n",
       "      <td>Skin-Melanoma</td>\n",
       "      <td>exp</td>\n",
       "      <td>5.010575e-13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DMRTB1</th>\n",
       "      <td>345197</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>0.057219</td>\n",
       "      <td>Panc-AdenoCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>9.937688e-13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACTRT2</th>\n",
       "      <td>109028</td>\n",
       "      <td>0.000031</td>\n",
       "      <td>0.027983</td>\n",
       "      <td>Skin-Melanoma</td>\n",
       "      <td>exp</td>\n",
       "      <td>3.201434e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ITPR2</th>\n",
       "      <td>145344</td>\n",
       "      <td>0.000034</td>\n",
       "      <td>0.660031</td>\n",
       "      <td>SoftTissue-Liposarc</td>\n",
       "      <td>exp</td>\n",
       "      <td>2.631870e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACVRL1</th>\n",
       "      <td>581377</td>\n",
       "      <td>0.000045</td>\n",
       "      <td>0.435663</td>\n",
       "      <td>Prost-AdenoCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>5.643940e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TPM2</th>\n",
       "      <td>254352</td>\n",
       "      <td>0.000060</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Breast-LobularCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.128742e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SH3BGR</th>\n",
       "      <td>36336</td>\n",
       "      <td>0.000061</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Panc-Endocrine</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.593320e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZFP69</th>\n",
       "      <td>145345</td>\n",
       "      <td>0.000072</td>\n",
       "      <td>0.691311</td>\n",
       "      <td>SoftTissue-Liposarc</td>\n",
       "      <td>exp</td>\n",
       "      <td>2.173972e-02</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                X         p         q              feature exp.nonexp  \\\n",
       "gene                                                                    \n",
       "BCL2            0  0.000000  0.000000           Lymph-BNHL        exp   \n",
       "ITLN1       54504  0.000000  0.000000        CNS-PiloAstro        exp   \n",
       "XIRP2      109020  0.000000  0.000000        Skin-Melanoma        exp   \n",
       "SOX18      272521  0.000000  0.000000          Eso-AdenoCA        exp   \n",
       "SIGLEC15   345195  0.000000  0.000000         Panc-AdenoCA        exp   \n",
       "TP53I3     345196  0.000000  0.000000         Panc-AdenoCA        exp   \n",
       "CALR       399696  0.000000  0.000000       Uterus-AdenoCA        exp   \n",
       "PPWD1      417864  0.000000  0.000000             Head-SCC        exp   \n",
       "PURA       563208  0.000000  0.000000       Breast-AdenoCA        exp   \n",
       "HIST1H2BK       1  0.000002  0.014953           Lymph-BNHL        exp   \n",
       "AKAP2      417865  0.000004  0.036627             Head-SCC        exp   \n",
       "NOL9            2  0.000008  0.051304           Lymph-BNHL        exp   \n",
       "NACA       109027  0.000017  0.016218        Skin-Melanoma        exp   \n",
       "DMRTB1     345197  0.000018  0.057219         Panc-AdenoCA        exp   \n",
       "ACTRT2     109028  0.000031  0.027983        Skin-Melanoma        exp   \n",
       "ITPR2      145344  0.000034  0.660031  SoftTissue-Liposarc        exp   \n",
       "ACVRL1     581377  0.000045  0.435663        Prost-AdenoCA        exp   \n",
       "TPM2       254352  0.000060  1.000000     Breast-LobularCA        exp   \n",
       "SH3BGR      36336  0.000061  1.000000       Panc-Endocrine        exp   \n",
       "ZFP69      145345  0.000072  0.691311  SoftTissue-Liposarc        exp   \n",
       "\n",
       "                    FDR  \n",
       "gene                     \n",
       "BCL2       1.518254e-18  \n",
       "ITLN1      1.518254e-18  \n",
       "XIRP2      1.518254e-18  \n",
       "SOX18      1.518254e-18  \n",
       "SIGLEC15   1.518254e-18  \n",
       "TP53I3     1.518254e-18  \n",
       "CALR       1.518254e-18  \n",
       "PPWD1      1.518254e-18  \n",
       "PURA       1.518254e-18  \n",
       "HIST1H2BK  5.211509e-18  \n",
       "AKAP2      3.059533e-17  \n",
       "NOL9       7.729580e-16  \n",
       "NACA       5.010575e-13  \n",
       "DMRTB1     9.937688e-13  \n",
       "ACTRT2     3.201434e-09  \n",
       "ITPR2      2.631870e-08  \n",
       "ACVRL1     5.643940e-06  \n",
       "TPM2       1.128742e-03  \n",
       "SH3BGR     1.593320e-03  \n",
       "ZFP69      2.173972e-02  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_syn"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Analyze individual genes**\n",
    "1. ITLN1: Cancer gene, but no literature support it's role in CNS-PiloAstro. The synonymous mutation is also found in TCGA\n",
    "    - High load, the same histology\n",
    "2. XIRP2: Surrogate of UV exposure. Could explain so many synonymous mutations? However, patients with mutations in these genes seem to liver longer,, the other 9 affected genes' ranking? and the patient disease subtype\n",
    "    - Same, highload\n",
    "3. SOX18: Had been found upregulated in other cancer types and related to poor prognostic. The HMG domain binds transcription factors.\n",
    "    - Same\n",
    "    - Highload, retrotransposon\n",
    "4. SIGLEC15: Related to Immune system, and seems high expression lower survival rate -- new star\n",
    "    - One of the major\n",
    "    - Not high load\n",
    "5. TP53I3: downstream of P53? seems may have a role but not fully studied\n",
    "    - One of major, Highload\n",
    "6. CALR: protein upregulated in subtypes? \n",
    "7. PPWD1: shallow work from other cancer types\n",
    "    - Same histology\n",
    "    - But syn patient do not have a high load, but only one patient\n",
    "8. HIST1H2BK: potential role in other tumor types\n",
    "    - Diffuse large B, follicular,\n",
    "    - High load\n",
    "9. AKAP2: R binding domain, RGS domain, interacts with Galpha. \n",
    "    - All 3 synonymous patients are primary tumors. \n",
    "    - Mutation load of syn patients on a higher end. Have high retrotransposon event signal\n",
    "10. NOL9: DLBCL, subtype, copynumber vairation\n",
    "    - Primary, Diffuse largeB, follicular\n",
    "    - One syn patient, higher than normal mutation load and CNA event. \n",
    "11. NACA: downregulated in melanoma\n",
    "    - High load, one histology\n",
    "12. DMRTB1: Maybe testis?\n",
    "    - Several histology types\n",
    "    - High load\n",
    "13. ACTRT2: nothing significant\n",
    "    - Mutation load of syn patients on a higher end\n",
    "14. ITPR2: upregulated in AML\n",
    "    - High load, one histology\n",
    "15. ACVRL1: ALK1, TGF receptor related. Found high mutation rate in prostate cancer. Kinase domain. TGF-beta expression related, so it is a interesting gene in prostate cancer\n",
    "    - All 3 synonymous patients are primary tumors. \n",
    "    - Mutation load of syn patients on a higher end.\n",
    "16. TPM2: Loss associated with RHOA activation in colorectal cancer. Isoforms found in breast cancer cell lines. downregulation of TPM is a sign of transformed cells.\n",
    "    - Lobular carcinoma, minor\n",
    "    - Not high load but only one\n",
    "17. SH3BGR:\n",
    "    - Different types\n",
    "    - Not highload but retrotransposon\n",
    "18. XFP69:\n",
    "    - Same, not highload but retrotransposon\n",
    "\n",
    "**Other questions to consider**  \n",
    "1. If mutaion in important protein domain?\n",
    "2. The histology subtype\n",
    "3. If the patients have other mutations, cnv, methylation\n",
    "4. If mutation in other cancer patient dataset"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**The histology subtypes of patients with the mutation**\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "# The sample information file\n",
    "dir_refs = '../anno_ref/proc_refs'\n",
    "df_info = pd.read_csv(os.path.join(dir_refs, 'PCAWG_sample_info.txt'), sep = '\\t')\n",
    "\n",
    "# Columns needed in the information dataframe\n",
    "num_cols = ['SV.events','Coding.SNVs', 'Non.coding.SNVs','CNA.events..do_not_use.', 'Retrotransposon.insertions',\\\n",
    "            'Mitochondrial.mutations','all.Indels']\n",
    "str_cols = ['dcc_specimen_type','histology_tier3','histology_tier4',\n",
    "        'tumour_histological_type',\\\n",
    "       'tumour_stage', 'tumour_histological_comment', 'specimen_donor_treatment_type']\n",
    "\n",
    "# The output directory for the histology and mutation anlyze result\n",
    "dir_out = './figure5/syn_candidate_anlyze'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "#histology tier3, 4, tumour_histological_type, SV.event, Coding.SNVs, CNA.events..do_not_use.\n",
    "# Interested in: All patients' histology (event number)in this cohort, patients with mutation in this gene,  syn patient\n",
    "\n",
    "def get_num_str(gene):\n",
    "    global df_syn\n",
    "    histology = df_syn.loc[gene,'feature'] #histology type\n",
    "    df_maf = pd.read_csv(os.path.join(dir_maf,feature_type, histology+'.csv'), sep = '\\t')\n",
    "    df_maf = df_maf.set_index(['Hugo_Symbol','Variant_Classification','Donor_ID' ])\n",
    "    df_gene_mut = df_maf.loc[pd.IndexSlice[gene, 'Silent',:],:]### All patients' specific histology in this histology type\n",
    "    ldonor = df_gene_mut.index.get_level_values('Donor_ID').unique().tolist()\n",
    "    \n",
    "    df_histology = df_info[df_info['histology_abbreviation'] == histology]\n",
    "    df_histology_syn = df_histology[df_histology['icgc_donor_id'].isin(ldonor)]\n",
    "\n",
    "    ### The tumor patient mutation information for synonymous patients and all patients\n",
    "    df_num_summ = pd.concat([pd.concat({'syn': df_histology_syn[num_cols].describe()}, names=['mut_status'])\\\n",
    "    ,pd.concat({'all': df_histology[num_cols].describe()}, names=['mut_status'])])\n",
    "\n",
    "    ### The tumor histology subtype information\n",
    "    df_str_info = pd.DataFrame(columns = str_cols, index = ['syn', 'all'])\n",
    "    for columns in str_cols:\n",
    "        if columns not in ['icgc_donor_id','histology_abbreviation','histology_tier2']:\n",
    "            df_str_info.loc['all', columns] = str(dict(Counter(df_histology[columns].tolist())))\n",
    "            df_str_info.loc['syn', columns] = str(dict(Counter(df_histology_syn[columns].tolist())))\n",
    "\n",
    "#     df_num_summ.to_csv(os.path.join(dir_out, f'num_{gene}.csv'))\n",
    "#     df_str_info.to_csv(os.path.join(dir_out, f'str_{gene}.csv'))\n",
    "\n",
    "\n",
    "# p = Pool(4)\n",
    "# p.map(get_num_str, df_syn.index.tolist())\n",
    "# p.close()\n",
    "# p.join()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I need to see if hypermutators are those synonymous patients"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "dir_sample = '../anno_ref/ICGC/mutational_signature'\n",
    "df_sample = pd.read_csv(os.path.join(dir_sample, 'SA_sample_classification.012718.txt'), sep = '\\t')\n",
    "\n",
    "exp_dir = '../anno_ref/ICGC/pcawg_rnaseq/'\n",
    "df_exp_info = pd.read_csv(os.path.join(exp_dir,'rnaseq.metadata.tsv'), sep = '\\t')\n",
    "\n",
    "sed_idx = df_sample[df_sample['class1'] == 'Secondary'].index\n",
    "samples = df_sample.loc[sed_idx, 'sample'].tolist()\n",
    "samples = [i.split('_')[-1] for i in samples]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "hype_samples = df_exp_info[df_exp_info['icgc_specimen_id'].isin(samples)]['icgc_donor_id'].tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "def check_hypermutator(gene):\n",
    "    global df_syn\n",
    "    histology = df_syn.loc[gene,'feature'] #histology type\n",
    "    df_maf = pd.read_csv(os.path.join(dir_maf,feature_type, histology+'.csv'), sep = '\\t')\n",
    "    df_maf = df_maf.set_index(['Hugo_Symbol','Variant_Classification','Donor_ID' ])\n",
    "    df_gene_mut = df_maf.loc[pd.IndexSlice[gene, 'Silent',:],:]### All patients' specific histology in this histology type\n",
    "    ldonor = df_gene_mut.index.get_level_values('Donor_ID').unique().tolist()\n",
    "    \n",
    "    n = 0\n",
    "    for donors in ldonor:\n",
    "        if donors in hype_samples:\n",
    "            n+=1\n",
    "    if n > 0:\n",
    "        print(f'There are {n} hypermutators out of {len(ldonor)} in {histology} for {gene}')\n",
    "\n",
    "# p = Pool(4)\n",
    "# p.map(check_hypermutator, df_syn.index.tolist())\n",
    "# p.close()\n",
    "# p.join()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
