{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os,pickle\n",
    "import numpy as np\n",
    "from collections import Counter\n",
    "from multiprocessing import Pool"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "out_dir = './figure5'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Read synonymous candidate list and create pivot table for genes\n",
    "cohorts = ['histology','organ','origin','system','pancancer']\n",
    "dir_out = './figure4/'\n",
    "\n",
    "dir_maf = '../maf_out/maf_cohorts_060121'\n",
    "feature_type = 'histology'\n",
    "\n",
    "# Read the significant gene dataframe after FDR calculation\n",
    "feature_type = 'histology';syn_nsyn = 'syn';run = 'cohort_072221';threshold = 1;\n",
    "df_syn = pd.read_csv(os.path.join(dir_out,feature_type+'.syn.df_all_forheatmap.'+run+'.'+str(threshold)+'.csv'),index_col = 0)\n",
    "df_syn = df_syn.set_index('gene')\n",
    "\n",
    "# For copy number variation\n",
    "# Read cnv file\n",
    "df_info = pd.read_csv('../anno_ref/proc_refs/PCAWG_sample_info.txt', sep = '\\t')\n",
    "df_cnv = pd.read_csv('../anno_ref/ICGC/consensus_cnv/consensus_CN.by_gene.170214.txt', sep = '\\t')\n",
    "df_cnv = df_cnv.set_index(['Gene Symbol', 'Locus ID', 'Cytoband'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>X</th>\n",
       "      <th>p</th>\n",
       "      <th>q</th>\n",
       "      <th>feature</th>\n",
       "      <th>exp.nonexp</th>\n",
       "      <th>FDR</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCL2</th>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.518254e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ITLN1</th>\n",
       "      <td>54504</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>CNS-PiloAstro</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.518254e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>XIRP2</th>\n",
       "      <td>109020</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Skin-Melanoma</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.518254e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SOX18</th>\n",
       "      <td>272521</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Eso-AdenoCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.518254e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SIGLEC15</th>\n",
       "      <td>345195</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Panc-AdenoCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.518254e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53I3</th>\n",
       "      <td>345196</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Panc-AdenoCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.518254e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CALR</th>\n",
       "      <td>399696</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Uterus-AdenoCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.518254e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PPWD1</th>\n",
       "      <td>417864</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Head-SCC</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.518254e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PURA</th>\n",
       "      <td>563208</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Breast-AdenoCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.518254e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HIST1H2BK</th>\n",
       "      <td>1</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>0.014953</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>exp</td>\n",
       "      <td>5.211509e-18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AKAP2</th>\n",
       "      <td>417865</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.036627</td>\n",
       "      <td>Head-SCC</td>\n",
       "      <td>exp</td>\n",
       "      <td>3.059533e-17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NOL9</th>\n",
       "      <td>2</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>0.051304</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>exp</td>\n",
       "      <td>7.729580e-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NACA</th>\n",
       "      <td>109027</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>0.016218</td>\n",
       "      <td>Skin-Melanoma</td>\n",
       "      <td>exp</td>\n",
       "      <td>5.010575e-13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DMRTB1</th>\n",
       "      <td>345197</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>0.057219</td>\n",
       "      <td>Panc-AdenoCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>9.937688e-13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACTRT2</th>\n",
       "      <td>109028</td>\n",
       "      <td>0.000031</td>\n",
       "      <td>0.027983</td>\n",
       "      <td>Skin-Melanoma</td>\n",
       "      <td>exp</td>\n",
       "      <td>3.201434e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ITPR2</th>\n",
       "      <td>145344</td>\n",
       "      <td>0.000034</td>\n",
       "      <td>0.660031</td>\n",
       "      <td>SoftTissue-Liposarc</td>\n",
       "      <td>exp</td>\n",
       "      <td>2.631870e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACVRL1</th>\n",
       "      <td>581377</td>\n",
       "      <td>0.000045</td>\n",
       "      <td>0.435663</td>\n",
       "      <td>Prost-AdenoCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>5.643940e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TPM2</th>\n",
       "      <td>254352</td>\n",
       "      <td>0.000060</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Breast-LobularCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.128742e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SH3BGR</th>\n",
       "      <td>36336</td>\n",
       "      <td>0.000061</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Panc-Endocrine</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.593320e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZFP69</th>\n",
       "      <td>145345</td>\n",
       "      <td>0.000072</td>\n",
       "      <td>0.691311</td>\n",
       "      <td>SoftTissue-Liposarc</td>\n",
       "      <td>exp</td>\n",
       "      <td>2.173972e-02</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                X         p         q              feature exp.nonexp  \\\n",
       "gene                                                                    \n",
       "BCL2            0  0.000000  0.000000           Lymph-BNHL        exp   \n",
       "ITLN1       54504  0.000000  0.000000        CNS-PiloAstro        exp   \n",
       "XIRP2      109020  0.000000  0.000000        Skin-Melanoma        exp   \n",
       "SOX18      272521  0.000000  0.000000          Eso-AdenoCA        exp   \n",
       "SIGLEC15   345195  0.000000  0.000000         Panc-AdenoCA        exp   \n",
       "TP53I3     345196  0.000000  0.000000         Panc-AdenoCA        exp   \n",
       "CALR       399696  0.000000  0.000000       Uterus-AdenoCA        exp   \n",
       "PPWD1      417864  0.000000  0.000000             Head-SCC        exp   \n",
       "PURA       563208  0.000000  0.000000       Breast-AdenoCA        exp   \n",
       "HIST1H2BK       1  0.000002  0.014953           Lymph-BNHL        exp   \n",
       "AKAP2      417865  0.000004  0.036627             Head-SCC        exp   \n",
       "NOL9            2  0.000008  0.051304           Lymph-BNHL        exp   \n",
       "NACA       109027  0.000017  0.016218        Skin-Melanoma        exp   \n",
       "DMRTB1     345197  0.000018  0.057219         Panc-AdenoCA        exp   \n",
       "ACTRT2     109028  0.000031  0.027983        Skin-Melanoma        exp   \n",
       "ITPR2      145344  0.000034  0.660031  SoftTissue-Liposarc        exp   \n",
       "ACVRL1     581377  0.000045  0.435663        Prost-AdenoCA        exp   \n",
       "TPM2       254352  0.000060  1.000000     Breast-LobularCA        exp   \n",
       "SH3BGR      36336  0.000061  1.000000       Panc-Endocrine        exp   \n",
       "ZFP69      145345  0.000072  0.691311  SoftTissue-Liposarc        exp   \n",
       "\n",
       "                    FDR  \n",
       "gene                     \n",
       "BCL2       1.518254e-18  \n",
       "ITLN1      1.518254e-18  \n",
       "XIRP2      1.518254e-18  \n",
       "SOX18      1.518254e-18  \n",
       "SIGLEC15   1.518254e-18  \n",
       "TP53I3     1.518254e-18  \n",
       "CALR       1.518254e-18  \n",
       "PPWD1      1.518254e-18  \n",
       "PURA       1.518254e-18  \n",
       "HIST1H2BK  5.211509e-18  \n",
       "AKAP2      3.059533e-17  \n",
       "NOL9       7.729580e-16  \n",
       "NACA       5.010575e-13  \n",
       "DMRTB1     9.937688e-13  \n",
       "ACTRT2     3.201434e-09  \n",
       "ITPR2      2.631870e-08  \n",
       "ACVRL1     5.643940e-06  \n",
       "TPM2       1.128742e-03  \n",
       "SH3BGR     1.593320e-03  \n",
       "ZFP69      2.173972e-02  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_syn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_mut(gene):\n",
    "    global df_syn\n",
    "    histology = df_syn.loc[gene,'feature'] #histology type\n",
    "    df_maf = pd.read_csv(os.path.join(dir_maf,feature_type, histology+'.csv'), sep = '\\t')\n",
    "    df_maf = df_maf.set_index(['Hugo_Symbol','Variant_Classification','Donor_ID' ])\n",
    "    ldonor_all = df_maf.index.get_level_values('Donor_ID').unique().tolist()\n",
    "    print(f'Total {len(ldonor_all)} patients')\n",
    "    df_gene_mut = df_maf.loc[pd.IndexSlice[gene, 'Silent',:],:]### All patients' specific histology in this histology type\n",
    "    ldonor = df_gene_mut.index.get_level_values('Donor_ID').unique().tolist()\n",
    "    print(f'{len(ldonor)} patients have synonymous mutations, they are: {ldonor}')\n",
    "    for muts in df_gene_mut['Genome_Change'].tolist():\n",
    "        chromosome = muts.split(':')[0].split('.')[-1].split('chr')[-1]\n",
    "        change = muts.split(':')[1]\n",
    "        mutation = chromosome+':g.'+change\n",
    "        print(mutation)\n",
    "\n",
    "    return df_maf, df_gene_mut, ldonor\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total 43 patients\n",
      "1 patients have synonymous mutations, they are: ['DO41217']\n",
      "19:g.13051179C>T\n",
      "19:g.13051188A>C\n",
      "19:g.13051197T>G\n"
     ]
    }
   ],
   "source": [
    "gene_name = 'CALR'\n",
    "df_all, df_mut, donor_list = get_mut(gene_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Hugo_Symbol  Variant_Classification  Donor_ID\n",
       "CALR         Silent                  DO41217     g.chr19:13051179C>T\n",
       "                                     DO41217     g.chr19:13051188A>C\n",
       "                                     DO41217     g.chr19:13051197T>G\n",
       "             Missense_Mutation       DO41217     g.chr19:13051215A>C\n",
       "Name: Genome_Change, dtype: object"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_all.loc[pd.IndexSlice[gene_name, :,donor_list],'Genome_Change']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get tumor aliquot id\n",
    "for donors in donor_list:\n",
    "    idx = df_info[df_info['icgc_donor_id'] == donors].index\n",
    "    aliquot_id = df_info.loc[idx, 'tumour_specimen_aliquot_id']\n",
    "    cnv = df_cnv.loc[pd.IndexSlice[gene_name,:,:],aliquot_id]\n",
    "cnv"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
