{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "### This notebook analyze candidate mutation positions and see if they lies in the splicing region\n",
    "import pandas as pd\n",
    "import os,pickle,re\n",
    "from tqdm import tqdm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Read the synonymous candidate list/maf directory\n",
    "dir_out_fig4 = './figure4/'\n",
    "feature_type = 'histology';syn_nsyn = 'syn';run = 'cohort_new_newcova_nohypermutator'\n",
    "df_syn = pd.read_csv(os.path.join(dir_out_fig4,feature_type+'.syn.df_all_forheatmap.'+run+'.csv'),index_col = 0)\n",
    "df_syn = df_syn.set_index('gene')\n",
    "\n",
    "dir_maf = '../data/maf/histology_nohypermutator'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>X</th>\n",
       "      <th>p</th>\n",
       "      <th>q</th>\n",
       "      <th>feature</th>\n",
       "      <th>exp.nonexp</th>\n",
       "      <th>FDR</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BCL2</th>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>exp</td>\n",
       "      <td>8.984726e-47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRSF2</th>\n",
       "      <td>2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>exp</td>\n",
       "      <td>8.984726e-47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ITLN1</th>\n",
       "      <td>52770</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>CNS-PiloAstro</td>\n",
       "      <td>exp</td>\n",
       "      <td>8.984726e-47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACTA1</th>\n",
       "      <td>211080</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Lymph-CLL</td>\n",
       "      <td>exp</td>\n",
       "      <td>8.984726e-47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SIGLEC15</th>\n",
       "      <td>316623</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Panc-AdenoCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>8.984726e-47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53I3</th>\n",
       "      <td>316624</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Panc-AdenoCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>8.984726e-47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LONRF3</th>\n",
       "      <td>351800</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Liver-HCC</td>\n",
       "      <td>exp</td>\n",
       "      <td>8.984726e-47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RTN4</th>\n",
       "      <td>351801</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Liver-HCC</td>\n",
       "      <td>exp</td>\n",
       "      <td>8.984726e-47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PPWD1</th>\n",
       "      <td>386980</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Head-SCC</td>\n",
       "      <td>exp</td>\n",
       "      <td>8.984726e-47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAGEC1</th>\n",
       "      <td>527700</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Breast-AdenoCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>8.984726e-47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PURA</th>\n",
       "      <td>527701</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Breast-AdenoCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>8.984726e-47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HIST1H2BK</th>\n",
       "      <td>3</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0.045887</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.983667e-40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NOL9</th>\n",
       "      <td>4</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.045887</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.175817e-39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZCCHC5</th>\n",
       "      <td>316625</td>\n",
       "      <td>0.000025</td>\n",
       "      <td>0.078039</td>\n",
       "      <td>Panc-AdenoCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>9.522483e-33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRX</th>\n",
       "      <td>158310</td>\n",
       "      <td>0.000029</td>\n",
       "      <td>0.533142</td>\n",
       "      <td>SoftTissue-Leiomyo</td>\n",
       "      <td>exp</td>\n",
       "      <td>5.522803e-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ITPR2</th>\n",
       "      <td>140720</td>\n",
       "      <td>0.000034</td>\n",
       "      <td>0.628776</td>\n",
       "      <td>SoftTissue-Liposarc</td>\n",
       "      <td>exp</td>\n",
       "      <td>2.212396e-28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SH3BGR</th>\n",
       "      <td>35180</td>\n",
       "      <td>0.000040</td>\n",
       "      <td>0.751429</td>\n",
       "      <td>Panc-Endocrine</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.929987e-25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACVRL1</th>\n",
       "      <td>545291</td>\n",
       "      <td>0.000043</td>\n",
       "      <td>0.397612</td>\n",
       "      <td>Prost-AdenoCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>2.121351e-24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSRB2</th>\n",
       "      <td>422160</td>\n",
       "      <td>0.000045</td>\n",
       "      <td>0.836424</td>\n",
       "      <td>Cervix-SCC</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.926509e-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TPM2</th>\n",
       "      <td>246260</td>\n",
       "      <td>0.000061</td>\n",
       "      <td>0.982699</td>\n",
       "      <td>Breast-LobularCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>4.230285e-17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZFP69</th>\n",
       "      <td>140721</td>\n",
       "      <td>0.000068</td>\n",
       "      <td>0.628776</td>\n",
       "      <td>SoftTissue-Liposarc</td>\n",
       "      <td>exp</td>\n",
       "      <td>8.832567e-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SOX18</th>\n",
       "      <td>263851</td>\n",
       "      <td>0.000074</td>\n",
       "      <td>0.607041</td>\n",
       "      <td>Eso-AdenoCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>8.348794e-13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NCAN</th>\n",
       "      <td>175900</td>\n",
       "      <td>0.000075</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Biliary-AdenoCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>2.594258e-12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HIST1H4E</th>\n",
       "      <td>369390</td>\n",
       "      <td>0.000076</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Uterus-AdenoCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>3.530040e-12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DMRTB1</th>\n",
       "      <td>316626</td>\n",
       "      <td>0.000080</td>\n",
       "      <td>0.197206</td>\n",
       "      <td>Panc-AdenoCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>4.289913e-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KLHL21</th>\n",
       "      <td>5</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>0.303637</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.129010e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ADAM17</th>\n",
       "      <td>140722</td>\n",
       "      <td>0.000101</td>\n",
       "      <td>0.628776</td>\n",
       "      <td>SoftTissue-Liposarc</td>\n",
       "      <td>exp</td>\n",
       "      <td>5.486659e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF648</th>\n",
       "      <td>510110</td>\n",
       "      <td>0.000105</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Stomach-AdenoCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>2.347573e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLRA3</th>\n",
       "      <td>246261</td>\n",
       "      <td>0.000105</td>\n",
       "      <td>0.982699</td>\n",
       "      <td>Breast-LobularCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>3.259279e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TMEM129</th>\n",
       "      <td>299030</td>\n",
       "      <td>0.000124</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>Thy-AdenoCA</td>\n",
       "      <td>exp</td>\n",
       "      <td>1.237774e-02</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                X         p         q              feature exp.nonexp  \\\n",
       "gene                                                                    \n",
       "BCL2            0  0.000000  0.000000           Lymph-BNHL        exp   \n",
       "SRSF2           2  0.000000  0.000000           Lymph-BNHL        exp   \n",
       "ITLN1       52770  0.000000  0.000000        CNS-PiloAstro        exp   \n",
       "ACTA1      211080  0.000000  0.000000            Lymph-CLL        exp   \n",
       "SIGLEC15   316623  0.000000  0.000000         Panc-AdenoCA        exp   \n",
       "TP53I3     316624  0.000000  0.000000         Panc-AdenoCA        exp   \n",
       "LONRF3     351800  0.000000  0.000000            Liver-HCC        exp   \n",
       "RTN4       351801  0.000000  0.000000            Liver-HCC        exp   \n",
       "PPWD1      386980  0.000000  0.000000             Head-SCC        exp   \n",
       "MAGEC1     527700  0.000000  0.000000       Breast-AdenoCA        exp   \n",
       "PURA       527701  0.000000  0.000000       Breast-AdenoCA        exp   \n",
       "HIST1H2BK       3  0.000011  0.045887           Lymph-BNHL        exp   \n",
       "NOL9            4  0.000012  0.045887           Lymph-BNHL        exp   \n",
       "ZCCHC5     316625  0.000025  0.078039         Panc-AdenoCA        exp   \n",
       "PRX        158310  0.000029  0.533142   SoftTissue-Leiomyo        exp   \n",
       "ITPR2      140720  0.000034  0.628776  SoftTissue-Liposarc        exp   \n",
       "SH3BGR      35180  0.000040  0.751429       Panc-Endocrine        exp   \n",
       "ACVRL1     545291  0.000043  0.397612        Prost-AdenoCA        exp   \n",
       "MSRB2      422160  0.000045  0.836424           Cervix-SCC        exp   \n",
       "TPM2       246260  0.000061  0.982699     Breast-LobularCA        exp   \n",
       "ZFP69      140721  0.000068  0.628776  SoftTissue-Liposarc        exp   \n",
       "SOX18      263851  0.000074  0.607041          Eso-AdenoCA        exp   \n",
       "NCAN       175900  0.000075  1.000000      Biliary-AdenoCA        exp   \n",
       "HIST1H4E   369390  0.000076  1.000000       Uterus-AdenoCA        exp   \n",
       "DMRTB1     316626  0.000080  0.197206         Panc-AdenoCA        exp   \n",
       "KLHL21          5  0.000098  0.303637           Lymph-BNHL        exp   \n",
       "ADAM17     140722  0.000101  0.628776  SoftTissue-Liposarc        exp   \n",
       "ZNF648     510110  0.000105  1.000000      Stomach-AdenoCA        exp   \n",
       "GLRA3      246261  0.000105  0.982699     Breast-LobularCA        exp   \n",
       "TMEM129    299030  0.000124  1.000000          Thy-AdenoCA        exp   \n",
       "\n",
       "                    FDR  \n",
       "gene                     \n",
       "BCL2       8.984726e-47  \n",
       "SRSF2      8.984726e-47  \n",
       "ITLN1      8.984726e-47  \n",
       "ACTA1      8.984726e-47  \n",
       "SIGLEC15   8.984726e-47  \n",
       "TP53I3     8.984726e-47  \n",
       "LONRF3     8.984726e-47  \n",
       "RTN4       8.984726e-47  \n",
       "PPWD1      8.984726e-47  \n",
       "MAGEC1     8.984726e-47  \n",
       "PURA       8.984726e-47  \n",
       "HIST1H2BK  1.983667e-40  \n",
       "NOL9       1.175817e-39  \n",
       "ZCCHC5     9.522483e-33  \n",
       "PRX        5.522803e-31  \n",
       "ITPR2      2.212396e-28  \n",
       "SH3BGR     1.929987e-25  \n",
       "ACVRL1     2.121351e-24  \n",
       "MSRB2      1.926509e-23  \n",
       "TPM2       4.230285e-17  \n",
       "ZFP69      8.832567e-15  \n",
       "SOX18      8.348794e-13  \n",
       "NCAN       2.594258e-12  \n",
       "HIST1H4E   3.530040e-12  \n",
       "DMRTB1     4.289913e-11  \n",
       "KLHL21     1.129010e-06  \n",
       "ADAM17     5.486659e-06  \n",
       "ZNF648     2.347573e-05  \n",
       "GLRA3      3.259279e-05  \n",
       "TMEM129    1.237774e-02  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_syn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read transcript info file\n",
    "dict_transcript_info = pickle.load(open('../data/proc_refs/dict_transcript_info_062121.pkl','rb'))\n",
    "dict_name = pickle.load(open(os.path.join('../data/proc_refs','dict_name_forcov_102121.pkl'), 'rb'))\n",
    "# dict_record = pickle.load(open('../anno_ref/proc_refs/dict_record_new.pkl','rb'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Find transcript for the gene\n",
    "# Create dictionary for candidate gene name - transcript id\n",
    "dict_name_id = {}\n",
    "for sig_genes in df_syn.index.tolist():\n",
    "    for transcript,gene in dict_name.items():\n",
    "        if gene == sig_genes:\n",
    "            dict_name_id[sig_genes] = transcript"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'strand': '-',\n",
       " 'chr': 'chr17',\n",
       " 'exon': [[74732881, 74733456], [74732236, 74732546], [74730201, 74731240]],\n",
       " 'CDS': [[74732881, 74733242], [74732246, 74732546]],\n",
       " 'UTR': [[74733243, 74733456], [74732236, 74732245], [74730201, 74731240]],\n",
       " 'transcript': [74730201, 74733456]}"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dict_transcript_info['ENST00000359995']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Get the synonymous mutations of the candidate gene in the cohort\n",
    "def get_syn_mut(gene):\n",
    "    global df_syn\n",
    "    \n",
    "    histology = df_syn.loc[gene,'feature'] #histology type\n",
    "    # open maf file\n",
    "    df_maf = pd.read_csv(os.path.join(dir_maf,histology+'.csv'), sep = '\\t')\n",
    "    df_maf = df_maf.set_index(['Hugo_Symbol','Variant_Classification','Donor_ID' ])\n",
    "    # total number of donors in this cohort\n",
    "    ldonor_all = df_maf.index.get_level_values('Donor_ID').unique().tolist()\n",
    "    #print(f'Total {len(ldonor_all)} patients')\n",
    "    \n",
    "    # Find synonymous mutations\n",
    "    df_gene_mut = df_maf.loc[pd.IndexSlice[gene, 'Silent',:],:]### All patients' specific histology in this histology type\n",
    "    # Number of patients have synonymous mutations\n",
    "    ldonor = df_gene_mut.index.get_level_values('Donor_ID').unique().tolist()\n",
    "    #print(f'{len(ldonor)} patients have synonymous mutations, they are: {ldonor}')\n",
    "\n",
    "    return df_gene_mut,histology"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Determine if the mutaiton position is close to the exon start-end\n",
    "def find_splicing_pos(mut_pos, cds, gene_strand):\n",
    "    flag = 0; effect = None\n",
    "    if gene_strand == '-':\n",
    "        if mut_pos< cds[1] and mut_pos > cds[1]-20:\n",
    "            effect = 'within 20bp of 5 prime splicing site';flag=1\n",
    "        elif mut_pos > cds[0] and mut_pos< cds[0] + 20:\n",
    "            effect = 'within 20bp of 3 prime splicing site';flag=1\n",
    "    else:\n",
    "        if mut_pos> cds[0] and mut_pos < cds[0]+20:\n",
    "            effect = 'within 20bp of 5 prime splicing site';flag = 1\n",
    "        elif mut_pos < cds[1] and mut_pos> cds[1] - 20:\n",
    "            effect = 'within 20bp of 3 prime splicing site';flag = 1\n",
    "    return flag, effect"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Determine if the mutaiton position is close to the exon start-end\n",
    "def find_initiation_pos(mut_pos, cds, gene_strand):\n",
    "    flag = 0; effect = None\n",
    "    # if gene_strand == '-':\n",
    "    #     if mut_pos< cds[1] and mut_pos > cds[1]-10:\n",
    "    #         effect = 'within 10bp of initiation site';flag = 1\n",
    "    #     elif mut_pos > cds[0] and mut_pos< cds[0] +20:\n",
    "    #         effect = 'within 20bp of 3 prime splicing site';flag = 1\n",
    "    # else:\n",
    "    if mut_pos > cds[0] and mut_pos< cds[0] + 10:\n",
    "        effect = 'within 10bp of initiation site';flag = 1\n",
    "    elif mut_pos < cds[1] and mut_pos> cds[1]- 20:\n",
    "        effect = 'within 20bp of 3 prime splicing site';flag = 1\n",
    "    return flag, effect"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_gene_df(gene, histology, mutation, exon, effect):\n",
    "    dict_gene_splice = {'Gene':[gene], 'Histology':[histology], 'Mutation':[mutation], 'Exon':[exon], 'Effect':[effect]}\n",
    "    df = pd.DataFrame.from_dict(dict_gene_splice)\n",
    "    print(dict_gene_splice)\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "searching for gene BCL2\n",
      "['g.chr18:60985471G>A', 'g.chr18:60985492C>T', 'g.chr18:60985573G>A', 'g.chr18:60985366A>G', 'g.chr18:60985435C>T', 'g.chr18:60985834C>T', 'g.chr18:60985876C>A', 'g.chr18:60985833G>A', 'g.chr18:60985359G>A', 'g.chr18:60985366A>T', 'g.chr18:60985861C>T', 'g.chr18:60985549G>A', 'g.chr18:60985711C>A', 'g.chr18:60985876C>T', 'g.chr18:60985528G>A', 'g.chr18:60985540G>A', 'g.chr18:60985713G>T', 'g.chr18:60985363C>T', 'g.chr18:60985417G>A', 'g.chr18:60985717T>C', 'g.chr18:60985795C>G', 'g.chr18:60985876C>G', 'g.chr18:60985582G>A', 'g.chr18:60985753G>A', 'g.chr18:60985384G>A', 'g.chr18:60985545G>A', 'g.chr18:60985738C>T', 'g.chr18:60985615C>T', 'g.chr18:60985444C>T', 'g.chr18:60985867G>A', 'g.chr18:60985735G>A', 'g.chr18:60985714G>A']\n",
      "searching for gene SRSF2\n",
      "['g.chr17:74732997G>C', 'g.chr17:74733006C>G', 'g.chr17:74733234G>A']\n",
      "{'Gene': ['SRSF2'], 'Histology': ['Lymph-BNHL'], 'Mutation': ['chr17:g.74733234G>A'], 'Exon': [2], 'Effect': ['within 20bp of 3 prime splicing site']}\n",
      "searching for gene ITLN1\n",
      "['g.chr1:160850421G>A', 'g.chr1:160851037G>A']\n",
      "searching for gene ACTA1\n",
      "['g.chr1:229567630A>G', 'g.chr1:229567636C>A', 'g.chr1:229567639C>A', 'g.chr1:229568045A>G', 'g.chr1:229568072G>C']\n",
      "{'Gene': ['ACTA1'], 'Histology': ['Lymph-CLL'], 'Mutation': ['chr1:g.229567630A>G'], 'Exon': [6], 'Effect': ['within 20bp of 5 prime splicing site']}\n",
      "{'Gene': ['ACTA1'], 'Histology': ['Lymph-CLL'], 'Mutation': ['chr1:g.229567636C>A'], 'Exon': [6], 'Effect': ['within 20bp of 5 prime splicing site']}\n",
      "{'Gene': ['ACTA1'], 'Histology': ['Lymph-CLL'], 'Mutation': ['chr1:g.229567639C>A'], 'Exon': [6], 'Effect': ['within 20bp of 5 prime splicing site']}\n",
      "searching for gene SIGLEC15\n",
      "['g.chr18:43418687G>C', 'g.chr18:43418708G>T']\n",
      "{'Gene': ['SIGLEC15'], 'Histology': ['Panc-AdenoCA'], 'Mutation': ['chr18:g.43418687G>C'], 'Exon': [4], 'Effect': ['within 20bp of 5 prime splicing site']}\n",
      "searching for gene TP53I3\n",
      "['g.chr2:24307101C>T', 'g.chr2:24307134G>A', 'g.chr2:24307122C>T']\n",
      "searching for gene LONRF3\n",
      "['g.chrX:118143154A>G', 'g.chrX:118143157A>G']\n",
      "searching for gene RTN4\n",
      "['g.chr2:55254047C>A', 'g.chr2:55254061G>T', 'g.chr2:55200752C>T']\n",
      "{'Gene': ['RTN4'], 'Histology': ['Liver-HCC'], 'Mutation': ['chr2:g.55200752C>T'], 'Exon': [6], 'Effect': ['within 20bp of 5 prime splicing site']}\n",
      "searching for gene PPWD1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/storage/home/yur97/anaconda3/envs/mSigsyn/lib/python3.6/site-packages/IPython/core/interactiveshell.py:3263: DtypeWarning: Columns (1,17,33,34,35,36) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  if (await self.run_code(code, result,  async_=asy)):\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['g.chr5:64881864T>A', 'g.chr5:64881882A>T']\n",
      "searching for gene MAGEC1\n",
      "['g.chrX:140993787G>A', 'g.chrX:140993823G>A']\n",
      "searching for gene PURA\n",
      "['g.chr5:139494369T>A', 'g.chr5:139494363T>C', 'g.chr5:139494468T>C', 'g.chr5:139494471G>C', 'g.chr5:139494492G>C', 'g.chr5:139494495T>C', 'g.chr5:139494501A>G']\n",
      "searching for gene HIST1H2BK\n",
      "['g.chr6:27114272C>G', 'g.chr6:27114271_27114272GC>AA', 'g.chr6:27114449G>A', 'g.chr6:27114467G>A', 'g.chr6:27114271G>A', 'g.chr6:27114440C>T']\n",
      "searching for gene NOL9\n",
      "['g.chr1:6614184G>A', 'g.chr1:6614410C>G', 'g.chr1:6614182C>G', 'g.chr1:6614362G>A', 'g.chr1:6614524G>A', 'g.chr1:6614329G>C', 'g.chr1:6614311G>A']\n",
      "searching for gene ZCCHC5\n",
      "['g.chrX:77913840C>T', 'g.chrX:77912781G>A', 'g.chrX:77912744G>A']\n",
      "searching for gene PRX\n",
      "['g.chr19:40902642C>T', 'g.chr19:40900053G>T']\n",
      "searching for gene ITPR2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/storage/home/yur97/anaconda3/envs/mSigsyn/lib/python3.6/site-packages/IPython/core/interactiveshell.py:3263: DtypeWarning: Columns (24) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  if (await self.run_code(code, result,  async_=asy)):\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['g.chr12:26755360C>T', 'g.chr12:26784883G>T', 'g.chr12:26784934G>T']\n",
      "searching for gene SH3BGR\n",
      "['g.chr21:40834315G>A', 'g.chr21:40834462C>T']\n",
      "{'Gene': ['SH3BGR'], 'Histology': ['Panc-Endocrine'], 'Mutation': ['chr21:g.40834315G>A'], 'Exon': [3], 'Effect': ['within 20bp of 5 prime splicing site']}\n",
      "searching for gene ACVRL1\n",
      "['g.chr12:52307539C>T', 'g.chr12:52307549T>C', 'g.chr12:52314611G>A']\n",
      "{'Gene': ['ACVRL1'], 'Histology': ['Prost-AdenoCA'], 'Mutation': ['chr12:g.52307539C>T'], 'Exon': [4], 'Effect': ['within 20bp of 3 prime splicing site']}\n",
      "{'Gene': ['ACVRL1'], 'Histology': ['Prost-AdenoCA'], 'Mutation': ['chr12:g.52307549T>C'], 'Exon': [4], 'Effect': ['within 20bp of 3 prime splicing site']}\n",
      "searching for gene MSRB2\n",
      "['g.chr10:23384576C>T', 'g.chr10:23384552C>G']\n",
      "searching for gene TPM2\n",
      "['g.chr9:35689758G>T', 'g.chr9:35689782C>G']\n",
      "searching for gene ZFP69\n",
      "['g.chr1:40961323C>G', 'g.chr1:40961446C>T']\n",
      "searching for gene SOX18\n",
      "['g.chr20:62680308C>T', 'g.chr20:62680134C>T', 'g.chr20:62680053G>A', 'g.chr20:62679855C>T']\n",
      "{'Gene': ['SOX18'], 'Histology': ['Eso-AdenoCA'], 'Mutation': ['chr20:g.62680308C>T'], 'Exon': [2], 'Effect': ['within 20bp of 5 prime splicing site']}\n",
      "searching for gene NCAN\n",
      "['g.chr19:19338265A>C', 'g.chr19:19360618A>G', 'g.chr19:19351488C>T']\n",
      "{'Gene': ['NCAN'], 'Histology': ['Biliary-AdenoCA'], 'Mutation': ['chr19:g.19351488C>T'], 'Exon': [12], 'Effect': ['within 20bp of 3 prime splicing site']}\n",
      "searching for gene HIST1H4E\n",
      "['g.chr6:26205019T>A', 'g.chr6:26205043G>T', 'g.chr6:26205082T>G']\n",
      "searching for gene DMRTB1\n",
      "['g.chr1:53930492C>T', 'g.chr1:53925543C>T', 'g.chr1:53925618G>A', 'g.chr1:53925309C>T']\n",
      "searching for gene KLHL21\n",
      "['g.chr1:6662178_6662179GG>AA', 'g.chr1:6662493G>A', 'g.chr1:6662178G>A', 'g.chr1:6662089C>G']\n",
      "searching for gene ADAM17\n",
      "['g.chr2:9630591A>G', 'g.chr2:9675993G>T']\n",
      "searching for gene ZNF648\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/storage/home/yur97/anaconda3/envs/mSigsyn/lib/python3.6/site-packages/IPython/core/interactiveshell.py:3263: DtypeWarning: Columns (17,24,28,33,34,35,36) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  if (await self.run_code(code, result,  async_=asy)):\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['g.chr1:182025565G>A', 'g.chr1:182025649G>A', 'g.chr1:182026306C>T']\n",
      "searching for gene GLRA3\n",
      "['g.chr4:175710034A>T', 'g.chr4:175710037A>T']\n",
      "searching for gene TMEM129\n",
      "['g.chr4:1720112G>A', 'g.chr4:1720274G>A']\n"
     ]
    }
   ],
   "source": [
    "df_splice = pd.DataFrame(columns = ['Gene','Histology', 'Mutation', 'Exon', 'Effect'])\n",
    "for genes in dict_name_id:\n",
    "    print(f'searching for gene {genes}')\n",
    "    transcript = dict_name_id[genes]\n",
    "    chromosome = dict_transcript_info[transcript]['chr']\n",
    "    CDS = dict_transcript_info[transcript]['CDS']\n",
    "    EXON = dict_transcript_info[transcript]['exon']\n",
    "    len_cds = len(CDS);len_exon = len(dict_transcript_info[transcript]['exon'])\n",
    "    # If there is extra UTR exon, add the index by +1 in the following session\n",
    "    n_plusidx = len_exon-len_cds+1\n",
    "    strand = dict_transcript_info[transcript]['strand']\n",
    "    df_synmut, his = get_syn_mut(genes)\n",
    "\n",
    "    # If negative strand then reverse the exon orders\n",
    "    # if strand == '-':\n",
    "    #     CDS.reverse()\n",
    "    print(df_synmut['Genome_Change'].unique().tolist())\n",
    "    for mutations in df_synmut['Genome_Change'].unique().tolist():\n",
    "        change = mutations.split(':')[1]\n",
    "        mutation = chromosome+':g.'+change\n",
    "        muts = int(re.findall(r'\\d+|X|Y', mutation)[1])\n",
    "        for idx,exons in enumerate(CDS):\n",
    "            if idx == 0: # Find initiation mutations\n",
    "                res, eff = find_initiation_pos(muts, exons, strand)\n",
    "                if res == 1:\n",
    "                    exn = idx +n_plusidx; df_gene_splice = make_gene_df(genes, his, mutation, exn, eff)\n",
    "                    df_splice = pd.concat([df_splice, df_gene_splice], axis = 0)\n",
    "            else:\n",
    "                res, eff = find_splicing_pos(muts, exons, strand)\n",
    "                if res ==1:\n",
    "                    exn = idx+n_plusidx; df_gene_splice = make_gene_df(genes, his, mutation, exn, eff)\n",
    "                    df_splice = pd.concat([df_splice, df_gene_splice], axis = 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_splice.to_csv('df_splice.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_splice = pd.read_csv('df_splice.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Histology</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>Exon</th>\n",
       "      <th>Effect</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>SRSF2</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>chr17:g.74733234G&gt;A</td>\n",
       "      <td>2</td>\n",
       "      <td>within 20bp of 3 prime splicing site</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>ACTA1</td>\n",
       "      <td>Lymph-CLL</td>\n",
       "      <td>chr1:g.229567630A&gt;G</td>\n",
       "      <td>6</td>\n",
       "      <td>within 20bp of 5 prime splicing site</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>ACTA1</td>\n",
       "      <td>Lymph-CLL</td>\n",
       "      <td>chr1:g.229567636C&gt;A</td>\n",
       "      <td>6</td>\n",
       "      <td>within 20bp of 5 prime splicing site</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>ACTA1</td>\n",
       "      <td>Lymph-CLL</td>\n",
       "      <td>chr1:g.229567639C&gt;A</td>\n",
       "      <td>6</td>\n",
       "      <td>within 20bp of 5 prime splicing site</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>SIGLEC15</td>\n",
       "      <td>Panc-AdenoCA</td>\n",
       "      <td>chr18:g.43418687G&gt;C</td>\n",
       "      <td>4</td>\n",
       "      <td>within 20bp of 5 prime splicing site</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0</td>\n",
       "      <td>RTN4</td>\n",
       "      <td>Liver-HCC</td>\n",
       "      <td>chr2:g.55200752C&gt;T</td>\n",
       "      <td>6</td>\n",
       "      <td>within 20bp of 5 prime splicing site</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>0</td>\n",
       "      <td>SH3BGR</td>\n",
       "      <td>Panc-Endocrine</td>\n",
       "      <td>chr21:g.40834315G&gt;A</td>\n",
       "      <td>3</td>\n",
       "      <td>within 20bp of 5 prime splicing site</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>0</td>\n",
       "      <td>ACVRL1</td>\n",
       "      <td>Prost-AdenoCA</td>\n",
       "      <td>chr12:g.52307539C&gt;T</td>\n",
       "      <td>4</td>\n",
       "      <td>within 20bp of 3 prime splicing site</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>0</td>\n",
       "      <td>ACVRL1</td>\n",
       "      <td>Prost-AdenoCA</td>\n",
       "      <td>chr12:g.52307549T&gt;C</td>\n",
       "      <td>4</td>\n",
       "      <td>within 20bp of 3 prime splicing site</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>0</td>\n",
       "      <td>SOX18</td>\n",
       "      <td>Eso-AdenoCA</td>\n",
       "      <td>chr20:g.62680308C&gt;T</td>\n",
       "      <td>2</td>\n",
       "      <td>within 20bp of 5 prime splicing site</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>0</td>\n",
       "      <td>NCAN</td>\n",
       "      <td>Biliary-AdenoCA</td>\n",
       "      <td>chr19:g.19351488C&gt;T</td>\n",
       "      <td>12</td>\n",
       "      <td>within 20bp of 3 prime splicing site</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Unnamed: 0      Gene        Histology             Mutation  Exon  \\\n",
       "0            0     SRSF2       Lymph-BNHL  chr17:g.74733234G>A     2   \n",
       "1            0     ACTA1        Lymph-CLL  chr1:g.229567630A>G     6   \n",
       "2            0     ACTA1        Lymph-CLL  chr1:g.229567636C>A     6   \n",
       "3            0     ACTA1        Lymph-CLL  chr1:g.229567639C>A     6   \n",
       "4            0  SIGLEC15     Panc-AdenoCA  chr18:g.43418687G>C     4   \n",
       "5            0      RTN4        Liver-HCC   chr2:g.55200752C>T     6   \n",
       "6            0    SH3BGR   Panc-Endocrine  chr21:g.40834315G>A     3   \n",
       "7            0    ACVRL1    Prost-AdenoCA  chr12:g.52307539C>T     4   \n",
       "8            0    ACVRL1    Prost-AdenoCA  chr12:g.52307549T>C     4   \n",
       "9            0     SOX18      Eso-AdenoCA  chr20:g.62680308C>T     2   \n",
       "10           0      NCAN  Biliary-AdenoCA  chr19:g.19351488C>T    12   \n",
       "\n",
       "                                  Effect  \n",
       "0   within 20bp of 3 prime splicing site  \n",
       "1   within 20bp of 5 prime splicing site  \n",
       "2   within 20bp of 5 prime splicing site  \n",
       "3   within 20bp of 5 prime splicing site  \n",
       "4   within 20bp of 5 prime splicing site  \n",
       "5   within 20bp of 5 prime splicing site  \n",
       "6   within 20bp of 5 prime splicing site  \n",
       "7   within 20bp of 3 prime splicing site  \n",
       "8   within 20bp of 3 prime splicing site  \n",
       "9   within 20bp of 5 prime splicing site  \n",
       "10  within 20bp of 3 prime splicing site  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_splice"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'BCL2': 'ENST00000398117',\n",
       " 'SRSF2': 'ENST00000359995',\n",
       " 'ITLN1': 'ENST00000326245',\n",
       " 'ACTA1': 'ENST00000366684',\n",
       " 'SIGLEC15': 'ENST00000389474',\n",
       " 'TP53I3': 'ENST00000335934',\n",
       " 'LONRF3': 'ENST00000371628',\n",
       " 'RTN4': 'ENST00000394609',\n",
       " 'PPWD1': 'ENST00000261308',\n",
       " 'MAGEC1': 'ENST00000285879',\n",
       " 'PURA': 'ENST00000331327',\n",
       " 'HIST1H2BK': 'ENST00000396891',\n",
       " 'NOL9': 'ENST00000377705',\n",
       " 'ZCCHC5': 'ENST00000321110',\n",
       " 'PRX': 'ENST00000291825',\n",
       " 'ITPR2': 'ENST00000381340',\n",
       " 'SH3BGR': 'ENST00000333634',\n",
       " 'ACVRL1': 'ENST00000388922',\n",
       " 'MSRB2': 'ENST00000376510',\n",
       " 'TPM2': 'ENST00000378292',\n",
       " 'ZFP69': 'ENST00000372706',\n",
       " 'SOX18': 'ENST00000340356',\n",
       " 'NCAN': 'ENST00000252575',\n",
       " 'HIST1H4E': 'ENST00000360441',\n",
       " 'DMRTB1': 'ENST00000371445',\n",
       " 'KLHL21': 'ENST00000377658',\n",
       " 'ADAM17': 'ENST00000310823',\n",
       " 'ZNF648': 'ENST00000339948',\n",
       " 'GLRA3': 'ENST00000274093',\n",
       " 'TMEM129': 'ENST00000382936'}"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dict_name_id"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
