{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# This script use ensemble api to check overlapping genes in the in the MutSig output\n",
    "# What really checked is the exact mutation that overlap with other genes\n",
    "# Author: Yiyun\n",
    "import json\n",
    "from os.path import join, isdir, exists\n",
    "from os import listdir\n",
    "import requests, sys\n",
    "import time\n",
    "import pandas as pd\n",
    "from multiprocessing import Pool\n",
    "import pickle\n",
    "import mmap\n",
    "from tqdm import tqdm\n",
    "\n",
    "server = \"https://grch37.rest.ensembl.org\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "nsyn_dir = '../mutsig_out/nsyn/out10212020'\n",
    "syn_dir = '../mutsig_out/syn/out10212020'\n",
    "org_dir = '../org_cov_maf'\n",
    "outdir = '../proc_09152020/dovgene'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# From forum https://stackoverflow.com/questions/667508/whats-a-good-rate-limiting-algorithm\n",
    "def RateLimited(maxPerSecond):\n",
    "    minInterval = 1.0 / float(maxPerSecond)\n",
    "    def decorate(func):\n",
    "        lastTimeCalled = [0.0]\n",
    "        def rateLimitedFunction(*args,**kargs):\n",
    "            elapsed = time.clock() - lastTimeCalled[0]\n",
    "            leftToWait = minInterval - elapsed\n",
    "            if leftToWait>0:\n",
    "                time.sleep(leftToWait)\n",
    "            ret = func(*args,**kargs)\n",
    "            lastTimeCalled[0] = time.clock()\n",
    "            return ret\n",
    "        return rateLimitedFunction\n",
    "    return decorate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# send the query to ensembl by position and variant, currently only work for SNP\n",
    "# return a dictionary of organ-gene-mutation-overlap gene-consequence\n",
    "@RateLimited(12)\n",
    "def get_overlapping_cons(muts):\n",
    "    ext = '/vep/human/region/'+muts+'/?'\n",
    "    r = requests.get(server+ext,headers={\"Content-Type\" : \"application/json\"})\n",
    "\n",
    "    if not r.ok:\n",
    "      r.raise_for_status()\n",
    "      sys.exit()\n",
    "\n",
    "    decoded = r.json()\n",
    "    \n",
    "    return decoded[0]['transcript_consequences']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def progress(n):\n",
    "    if n%500==0:\n",
    "        print \"finish reading \"+ str(n)+ ' mutations'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def get_targe_gene(org):\n",
    "    organ_dir = join(syn_dir,org)\n",
    "    for f in listdir(organ_dir):\n",
    "        if f.endswith('sig_genes.txt'):\n",
    "            f_dir = join(organ_dir,f)\n",
    "            df_org = pd.read_csv(f_dir,sep = '\\t', index_col=None)\n",
    "            lg = df_org['gene'].tolist()\n",
    "    \n",
    "    # divide genelist into 3 list\n",
    "    chunks = [lg[x] for x in xrange(0, len(lg), 200)]\n",
    "    \n",
    "#     tc = chunks[0]\n",
    "#     bc = chunks[-1]\n",
    "#     mc = chunks[int(len(chunks)/2)]\n",
    "    \n",
    "    return chunks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# This function get a temporaty maf file\n",
    "def get_variant_maf(org,lgene):\n",
    "    \n",
    "    dmut = {}\n",
    "    \n",
    "    maf = join(org_dir,org,'merged_tumor_covmerged_'+org+'.txt_filtered.maf')\n",
    "\n",
    "    with open(maf,'r+b') as f:\n",
    "        next(f)\n",
    "        mmaf = mmap.mmap(f.fileno(), 0)\n",
    "        for lines in iter(mmaf.readline, \"\"): \n",
    "            name,chrom,startp,endp,strand,vc,vt, ref, tum1, tum2= lines.split('\\t')[0:10]\n",
    "        # get variant\n",
    "            if any(name in l for l in lgene) and vt == 'SNP' and vc != 'RNA' and vc!= 'lincRNA' and name != 'Unknown':\n",
    "                if tum1 == ref and tum2 != ref:\n",
    "                    var = tum2\n",
    "                elif tum2 == ref and tum1 != ref:\n",
    "                    var = tum1\n",
    "                elif tum2 != ref and tum1 != ref:\n",
    "                    var = tum1\n",
    "                else:\n",
    "                    print \"NO VARIANT\" + name +'\\t'+ starp\n",
    "                    continue\n",
    "                    \n",
    "                mut = str(chrom)+':'+str(startp)+'-'+str(startp)+'/'+str(var)\n",
    "                if name not in dmut:\n",
    "                    dmut[name] = {}\n",
    "                if mut not in dmut[name]:\n",
    "                    dmut[name][mut] = {}\n",
    "                if vc not in dmut[name][mut]:\n",
    "                    dmut[name][mut] = {}\n",
    "                    dmut[name][mut]['vc'] = vc\n",
    "                \n",
    "        mmaf.close()\n",
    "        \n",
    "    pickle.dump(dmut, open(org+'_tmp.pkl','wb'))\n",
    "    \n",
    "    return dmut"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# this function go through maf file and get potential overlapping genes for mutation\n",
    "def get_variant(org):\n",
    "    print 'Start: '+org\n",
    "    global org_dir, outdir\n",
    "    # a numerator for mutations\n",
    "    n = 0\n",
    "    \n",
    "    listgene = get_targe_gene(org)\n",
    "    dict_mut = get_variant_maf(org,listgene)\n",
    "    \n",
    "    # open maf file and get variant\n",
    "    outf = join(outdir,org+'_102120.tsv')\n",
    "    lf = []\n",
    "    if exists(outf):\n",
    "        with open(outf,'r') as fo:\n",
    "            for lines in fo:\n",
    "                gpos = lines.split('\\t')[1].strip('\\n')\n",
    "                lf.append(gpos)\n",
    "    else:\n",
    "        fo = open(outf,'wb')\n",
    "        fo.close()\n",
    "            \n",
    "    for gene in dict_mut: \n",
    "        for mut in dict_mut[gene]:\n",
    "            n+=1\n",
    "            progress(n)\n",
    "            \n",
    "            if mut in lf:\n",
    "                continue\n",
    "\n",
    "            # ------send to ensembl api ------\n",
    "\n",
    "            res = get_overlapping_cons(mut)\n",
    "            for cons in res:\n",
    "                # if it's not the target gene, append to dictionary\n",
    "                if cons['gene_symbol'] != gene: \n",
    "                    ovpgene = cons['gene_symbol']\n",
    "\n",
    "                    with open(outf, 'a') as of:\n",
    "                         of.write(gene+'\\t'+mut+'\\t'+str(dict_mut[gene][mut]['vc'])+'\\t'+ovpgene+'\\t'+str(cons['consequence_terms'])+'\\n')\n",
    "                else:\n",
    "                    with open(outf, 'a') as of:\n",
    "                         of.write(gene+'\\t'+mut+'\\n')\n",
    "\n",
    "    \n",
    "    print 'Finish: '+ org"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Ovary', 'Thy', 'Biliary', 'Breast', 'Panc', 'Lung', 'Liver', 'CNS', 'Lymph', 'Bone', 'Eso', 'ColoRect', 'Kidney', 'Myeloid', 'Stomach', 'Prost', 'Skin', 'Head', 'Uterus', 'SoftTissue', 'Cervix', 'Bladder']\n",
      "Start: Ovary\n",
      "Start: Liver\n",
      "Start: Panc\n",
      "Start: Biliary\n",
      "finish reading 500 mutations\n",
      "finish reading 1000 mutations\n",
      "finish reading 1500 mutations\n",
      "Finish: Biliary\n",
      "Start: Breast\n",
      "finish reading 500 mutations\n",
      "finish reading 1000 mutations\n",
      "finish reading 1500 mutations\n",
      "Finish: Ovary\n",
      "Start: Thy\n",
      "Finish: Thy\n",
      "Start: Lymph\n",
      "finish reading 500 mutations\n",
      "finish reading 1000 mutations\n",
      "finish reading 1500 mutations\n",
      "finish reading 2000 mutations\n",
      "finish reading 2500 mutations\n",
      "finish reading 3000 mutations\n",
      "finish reading 3500 mutations\n",
      "Finish: Panc\n",
      "Start: Lung\n",
      "finish reading 500 mutations\n",
      "finish reading 1000 mutations\n",
      "finish reading 1500 mutations\n",
      "finish reading 2000 mutations\n",
      "finish reading 2500 mutations\n",
      "finish reading 3000 mutations\n",
      "finish reading 3500 mutations\n",
      "finish reading 4000 mutations\n",
      "finish reading 4500 mutations\n",
      "Finish: Breast\n",
      "Start: Eso\n",
      "finish reading 500 mutations\n",
      "finish reading 1000 mutations\n",
      "finish reading 1500 mutations\n",
      "finish reading 2000 mutations\n",
      "finish reading 2500 mutations\n",
      "finish reading 3000 mutations\n",
      "finish reading 3500 mutations\n",
      "finish reading 4000 mutations\n",
      "finish reading 4500 mutations\n",
      "finish reading 5000 mutations\n",
      "Finish: Lymph\n",
      "Start: Bone\n",
      "finish reading 500 mutations\n",
      "Finish: Bone\n",
      "Start: Kidney\n",
      "finish reading 500 mutations\n",
      "finish reading 1000 mutations\n",
      "finish reading 1500 mutations\n",
      "finish reading 2000 mutations\n",
      "finish reading 2500 mutations\n",
      "finish reading 3000 mutations\n",
      "finish reading 3500 mutations\n",
      "finish reading 4000 mutations\n",
      "finish reading 4500 mutations\n",
      "finish reading 500 mutations\n",
      "finish reading 1000 mutations\n",
      "finish reading 5000 mutations\n",
      "finish reading 1500 mutations\n",
      "finish reading 2000 mutations\n",
      "finish reading 5500 mutations\n",
      "finish reading 2500 mutations\n",
      "finish reading 3000 mutations\n",
      "finish reading 6000 mutations\n",
      "Finish: Kidney\n",
      "Start: Myeloid\n",
      "finish reading 6500 mutations\n",
      "Finish: Myeloid\n",
      "Start: Stomach\n",
      "finish reading 7000 mutations\n",
      "finish reading 7500 mutations\n",
      "finish reading 8000 mutations\n",
      "finish reading 8500 mutations\n",
      "finish reading 9000 mutations\n",
      "finish reading 9500 mutations\n",
      "finish reading 10000 mutations\n",
      "finish reading 10500 mutations\n",
      "finish reading 11000 mutations\n",
      "finish reading 11500 mutations\n",
      "Finish: Liver\n",
      "Start: CNS\n",
      "finish reading 500 mutations\n",
      "finish reading 1000 mutations\n",
      "finish reading 1500 mutations\n",
      "finish reading 2000 mutations\n",
      "finish reading 2500 mutations\n",
      "finish reading 3000 mutations\n",
      "finish reading 3500 mutations\n",
      "finish reading 4000 mutations\n",
      "finish reading 4500 mutations\n",
      "finish reading 5000 mutations\n",
      "finish reading 5500 mutations\n",
      "finish reading 6000 mutations\n",
      "finish reading 6500 mutations\n",
      "finish reading 7000 mutations\n",
      "finish reading 500 mutations\n",
      "finish reading 7500 mutations\n",
      "finish reading 1000 mutations\n",
      "finish reading 1500 mutations\n",
      "finish reading 8000 mutations\n",
      "finish reading 2000 mutations\n",
      "finish reading 8500 mutations\n",
      "finish reading 2500 mutations\n",
      "Finish: Eso\n",
      "Start: ColoRect\n",
      "finish reading 3000 mutations\n",
      "finish reading 3500 mutations\n",
      "finish reading 4000 mutations\n",
      "finish reading 4500 mutations\n",
      "finish reading 5000 mutations\n",
      "finish reading 5500 mutations\n",
      "finish reading 500 mutations\n",
      "finish reading 6000 mutations\n",
      "finish reading 1000 mutations\n",
      "finish reading 1500 mutations\n",
      "Finish: CNS\n",
      "Start: Skin\n",
      "finish reading 6500 mutations\n",
      "finish reading 7000 mutations\n",
      "finish reading 7500 mutations\n",
      "finish reading 8000 mutations\n",
      "finish reading 8500 mutations\n",
      "Finish: Lung\n",
      "Start: Uterus\n",
      "finish reading 500 mutations\n",
      "finish reading 1000 mutations\n",
      "finish reading 1500 mutations\n",
      "finish reading 2000 mutations\n",
      "finish reading 2500 mutations\n",
      "finish reading 3000 mutations\n",
      "finish reading 3500 mutations\n",
      "finish reading 4000 mutations\n",
      "finish reading 4500 mutations\n",
      "finish reading 5000 mutations\n",
      "finish reading 5500 mutations\n",
      "finish reading 6000 mutations\n",
      "Finish: Stomach\n",
      "Start: Prost\n",
      "finish reading 500 mutations\n",
      "finish reading 1000 mutations\n",
      "finish reading 1500 mutations\n",
      "finish reading 2000 mutations\n",
      "Finish: Uterus\n",
      "Start: SoftTissue\n",
      "Start: Cervix\n"
     ]
    },
    {
     "ename": "OSError",
     "evalue": "[Errno 2] No such file or directory: '../mutsig_out/syn/out10212020/SoftTissue'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m\u001b[0m",
      "\u001b[0;31mOSError\u001b[0mTraceback (most recent call last)",
      "\u001b[0;32m<ipython-input-24-64ac6039d4de>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0;32mprint\u001b[0m \u001b[0mlorg\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mp\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mPool\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m4\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 5\u001b[0;31m \u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmap\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mget_variant\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlorg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      6\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/aci/sw/python/2.7.14_anaconda-5.0.1/lib/python2.7/multiprocessing/pool.pyc\u001b[0m in \u001b[0;36mmap\u001b[0;34m(self, func, iterable, chunksize)\u001b[0m\n\u001b[1;32m    251\u001b[0m         '''\n\u001b[1;32m    252\u001b[0m         \u001b[0;32massert\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_state\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0mRUN\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 253\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmap_async\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfunc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0miterable\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mchunksize\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    254\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    255\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mimap\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0miterable\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mchunksize\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/aci/sw/python/2.7.14_anaconda-5.0.1/lib/python2.7/multiprocessing/pool.pyc\u001b[0m in \u001b[0;36mget\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    570\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_value\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    571\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 572\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_value\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    573\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    574\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_set\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mi\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mobj\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mOSError\u001b[0m: [Errno 2] No such file or directory: '../mutsig_out/syn/out10212020/SoftTissue'"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "finish reading 500 mutations\n",
      "finish reading 1000 mutations\n",
      "finish reading 1500 mutations\n",
      "Finish: Cervix\n",
      "Start: Bladder\n",
      "finish reading 500 mutations\n",
      "finish reading 1000 mutations\n",
      "finish reading 1500 mutations\n",
      "Finish: Bladder\n",
      "finish reading 500 mutations\n",
      "finish reading 1000 mutations\n",
      "finish reading 1500 mutations\n",
      "finish reading 2000 mutations\n",
      "finish reading 2500 mutations\n",
      "finish reading 3000 mutations\n",
      "finish reading 3500 mutations\n",
      "finish reading 500 mutations\n",
      "finish reading 1000 mutations\n",
      "finish reading 1500 mutations\n",
      "finish reading 2000 mutations\n",
      "finish reading 2500 mutations\n",
      "finish reading 3000 mutations\n",
      "finish reading 3500 mutations\n",
      "finish reading 4000 mutations\n",
      "finish reading 4500 mutations\n",
      "finish reading 5000 mutations\n",
      "finish reading 5500 mutations\n",
      "finish reading 6000 mutations\n",
      "finish reading 4000 mutations\n",
      "finish reading 6500 mutations\n",
      "Finish: Prost\n",
      "finish reading 7000 mutations\n",
      "finish reading 4500 mutations\n",
      "finish reading 7500 mutations\n"
     ]
    }
   ],
   "source": [
    "lorg = listdir(org_dir)\n",
    "lorg = [l for l in lorg if not l.startswith('.')]\n",
    "print lorg\n",
    "p = Pool(4)\n",
    "p.map(get_variant, lorg)\n",
    "p.join"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
