{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "# This notebook compare the synonsymou mutations of patients with follicular lymphoma /wo\n",
    "# Author: Yiyun\n",
    "import mmap\n",
    "from os import listdir\n",
    "from os.path import join\n",
    "from collections import Counter\n",
    "import requests, sys\n",
    "import re\n",
    "import pandas as pd\n",
    "import time\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "server = \"https://grch37.rest.ensembl.org\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "flymph = '../org_cov_maf/Lymph/merged_tumor_covmerged_Lymph.txt_filtered.maf'\n",
    "finfo = '../anno_ref/PCAWG_sample_info.txt'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "# get the count for syonymous mutations in patients\n",
    "def get_count(dictionary):\n",
    "    ls = [] # synonymous mutations list\n",
    "    for k, v in dictionary.iteritems():\n",
    "        for mut in v:\n",
    "            ls.append(mut)\n",
    "    \n",
    "    return ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "# A rate limit function\n",
    "# From forum https://stackoverflow.com/questions/667508/whats-a-good-rate-limiting-algorithm\n",
    "def RateLimited(maxPerSecond):\n",
    "    minInterval = 1.0 / float(maxPerSecond)\n",
    "    def decorate(func):\n",
    "        lastTimeCalled = [0.0]\n",
    "        def rateLimitedFunction(*args,**kargs):\n",
    "            elapsed = time.clock() - lastTimeCalled[0]\n",
    "            leftToWait = minInterval - elapsed\n",
    "            if leftToWait>0:\n",
    "                time.sleep(leftToWait)\n",
    "            ret = func(*args,**kargs)\n",
    "            lastTimeCalled[0] = time.clock()\n",
    "            return ret\n",
    "        return rateLimitedFunction\n",
    "    return decorate\n",
    "\n",
    "# send the query to ensembl by position and variant, currently only work for SNP\n",
    "# return a dictionary of organ-gene-mutation-overlap gene-consequence\n",
    "@RateLimited(12)\n",
    "def get_overlapping_cons(muts):\n",
    "    ext = '/vep/human/region/'+muts+'/?'\n",
    "    r = requests.get(server+ext,headers={\"Content-Type\" : \"application/json\"})\n",
    "\n",
    "    if not r.ok:\n",
    "      r.raise_for_status()\n",
    "      sys.exit()\n",
    "\n",
    "    decoded = r.json()\n",
    "    \n",
    "    return decoded\n",
    "\n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Find all follicular lymphoma patients  \n",
    "Note: None of the follicular lymphoma patients are from TCGA  \n",
    "All from MALY-DE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "# get patient tumor id with follicular lymphoma\n",
    "lfp = []\n",
    "with open(finfo,'r') as f:\n",
    "    for lines in f:\n",
    "        line_split = lines.split('\\t')\n",
    "        if 'Follicular lymphoma' in line_split[14]:\n",
    "            lfp.append(line_split[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "36"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(lfp) #number of follicular lymphoma patient"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Find the driver mut in patients"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "fdriver = '../anno_ref/icgc/driver_mutation/TableS3_panorama_driver_mutations_ICGC_samples.public.tsv'\n",
    "# initiate a datastructure to store driver\n",
    "driver_col = ['patient','ttype','chr','pos','res','alt','gene','driver','driver_statement','category','top_category','biallelic' ]\n",
    "df_driver = pd.DataFrame(columns = driver_col)\n",
    "\n",
    "with open(fdriver,'r') as driverf:\n",
    "    for lines in driverf:\n",
    "        line_split = lines.split('\\t')\n",
    "        patient_id, ttype,chrom,pos,res,alt,gene,driver,driver_statement,category, top_category, biallelic = line_split[0:13]\n",
    "        \n",
    "        a_series = pd.Series(line_split, index = df_driver.columns)\n",
    "\n",
    "        if patient_id in lfp:\n",
    "            df_driver = df_driver.append(a_series, ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>ttype</th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>res</th>\n",
       "      <th>alt</th>\n",
       "      <th>gene</th>\n",
       "      <th>driver</th>\n",
       "      <th>driver_statement</th>\n",
       "      <th>category</th>\n",
       "      <th>top_category</th>\n",
       "      <th>biallelic</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>02917220-6a7a-46a1-8656-907e96bef88e</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>03ad38a6-0902-4aaa-84a3-91ea88fa9883</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>15678c3d-ce61-454d-849e-1d4ca358909b</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1d1eeba2-4d7b-4380-b0b6-26d246bc4158</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>232b7754-b3c3-4530-9d24-3af40d9b2816</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>3e7ccab5-5b1d-4147-b907-77cab8f0837e</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>3e94aa64-7dd3-4d63-a5de-b4050d3dfafa</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>4a4309b0-2aca-4d6a-ac50-039b28406bba</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL6_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL6</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>4d72d06d-b6af-4294-9eef-6e61e067761f</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>4e7cdeda-6dc1-4f17-b853-72a68e5aa7e1</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL6_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL6</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>4e7e6e1f-c648-446f-bdf6-0b1fcc6dfa83</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>56e57223-264e-489d-bc9a-a07789065c40</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>5f18b334-6616-406f-8f33-0592b56c14e6</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>626c9788-d70c-449b-b9dc-5e3f6fc00991</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>650fe009-da01-4717-89df-9c95fafe3d7e</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>81b4dd02-34ec-4ddc-8818-ed888a60ad88</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>81cc0f39-6677-4f2e-9a75-d30152b188f3</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>866ecfe7-caa6-4565-9418-6b9d6c8a3b43</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>8fc11cb6-9d79-41c3-bb45-0dadda890e8d</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>9e842227-2bc6-4185-bedb-abeb03836b51</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>baa02e74-9cae-4996-858e-17d9f52d924b</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>bd6bd940-ef06-433a-80d9-dca5683e9cd2</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>bdbcffb0-c7da-4e7d-8002-9feffe48533c</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>d6ee68e5-ceb3-4a63-b3bd-e64fe5398102</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>e2b09705-c5c8-48ee-a90e-19648a7bf2cb</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>e98e4e23-a139-4fff-b017-35f3f976eddd</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ebe0ed67-2d3f-45cd-8f9b-4912595b16a0</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>f047cb85-8f88-45d5-83c3-0416c9f3fff7</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>f04aecf0-eb12-4ab9-928e-7bcf201b0b40</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>f2c08ac1-4755-4c36-a11d-338d8df9fa0d</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL6_IGH</td>\n",
       "      <td>x</td>\n",
       "      <td>BCL6</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>cis-activating_SV</td>\n",
       "      <td>SV</td>\n",
       "      <td>no\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>231</th>\n",
       "      <td>3e7ccab5-5b1d-4147-b907-77cab8f0837e</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>B2M</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SMutation\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232</th>\n",
       "      <td>3e7ccab5-5b1d-4147-b907-77cab8f0837e</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>TNFRSF14</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SMutation\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233</th>\n",
       "      <td>4e7cdeda-6dc1-4f17-b853-72a68e5aa7e1</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>TP53</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SMutation\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234</th>\n",
       "      <td>4e7e6e1f-c648-446f-bdf6-0b1fcc6dfa83</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>CREBBP</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SMutation\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>235</th>\n",
       "      <td>56e57223-264e-489d-bc9a-a07789065c40</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>ARID1B</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SDeletion\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>626c9788-d70c-449b-b9dc-5e3f6fc00991</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>TMSB4X</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SMutation\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>237</th>\n",
       "      <td>626c9788-d70c-449b-b9dc-5e3f6fc00991</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>TP53</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SMutation\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>238</th>\n",
       "      <td>650fe009-da01-4717-89df-9c95fafe3d7e</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>CREBBP</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SMutation\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>239</th>\n",
       "      <td>687baf0b-aed9-4ab2-adb6-556a1d177a24</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>TNFRSF14</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SMutation\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>240</th>\n",
       "      <td>81cc0f39-6677-4f2e-9a75-d30152b188f3</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>CREBBP</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SMutation\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241</th>\n",
       "      <td>81cc0f39-6677-4f2e-9a75-d30152b188f3</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>TNFRSF14</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SMutation\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>242</th>\n",
       "      <td>866ecfe7-caa6-4565-9418-6b9d6c8a3b43</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>CREBBP</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SMutation\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>243</th>\n",
       "      <td>8fc11cb6-9d79-41c3-bb45-0dadda890e8d</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>CREBBP</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SMutation\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>244</th>\n",
       "      <td>8fc11cb6-9d79-41c3-bb45-0dadda890e8d</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>TNFAIP3</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SDeletion\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>245</th>\n",
       "      <td>8fc11cb6-9d79-41c3-bb45-0dadda890e8d</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>TP53</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SMutation\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246</th>\n",
       "      <td>9c27fedd-b1b3-4af0-9e9b-20271854db08</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>ARID5B</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SMutation\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>247</th>\n",
       "      <td>ac2c8928-33a2-4aa0-8bc1-cdfa1ca0b56a</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>CDKN2A</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SDeletion\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248</th>\n",
       "      <td>bd6bd940-ef06-433a-80d9-dca5683e9cd2</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>CDKN2A</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SDeletion\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249</th>\n",
       "      <td>bd6bd940-ef06-433a-80d9-dca5683e9cd2</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>CDKN2B</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SDeletion\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>250</th>\n",
       "      <td>bdbcffb0-c7da-4e7d-8002-9feffe48533c</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>CREBBP</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SMutation\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>251</th>\n",
       "      <td>db9ce6c6-529c-4da6-92c4-f4f3fc3fe109</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>TNFRSF14</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SMutation\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>252</th>\n",
       "      <td>ebe0ed67-2d3f-45cd-8f9b-4912595b16a0</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>FAS</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SDeletion\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>253</th>\n",
       "      <td>ebe0ed67-2d3f-45cd-8f9b-4912595b16a0</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>MSR1</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SSV\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>254</th>\n",
       "      <td>ebe0ed67-2d3f-45cd-8f9b-4912595b16a0</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>TMEM30A</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SMutation\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>255</th>\n",
       "      <td>ebe0ed67-2d3f-45cd-8f9b-4912595b16a0</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>TMSB4X</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SMutation\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>256</th>\n",
       "      <td>f047cb85-8f88-45d5-83c3-0416c9f3fff7</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>TNFRSF14</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SMutation\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>257</th>\n",
       "      <td>f04aecf0-eb12-4ab9-928e-7bcf201b0b40</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>CREBBP</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SMutation\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>258</th>\n",
       "      <td>f2c08ac1-4755-4c36-a11d-338d8df9fa0d</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>CREBBP</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SMutation\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>259</th>\n",
       "      <td>f2c08ac1-4755-4c36-a11d-338d8df9fa0d</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>TNFRSF14</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_deletion</td>\n",
       "      <td>CNA</td>\n",
       "      <td>SDeletion/SMutation\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>260</th>\n",
       "      <td>ebe0ed67-2d3f-45cd-8f9b-4912595b16a0</td>\n",
       "      <td>Lymph-BNHL</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>x</td>\n",
       "      <td>MSR1</td>\n",
       "      <td>known</td>\n",
       "      <td>known</td>\n",
       "      <td>coding_tsg_breakpoint</td>\n",
       "      <td>SV</td>\n",
       "      <td>SDeletion/SSV\\n</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>261 rows Ã— 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  patient       ttype chr pos       res alt  \\\n",
       "0    02917220-6a7a-46a1-8656-907e96bef88e  Lymph-BNHL   x   x  BCL2_IGH   x   \n",
       "1    03ad38a6-0902-4aaa-84a3-91ea88fa9883  Lymph-BNHL   x   x  BCL2_IGH   x   \n",
       "2    15678c3d-ce61-454d-849e-1d4ca358909b  Lymph-BNHL   x   x  BCL2_IGH   x   \n",
       "3    1d1eeba2-4d7b-4380-b0b6-26d246bc4158  Lymph-BNHL   x   x  BCL2_IGH   x   \n",
       "4    232b7754-b3c3-4530-9d24-3af40d9b2816  Lymph-BNHL   x   x  BCL2_IGH   x   \n",
       "5    3e7ccab5-5b1d-4147-b907-77cab8f0837e  Lymph-BNHL   x   x  BCL2_IGH   x   \n",
       "6    3e94aa64-7dd3-4d63-a5de-b4050d3dfafa  Lymph-BNHL   x   x  BCL2_IGH   x   \n",
       "7    4a4309b0-2aca-4d6a-ac50-039b28406bba  Lymph-BNHL   x   x  BCL6_IGH   x   \n",
       "8    4d72d06d-b6af-4294-9eef-6e61e067761f  Lymph-BNHL   x   x  BCL2_IGH   x   \n",
       "9    4e7cdeda-6dc1-4f17-b853-72a68e5aa7e1  Lymph-BNHL   x   x  BCL6_IGH   x   \n",
       "10   4e7e6e1f-c648-446f-bdf6-0b1fcc6dfa83  Lymph-BNHL   x   x  BCL2_IGH   x   \n",
       "11   56e57223-264e-489d-bc9a-a07789065c40  Lymph-BNHL   x   x  BCL2_IGH   x   \n",
       "12   5f18b334-6616-406f-8f33-0592b56c14e6  Lymph-BNHL   x   x  BCL2_IGH   x   \n",
       "13   626c9788-d70c-449b-b9dc-5e3f6fc00991  Lymph-BNHL   x   x  BCL2_IGH   x   \n",
       "14   650fe009-da01-4717-89df-9c95fafe3d7e  Lymph-BNHL   x   x  BCL2_IGH   x   \n",
       "15   81b4dd02-34ec-4ddc-8818-ed888a60ad88  Lymph-BNHL   x   x  BCL2_IGH   x   \n",
       "16   81cc0f39-6677-4f2e-9a75-d30152b188f3  Lymph-BNHL   x   x  BCL2_IGH   x   \n",
       "17   866ecfe7-caa6-4565-9418-6b9d6c8a3b43  Lymph-BNHL   x   x  BCL2_IGH   x   \n",
       "18   8fc11cb6-9d79-41c3-bb45-0dadda890e8d  Lymph-BNHL   x   x  BCL2_IGH   x   \n",
       "19   9e842227-2bc6-4185-bedb-abeb03836b51  Lymph-BNHL   x   x  BCL2_IGH   x   \n",
       "20   baa02e74-9cae-4996-858e-17d9f52d924b  Lymph-BNHL   x   x  BCL2_IGH   x   \n",
       "21   bd6bd940-ef06-433a-80d9-dca5683e9cd2  Lymph-BNHL   x   x  BCL2_IGH   x   \n",
       "22   bdbcffb0-c7da-4e7d-8002-9feffe48533c  Lymph-BNHL   x   x  BCL2_IGH   x   \n",
       "23   d6ee68e5-ceb3-4a63-b3bd-e64fe5398102  Lymph-BNHL   x   x  BCL2_IGH   x   \n",
       "24   e2b09705-c5c8-48ee-a90e-19648a7bf2cb  Lymph-BNHL   x   x  BCL2_IGH   x   \n",
       "25   e98e4e23-a139-4fff-b017-35f3f976eddd  Lymph-BNHL   x   x  BCL2_IGH   x   \n",
       "26   ebe0ed67-2d3f-45cd-8f9b-4912595b16a0  Lymph-BNHL   x   x  BCL2_IGH   x   \n",
       "27   f047cb85-8f88-45d5-83c3-0416c9f3fff7  Lymph-BNHL   x   x  BCL2_IGH   x   \n",
       "28   f04aecf0-eb12-4ab9-928e-7bcf201b0b40  Lymph-BNHL   x   x  BCL2_IGH   x   \n",
       "29   f2c08ac1-4755-4c36-a11d-338d8df9fa0d  Lymph-BNHL   x   x  BCL6_IGH   x   \n",
       "..                                    ...         ...  ..  ..       ...  ..   \n",
       "231  3e7ccab5-5b1d-4147-b907-77cab8f0837e  Lymph-BNHL   x   x         x   x   \n",
       "232  3e7ccab5-5b1d-4147-b907-77cab8f0837e  Lymph-BNHL   x   x         x   x   \n",
       "233  4e7cdeda-6dc1-4f17-b853-72a68e5aa7e1  Lymph-BNHL   x   x         x   x   \n",
       "234  4e7e6e1f-c648-446f-bdf6-0b1fcc6dfa83  Lymph-BNHL   x   x         x   x   \n",
       "235  56e57223-264e-489d-bc9a-a07789065c40  Lymph-BNHL   x   x         x   x   \n",
       "236  626c9788-d70c-449b-b9dc-5e3f6fc00991  Lymph-BNHL   x   x         x   x   \n",
       "237  626c9788-d70c-449b-b9dc-5e3f6fc00991  Lymph-BNHL   x   x         x   x   \n",
       "238  650fe009-da01-4717-89df-9c95fafe3d7e  Lymph-BNHL   x   x         x   x   \n",
       "239  687baf0b-aed9-4ab2-adb6-556a1d177a24  Lymph-BNHL   x   x         x   x   \n",
       "240  81cc0f39-6677-4f2e-9a75-d30152b188f3  Lymph-BNHL   x   x         x   x   \n",
       "241  81cc0f39-6677-4f2e-9a75-d30152b188f3  Lymph-BNHL   x   x         x   x   \n",
       "242  866ecfe7-caa6-4565-9418-6b9d6c8a3b43  Lymph-BNHL   x   x         x   x   \n",
       "243  8fc11cb6-9d79-41c3-bb45-0dadda890e8d  Lymph-BNHL   x   x         x   x   \n",
       "244  8fc11cb6-9d79-41c3-bb45-0dadda890e8d  Lymph-BNHL   x   x         x   x   \n",
       "245  8fc11cb6-9d79-41c3-bb45-0dadda890e8d  Lymph-BNHL   x   x         x   x   \n",
       "246  9c27fedd-b1b3-4af0-9e9b-20271854db08  Lymph-BNHL   x   x         x   x   \n",
       "247  ac2c8928-33a2-4aa0-8bc1-cdfa1ca0b56a  Lymph-BNHL   x   x         x   x   \n",
       "248  bd6bd940-ef06-433a-80d9-dca5683e9cd2  Lymph-BNHL   x   x         x   x   \n",
       "249  bd6bd940-ef06-433a-80d9-dca5683e9cd2  Lymph-BNHL   x   x         x   x   \n",
       "250  bdbcffb0-c7da-4e7d-8002-9feffe48533c  Lymph-BNHL   x   x         x   x   \n",
       "251  db9ce6c6-529c-4da6-92c4-f4f3fc3fe109  Lymph-BNHL   x   x         x   x   \n",
       "252  ebe0ed67-2d3f-45cd-8f9b-4912595b16a0  Lymph-BNHL   x   x         x   x   \n",
       "253  ebe0ed67-2d3f-45cd-8f9b-4912595b16a0  Lymph-BNHL   x   x         x   x   \n",
       "254  ebe0ed67-2d3f-45cd-8f9b-4912595b16a0  Lymph-BNHL   x   x         x   x   \n",
       "255  ebe0ed67-2d3f-45cd-8f9b-4912595b16a0  Lymph-BNHL   x   x         x   x   \n",
       "256  f047cb85-8f88-45d5-83c3-0416c9f3fff7  Lymph-BNHL   x   x         x   x   \n",
       "257  f04aecf0-eb12-4ab9-928e-7bcf201b0b40  Lymph-BNHL   x   x         x   x   \n",
       "258  f2c08ac1-4755-4c36-a11d-338d8df9fa0d  Lymph-BNHL   x   x         x   x   \n",
       "259  f2c08ac1-4755-4c36-a11d-338d8df9fa0d  Lymph-BNHL   x   x         x   x   \n",
       "260  ebe0ed67-2d3f-45cd-8f9b-4912595b16a0  Lymph-BNHL   x   x         x   x   \n",
       "\n",
       "         gene driver driver_statement               category top_category  \\\n",
       "0        BCL2  known            known      cis-activating_SV           SV   \n",
       "1        BCL2  known            known      cis-activating_SV           SV   \n",
       "2        BCL2  known            known      cis-activating_SV           SV   \n",
       "3        BCL2  known            known      cis-activating_SV           SV   \n",
       "4        BCL2  known            known      cis-activating_SV           SV   \n",
       "5        BCL2  known            known      cis-activating_SV           SV   \n",
       "6        BCL2  known            known      cis-activating_SV           SV   \n",
       "7        BCL6  known            known      cis-activating_SV           SV   \n",
       "8        BCL2  known            known      cis-activating_SV           SV   \n",
       "9        BCL6  known            known      cis-activating_SV           SV   \n",
       "10       BCL2  known            known      cis-activating_SV           SV   \n",
       "11       BCL2  known            known      cis-activating_SV           SV   \n",
       "12       BCL2  known            known      cis-activating_SV           SV   \n",
       "13       BCL2  known            known      cis-activating_SV           SV   \n",
       "14       BCL2  known            known      cis-activating_SV           SV   \n",
       "15       BCL2  known            known      cis-activating_SV           SV   \n",
       "16       BCL2  known            known      cis-activating_SV           SV   \n",
       "17       BCL2  known            known      cis-activating_SV           SV   \n",
       "18       BCL2  known            known      cis-activating_SV           SV   \n",
       "19       BCL2  known            known      cis-activating_SV           SV   \n",
       "20       BCL2  known            known      cis-activating_SV           SV   \n",
       "21       BCL2  known            known      cis-activating_SV           SV   \n",
       "22       BCL2  known            known      cis-activating_SV           SV   \n",
       "23       BCL2  known            known      cis-activating_SV           SV   \n",
       "24       BCL2  known            known      cis-activating_SV           SV   \n",
       "25       BCL2  known            known      cis-activating_SV           SV   \n",
       "26       BCL2  known            known      cis-activating_SV           SV   \n",
       "27       BCL2  known            known      cis-activating_SV           SV   \n",
       "28       BCL2  known            known      cis-activating_SV           SV   \n",
       "29       BCL6  known            known      cis-activating_SV           SV   \n",
       "..        ...    ...              ...                    ...          ...   \n",
       "231       B2M  known            known        coding_deletion          CNA   \n",
       "232  TNFRSF14  known            known        coding_deletion          CNA   \n",
       "233      TP53  known            known        coding_deletion          CNA   \n",
       "234    CREBBP  known            known        coding_deletion          CNA   \n",
       "235    ARID1B  known            known        coding_deletion          CNA   \n",
       "236    TMSB4X  known            known        coding_deletion          CNA   \n",
       "237      TP53  known            known        coding_deletion          CNA   \n",
       "238    CREBBP  known            known        coding_deletion          CNA   \n",
       "239  TNFRSF14  known            known        coding_deletion          CNA   \n",
       "240    CREBBP  known            known        coding_deletion          CNA   \n",
       "241  TNFRSF14  known            known        coding_deletion          CNA   \n",
       "242    CREBBP  known            known        coding_deletion          CNA   \n",
       "243    CREBBP  known            known        coding_deletion          CNA   \n",
       "244   TNFAIP3  known            known        coding_deletion          CNA   \n",
       "245      TP53  known            known        coding_deletion          CNA   \n",
       "246    ARID5B  known            known        coding_deletion          CNA   \n",
       "247    CDKN2A  known            known        coding_deletion          CNA   \n",
       "248    CDKN2A  known            known        coding_deletion          CNA   \n",
       "249    CDKN2B  known            known        coding_deletion          CNA   \n",
       "250    CREBBP  known            known        coding_deletion          CNA   \n",
       "251  TNFRSF14  known            known        coding_deletion          CNA   \n",
       "252       FAS  known            known        coding_deletion          CNA   \n",
       "253      MSR1  known            known        coding_deletion          CNA   \n",
       "254   TMEM30A  known            known        coding_deletion          CNA   \n",
       "255    TMSB4X  known            known        coding_deletion          CNA   \n",
       "256  TNFRSF14  known            known        coding_deletion          CNA   \n",
       "257    CREBBP  known            known        coding_deletion          CNA   \n",
       "258    CREBBP  known            known        coding_deletion          CNA   \n",
       "259  TNFRSF14  known            known        coding_deletion          CNA   \n",
       "260      MSR1  known            known  coding_tsg_breakpoint           SV   \n",
       "\n",
       "                 biallelic  \n",
       "0                     no\\n  \n",
       "1                     no\\n  \n",
       "2                     no\\n  \n",
       "3                     no\\n  \n",
       "4                     no\\n  \n",
       "5                     no\\n  \n",
       "6                     no\\n  \n",
       "7                     no\\n  \n",
       "8                     no\\n  \n",
       "9                     no\\n  \n",
       "10                    no\\n  \n",
       "11                    no\\n  \n",
       "12                    no\\n  \n",
       "13                    no\\n  \n",
       "14                    no\\n  \n",
       "15                    no\\n  \n",
       "16                    no\\n  \n",
       "17                    no\\n  \n",
       "18                    no\\n  \n",
       "19                    no\\n  \n",
       "20                    no\\n  \n",
       "21                    no\\n  \n",
       "22                    no\\n  \n",
       "23                    no\\n  \n",
       "24                    no\\n  \n",
       "25                    no\\n  \n",
       "26                    no\\n  \n",
       "27                    no\\n  \n",
       "28                    no\\n  \n",
       "29                    no\\n  \n",
       "..                     ...  \n",
       "231  SDeletion/SMutation\\n  \n",
       "232  SDeletion/SMutation\\n  \n",
       "233  SDeletion/SMutation\\n  \n",
       "234  SDeletion/SMutation\\n  \n",
       "235  SDeletion/SDeletion\\n  \n",
       "236  SDeletion/SMutation\\n  \n",
       "237  SDeletion/SMutation\\n  \n",
       "238  SDeletion/SMutation\\n  \n",
       "239  SDeletion/SMutation\\n  \n",
       "240  SDeletion/SMutation\\n  \n",
       "241  SDeletion/SMutation\\n  \n",
       "242  SDeletion/SMutation\\n  \n",
       "243  SDeletion/SMutation\\n  \n",
       "244  SDeletion/SDeletion\\n  \n",
       "245  SDeletion/SMutation\\n  \n",
       "246  SDeletion/SMutation\\n  \n",
       "247  SDeletion/SDeletion\\n  \n",
       "248  SDeletion/SDeletion\\n  \n",
       "249  SDeletion/SDeletion\\n  \n",
       "250  SDeletion/SMutation\\n  \n",
       "251  SDeletion/SMutation\\n  \n",
       "252  SDeletion/SDeletion\\n  \n",
       "253        SDeletion/SSV\\n  \n",
       "254  SDeletion/SMutation\\n  \n",
       "255  SDeletion/SMutation\\n  \n",
       "256  SDeletion/SMutation\\n  \n",
       "257  SDeletion/SMutation\\n  \n",
       "258  SDeletion/SMutation\\n  \n",
       "259  SDeletion/SMutation\\n  \n",
       "260        SDeletion/SSV\\n  \n",
       "\n",
       "[261 rows x 12 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_driver"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Find all Lymphoma patients and their silent mutations and other mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "df_mutbcl2 = get_dfmut('BCL2')\n",
    "df_mutpim1 = get_dfmut('PIM1')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "def get_dfmut(genename):\n",
    "    # create empty pandas dataframe \n",
    "    cols = ['type','gene','fusion','sv','tumor_exp','normal_exp']\n",
    "    df_mut = pd.DataFrame(columns = cols)\n",
    "\n",
    "    # In mutations file, find the patients mutations\n",
    "    with open(flymph,'r+b') as maf:\n",
    "        next(maf)\n",
    "        mmaf = mmap.mmap(maf.fileno(), 0)\n",
    "        for line in iter(mmaf.readline, \"\"):\n",
    "            lines_split = line.split('\\t')\n",
    "            mafg,chrm, sp,ep, strand, vc,vt = lines_split[0:7]\n",
    "            gmut = lines_split[14]\n",
    "            pt = lines_split[12]  \n",
    "\n",
    "            # Add patients to dataframe\n",
    "            if pt not in df_mut.index:\n",
    "                df_mut.loc[pt,'type'] = None\n",
    "            if pt in lfp:\n",
    "                df_mut.loc[pt,'type'] = 'Follicular'\n",
    "            else:\n",
    "                df_mut.loc[pt,'type'] = 'Not Follicular'\n",
    "\n",
    "            # Only record BCL2 mutations\n",
    "            if mafg == genename:\n",
    "                df_mut.loc[pt,'gene'] = mafg\n",
    "\n",
    "                # Add mutation column\n",
    "                if vc not in df_mut.columns:  df_mut.insert(2, vc, None)\n",
    "\n",
    "                # Append mutation to cell\n",
    "                if not pd.isnull(df_mut.loc[pt,[vc]].values):\n",
    "                    df_mut.loc[pt,[vc]] += ','+gmut\n",
    "                else:\n",
    "                    df_mut.loc[pt,[vc]] = gmut\n",
    "\n",
    "        mmaf.close()\n",
    "    return df_mut"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "eba6d269-27f8-4e50-b59a-5d639bdd5cff                                   g.chr6:37140782G>A\n",
       "1b1780d5-06da-40ee-9e15-02631a68027b                                                  NaN\n",
       "a29278af-7ecf-403e-b6a9-623ea7879d05                                   g.chr6:37138424C>T\n",
       "e89e9c69-ffcd-4a4c-818d-1dee43ddc76a                                                  NaN\n",
       "fa676301-902f-473f-8313-5bff34ae549a                                                  NaN\n",
       "07835447-5d7e-4828-80fd-89b063989c9a    g.chr18:60985573G>A,g.chr6:37138934C>T,g.chr6:...\n",
       "0e7f46ca-6f5c-4538-b6d6-00af65d57fcf                                                  NaN\n",
       "124d5b04-4588-44c3-9c00-61e064cf1bdb                g.chr6:37139209G>A,g.chr6:37139242G>A\n",
       "140d5fa9-afbe-444e-a7e7-6a4cb4ab2923                                                  NaN\n",
       "1a4633c4-72a0-4e30-8c4c-345e04337627                                   g.chr6:37138805C>T\n",
       "41dba8d1-8aad-474c-a82b-20b28ab818bd                                  g.chr18:60985834C>T\n",
       "461df2ae-fcf1-4b93-be0a-c14954fe7c42                                                  NaN\n",
       "4e7cdeda-6dc1-4f17-b853-72a68e5aa7e1    g.chr6:37138402C>T,g.chr6:37138423G>A,g.chr6:3...\n",
       "56e57223-264e-489d-bc9a-a07789065c40              g.chr18:60985549G>A,g.chr18:60985876C>T\n",
       "626c9788-d70c-449b-b9dc-5e3f6fc00991              g.chr18:60985528G>A,g.chr18:60985834C>T\n",
       "76788f17-257d-47c7-ad85-ac5f959216b5                                                  NaN\n",
       "78bd2aa8-e5f6-4b4b-96f0-d406fbdbcee8                                                  NaN\n",
       "82b8cda8-fbff-455e-b0db-7ff6528bd6c8                                                  NaN\n",
       "866ecfe7-caa6-4565-9418-6b9d6c8a3b43                                                  NaN\n",
       "8fc11cb6-9d79-41c3-bb45-0dadda890e8d    g.chr18:60985363C>T,g.chr18:60985417G>A,g.chr1...\n",
       "929f2216-6ab3-414f-a814-49b69d4b49d9    g.chr6:37138402C>T,g.chr6:37138423G>A,g.chr6:3...\n",
       "995a1ad2-faca-4a37-a59d-e62455985afb                g.chr6:37138553G>A,g.chr6:37138640C>T\n",
       "9b3e7a03-cf8c-4da8-bc36-20129d7783ba                                  g.chr18:60985876C>G\n",
       "9c27fedd-b1b3-4af0-9e9b-20271854db08                                                  NaN\n",
       "9e842227-2bc6-4185-bedb-abeb03836b51    g.chr18:60985492C>T,g.chr18:60985582G>A,g.chr1...\n",
       "abedd46c-47b8-4242-adb6-12835400cb3f                                                  NaN\n",
       "ac2c8928-33a2-4aa0-8bc1-cdfa1ca0b56a                                                  NaN\n",
       "b9dbc78e-44ce-427c-bbeb-de8d097bde2c                                   g.chr6:37139012C>T\n",
       "baa02e74-9cae-4996-858e-17d9f52d924b              g.chr18:60985384G>A,g.chr18:60985545G>A\n",
       "d733cf1a-4c42-4def-b6cb-5ef2c6a27fcd                g.chr6:37138937C>T,g.chr6:37139077G>A\n",
       "e6168db0-d12f-4218-8ab9-bd704201cb2b                                                  NaN\n",
       "e84e0649-a2e8-4873-9cb6-1aa65601ae3a                                   g.chr6:37139146G>A\n",
       "f047cb85-8f88-45d5-83c3-0416c9f3fff7    g.chr18:60985435C>T,g.chr18:60985444C>T,g.chr1...\n",
       "fce8d8c6-f2a0-43a8-9a7a-b9c519a6686c                                                  NaN\n",
       "fdf7dfb7-9285-46b3-be60-672164507705                                  g.chr18:60985714G>A\n",
       "Name: Silent, dtype: object"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_mut[(df_mut['gene'] == 'PIM1')]['Silent']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "# This shows all BCL silent mutations of patients\n",
    "df_mut_bcls = df_mut[df_mut['Silent'].notnull()]['Silent'].str.split(',', expand=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Count synonymous mutations in follicular lymphoma/ non follicular lymphoma patient dictionary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "33 synonymous mutations in 36 follicular lymphoma patients\n"
     ]
    }
   ],
   "source": [
    "fsilent = df_mut[(df_mut['gene'] == 'BCL2') & (df_mut['type'] == 'Follicular')]['Silent'].values.tolist()\n",
    "ls = []\n",
    "for v in fsilent:\n",
    "    if not pd.isnull(v):\n",
    "        mutation = re.findall('g.chr\\d+:\\d+[A-Z]+>[A-Z]+', v)\n",
    "        for k in mutation:\n",
    "            ls.append(k)\n",
    "\n",
    "print \"%d synonymous mutations in %d follicular lymphoma patients\"%(len(ls),len(lfp))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get overlapping genes of synonymous mutations in follicular lymphoma patientss"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "# This is the same code as above, but for all synonymous mutations\n",
    "fsilent = df_mut[df_mut['gene'] == 'BCL2']['Silent'].values.tolist()\n",
    "ls = []\n",
    "for v in fsilent:\n",
    "    if not pd.isnull(v):\n",
    "        mutation = re.findall('g.chr\\d+:\\d+[A-Z]+>[A-Z]+', v)\n",
    "        for k in mutation:\n",
    "            ls.append(k)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "\n",
    "# get overlapping gene from the consequence of mutation\n",
    "def get_cons(res,name):\n",
    "    lm = []\n",
    "    global dmut, mut\n",
    "    for cons in res:\n",
    "        # if it's not the target gene, append to dictionary\n",
    "        if cons['gene_symbol'] != name: \n",
    "            ovpgene = cons['gene_symbol']\n",
    "            lm.append(ovpgene)\n",
    "    if lm:\n",
    "        print \"Found overlapping gene %s for %s\"%(str(set(lm)),mut)\n",
    "    else:\n",
    "        print \"No overlapping gene found for %s\"%(mut)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18:60985876-60985876/T\n",
      "8 24 C/T\n",
      "MODIFIER [u'regulatory_region_variant'] promoter\n",
      "G\n",
      "18:60985444-60985444/T\n",
      "152 456 C/T\n",
      "MODIFIER [u'regulatory_region_variant'] promoter\n",
      "E\n",
      "18:60985471-60985471/A\n",
      "143 429 G/A\n",
      "MODIFIER [u'regulatory_region_variant'] promoter\n",
      "N\n",
      "18:60985492-60985492/T\n",
      "136 408 C/T\n",
      "MODIFIER [u'regulatory_region_variant'] promoter\n",
      "E\n",
      "18:60985573-60985573/A\n",
      "109 327 G/A\n",
      "MODIFIER [u'regulatory_region_variant'] promoter\n",
      "R\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0mTraceback (most recent call last)",
      "\u001b[0;32m<ipython-input-14-bdc684ecd4a4>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      5\u001b[0m     \u001b[0mvar\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmuts\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m     \u001b[0mmut\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mchrom\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m':'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstartp\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m'-'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstartp\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m'/'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvar\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 7\u001b[0;31m     \u001b[0mres\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mget_overlapping_cons\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmut\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      8\u001b[0m     \u001b[0mcdnapos\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mres\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'transcript_consequences'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'cdna_end'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m     \u001b[0mproteinpos\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mres\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'transcript_consequences'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'protein_end'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-4-2425f20b6540>\u001b[0m in \u001b[0;36mrateLimitedFunction\u001b[0;34m(*args, **kargs)\u001b[0m\n\u001b[1;32m     10\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mleftToWait\u001b[0m\u001b[0;34m>\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     11\u001b[0m                 \u001b[0mtime\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msleep\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mleftToWait\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 12\u001b[0;31m             \u001b[0mret\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m**\u001b[0m\u001b[0mkargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     13\u001b[0m             \u001b[0mlastTimeCalled\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtime\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mclock\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     14\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mret\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-4-2425f20b6540>\u001b[0m in \u001b[0;36mget_overlapping_cons\u001b[0;34m(muts)\u001b[0m\n\u001b[1;32m     21\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mget_overlapping_cons\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmuts\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     22\u001b[0m     \u001b[0mext\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'/vep/human/region/'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mmuts\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m'/?'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 23\u001b[0;31m     \u001b[0mr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mrequests\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mserver\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mext\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mheaders\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m{\u001b[0m\u001b[0;34m\"Content-Type\"\u001b[0m \u001b[0;34m:\u001b[0m \u001b[0;34m\"application/json\"\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     24\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     25\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mok\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/aci/sw/python/2.7.14_anaconda-5.0.1/lib/python2.7/site-packages/requests/api.pyc\u001b[0m in \u001b[0;36mget\u001b[0;34m(url, params, **kwargs)\u001b[0m\n\u001b[1;32m     70\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     71\u001b[0m     \u001b[0mkwargs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msetdefault\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'allow_redirects'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 72\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mrequest\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'get'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mparams\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mparams\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     73\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     74\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/aci/sw/python/2.7.14_anaconda-5.0.1/lib/python2.7/site-packages/requests/api.pyc\u001b[0m in \u001b[0;36mrequest\u001b[0;34m(method, url, **kwargs)\u001b[0m\n\u001b[1;32m     56\u001b[0m     \u001b[0;31m# cases, and look like a memory leak in others.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     57\u001b[0m     \u001b[0;32mwith\u001b[0m \u001b[0msessions\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSession\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0msession\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 58\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0msession\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     59\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     60\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/aci/sw/python/2.7.14_anaconda-5.0.1/lib/python2.7/site-packages/requests/sessions.pyc\u001b[0m in \u001b[0;36mrequest\u001b[0;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[1;32m    506\u001b[0m         }\n\u001b[1;32m    507\u001b[0m         \u001b[0msend_kwargs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mupdate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msettings\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 508\u001b[0;31m         \u001b[0mresp\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mprep\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0msend_kwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    509\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    510\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mresp\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/aci/sw/python/2.7.14_anaconda-5.0.1/lib/python2.7/site-packages/requests/sessions.pyc\u001b[0m in \u001b[0;36msend\u001b[0;34m(self, request, **kwargs)\u001b[0m\n\u001b[1;32m    616\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    617\u001b[0m         \u001b[0;31m# Send the request\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 618\u001b[0;31m         \u001b[0mr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0madapter\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    619\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    620\u001b[0m         \u001b[0;31m# Total elapsed time of the request (approximately)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/aci/sw/python/2.7.14_anaconda-5.0.1/lib/python2.7/site-packages/requests/adapters.pyc\u001b[0m in \u001b[0;36msend\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    438\u001b[0m                     \u001b[0mdecode_content\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mFalse\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    439\u001b[0m                     \u001b[0mretries\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmax_retries\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 440\u001b[0;31m                     \u001b[0mtimeout\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    441\u001b[0m                 )\n\u001b[1;32m    442\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/aci/sw/python/2.7.14_anaconda-5.0.1/lib/python2.7/site-packages/urllib3/connectionpool.pyc\u001b[0m in \u001b[0;36murlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001b[0m\n\u001b[1;32m    599\u001b[0m                                                   \u001b[0mtimeout\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtimeout_obj\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    600\u001b[0m                                                   \u001b[0mbody\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mbody\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mheaders\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mheaders\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 601\u001b[0;31m                                                   chunked=chunked)\n\u001b[0m\u001b[1;32m    602\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    603\u001b[0m             \u001b[0;31m# If we're going to release the connection in ``finally:``, then\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/aci/sw/python/2.7.14_anaconda-5.0.1/lib/python2.7/site-packages/urllib3/connectionpool.pyc\u001b[0m in \u001b[0;36m_make_request\u001b[0;34m(self, conn, method, url, timeout, chunked, **httplib_request_kw)\u001b[0m\n\u001b[1;32m    378\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    379\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m  \u001b[0;31m# Python 2.7, use buffering of HTTP responses\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 380\u001b[0;31m                 \u001b[0mhttplib_response\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mconn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgetresponse\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbuffering\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    381\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mTypeError\u001b[0m\u001b[0;34m:\u001b[0m  \u001b[0;31m# Python 2.6 and older, Python 3\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    382\u001b[0m                 \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/aci/sw/python/2.7.14_anaconda-5.0.1/lib/python2.7/httplib.pyc\u001b[0m in \u001b[0;36mgetresponse\u001b[0;34m(self, buffering)\u001b[0m\n\u001b[1;32m   1119\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1120\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1121\u001b[0;31m             \u001b[0mresponse\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbegin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1122\u001b[0m             \u001b[0;32massert\u001b[0m \u001b[0mresponse\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwill_close\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0m_UNKNOWN\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1123\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__state\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_CS_IDLE\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/aci/sw/python/2.7.14_anaconda-5.0.1/lib/python2.7/httplib.pyc\u001b[0m in \u001b[0;36mbegin\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    436\u001b[0m         \u001b[0;31m# read until we get a non-100 response\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    437\u001b[0m         \u001b[0;32mwhile\u001b[0m \u001b[0mTrue\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 438\u001b[0;31m             \u001b[0mversion\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstatus\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mreason\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_read_status\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    439\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mstatus\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0mCONTINUE\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    440\u001b[0m                 \u001b[0;32mbreak\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/aci/sw/python/2.7.14_anaconda-5.0.1/lib/python2.7/httplib.pyc\u001b[0m in \u001b[0;36m_read_status\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    392\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_read_status\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    393\u001b[0m         \u001b[0;31m# Initialize with Simple-Response defaults\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 394\u001b[0;31m         \u001b[0mline\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreadline\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0m_MAXLINE\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    395\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0m_MAXLINE\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    396\u001b[0m             \u001b[0;32mraise\u001b[0m \u001b[0mLineTooLong\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"header line\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/aci/sw/python/2.7.14_anaconda-5.0.1/lib/python2.7/socket.pyc\u001b[0m in \u001b[0;36mreadline\u001b[0;34m(self, size)\u001b[0m\n\u001b[1;32m    478\u001b[0m             \u001b[0;32mwhile\u001b[0m \u001b[0mTrue\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    479\u001b[0m                 \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 480\u001b[0;31m                     \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_sock\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrecv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_rbufsize\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    481\u001b[0m                 \u001b[0;32mexcept\u001b[0m \u001b[0merror\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    482\u001b[0m                     \u001b[0;32mif\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0mEINTR\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/aci/sw/python/2.7.14_anaconda-5.0.1/lib/python2.7/site-packages/urllib3/contrib/pyopenssl.pyc\u001b[0m in \u001b[0;36mrecv\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m    256\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mrecv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    257\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 258\u001b[0;31m             \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mconnection\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrecv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    259\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mOpenSSL\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSSL\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSysCallError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    260\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msuppress_ragged_eofs\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0margs\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'Unexpected EOF'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/aci/sw/python/2.7.14_anaconda-5.0.1/lib/python2.7/site-packages/OpenSSL/SSL.pyc\u001b[0m in \u001b[0;36mrecv\u001b[0;34m(self, bufsiz, flags)\u001b[0m\n\u001b[1;32m   1591\u001b[0m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_lib\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSSL_peek\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_ssl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbuf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbufsiz\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1592\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1593\u001b[0;31m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_lib\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSSL_read\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_ssl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbuf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbufsiz\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1594\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_raise_ssl_error\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_ssl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1595\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0m_ffi\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuffer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbuf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "# get the consequence of synonymous mutations in follicular lymphoma patientss\n",
    "for muts in ls:\n",
    "    chrom = re.findall(r'\\d+', muts.split(':')[0])[0]\n",
    "    startp = re.findall(r'\\d+', muts.split(':')[1])[0]\n",
    "    var = muts[-1]\n",
    "    mut = str(chrom)+':'+str(startp)+'-'+str(startp)+'/'+str(var)\n",
    "    res = get_overlapping_cons(mut) \n",
    "    cdnapos = res[0]['transcript_consequences'][0]['cdna_end']\n",
    "    proteinpos = res[0]['transcript_consequences'][0]['protein_end']\n",
    "    cdspos = res[0]['transcript_consequences'][0]['cds_end']\n",
    "    codon = res[0]['transcript_consequences'][0]['codons']\n",
    "    aa = res[0]['transcript_consequences'][0]['amino_acids']\n",
    "    allele = res[0]['allele_string']\n",
    "    reg_impact = res[0]['regulatory_feature_consequences'][0]['impact']\n",
    "    reg_term = res[0]['regulatory_feature_consequences'][0]['consequence_terms']\n",
    "    reg_type = res[0]['regulatory_feature_consequences'][0]['biotype']\n",
    "    print mut\n",
    "    print proteinpos,cdspos,allele\n",
    "    print reg_impact,reg_term,reg_type\n",
    "    print aa\n",
    "#     get_cons(res,'BCL2')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>type</th>\n",
       "      <th>gene</th>\n",
       "      <th>3'UTR</th>\n",
       "      <th>De_novo_Start_OutOfFrame</th>\n",
       "      <th>Missense_Mutation</th>\n",
       "      <th>IGR</th>\n",
       "      <th>5'Flank</th>\n",
       "      <th>De_novo_Start_InFrame</th>\n",
       "      <th>5'UTR</th>\n",
       "      <th>Silent</th>\n",
       "      <th>Intron</th>\n",
       "      <th>fusion</th>\n",
       "      <th>sv</th>\n",
       "      <th>tumor_exp</th>\n",
       "      <th>normal_exp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>04b570c2-3224-4e9b-81cc-089b4a7ff07a</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>05da3161-1444-4188-914d-19762de3a443</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>08f7812b-0d74-42ba-985b-d0a027e8a80c</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0b6cd7df-6970-4d60-b7b5-85002a7d8781</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1157a548-6770-4c89-a84c-bcf6f0e96ea8</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1193a9c4-5aab-4cd7-a690-60c96bd1172d</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132f7f2a-b902-4343-aa08-cf6a7af10b9a</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>g.chr18:60987915T&gt;C,g.chr18:60988035C&gt;G,g.chr1...</td>\n",
       "      <td>g.chr18:60986212A&gt;C</td>\n",
       "      <td>g.chr18:60986154A&gt;G,g.chr18:60986235T&gt;C,g.chr1...</td>\n",
       "      <td>g.chr18:60985876C&gt;T</td>\n",
       "      <td>g.chr18:60983860A&gt;G,g.chr18:60983904A&gt;G,g.chr1...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140d6ef7-0217-4d16-9add-215d0d14ba06</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14cbcbb1-028b-471a-bb4f-8b29cfa5e4eb</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60819992G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15563a10-47a8-401d-a507-2c7101f7ae60</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19cd4360-8392-4bc2-ae88-fdc1335d886b</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1bdfaa8a-1b7e-4251-b135-ce5e4568371b</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1de83d45-8492-4b83-9055-a9f43b823f91</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21e7e43e-f9b1-4351-a6d9-b9cd06d15f68</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22cd3875-a123-4f0d-8cd1-371958208ad6</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26c301a6-a597-4c24-9f6c-2293e5ae41c7</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>278b2498-1d64-493b-ac43-3489ec86f313</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60958715A&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2a7f6aac-8f94-4815-8b94-0ea7db71135c</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2df17728-33e8-45ab-833b-ef68a5482daa</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>367fdef3-d7af-4996-932f-cc5087e843dc</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39a5d94d-e8c8-4057-be96-362ffbafb94d</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3de261ea-6478-4f96-8ca1-30be130e945f</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3e8a2c90-e747-4a22-bc9e-0b062479fec2</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3ee54a3c-69ba-479a-b1fe-79917e1b6dd5</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3f72f750-5666-44e7-acaf-a912d89475be</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4808f9e1-452d-45eb-9b87-2f87c67d3452</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60907169A&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4b7a704b-e516-4275-be11-45cc935237e2</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4b8c6b0c-0d0c-4834-a2f3-691e0919b902</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4b91ece6-c9b2-4889-b18c-c63eb58eb061</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>b9dbc78e-44ce-427c-bbeb-de8d097bde2c</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60813158C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>baa02e74-9cae-4996-858e-17d9f52d924b</th>\n",
       "      <td>Follicular</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60985727T&gt;G,g.chr18:60985760C&gt;T,g.chr1...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60988134C&gt;G,g.chr18:60988165C&gt;T,g.chr1...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60985900C&gt;G,g.chr18:60985910G&gt;C,g.chr1...</td>\n",
       "      <td>g.chr18:60985384G&gt;A,g.chr18:60985545G&gt;A</td>\n",
       "      <td>g.chr18:60805688_60805689AA&gt;TG,g.chr18:6080574...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bc1d5327-2e76-4e0e-b749-72a559469d0d</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60882379G&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bd6bd940-ef06-433a-80d9-dca5683e9cd2</th>\n",
       "      <td>Follicular</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60985813C&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60987879C&gt;T,g.chr18:60988085C&gt;T,g.chr1...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60985932G&gt;A,g.chr18:60986057T&gt;G,g.chr1...</td>\n",
       "      <td>g.chr18:60985738C&gt;T</td>\n",
       "      <td>g.chr18:60984901C&gt;T,g.chr18:60984926G&gt;T,g.chr1...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>bdbcffb0-c7da-4e7d-8002-9feffe48533c</th>\n",
       "      <td>Follicular</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60988066C&gt;G,g.chr18:60988229A&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60986148G&gt;C,g.chr18:60986652C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60984884C&gt;G,g.chr18:60985172G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>c66926a8-474d-482c-bfb7-043029c8b737</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cc7bba1d-8990-4557-a565-9cff1085f023</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ce85ccf3-6621-4976-b187-2f28d45dd4d9</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>d31cce9d-1b38-44fd-9378-0d128e12a011</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60878461T&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>d6ee68e5-ceb3-4a63-b3bd-e64fe5398102</th>\n",
       "      <td>Follicular</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60988053A&gt;T,g.chr18:60988291C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60986254G&gt;A,g.chr18:60986378T&gt;C,g.chr1...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60985015C&gt;G,g.chr18:60985034G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>d733cf1a-4c42-4def-b6cb-5ef2c6a27fcd</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60958355G&gt;A,g.chr18:60982905A&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>db9ce6c6-529c-4da6-92c4-f4f3fc3fe109</th>\n",
       "      <td>Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dbae3c4e-6d66-483a-a611-7285067bf85c</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60983090_60983091insTA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dc4bc4c4-2cc1-4a2e-a9f2-613088af17bf</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60962249T&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dd8f3e1c-6ed6-41c2-957d-6f679a3cb3ca</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>df814571-57f1-4e55-bc1a-c892c568c4ba</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>e2b09705-c5c8-48ee-a90e-19648a7bf2cb</th>\n",
       "      <td>Follicular</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60987916T&gt;C,g.chr18:60988032T&gt;C,g.chr1...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60986022T&gt;A,g.chr18:60986046C&gt;G,g.chr1...</td>\n",
       "      <td>g.chr18:60985615C&gt;T</td>\n",
       "      <td>g.chr18:60806367A&gt;G,g.chr18:60983695T&gt;C,g.chr1...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>e2fa7251-507e-4d76-95a3-a228adc3885a</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>e6168db0-d12f-4218-8ab9-bd704201cb2b</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>e84e0649-a2e8-4873-9cb6-1aa65601ae3a</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>e98e4e23-a139-4fff-b017-35f3f976eddd</th>\n",
       "      <td>Follicular</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60986107A&gt;T</td>\n",
       "      <td>g.chr18:60985726A&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60987989C&gt;G,g.chr18:60987993C&gt;T,g.chr1...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60985947G&gt;A,g.chr18:60986070T&gt;G,g.chr1...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60984962A&gt;G,g.chr18:60985015C&gt;G,g.chr1...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ebe0ed67-2d3f-45cd-8f9b-4912595b16a0</th>\n",
       "      <td>Follicular</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60985411G&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60987900G&gt;A,g.chr18:60988035C&gt;T,g.chr1...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60985917C&gt;T,g.chr18:60986018T&gt;G,g.chr1...</td>\n",
       "      <td>g.chr18:60985834C&gt;T</td>\n",
       "      <td>g.chr18:60805516G&gt;A,g.chr18:60984920T&gt;C,g.chr1...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>f047cb85-8f88-45d5-83c3-0416c9f3fff7</th>\n",
       "      <td>Follicular</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60987926G&gt;A,g.chr18:60988031C&gt;T,g.chr1...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60986010_60986011GC&gt;AA,g.chr18:6098613...</td>\n",
       "      <td>g.chr18:60985435C&gt;T,g.chr18:60985444C&gt;T,g.chr1...</td>\n",
       "      <td>g.chr18:60823025C&gt;G,g.chr18:60873794A&gt;C,g.chr1...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>f04aecf0-eb12-4ab9-928e-7bcf201b0b40</th>\n",
       "      <td>Follicular</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60986153T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60988121C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60986169C&gt;G</td>\n",
       "      <td>g.chr18:60985735G&gt;A</td>\n",
       "      <td>g.chr18:60984967G&gt;C,g.chr18:60985042T&gt;A,g.chr1...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>f2c08ac1-4755-4c36-a11d-338d8df9fa0d</th>\n",
       "      <td>Follicular</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60987880A&gt;C,g.chr18:60987913_60987914C...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60985900_60985901CC&gt;TT,g.chr18:6098590...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>f60007b3-0603-4ad1-8b47-d2d4c4c621e4</th>\n",
       "      <td>Follicular</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60985861C&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60987893G&gt;C,g.chr18:60987975C&gt;G,g.chr1...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60985916G&gt;C,g.chr18:60985981_60985982A...</td>\n",
       "      <td>g.chr18:60985444C&gt;T</td>\n",
       "      <td>g.chr18:60967792A&gt;T,g.chr18:60983996C&gt;A,g.chr1...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>f9837a56-7244-4846-a63d-266e6a92f168</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>f9dc999f-6dde-448d-9cf1-2897ddcf7b0b</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60989272G&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60825242G&gt;C,g.chr18:60828405T&gt;A,g.chr1...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fce8d8c6-f2a0-43a8-9a7a-b9c519a6686c</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>fdf7dfb7-9285-46b3-be60-672164507705</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60985725G&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60987912_60987913CC&gt;AA,g.chr18:6098799...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>g.chr18:60985993G&gt;C,g.chr18:60986159T&gt;A,g.chr1...</td>\n",
       "      <td>g.chr18:60985714G&gt;A</td>\n",
       "      <td>g.chr18:60805496C&gt;T,g.chr18:60805509C&gt;T,g.chr1...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>197 rows Ã— 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                type  gene 3'UTR  \\\n",
       "Tumor_Sample_Barcode                  Not Follicular   NaN  None   \n",
       "04b570c2-3224-4e9b-81cc-089b4a7ff07a  Not Follicular   NaN  None   \n",
       "05da3161-1444-4188-914d-19762de3a443  Not Follicular   NaN  None   \n",
       "08f7812b-0d74-42ba-985b-d0a027e8a80c  Not Follicular   NaN  None   \n",
       "0b6cd7df-6970-4d60-b7b5-85002a7d8781  Not Follicular   NaN  None   \n",
       "1157a548-6770-4c89-a84c-bcf6f0e96ea8  Not Follicular   NaN  None   \n",
       "1193a9c4-5aab-4cd7-a690-60c96bd1172d  Not Follicular   NaN  None   \n",
       "132f7f2a-b902-4343-aa08-cf6a7af10b9a  Not Follicular  BCL2  None   \n",
       "140d6ef7-0217-4d16-9add-215d0d14ba06  Not Follicular   NaN  None   \n",
       "14cbcbb1-028b-471a-bb4f-8b29cfa5e4eb  Not Follicular  BCL2  None   \n",
       "15563a10-47a8-401d-a507-2c7101f7ae60  Not Follicular   NaN  None   \n",
       "19cd4360-8392-4bc2-ae88-fdc1335d886b  Not Follicular   NaN  None   \n",
       "1bdfaa8a-1b7e-4251-b135-ce5e4568371b  Not Follicular   NaN  None   \n",
       "1de83d45-8492-4b83-9055-a9f43b823f91  Not Follicular   NaN  None   \n",
       "21e7e43e-f9b1-4351-a6d9-b9cd06d15f68  Not Follicular   NaN  None   \n",
       "22cd3875-a123-4f0d-8cd1-371958208ad6  Not Follicular   NaN  None   \n",
       "26c301a6-a597-4c24-9f6c-2293e5ae41c7  Not Follicular   NaN  None   \n",
       "278b2498-1d64-493b-ac43-3489ec86f313  Not Follicular  BCL2  None   \n",
       "2a7f6aac-8f94-4815-8b94-0ea7db71135c  Not Follicular   NaN  None   \n",
       "2df17728-33e8-45ab-833b-ef68a5482daa  Not Follicular   NaN  None   \n",
       "367fdef3-d7af-4996-932f-cc5087e843dc  Not Follicular   NaN  None   \n",
       "39a5d94d-e8c8-4057-be96-362ffbafb94d  Not Follicular   NaN  None   \n",
       "3de261ea-6478-4f96-8ca1-30be130e945f  Not Follicular   NaN  None   \n",
       "3e8a2c90-e747-4a22-bc9e-0b062479fec2  Not Follicular   NaN  None   \n",
       "3ee54a3c-69ba-479a-b1fe-79917e1b6dd5  Not Follicular   NaN  None   \n",
       "3f72f750-5666-44e7-acaf-a912d89475be  Not Follicular   NaN  None   \n",
       "4808f9e1-452d-45eb-9b87-2f87c67d3452  Not Follicular  BCL2  None   \n",
       "4b7a704b-e516-4275-be11-45cc935237e2  Not Follicular   NaN  None   \n",
       "4b8c6b0c-0d0c-4834-a2f3-691e0919b902  Not Follicular   NaN  None   \n",
       "4b91ece6-c9b2-4889-b18c-c63eb58eb061  Not Follicular   NaN  None   \n",
       "...                                              ...   ...   ...   \n",
       "b9dbc78e-44ce-427c-bbeb-de8d097bde2c  Not Follicular  BCL2   NaN   \n",
       "baa02e74-9cae-4996-858e-17d9f52d924b      Follicular  BCL2   NaN   \n",
       "bc1d5327-2e76-4e0e-b749-72a559469d0d  Not Follicular  BCL2   NaN   \n",
       "bd6bd940-ef06-433a-80d9-dca5683e9cd2      Follicular  BCL2   NaN   \n",
       "bdbcffb0-c7da-4e7d-8002-9feffe48533c      Follicular  BCL2   NaN   \n",
       "c66926a8-474d-482c-bfb7-043029c8b737  Not Follicular   NaN   NaN   \n",
       "cc7bba1d-8990-4557-a565-9cff1085f023  Not Follicular   NaN   NaN   \n",
       "ce85ccf3-6621-4976-b187-2f28d45dd4d9  Not Follicular   NaN   NaN   \n",
       "d31cce9d-1b38-44fd-9378-0d128e12a011  Not Follicular  BCL2   NaN   \n",
       "d6ee68e5-ceb3-4a63-b3bd-e64fe5398102      Follicular  BCL2   NaN   \n",
       "d733cf1a-4c42-4def-b6cb-5ef2c6a27fcd  Not Follicular  BCL2   NaN   \n",
       "db9ce6c6-529c-4da6-92c4-f4f3fc3fe109      Follicular   NaN   NaN   \n",
       "dbae3c4e-6d66-483a-a611-7285067bf85c  Not Follicular  BCL2   NaN   \n",
       "dc4bc4c4-2cc1-4a2e-a9f2-613088af17bf  Not Follicular  BCL2   NaN   \n",
       "dd8f3e1c-6ed6-41c2-957d-6f679a3cb3ca  Not Follicular   NaN   NaN   \n",
       "df814571-57f1-4e55-bc1a-c892c568c4ba  Not Follicular   NaN   NaN   \n",
       "e2b09705-c5c8-48ee-a90e-19648a7bf2cb      Follicular  BCL2   NaN   \n",
       "e2fa7251-507e-4d76-95a3-a228adc3885a  Not Follicular   NaN   NaN   \n",
       "e6168db0-d12f-4218-8ab9-bd704201cb2b  Not Follicular   NaN   NaN   \n",
       "e84e0649-a2e8-4873-9cb6-1aa65601ae3a  Not Follicular   NaN   NaN   \n",
       "e98e4e23-a139-4fff-b017-35f3f976eddd      Follicular  BCL2   NaN   \n",
       "ebe0ed67-2d3f-45cd-8f9b-4912595b16a0      Follicular  BCL2   NaN   \n",
       "f047cb85-8f88-45d5-83c3-0416c9f3fff7      Follicular  BCL2   NaN   \n",
       "f04aecf0-eb12-4ab9-928e-7bcf201b0b40      Follicular  BCL2   NaN   \n",
       "f2c08ac1-4755-4c36-a11d-338d8df9fa0d      Follicular  BCL2   NaN   \n",
       "f60007b3-0603-4ad1-8b47-d2d4c4c621e4      Follicular  BCL2   NaN   \n",
       "f9837a56-7244-4846-a63d-266e6a92f168  Not Follicular   NaN   NaN   \n",
       "f9dc999f-6dde-448d-9cf1-2897ddcf7b0b  Not Follicular  BCL2   NaN   \n",
       "fce8d8c6-f2a0-43a8-9a7a-b9c519a6686c  Not Follicular   NaN   NaN   \n",
       "fdf7dfb7-9285-46b3-be60-672164507705  Not Follicular  BCL2   NaN   \n",
       "\n",
       "                                     De_novo_Start_OutOfFrame  \\\n",
       "Tumor_Sample_Barcode                                     None   \n",
       "04b570c2-3224-4e9b-81cc-089b4a7ff07a                     None   \n",
       "05da3161-1444-4188-914d-19762de3a443                     None   \n",
       "08f7812b-0d74-42ba-985b-d0a027e8a80c                     None   \n",
       "0b6cd7df-6970-4d60-b7b5-85002a7d8781                     None   \n",
       "1157a548-6770-4c89-a84c-bcf6f0e96ea8                     None   \n",
       "1193a9c4-5aab-4cd7-a690-60c96bd1172d                     None   \n",
       "132f7f2a-b902-4343-aa08-cf6a7af10b9a                     None   \n",
       "140d6ef7-0217-4d16-9add-215d0d14ba06                     None   \n",
       "14cbcbb1-028b-471a-bb4f-8b29cfa5e4eb                     None   \n",
       "15563a10-47a8-401d-a507-2c7101f7ae60                     None   \n",
       "19cd4360-8392-4bc2-ae88-fdc1335d886b                     None   \n",
       "1bdfaa8a-1b7e-4251-b135-ce5e4568371b                     None   \n",
       "1de83d45-8492-4b83-9055-a9f43b823f91                     None   \n",
       "21e7e43e-f9b1-4351-a6d9-b9cd06d15f68                     None   \n",
       "22cd3875-a123-4f0d-8cd1-371958208ad6                     None   \n",
       "26c301a6-a597-4c24-9f6c-2293e5ae41c7                     None   \n",
       "278b2498-1d64-493b-ac43-3489ec86f313                     None   \n",
       "2a7f6aac-8f94-4815-8b94-0ea7db71135c                     None   \n",
       "2df17728-33e8-45ab-833b-ef68a5482daa                     None   \n",
       "367fdef3-d7af-4996-932f-cc5087e843dc                     None   \n",
       "39a5d94d-e8c8-4057-be96-362ffbafb94d                     None   \n",
       "3de261ea-6478-4f96-8ca1-30be130e945f                     None   \n",
       "3e8a2c90-e747-4a22-bc9e-0b062479fec2                     None   \n",
       "3ee54a3c-69ba-479a-b1fe-79917e1b6dd5                     None   \n",
       "3f72f750-5666-44e7-acaf-a912d89475be                     None   \n",
       "4808f9e1-452d-45eb-9b87-2f87c67d3452                     None   \n",
       "4b7a704b-e516-4275-be11-45cc935237e2                     None   \n",
       "4b8c6b0c-0d0c-4834-a2f3-691e0919b902                     None   \n",
       "4b91ece6-c9b2-4889-b18c-c63eb58eb061                     None   \n",
       "...                                                       ...   \n",
       "b9dbc78e-44ce-427c-bbeb-de8d097bde2c                      NaN   \n",
       "baa02e74-9cae-4996-858e-17d9f52d924b                      NaN   \n",
       "bc1d5327-2e76-4e0e-b749-72a559469d0d                      NaN   \n",
       "bd6bd940-ef06-433a-80d9-dca5683e9cd2                      NaN   \n",
       "bdbcffb0-c7da-4e7d-8002-9feffe48533c                      NaN   \n",
       "c66926a8-474d-482c-bfb7-043029c8b737                      NaN   \n",
       "cc7bba1d-8990-4557-a565-9cff1085f023                      NaN   \n",
       "ce85ccf3-6621-4976-b187-2f28d45dd4d9                      NaN   \n",
       "d31cce9d-1b38-44fd-9378-0d128e12a011                      NaN   \n",
       "d6ee68e5-ceb3-4a63-b3bd-e64fe5398102                      NaN   \n",
       "d733cf1a-4c42-4def-b6cb-5ef2c6a27fcd                      NaN   \n",
       "db9ce6c6-529c-4da6-92c4-f4f3fc3fe109                      NaN   \n",
       "dbae3c4e-6d66-483a-a611-7285067bf85c                      NaN   \n",
       "dc4bc4c4-2cc1-4a2e-a9f2-613088af17bf                      NaN   \n",
       "dd8f3e1c-6ed6-41c2-957d-6f679a3cb3ca                      NaN   \n",
       "df814571-57f1-4e55-bc1a-c892c568c4ba                      NaN   \n",
       "e2b09705-c5c8-48ee-a90e-19648a7bf2cb                      NaN   \n",
       "e2fa7251-507e-4d76-95a3-a228adc3885a                      NaN   \n",
       "e6168db0-d12f-4218-8ab9-bd704201cb2b                      NaN   \n",
       "e84e0649-a2e8-4873-9cb6-1aa65601ae3a                      NaN   \n",
       "e98e4e23-a139-4fff-b017-35f3f976eddd      g.chr18:60986107A>T   \n",
       "ebe0ed67-2d3f-45cd-8f9b-4912595b16a0                      NaN   \n",
       "f047cb85-8f88-45d5-83c3-0416c9f3fff7                      NaN   \n",
       "f04aecf0-eb12-4ab9-928e-7bcf201b0b40      g.chr18:60986153T>C   \n",
       "f2c08ac1-4755-4c36-a11d-338d8df9fa0d                      NaN   \n",
       "f60007b3-0603-4ad1-8b47-d2d4c4c621e4                      NaN   \n",
       "f9837a56-7244-4846-a63d-266e6a92f168                      NaN   \n",
       "f9dc999f-6dde-448d-9cf1-2897ddcf7b0b                      NaN   \n",
       "fce8d8c6-f2a0-43a8-9a7a-b9c519a6686c                      NaN   \n",
       "fdf7dfb7-9285-46b3-be60-672164507705                      NaN   \n",
       "\n",
       "                                                                      Missense_Mutation  \\\n",
       "Tumor_Sample_Barcode                                                               None   \n",
       "04b570c2-3224-4e9b-81cc-089b4a7ff07a                                               None   \n",
       "05da3161-1444-4188-914d-19762de3a443                                               None   \n",
       "08f7812b-0d74-42ba-985b-d0a027e8a80c                                               None   \n",
       "0b6cd7df-6970-4d60-b7b5-85002a7d8781                                               None   \n",
       "1157a548-6770-4c89-a84c-bcf6f0e96ea8                                               None   \n",
       "1193a9c4-5aab-4cd7-a690-60c96bd1172d                                               None   \n",
       "132f7f2a-b902-4343-aa08-cf6a7af10b9a                                               None   \n",
       "140d6ef7-0217-4d16-9add-215d0d14ba06                                               None   \n",
       "14cbcbb1-028b-471a-bb4f-8b29cfa5e4eb                                               None   \n",
       "15563a10-47a8-401d-a507-2c7101f7ae60                                               None   \n",
       "19cd4360-8392-4bc2-ae88-fdc1335d886b                                               None   \n",
       "1bdfaa8a-1b7e-4251-b135-ce5e4568371b                                               None   \n",
       "1de83d45-8492-4b83-9055-a9f43b823f91                                               None   \n",
       "21e7e43e-f9b1-4351-a6d9-b9cd06d15f68                                               None   \n",
       "22cd3875-a123-4f0d-8cd1-371958208ad6                                               None   \n",
       "26c301a6-a597-4c24-9f6c-2293e5ae41c7                                               None   \n",
       "278b2498-1d64-493b-ac43-3489ec86f313                                               None   \n",
       "2a7f6aac-8f94-4815-8b94-0ea7db71135c                                               None   \n",
       "2df17728-33e8-45ab-833b-ef68a5482daa                                               None   \n",
       "367fdef3-d7af-4996-932f-cc5087e843dc                                               None   \n",
       "39a5d94d-e8c8-4057-be96-362ffbafb94d                                               None   \n",
       "3de261ea-6478-4f96-8ca1-30be130e945f                                               None   \n",
       "3e8a2c90-e747-4a22-bc9e-0b062479fec2                                               None   \n",
       "3ee54a3c-69ba-479a-b1fe-79917e1b6dd5                                               None   \n",
       "3f72f750-5666-44e7-acaf-a912d89475be                                               None   \n",
       "4808f9e1-452d-45eb-9b87-2f87c67d3452                                               None   \n",
       "4b7a704b-e516-4275-be11-45cc935237e2                                               None   \n",
       "4b8c6b0c-0d0c-4834-a2f3-691e0919b902                                               None   \n",
       "4b91ece6-c9b2-4889-b18c-c63eb58eb061                                               None   \n",
       "...                                                                                 ...   \n",
       "b9dbc78e-44ce-427c-bbeb-de8d097bde2c                                                NaN   \n",
       "baa02e74-9cae-4996-858e-17d9f52d924b  g.chr18:60985727T>G,g.chr18:60985760C>T,g.chr1...   \n",
       "bc1d5327-2e76-4e0e-b749-72a559469d0d                                                NaN   \n",
       "bd6bd940-ef06-433a-80d9-dca5683e9cd2                                g.chr18:60985813C>A   \n",
       "bdbcffb0-c7da-4e7d-8002-9feffe48533c                                                NaN   \n",
       "c66926a8-474d-482c-bfb7-043029c8b737                                                NaN   \n",
       "cc7bba1d-8990-4557-a565-9cff1085f023                                                NaN   \n",
       "ce85ccf3-6621-4976-b187-2f28d45dd4d9                                                NaN   \n",
       "d31cce9d-1b38-44fd-9378-0d128e12a011                                                NaN   \n",
       "d6ee68e5-ceb3-4a63-b3bd-e64fe5398102                                                NaN   \n",
       "d733cf1a-4c42-4def-b6cb-5ef2c6a27fcd                                                NaN   \n",
       "db9ce6c6-529c-4da6-92c4-f4f3fc3fe109                                                NaN   \n",
       "dbae3c4e-6d66-483a-a611-7285067bf85c                                                NaN   \n",
       "dc4bc4c4-2cc1-4a2e-a9f2-613088af17bf                                                NaN   \n",
       "dd8f3e1c-6ed6-41c2-957d-6f679a3cb3ca                                                NaN   \n",
       "df814571-57f1-4e55-bc1a-c892c568c4ba                                                NaN   \n",
       "e2b09705-c5c8-48ee-a90e-19648a7bf2cb                                                NaN   \n",
       "e2fa7251-507e-4d76-95a3-a228adc3885a                                                NaN   \n",
       "e6168db0-d12f-4218-8ab9-bd704201cb2b                                                NaN   \n",
       "e84e0649-a2e8-4873-9cb6-1aa65601ae3a                                                NaN   \n",
       "e98e4e23-a139-4fff-b017-35f3f976eddd                                g.chr18:60985726A>C   \n",
       "ebe0ed67-2d3f-45cd-8f9b-4912595b16a0                                g.chr18:60985411G>C   \n",
       "f047cb85-8f88-45d5-83c3-0416c9f3fff7                                                NaN   \n",
       "f04aecf0-eb12-4ab9-928e-7bcf201b0b40                                                NaN   \n",
       "f2c08ac1-4755-4c36-a11d-338d8df9fa0d                                                NaN   \n",
       "f60007b3-0603-4ad1-8b47-d2d4c4c621e4                                g.chr18:60985861C>G   \n",
       "f9837a56-7244-4846-a63d-266e6a92f168                                                NaN   \n",
       "f9dc999f-6dde-448d-9cf1-2897ddcf7b0b                                                NaN   \n",
       "fce8d8c6-f2a0-43a8-9a7a-b9c519a6686c                                                NaN   \n",
       "fdf7dfb7-9285-46b3-be60-672164507705                                g.chr18:60985725G>C   \n",
       "\n",
       "                                       IGR  \\\n",
       "Tumor_Sample_Barcode                  None   \n",
       "04b570c2-3224-4e9b-81cc-089b4a7ff07a  None   \n",
       "05da3161-1444-4188-914d-19762de3a443  None   \n",
       "08f7812b-0d74-42ba-985b-d0a027e8a80c  None   \n",
       "0b6cd7df-6970-4d60-b7b5-85002a7d8781  None   \n",
       "1157a548-6770-4c89-a84c-bcf6f0e96ea8  None   \n",
       "1193a9c4-5aab-4cd7-a690-60c96bd1172d  None   \n",
       "132f7f2a-b902-4343-aa08-cf6a7af10b9a  None   \n",
       "140d6ef7-0217-4d16-9add-215d0d14ba06  None   \n",
       "14cbcbb1-028b-471a-bb4f-8b29cfa5e4eb  None   \n",
       "15563a10-47a8-401d-a507-2c7101f7ae60  None   \n",
       "19cd4360-8392-4bc2-ae88-fdc1335d886b  None   \n",
       "1bdfaa8a-1b7e-4251-b135-ce5e4568371b  None   \n",
       "1de83d45-8492-4b83-9055-a9f43b823f91  None   \n",
       "21e7e43e-f9b1-4351-a6d9-b9cd06d15f68  None   \n",
       "22cd3875-a123-4f0d-8cd1-371958208ad6  None   \n",
       "26c301a6-a597-4c24-9f6c-2293e5ae41c7  None   \n",
       "278b2498-1d64-493b-ac43-3489ec86f313  None   \n",
       "2a7f6aac-8f94-4815-8b94-0ea7db71135c  None   \n",
       "2df17728-33e8-45ab-833b-ef68a5482daa  None   \n",
       "367fdef3-d7af-4996-932f-cc5087e843dc  None   \n",
       "39a5d94d-e8c8-4057-be96-362ffbafb94d  None   \n",
       "3de261ea-6478-4f96-8ca1-30be130e945f  None   \n",
       "3e8a2c90-e747-4a22-bc9e-0b062479fec2  None   \n",
       "3ee54a3c-69ba-479a-b1fe-79917e1b6dd5  None   \n",
       "3f72f750-5666-44e7-acaf-a912d89475be  None   \n",
       "4808f9e1-452d-45eb-9b87-2f87c67d3452  None   \n",
       "4b7a704b-e516-4275-be11-45cc935237e2  None   \n",
       "4b8c6b0c-0d0c-4834-a2f3-691e0919b902  None   \n",
       "4b91ece6-c9b2-4889-b18c-c63eb58eb061  None   \n",
       "...                                    ...   \n",
       "b9dbc78e-44ce-427c-bbeb-de8d097bde2c   NaN   \n",
       "baa02e74-9cae-4996-858e-17d9f52d924b   NaN   \n",
       "bc1d5327-2e76-4e0e-b749-72a559469d0d   NaN   \n",
       "bd6bd940-ef06-433a-80d9-dca5683e9cd2   NaN   \n",
       "bdbcffb0-c7da-4e7d-8002-9feffe48533c   NaN   \n",
       "c66926a8-474d-482c-bfb7-043029c8b737   NaN   \n",
       "cc7bba1d-8990-4557-a565-9cff1085f023   NaN   \n",
       "ce85ccf3-6621-4976-b187-2f28d45dd4d9   NaN   \n",
       "d31cce9d-1b38-44fd-9378-0d128e12a011   NaN   \n",
       "d6ee68e5-ceb3-4a63-b3bd-e64fe5398102   NaN   \n",
       "d733cf1a-4c42-4def-b6cb-5ef2c6a27fcd   NaN   \n",
       "db9ce6c6-529c-4da6-92c4-f4f3fc3fe109   NaN   \n",
       "dbae3c4e-6d66-483a-a611-7285067bf85c   NaN   \n",
       "dc4bc4c4-2cc1-4a2e-a9f2-613088af17bf   NaN   \n",
       "dd8f3e1c-6ed6-41c2-957d-6f679a3cb3ca   NaN   \n",
       "df814571-57f1-4e55-bc1a-c892c568c4ba   NaN   \n",
       "e2b09705-c5c8-48ee-a90e-19648a7bf2cb   NaN   \n",
       "e2fa7251-507e-4d76-95a3-a228adc3885a   NaN   \n",
       "e6168db0-d12f-4218-8ab9-bd704201cb2b   NaN   \n",
       "e84e0649-a2e8-4873-9cb6-1aa65601ae3a   NaN   \n",
       "e98e4e23-a139-4fff-b017-35f3f976eddd   NaN   \n",
       "ebe0ed67-2d3f-45cd-8f9b-4912595b16a0   NaN   \n",
       "f047cb85-8f88-45d5-83c3-0416c9f3fff7   NaN   \n",
       "f04aecf0-eb12-4ab9-928e-7bcf201b0b40   NaN   \n",
       "f2c08ac1-4755-4c36-a11d-338d8df9fa0d   NaN   \n",
       "f60007b3-0603-4ad1-8b47-d2d4c4c621e4   NaN   \n",
       "f9837a56-7244-4846-a63d-266e6a92f168   NaN   \n",
       "f9dc999f-6dde-448d-9cf1-2897ddcf7b0b   NaN   \n",
       "fce8d8c6-f2a0-43a8-9a7a-b9c519a6686c   NaN   \n",
       "fdf7dfb7-9285-46b3-be60-672164507705   NaN   \n",
       "\n",
       "                                                                                5'Flank  \\\n",
       "Tumor_Sample_Barcode                                                               None   \n",
       "04b570c2-3224-4e9b-81cc-089b4a7ff07a                                               None   \n",
       "05da3161-1444-4188-914d-19762de3a443                                               None   \n",
       "08f7812b-0d74-42ba-985b-d0a027e8a80c                                               None   \n",
       "0b6cd7df-6970-4d60-b7b5-85002a7d8781                                               None   \n",
       "1157a548-6770-4c89-a84c-bcf6f0e96ea8                                               None   \n",
       "1193a9c4-5aab-4cd7-a690-60c96bd1172d                                               None   \n",
       "132f7f2a-b902-4343-aa08-cf6a7af10b9a  g.chr18:60987915T>C,g.chr18:60988035C>G,g.chr1...   \n",
       "140d6ef7-0217-4d16-9add-215d0d14ba06                                                NaN   \n",
       "14cbcbb1-028b-471a-bb4f-8b29cfa5e4eb                                                NaN   \n",
       "15563a10-47a8-401d-a507-2c7101f7ae60                                                NaN   \n",
       "19cd4360-8392-4bc2-ae88-fdc1335d886b                                                NaN   \n",
       "1bdfaa8a-1b7e-4251-b135-ce5e4568371b                                                NaN   \n",
       "1de83d45-8492-4b83-9055-a9f43b823f91                                                NaN   \n",
       "21e7e43e-f9b1-4351-a6d9-b9cd06d15f68                                                NaN   \n",
       "22cd3875-a123-4f0d-8cd1-371958208ad6                                                NaN   \n",
       "26c301a6-a597-4c24-9f6c-2293e5ae41c7                                                NaN   \n",
       "278b2498-1d64-493b-ac43-3489ec86f313                                                NaN   \n",
       "2a7f6aac-8f94-4815-8b94-0ea7db71135c                                                NaN   \n",
       "2df17728-33e8-45ab-833b-ef68a5482daa                                                NaN   \n",
       "367fdef3-d7af-4996-932f-cc5087e843dc                                                NaN   \n",
       "39a5d94d-e8c8-4057-be96-362ffbafb94d                                                NaN   \n",
       "3de261ea-6478-4f96-8ca1-30be130e945f                                                NaN   \n",
       "3e8a2c90-e747-4a22-bc9e-0b062479fec2                                                NaN   \n",
       "3ee54a3c-69ba-479a-b1fe-79917e1b6dd5                                                NaN   \n",
       "3f72f750-5666-44e7-acaf-a912d89475be                                                NaN   \n",
       "4808f9e1-452d-45eb-9b87-2f87c67d3452                                                NaN   \n",
       "4b7a704b-e516-4275-be11-45cc935237e2                                                NaN   \n",
       "4b8c6b0c-0d0c-4834-a2f3-691e0919b902                                                NaN   \n",
       "4b91ece6-c9b2-4889-b18c-c63eb58eb061                                                NaN   \n",
       "...                                                                                 ...   \n",
       "b9dbc78e-44ce-427c-bbeb-de8d097bde2c                                                NaN   \n",
       "baa02e74-9cae-4996-858e-17d9f52d924b  g.chr18:60988134C>G,g.chr18:60988165C>T,g.chr1...   \n",
       "bc1d5327-2e76-4e0e-b749-72a559469d0d                                                NaN   \n",
       "bd6bd940-ef06-433a-80d9-dca5683e9cd2  g.chr18:60987879C>T,g.chr18:60988085C>T,g.chr1...   \n",
       "bdbcffb0-c7da-4e7d-8002-9feffe48533c            g.chr18:60988066C>G,g.chr18:60988229A>T   \n",
       "c66926a8-474d-482c-bfb7-043029c8b737                                                NaN   \n",
       "cc7bba1d-8990-4557-a565-9cff1085f023                                                NaN   \n",
       "ce85ccf3-6621-4976-b187-2f28d45dd4d9                                                NaN   \n",
       "d31cce9d-1b38-44fd-9378-0d128e12a011                                                NaN   \n",
       "d6ee68e5-ceb3-4a63-b3bd-e64fe5398102            g.chr18:60988053A>T,g.chr18:60988291C>T   \n",
       "d733cf1a-4c42-4def-b6cb-5ef2c6a27fcd                                                NaN   \n",
       "db9ce6c6-529c-4da6-92c4-f4f3fc3fe109                                                NaN   \n",
       "dbae3c4e-6d66-483a-a611-7285067bf85c                                                NaN   \n",
       "dc4bc4c4-2cc1-4a2e-a9f2-613088af17bf                                                NaN   \n",
       "dd8f3e1c-6ed6-41c2-957d-6f679a3cb3ca                                                NaN   \n",
       "df814571-57f1-4e55-bc1a-c892c568c4ba                                                NaN   \n",
       "e2b09705-c5c8-48ee-a90e-19648a7bf2cb  g.chr18:60987916T>C,g.chr18:60988032T>C,g.chr1...   \n",
       "e2fa7251-507e-4d76-95a3-a228adc3885a                                                NaN   \n",
       "e6168db0-d12f-4218-8ab9-bd704201cb2b                                                NaN   \n",
       "e84e0649-a2e8-4873-9cb6-1aa65601ae3a                                                NaN   \n",
       "e98e4e23-a139-4fff-b017-35f3f976eddd  g.chr18:60987989C>G,g.chr18:60987993C>T,g.chr1...   \n",
       "ebe0ed67-2d3f-45cd-8f9b-4912595b16a0  g.chr18:60987900G>A,g.chr18:60988035C>T,g.chr1...   \n",
       "f047cb85-8f88-45d5-83c3-0416c9f3fff7  g.chr18:60987926G>A,g.chr18:60988031C>T,g.chr1...   \n",
       "f04aecf0-eb12-4ab9-928e-7bcf201b0b40                                g.chr18:60988121C>T   \n",
       "f2c08ac1-4755-4c36-a11d-338d8df9fa0d  g.chr18:60987880A>C,g.chr18:60987913_60987914C...   \n",
       "f60007b3-0603-4ad1-8b47-d2d4c4c621e4  g.chr18:60987893G>C,g.chr18:60987975C>G,g.chr1...   \n",
       "f9837a56-7244-4846-a63d-266e6a92f168                                                NaN   \n",
       "f9dc999f-6dde-448d-9cf1-2897ddcf7b0b                                g.chr18:60989272G>C   \n",
       "fce8d8c6-f2a0-43a8-9a7a-b9c519a6686c                                                NaN   \n",
       "fdf7dfb7-9285-46b3-be60-672164507705  g.chr18:60987912_60987913CC>AA,g.chr18:6098799...   \n",
       "\n",
       "                                     De_novo_Start_InFrame  \\\n",
       "Tumor_Sample_Barcode                                  None   \n",
       "04b570c2-3224-4e9b-81cc-089b4a7ff07a                  None   \n",
       "05da3161-1444-4188-914d-19762de3a443                  None   \n",
       "08f7812b-0d74-42ba-985b-d0a027e8a80c                  None   \n",
       "0b6cd7df-6970-4d60-b7b5-85002a7d8781                  None   \n",
       "1157a548-6770-4c89-a84c-bcf6f0e96ea8                  None   \n",
       "1193a9c4-5aab-4cd7-a690-60c96bd1172d                  None   \n",
       "132f7f2a-b902-4343-aa08-cf6a7af10b9a   g.chr18:60986212A>C   \n",
       "140d6ef7-0217-4d16-9add-215d0d14ba06                   NaN   \n",
       "14cbcbb1-028b-471a-bb4f-8b29cfa5e4eb                   NaN   \n",
       "15563a10-47a8-401d-a507-2c7101f7ae60                   NaN   \n",
       "19cd4360-8392-4bc2-ae88-fdc1335d886b                   NaN   \n",
       "1bdfaa8a-1b7e-4251-b135-ce5e4568371b                   NaN   \n",
       "1de83d45-8492-4b83-9055-a9f43b823f91                   NaN   \n",
       "21e7e43e-f9b1-4351-a6d9-b9cd06d15f68                   NaN   \n",
       "22cd3875-a123-4f0d-8cd1-371958208ad6                   NaN   \n",
       "26c301a6-a597-4c24-9f6c-2293e5ae41c7                   NaN   \n",
       "278b2498-1d64-493b-ac43-3489ec86f313                   NaN   \n",
       "2a7f6aac-8f94-4815-8b94-0ea7db71135c                   NaN   \n",
       "2df17728-33e8-45ab-833b-ef68a5482daa                   NaN   \n",
       "367fdef3-d7af-4996-932f-cc5087e843dc                   NaN   \n",
       "39a5d94d-e8c8-4057-be96-362ffbafb94d                   NaN   \n",
       "3de261ea-6478-4f96-8ca1-30be130e945f                   NaN   \n",
       "3e8a2c90-e747-4a22-bc9e-0b062479fec2                   NaN   \n",
       "3ee54a3c-69ba-479a-b1fe-79917e1b6dd5                   NaN   \n",
       "3f72f750-5666-44e7-acaf-a912d89475be                   NaN   \n",
       "4808f9e1-452d-45eb-9b87-2f87c67d3452                   NaN   \n",
       "4b7a704b-e516-4275-be11-45cc935237e2                   NaN   \n",
       "4b8c6b0c-0d0c-4834-a2f3-691e0919b902                   NaN   \n",
       "4b91ece6-c9b2-4889-b18c-c63eb58eb061                   NaN   \n",
       "...                                                    ...   \n",
       "b9dbc78e-44ce-427c-bbeb-de8d097bde2c                   NaN   \n",
       "baa02e74-9cae-4996-858e-17d9f52d924b                   NaN   \n",
       "bc1d5327-2e76-4e0e-b749-72a559469d0d                   NaN   \n",
       "bd6bd940-ef06-433a-80d9-dca5683e9cd2                   NaN   \n",
       "bdbcffb0-c7da-4e7d-8002-9feffe48533c                   NaN   \n",
       "c66926a8-474d-482c-bfb7-043029c8b737                   NaN   \n",
       "cc7bba1d-8990-4557-a565-9cff1085f023                   NaN   \n",
       "ce85ccf3-6621-4976-b187-2f28d45dd4d9                   NaN   \n",
       "d31cce9d-1b38-44fd-9378-0d128e12a011                   NaN   \n",
       "d6ee68e5-ceb3-4a63-b3bd-e64fe5398102                   NaN   \n",
       "d733cf1a-4c42-4def-b6cb-5ef2c6a27fcd                   NaN   \n",
       "db9ce6c6-529c-4da6-92c4-f4f3fc3fe109                   NaN   \n",
       "dbae3c4e-6d66-483a-a611-7285067bf85c                   NaN   \n",
       "dc4bc4c4-2cc1-4a2e-a9f2-613088af17bf                   NaN   \n",
       "dd8f3e1c-6ed6-41c2-957d-6f679a3cb3ca                   NaN   \n",
       "df814571-57f1-4e55-bc1a-c892c568c4ba                   NaN   \n",
       "e2b09705-c5c8-48ee-a90e-19648a7bf2cb                   NaN   \n",
       "e2fa7251-507e-4d76-95a3-a228adc3885a                   NaN   \n",
       "e6168db0-d12f-4218-8ab9-bd704201cb2b                   NaN   \n",
       "e84e0649-a2e8-4873-9cb6-1aa65601ae3a                   NaN   \n",
       "e98e4e23-a139-4fff-b017-35f3f976eddd                   NaN   \n",
       "ebe0ed67-2d3f-45cd-8f9b-4912595b16a0                   NaN   \n",
       "f047cb85-8f88-45d5-83c3-0416c9f3fff7                   NaN   \n",
       "f04aecf0-eb12-4ab9-928e-7bcf201b0b40                   NaN   \n",
       "f2c08ac1-4755-4c36-a11d-338d8df9fa0d                   NaN   \n",
       "f60007b3-0603-4ad1-8b47-d2d4c4c621e4                   NaN   \n",
       "f9837a56-7244-4846-a63d-266e6a92f168                   NaN   \n",
       "f9dc999f-6dde-448d-9cf1-2897ddcf7b0b                   NaN   \n",
       "fce8d8c6-f2a0-43a8-9a7a-b9c519a6686c                   NaN   \n",
       "fdf7dfb7-9285-46b3-be60-672164507705                   NaN   \n",
       "\n",
       "                                                                                  5'UTR  \\\n",
       "Tumor_Sample_Barcode                                                               None   \n",
       "04b570c2-3224-4e9b-81cc-089b4a7ff07a                                               None   \n",
       "05da3161-1444-4188-914d-19762de3a443                                               None   \n",
       "08f7812b-0d74-42ba-985b-d0a027e8a80c                                               None   \n",
       "0b6cd7df-6970-4d60-b7b5-85002a7d8781                                               None   \n",
       "1157a548-6770-4c89-a84c-bcf6f0e96ea8                                               None   \n",
       "1193a9c4-5aab-4cd7-a690-60c96bd1172d                                               None   \n",
       "132f7f2a-b902-4343-aa08-cf6a7af10b9a  g.chr18:60986154A>G,g.chr18:60986235T>C,g.chr1...   \n",
       "140d6ef7-0217-4d16-9add-215d0d14ba06                                                NaN   \n",
       "14cbcbb1-028b-471a-bb4f-8b29cfa5e4eb                                                NaN   \n",
       "15563a10-47a8-401d-a507-2c7101f7ae60                                                NaN   \n",
       "19cd4360-8392-4bc2-ae88-fdc1335d886b                                                NaN   \n",
       "1bdfaa8a-1b7e-4251-b135-ce5e4568371b                                                NaN   \n",
       "1de83d45-8492-4b83-9055-a9f43b823f91                                                NaN   \n",
       "21e7e43e-f9b1-4351-a6d9-b9cd06d15f68                                                NaN   \n",
       "22cd3875-a123-4f0d-8cd1-371958208ad6                                                NaN   \n",
       "26c301a6-a597-4c24-9f6c-2293e5ae41c7                                                NaN   \n",
       "278b2498-1d64-493b-ac43-3489ec86f313                                                NaN   \n",
       "2a7f6aac-8f94-4815-8b94-0ea7db71135c                                                NaN   \n",
       "2df17728-33e8-45ab-833b-ef68a5482daa                                                NaN   \n",
       "367fdef3-d7af-4996-932f-cc5087e843dc                                                NaN   \n",
       "39a5d94d-e8c8-4057-be96-362ffbafb94d                                                NaN   \n",
       "3de261ea-6478-4f96-8ca1-30be130e945f                                                NaN   \n",
       "3e8a2c90-e747-4a22-bc9e-0b062479fec2                                                NaN   \n",
       "3ee54a3c-69ba-479a-b1fe-79917e1b6dd5                                                NaN   \n",
       "3f72f750-5666-44e7-acaf-a912d89475be                                                NaN   \n",
       "4808f9e1-452d-45eb-9b87-2f87c67d3452                                                NaN   \n",
       "4b7a704b-e516-4275-be11-45cc935237e2                                                NaN   \n",
       "4b8c6b0c-0d0c-4834-a2f3-691e0919b902                                                NaN   \n",
       "4b91ece6-c9b2-4889-b18c-c63eb58eb061                                                NaN   \n",
       "...                                                                                 ...   \n",
       "b9dbc78e-44ce-427c-bbeb-de8d097bde2c                                                NaN   \n",
       "baa02e74-9cae-4996-858e-17d9f52d924b  g.chr18:60985900C>G,g.chr18:60985910G>C,g.chr1...   \n",
       "bc1d5327-2e76-4e0e-b749-72a559469d0d                                                NaN   \n",
       "bd6bd940-ef06-433a-80d9-dca5683e9cd2  g.chr18:60985932G>A,g.chr18:60986057T>G,g.chr1...   \n",
       "bdbcffb0-c7da-4e7d-8002-9feffe48533c            g.chr18:60986148G>C,g.chr18:60986652C>T   \n",
       "c66926a8-474d-482c-bfb7-043029c8b737                                                NaN   \n",
       "cc7bba1d-8990-4557-a565-9cff1085f023                                                NaN   \n",
       "ce85ccf3-6621-4976-b187-2f28d45dd4d9                                                NaN   \n",
       "d31cce9d-1b38-44fd-9378-0d128e12a011                                                NaN   \n",
       "d6ee68e5-ceb3-4a63-b3bd-e64fe5398102  g.chr18:60986254G>A,g.chr18:60986378T>C,g.chr1...   \n",
       "d733cf1a-4c42-4def-b6cb-5ef2c6a27fcd                                                NaN   \n",
       "db9ce6c6-529c-4da6-92c4-f4f3fc3fe109                                                NaN   \n",
       "dbae3c4e-6d66-483a-a611-7285067bf85c                                                NaN   \n",
       "dc4bc4c4-2cc1-4a2e-a9f2-613088af17bf                                                NaN   \n",
       "dd8f3e1c-6ed6-41c2-957d-6f679a3cb3ca                                                NaN   \n",
       "df814571-57f1-4e55-bc1a-c892c568c4ba                                                NaN   \n",
       "e2b09705-c5c8-48ee-a90e-19648a7bf2cb  g.chr18:60986022T>A,g.chr18:60986046C>G,g.chr1...   \n",
       "e2fa7251-507e-4d76-95a3-a228adc3885a                                                NaN   \n",
       "e6168db0-d12f-4218-8ab9-bd704201cb2b                                                NaN   \n",
       "e84e0649-a2e8-4873-9cb6-1aa65601ae3a                                                NaN   \n",
       "e98e4e23-a139-4fff-b017-35f3f976eddd  g.chr18:60985947G>A,g.chr18:60986070T>G,g.chr1...   \n",
       "ebe0ed67-2d3f-45cd-8f9b-4912595b16a0  g.chr18:60985917C>T,g.chr18:60986018T>G,g.chr1...   \n",
       "f047cb85-8f88-45d5-83c3-0416c9f3fff7  g.chr18:60986010_60986011GC>AA,g.chr18:6098613...   \n",
       "f04aecf0-eb12-4ab9-928e-7bcf201b0b40                                g.chr18:60986169C>G   \n",
       "f2c08ac1-4755-4c36-a11d-338d8df9fa0d  g.chr18:60985900_60985901CC>TT,g.chr18:6098590...   \n",
       "f60007b3-0603-4ad1-8b47-d2d4c4c621e4  g.chr18:60985916G>C,g.chr18:60985981_60985982A...   \n",
       "f9837a56-7244-4846-a63d-266e6a92f168                                                NaN   \n",
       "f9dc999f-6dde-448d-9cf1-2897ddcf7b0b                                                NaN   \n",
       "fce8d8c6-f2a0-43a8-9a7a-b9c519a6686c                                                NaN   \n",
       "fdf7dfb7-9285-46b3-be60-672164507705  g.chr18:60985993G>C,g.chr18:60986159T>A,g.chr1...   \n",
       "\n",
       "                                                                                 Silent  \\\n",
       "Tumor_Sample_Barcode                                                               None   \n",
       "04b570c2-3224-4e9b-81cc-089b4a7ff07a                                               None   \n",
       "05da3161-1444-4188-914d-19762de3a443                                               None   \n",
       "08f7812b-0d74-42ba-985b-d0a027e8a80c                                               None   \n",
       "0b6cd7df-6970-4d60-b7b5-85002a7d8781                                               None   \n",
       "1157a548-6770-4c89-a84c-bcf6f0e96ea8                                               None   \n",
       "1193a9c4-5aab-4cd7-a690-60c96bd1172d                                               None   \n",
       "132f7f2a-b902-4343-aa08-cf6a7af10b9a                                g.chr18:60985876C>T   \n",
       "140d6ef7-0217-4d16-9add-215d0d14ba06                                                NaN   \n",
       "14cbcbb1-028b-471a-bb4f-8b29cfa5e4eb                                                NaN   \n",
       "15563a10-47a8-401d-a507-2c7101f7ae60                                                NaN   \n",
       "19cd4360-8392-4bc2-ae88-fdc1335d886b                                                NaN   \n",
       "1bdfaa8a-1b7e-4251-b135-ce5e4568371b                                                NaN   \n",
       "1de83d45-8492-4b83-9055-a9f43b823f91                                                NaN   \n",
       "21e7e43e-f9b1-4351-a6d9-b9cd06d15f68                                                NaN   \n",
       "22cd3875-a123-4f0d-8cd1-371958208ad6                                                NaN   \n",
       "26c301a6-a597-4c24-9f6c-2293e5ae41c7                                                NaN   \n",
       "278b2498-1d64-493b-ac43-3489ec86f313                                                NaN   \n",
       "2a7f6aac-8f94-4815-8b94-0ea7db71135c                                                NaN   \n",
       "2df17728-33e8-45ab-833b-ef68a5482daa                                                NaN   \n",
       "367fdef3-d7af-4996-932f-cc5087e843dc                                                NaN   \n",
       "39a5d94d-e8c8-4057-be96-362ffbafb94d                                                NaN   \n",
       "3de261ea-6478-4f96-8ca1-30be130e945f                                                NaN   \n",
       "3e8a2c90-e747-4a22-bc9e-0b062479fec2                                                NaN   \n",
       "3ee54a3c-69ba-479a-b1fe-79917e1b6dd5                                                NaN   \n",
       "3f72f750-5666-44e7-acaf-a912d89475be                                                NaN   \n",
       "4808f9e1-452d-45eb-9b87-2f87c67d3452                                                NaN   \n",
       "4b7a704b-e516-4275-be11-45cc935237e2                                                NaN   \n",
       "4b8c6b0c-0d0c-4834-a2f3-691e0919b902                                                NaN   \n",
       "4b91ece6-c9b2-4889-b18c-c63eb58eb061                                                NaN   \n",
       "...                                                                                 ...   \n",
       "b9dbc78e-44ce-427c-bbeb-de8d097bde2c                                                NaN   \n",
       "baa02e74-9cae-4996-858e-17d9f52d924b            g.chr18:60985384G>A,g.chr18:60985545G>A   \n",
       "bc1d5327-2e76-4e0e-b749-72a559469d0d                                                NaN   \n",
       "bd6bd940-ef06-433a-80d9-dca5683e9cd2                                g.chr18:60985738C>T   \n",
       "bdbcffb0-c7da-4e7d-8002-9feffe48533c                                                NaN   \n",
       "c66926a8-474d-482c-bfb7-043029c8b737                                                NaN   \n",
       "cc7bba1d-8990-4557-a565-9cff1085f023                                                NaN   \n",
       "ce85ccf3-6621-4976-b187-2f28d45dd4d9                                                NaN   \n",
       "d31cce9d-1b38-44fd-9378-0d128e12a011                                                NaN   \n",
       "d6ee68e5-ceb3-4a63-b3bd-e64fe5398102                                                NaN   \n",
       "d733cf1a-4c42-4def-b6cb-5ef2c6a27fcd                                                NaN   \n",
       "db9ce6c6-529c-4da6-92c4-f4f3fc3fe109                                                NaN   \n",
       "dbae3c4e-6d66-483a-a611-7285067bf85c                                                NaN   \n",
       "dc4bc4c4-2cc1-4a2e-a9f2-613088af17bf                                                NaN   \n",
       "dd8f3e1c-6ed6-41c2-957d-6f679a3cb3ca                                                NaN   \n",
       "df814571-57f1-4e55-bc1a-c892c568c4ba                                                NaN   \n",
       "e2b09705-c5c8-48ee-a90e-19648a7bf2cb                                g.chr18:60985615C>T   \n",
       "e2fa7251-507e-4d76-95a3-a228adc3885a                                                NaN   \n",
       "e6168db0-d12f-4218-8ab9-bd704201cb2b                                                NaN   \n",
       "e84e0649-a2e8-4873-9cb6-1aa65601ae3a                                                NaN   \n",
       "e98e4e23-a139-4fff-b017-35f3f976eddd                                                NaN   \n",
       "ebe0ed67-2d3f-45cd-8f9b-4912595b16a0                                g.chr18:60985834C>T   \n",
       "f047cb85-8f88-45d5-83c3-0416c9f3fff7  g.chr18:60985435C>T,g.chr18:60985444C>T,g.chr1...   \n",
       "f04aecf0-eb12-4ab9-928e-7bcf201b0b40                                g.chr18:60985735G>A   \n",
       "f2c08ac1-4755-4c36-a11d-338d8df9fa0d                                                NaN   \n",
       "f60007b3-0603-4ad1-8b47-d2d4c4c621e4                                g.chr18:60985444C>T   \n",
       "f9837a56-7244-4846-a63d-266e6a92f168                                                NaN   \n",
       "f9dc999f-6dde-448d-9cf1-2897ddcf7b0b                                                NaN   \n",
       "fce8d8c6-f2a0-43a8-9a7a-b9c519a6686c                                                NaN   \n",
       "fdf7dfb7-9285-46b3-be60-672164507705                                g.chr18:60985714G>A   \n",
       "\n",
       "                                                                                 Intron  \\\n",
       "Tumor_Sample_Barcode                                                               None   \n",
       "04b570c2-3224-4e9b-81cc-089b4a7ff07a                                               None   \n",
       "05da3161-1444-4188-914d-19762de3a443                                               None   \n",
       "08f7812b-0d74-42ba-985b-d0a027e8a80c                                               None   \n",
       "0b6cd7df-6970-4d60-b7b5-85002a7d8781                                               None   \n",
       "1157a548-6770-4c89-a84c-bcf6f0e96ea8                                               None   \n",
       "1193a9c4-5aab-4cd7-a690-60c96bd1172d                                               None   \n",
       "132f7f2a-b902-4343-aa08-cf6a7af10b9a  g.chr18:60983860A>G,g.chr18:60983904A>G,g.chr1...   \n",
       "140d6ef7-0217-4d16-9add-215d0d14ba06                                                NaN   \n",
       "14cbcbb1-028b-471a-bb4f-8b29cfa5e4eb                                g.chr18:60819992G>A   \n",
       "15563a10-47a8-401d-a507-2c7101f7ae60                                                NaN   \n",
       "19cd4360-8392-4bc2-ae88-fdc1335d886b                                                NaN   \n",
       "1bdfaa8a-1b7e-4251-b135-ce5e4568371b                                                NaN   \n",
       "1de83d45-8492-4b83-9055-a9f43b823f91                                                NaN   \n",
       "21e7e43e-f9b1-4351-a6d9-b9cd06d15f68                                                NaN   \n",
       "22cd3875-a123-4f0d-8cd1-371958208ad6                                                NaN   \n",
       "26c301a6-a597-4c24-9f6c-2293e5ae41c7                                                NaN   \n",
       "278b2498-1d64-493b-ac43-3489ec86f313                                g.chr18:60958715A>T   \n",
       "2a7f6aac-8f94-4815-8b94-0ea7db71135c                                                NaN   \n",
       "2df17728-33e8-45ab-833b-ef68a5482daa                                                NaN   \n",
       "367fdef3-d7af-4996-932f-cc5087e843dc                                                NaN   \n",
       "39a5d94d-e8c8-4057-be96-362ffbafb94d                                                NaN   \n",
       "3de261ea-6478-4f96-8ca1-30be130e945f                                                NaN   \n",
       "3e8a2c90-e747-4a22-bc9e-0b062479fec2                                                NaN   \n",
       "3ee54a3c-69ba-479a-b1fe-79917e1b6dd5                                                NaN   \n",
       "3f72f750-5666-44e7-acaf-a912d89475be                                                NaN   \n",
       "4808f9e1-452d-45eb-9b87-2f87c67d3452                                g.chr18:60907169A>C   \n",
       "4b7a704b-e516-4275-be11-45cc935237e2                                                NaN   \n",
       "4b8c6b0c-0d0c-4834-a2f3-691e0919b902                                                NaN   \n",
       "4b91ece6-c9b2-4889-b18c-c63eb58eb061                                                NaN   \n",
       "...                                                                                 ...   \n",
       "b9dbc78e-44ce-427c-bbeb-de8d097bde2c                                g.chr18:60813158C>T   \n",
       "baa02e74-9cae-4996-858e-17d9f52d924b  g.chr18:60805688_60805689AA>TG,g.chr18:6080574...   \n",
       "bc1d5327-2e76-4e0e-b749-72a559469d0d                                g.chr18:60882379G>C   \n",
       "bd6bd940-ef06-433a-80d9-dca5683e9cd2  g.chr18:60984901C>T,g.chr18:60984926G>T,g.chr1...   \n",
       "bdbcffb0-c7da-4e7d-8002-9feffe48533c            g.chr18:60984884C>G,g.chr18:60985172G>A   \n",
       "c66926a8-474d-482c-bfb7-043029c8b737                                                NaN   \n",
       "cc7bba1d-8990-4557-a565-9cff1085f023                                                NaN   \n",
       "ce85ccf3-6621-4976-b187-2f28d45dd4d9                                                NaN   \n",
       "d31cce9d-1b38-44fd-9378-0d128e12a011                                g.chr18:60878461T>A   \n",
       "d6ee68e5-ceb3-4a63-b3bd-e64fe5398102            g.chr18:60985015C>G,g.chr18:60985034G>A   \n",
       "d733cf1a-4c42-4def-b6cb-5ef2c6a27fcd            g.chr18:60958355G>A,g.chr18:60982905A>G   \n",
       "db9ce6c6-529c-4da6-92c4-f4f3fc3fe109                                                NaN   \n",
       "dbae3c4e-6d66-483a-a611-7285067bf85c                     g.chr18:60983090_60983091insTA   \n",
       "dc4bc4c4-2cc1-4a2e-a9f2-613088af17bf                                g.chr18:60962249T>A   \n",
       "dd8f3e1c-6ed6-41c2-957d-6f679a3cb3ca                                                NaN   \n",
       "df814571-57f1-4e55-bc1a-c892c568c4ba                                                NaN   \n",
       "e2b09705-c5c8-48ee-a90e-19648a7bf2cb  g.chr18:60806367A>G,g.chr18:60983695T>C,g.chr1...   \n",
       "e2fa7251-507e-4d76-95a3-a228adc3885a                                                NaN   \n",
       "e6168db0-d12f-4218-8ab9-bd704201cb2b                                                NaN   \n",
       "e84e0649-a2e8-4873-9cb6-1aa65601ae3a                                                NaN   \n",
       "e98e4e23-a139-4fff-b017-35f3f976eddd  g.chr18:60984962A>G,g.chr18:60985015C>G,g.chr1...   \n",
       "ebe0ed67-2d3f-45cd-8f9b-4912595b16a0  g.chr18:60805516G>A,g.chr18:60984920T>C,g.chr1...   \n",
       "f047cb85-8f88-45d5-83c3-0416c9f3fff7  g.chr18:60823025C>G,g.chr18:60873794A>C,g.chr1...   \n",
       "f04aecf0-eb12-4ab9-928e-7bcf201b0b40  g.chr18:60984967G>C,g.chr18:60985042T>A,g.chr1...   \n",
       "f2c08ac1-4755-4c36-a11d-338d8df9fa0d                                                NaN   \n",
       "f60007b3-0603-4ad1-8b47-d2d4c4c621e4  g.chr18:60967792A>T,g.chr18:60983996C>A,g.chr1...   \n",
       "f9837a56-7244-4846-a63d-266e6a92f168                                                NaN   \n",
       "f9dc999f-6dde-448d-9cf1-2897ddcf7b0b  g.chr18:60825242G>C,g.chr18:60828405T>A,g.chr1...   \n",
       "fce8d8c6-f2a0-43a8-9a7a-b9c519a6686c                                                NaN   \n",
       "fdf7dfb7-9285-46b3-be60-672164507705  g.chr18:60805496C>T,g.chr18:60805509C>T,g.chr1...   \n",
       "\n",
       "                                     fusion   sv tumor_exp normal_exp  \n",
       "Tumor_Sample_Barcode                    NaN  NaN       NaN        NaN  \n",
       "04b570c2-3224-4e9b-81cc-089b4a7ff07a    NaN  NaN       NaN        NaN  \n",
       "05da3161-1444-4188-914d-19762de3a443    NaN  NaN       NaN        NaN  \n",
       "08f7812b-0d74-42ba-985b-d0a027e8a80c    NaN  NaN       NaN        NaN  \n",
       "0b6cd7df-6970-4d60-b7b5-85002a7d8781    NaN  NaN       NaN        NaN  \n",
       "1157a548-6770-4c89-a84c-bcf6f0e96ea8    NaN  NaN       NaN        NaN  \n",
       "1193a9c4-5aab-4cd7-a690-60c96bd1172d    NaN  NaN       NaN        NaN  \n",
       "132f7f2a-b902-4343-aa08-cf6a7af10b9a    NaN  NaN       NaN        NaN  \n",
       "140d6ef7-0217-4d16-9add-215d0d14ba06    NaN  NaN       NaN        NaN  \n",
       "14cbcbb1-028b-471a-bb4f-8b29cfa5e4eb    NaN  NaN       NaN        NaN  \n",
       "15563a10-47a8-401d-a507-2c7101f7ae60    NaN  NaN       NaN        NaN  \n",
       "19cd4360-8392-4bc2-ae88-fdc1335d886b    NaN  NaN       NaN        NaN  \n",
       "1bdfaa8a-1b7e-4251-b135-ce5e4568371b    NaN  NaN       NaN        NaN  \n",
       "1de83d45-8492-4b83-9055-a9f43b823f91    NaN  NaN       NaN        NaN  \n",
       "21e7e43e-f9b1-4351-a6d9-b9cd06d15f68    NaN  NaN       NaN        NaN  \n",
       "22cd3875-a123-4f0d-8cd1-371958208ad6    NaN  NaN       NaN        NaN  \n",
       "26c301a6-a597-4c24-9f6c-2293e5ae41c7    NaN  NaN       NaN        NaN  \n",
       "278b2498-1d64-493b-ac43-3489ec86f313    NaN  NaN       NaN        NaN  \n",
       "2a7f6aac-8f94-4815-8b94-0ea7db71135c    NaN  NaN       NaN        NaN  \n",
       "2df17728-33e8-45ab-833b-ef68a5482daa    NaN  NaN       NaN        NaN  \n",
       "367fdef3-d7af-4996-932f-cc5087e843dc    NaN  NaN       NaN        NaN  \n",
       "39a5d94d-e8c8-4057-be96-362ffbafb94d    NaN  NaN       NaN        NaN  \n",
       "3de261ea-6478-4f96-8ca1-30be130e945f    NaN  NaN       NaN        NaN  \n",
       "3e8a2c90-e747-4a22-bc9e-0b062479fec2    NaN  NaN       NaN        NaN  \n",
       "3ee54a3c-69ba-479a-b1fe-79917e1b6dd5    NaN  NaN       NaN        NaN  \n",
       "3f72f750-5666-44e7-acaf-a912d89475be    NaN  NaN       NaN        NaN  \n",
       "4808f9e1-452d-45eb-9b87-2f87c67d3452    NaN  NaN       NaN        NaN  \n",
       "4b7a704b-e516-4275-be11-45cc935237e2    NaN  NaN       NaN        NaN  \n",
       "4b8c6b0c-0d0c-4834-a2f3-691e0919b902    NaN  NaN       NaN        NaN  \n",
       "4b91ece6-c9b2-4889-b18c-c63eb58eb061    NaN  NaN       NaN        NaN  \n",
       "...                                     ...  ...       ...        ...  \n",
       "b9dbc78e-44ce-427c-bbeb-de8d097bde2c    NaN  NaN       NaN        NaN  \n",
       "baa02e74-9cae-4996-858e-17d9f52d924b    NaN  NaN       NaN        NaN  \n",
       "bc1d5327-2e76-4e0e-b749-72a559469d0d    NaN  NaN       NaN        NaN  \n",
       "bd6bd940-ef06-433a-80d9-dca5683e9cd2    NaN  NaN       NaN        NaN  \n",
       "bdbcffb0-c7da-4e7d-8002-9feffe48533c    NaN  NaN       NaN        NaN  \n",
       "c66926a8-474d-482c-bfb7-043029c8b737    NaN  NaN       NaN        NaN  \n",
       "cc7bba1d-8990-4557-a565-9cff1085f023    NaN  NaN       NaN        NaN  \n",
       "ce85ccf3-6621-4976-b187-2f28d45dd4d9    NaN  NaN       NaN        NaN  \n",
       "d31cce9d-1b38-44fd-9378-0d128e12a011    NaN  NaN       NaN        NaN  \n",
       "d6ee68e5-ceb3-4a63-b3bd-e64fe5398102    NaN  NaN       NaN        NaN  \n",
       "d733cf1a-4c42-4def-b6cb-5ef2c6a27fcd    NaN  NaN       NaN        NaN  \n",
       "db9ce6c6-529c-4da6-92c4-f4f3fc3fe109    NaN  NaN       NaN        NaN  \n",
       "dbae3c4e-6d66-483a-a611-7285067bf85c    NaN  NaN       NaN        NaN  \n",
       "dc4bc4c4-2cc1-4a2e-a9f2-613088af17bf    NaN  NaN       NaN        NaN  \n",
       "dd8f3e1c-6ed6-41c2-957d-6f679a3cb3ca    NaN  NaN       NaN        NaN  \n",
       "df814571-57f1-4e55-bc1a-c892c568c4ba    NaN  NaN       NaN        NaN  \n",
       "e2b09705-c5c8-48ee-a90e-19648a7bf2cb    NaN  NaN       NaN        NaN  \n",
       "e2fa7251-507e-4d76-95a3-a228adc3885a    NaN  NaN       NaN        NaN  \n",
       "e6168db0-d12f-4218-8ab9-bd704201cb2b    NaN  NaN       NaN        NaN  \n",
       "e84e0649-a2e8-4873-9cb6-1aa65601ae3a    NaN  NaN       NaN        NaN  \n",
       "e98e4e23-a139-4fff-b017-35f3f976eddd    NaN  NaN       NaN        NaN  \n",
       "ebe0ed67-2d3f-45cd-8f9b-4912595b16a0    NaN  NaN       NaN        NaN  \n",
       "f047cb85-8f88-45d5-83c3-0416c9f3fff7    NaN  NaN       NaN        NaN  \n",
       "f04aecf0-eb12-4ab9-928e-7bcf201b0b40    NaN  NaN       NaN        NaN  \n",
       "f2c08ac1-4755-4c36-a11d-338d8df9fa0d    NaN  NaN       NaN        NaN  \n",
       "f60007b3-0603-4ad1-8b47-d2d4c4c621e4    NaN  NaN       NaN        NaN  \n",
       "f9837a56-7244-4846-a63d-266e6a92f168    NaN  NaN       NaN        NaN  \n",
       "f9dc999f-6dde-448d-9cf1-2897ddcf7b0b    NaN  NaN       NaN        NaN  \n",
       "fce8d8c6-f2a0-43a8-9a7a-b9c519a6686c    NaN  NaN       NaN        NaN  \n",
       "fdf7dfb7-9285-46b3-be60-672164507705    NaN  NaN       NaN        NaN  \n",
       "\n",
       "[197 rows x 15 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_mut"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "g.chr18:60793942A>C 3'UTR\n",
      "g.chr18:60793510T>C 3'UTR\n",
      "g.chr18:60793704G>A 3'UTR\n",
      "g.chr18:60794375G>A 3'UTR\n",
      "g.chr18:60794650G>A 3'UTR\n",
      "g.chr18:60794752T>C 3'UTR\n",
      "g.chr18:60794785A>G 3'UTR\n",
      "g.chr18:60794861C>T 3'UTR\n",
      "g.chr18:60794942C>G 3'UTR\n",
      "g.chr18:60795751C>G 3'UTR\n",
      "g.chr18:60986924G>C De_novo_Start_OutOfFrame\n",
      "g.chr18:60986160A>C De_novo_Start_OutOfFrame\n",
      "g.chr18:60986716G>T De_novo_Start_OutOfFrame\n",
      "g.chr18:60986100T>C De_novo_Start_OutOfFrame\n",
      "g.chr18:60986272C>T De_novo_Start_OutOfFrame\n",
      "g.chr18:60986100T>C De_novo_Start_OutOfFrame\n",
      "g.chr18:60987124C>A De_novo_Start_OutOfFrame\n",
      "g.chr18:60986107A>T De_novo_Start_OutOfFrame\n",
      "g.chr18:60986153T>C De_novo_Start_OutOfFrame\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 986, u'impact': u'MODERATE', u'protein_start': 29, u'consequence_terms': [u'missense_variant'], u'codons': u'gaG/gaC', u'gene_symbol': u'BCL2', u'sift_score': 0.73, u'polyphen_score': 0.022, u'variant_allele': u'G', u'cds_end': 87, u'cds_start': 87, u'polyphen_prediction': u'benign', u'cdna_end': 986, u'amino_acids': u'E/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 29}, {u'polyphen_score': 0.022, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1549, u'cds_start': 87, u'protein_start': 29, u'codons': u'gaG/gaC', u'gene_symbol': u'BCL2', u'sift_score': 0.73, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'G', u'cds_end': 87, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1549, u'amino_acids': u'E/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 29}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 580, u'cds_start': 87, u'protein_start': 29, u'consequence_terms': [u'missense_variant'], u'codons': u'gaG/gaC', u'gene_symbol': u'BCL2', u'sift_score': 0.66, u'polyphen_score': 0.039, u'variant_allele': u'G', u'cds_end': 87, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 580, u'amino_acids': u'E/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 29}, {u'impact': u'MODIFIER', u'distance': 4498, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'G', u'strand': -1}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 233, u'cds_start': 87, u'protein_start': 29, u'codons': u'gaG/gaC', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.66, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'G', u'cds_end': 87, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 233, u'amino_acids': u'E/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'polyphen_score': 0.039, u'strand': -1, u'protein_end': 29}]\n",
      "[{u'polyphen_score': 0.045, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1368, u'cds_start': 469, u'protein_start': 157, u'codons': u'Atg/Ttg', u'gene_symbol': u'BCL2', u'sift_score': 0.62, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'A', u'cds_end': 469, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1368, u'amino_acids': u'M/L', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 157}, {u'polyphen_score': 0.045, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1931, u'impact': u'MODERATE', u'protein_start': 157, u'codons': u'Atg/Ttg', u'gene_symbol': u'BCL2', u'sift_score': 0.62, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'A', u'cds_end': 469, u'cds_start': 469, u'polyphen_prediction': u'benign', u'cdna_end': 1931, u'amino_acids': u'M/L', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 157}, {u'polyphen_score': 0.014, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 962, u'impact': u'MODERATE', u'protein_start': 157, u'codons': u'Atg/Ttg', u'gene_symbol': u'BCL2', u'sift_score': 1, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'A', u'cds_end': 469, u'cds_start': 469, u'polyphen_prediction': u'benign', u'cdna_end': 962, u'amino_acids': u'M/L', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 157}, {u'impact': u'MODIFIER', u'distance': 4116, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 615, u'cds_start': 469, u'protein_start': 157, u'codons': u'Atg/Ttg', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 1, u'polyphen_score': 0.014, u'variant_allele': u'A', u'cds_end': 469, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 615, u'amino_acids': u'M/L', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 157}]\n",
      "[{u'polyphen_score': 0.048, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1237, u'cds_start': 338, u'protein_start': 113, u'codons': u'gCc/gGc', u'gene_symbol': u'BCL2', u'sift_score': 0.06, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 338, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1237, u'amino_acids': u'A/G', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 113}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1800, u'cds_start': 338, u'protein_start': 113, u'codons': u'gCc/gGc', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.06, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 338, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1800, u'amino_acids': u'A/G', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'polyphen_score': 0.048, u'strand': -1, u'protein_end': 113}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 831, u'impact': u'MODERATE', u'protein_start': 113, u'codons': u'gCc/gGc', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.06, u'polyphen_score': 0.072, u'variant_allele': u'C', u'cds_end': 338, u'cds_start': 338, u'polyphen_prediction': u'benign', u'cdna_end': 831, u'amino_acids': u'A/G', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 113}, {u'impact': u'MODIFIER', u'distance': 4247, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'C', u'strand': -1}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 484, u'impact': u'MODERATE', u'protein_start': 113, u'consequence_terms': [u'missense_variant'], u'codons': u'gCc/gGc', u'gene_symbol': u'BCL2', u'sift_score': 0.06, u'polyphen_score': 0.072, u'variant_allele': u'C', u'cds_end': 338, u'cds_start': 338, u'polyphen_prediction': u'benign', u'cdna_end': 484, u'amino_acids': u'A/G', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 113}]\n",
      "[{u'polyphen_score': 0.003, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1039, u'cds_start': 140, u'protein_start': 47, u'codons': u'gGc/gCc', u'gene_symbol': u'BCL2', u'sift_score': 0.39, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'G', u'cds_end': 140, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1039, u'amino_acids': u'G/A', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 47}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1602, u'cds_start': 140, u'protein_start': 47, u'consequence_terms': [u'missense_variant'], u'codons': u'gGc/gCc', u'gene_symbol': u'BCL2', u'sift_score': 0.39, u'polyphen_score': 0.003, u'variant_allele': u'G', u'cds_end': 140, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1602, u'amino_acids': u'G/A', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 47}, {u'polyphen_score': 0.005, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 633, u'cds_start': 140, u'protein_start': 47, u'codons': u'gGc/gCc', u'gene_symbol': u'BCL2', u'sift_score': 0.36, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'G', u'cds_end': 140, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 633, u'amino_acids': u'G/A', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 47}, {u'impact': u'MODIFIER', u'distance': 4445, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'G', u'strand': -1}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 286, u'cds_start': 140, u'protein_start': 47, u'codons': u'gGc/gCc', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.36, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'G', u'cds_end': 140, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 286, u'amino_acids': u'G/A', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'polyphen_score': 0.005, u'strand': -1, u'protein_end': 47}]\n",
      "[{u'polyphen_score': 0.617, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 939, u'cds_start': 40, u'protein_start': 14, u'codons': u'Ata/Cta', u'gene_symbol': u'BCL2', u'sift_score': 0.3, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'G', u'cds_end': 40, u'impact': u'MODERATE', u'polyphen_prediction': u'possibly_damaging', u'cdna_end': 939, u'amino_acids': u'I/L', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 14}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1502, u'cds_start': 40, u'protein_start': 14, u'consequence_terms': [u'missense_variant'], u'codons': u'Ata/Cta', u'gene_symbol': u'BCL2', u'sift_score': 0.3, u'polyphen_score': 0.617, u'variant_allele': u'G', u'cds_end': 40, u'impact': u'MODERATE', u'polyphen_prediction': u'possibly_damaging', u'cdna_end': 1502, u'amino_acids': u'I/L', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 14}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 533, u'impact': u'MODERATE', u'protein_start': 14, u'codons': u'Ata/Cta', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.58, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'G', u'cds_end': 40, u'cds_start': 40, u'polyphen_prediction': u'probably_damaging', u'cdna_end': 533, u'amino_acids': u'I/L', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'polyphen_score': 0.998, u'strand': -1, u'protein_end': 14}, {u'impact': u'MODIFIER', u'distance': 4545, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'G', u'strand': -1}, {u'polyphen_score': 0.998, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 186, u'impact': u'MODERATE', u'protein_start': 14, u'codons': u'Ata/Cta', u'gene_symbol': u'BCL2', u'sift_score': 0.58, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'G', u'cds_end': 40, u'cds_start': 40, u'polyphen_prediction': u'probably_damaging', u'cdna_end': 186, u'amino_acids': u'I/L', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 14}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1422, u'impact': u'MODERATE', u'protein_start': 175, u'codons': u'Ctg/Gtg', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.29, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 523, u'cds_start': 523, u'polyphen_prediction': u'benign', u'cdna_end': 1422, u'amino_acids': u'L/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'polyphen_score': 0.003, u'strand': -1, u'protein_end': 175}, {u'polyphen_score': 0.003, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1985, u'cds_start': 523, u'protein_start': 175, u'codons': u'Ctg/Gtg', u'gene_symbol': u'BCL2', u'sift_score': 0.29, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 523, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1985, u'amino_acids': u'L/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 175}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1016, u'impact': u'MODERATE', u'protein_start': 175, u'codons': u'Ctg/Gtg', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.29, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 523, u'cds_start': 523, u'polyphen_prediction': u'benign', u'cdna_end': 1016, u'amino_acids': u'L/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'polyphen_score': 0.007, u'strand': -1, u'protein_end': 175}, {u'impact': u'MODIFIER', u'distance': 4062, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'C', u'strand': -1}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 669, u'cds_start': 523, u'protein_start': 175, u'consequence_terms': [u'missense_variant'], u'codons': u'Ctg/Gtg', u'gene_symbol': u'BCL2', u'sift_score': 0.29, u'polyphen_score': 0.007, u'variant_allele': u'C', u'cds_end': 523, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 669, u'amino_acids': u'L/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 175}]\n",
      "[{u'polyphen_score': 0.013, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 909, u'impact': u'MODERATE', u'protein_start': 4, u'codons': u'Gct/Cct', u'gene_symbol': u'BCL2', u'sift_score': 0.47, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'G', u'cds_end': 10, u'cds_start': 10, u'polyphen_prediction': u'benign', u'cdna_end': 909, u'amino_acids': u'A/P', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 4}, {u'polyphen_score': 0.013, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1472, u'impact': u'MODERATE', u'protein_start': 4, u'codons': u'Gct/Cct', u'gene_symbol': u'BCL2', u'sift_score': 0.47, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'G', u'cds_end': 10, u'cds_start': 10, u'polyphen_prediction': u'benign', u'cdna_end': 1472, u'amino_acids': u'A/P', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 4}, {u'polyphen_score': 0.003, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 503, u'impact': u'MODERATE', u'protein_start': 4, u'codons': u'Gct/Cct', u'gene_symbol': u'BCL2', u'sift_score': 0.5, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'G', u'cds_end': 10, u'cds_start': 10, u'polyphen_prediction': u'benign', u'cdna_end': 503, u'amino_acids': u'A/P', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 4}, {u'impact': u'MODIFIER', u'distance': 4575, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'G', u'strand': -1}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 156, u'cds_start': 10, u'protein_start': 4, u'codons': u'Gct/Cct', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.5, u'polyphen_score': 0.003, u'variant_allele': u'G', u'cds_end': 10, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 156, u'amino_acids': u'A/P', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 4}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1291, u'cds_start': 392, u'protein_start': 131, u'consequence_terms': [u'missense_variant'], u'codons': u'gCc/gTc', u'gene_symbol': u'BCL2', u'sift_score': 0.4, u'polyphen_score': 0.055, u'variant_allele': u'A', u'cds_end': 392, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1291, u'amino_acids': u'A/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 131}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1854, u'impact': u'MODERATE', u'protein_start': 131, u'consequence_terms': [u'missense_variant'], u'codons': u'gCc/gTc', u'gene_symbol': u'BCL2', u'sift_score': 0.4, u'polyphen_score': 0.055, u'variant_allele': u'A', u'cds_end': 392, u'cds_start': 392, u'polyphen_prediction': u'benign', u'cdna_end': 1854, u'amino_acids': u'A/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 131}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 885, u'impact': u'MODERATE', u'protein_start': 131, u'codons': u'gCc/gTc', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.39, u'polyphen_score': 0.123, u'variant_allele': u'A', u'cds_end': 392, u'cds_start': 392, u'polyphen_prediction': u'benign', u'cdna_end': 885, u'amino_acids': u'A/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 131}, {u'impact': u'MODIFIER', u'distance': 4193, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 538, u'cds_start': 392, u'protein_start': 131, u'codons': u'gCc/gTc', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.39, u'polyphen_score': 0.123, u'variant_allele': u'A', u'cds_end': 392, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 538, u'amino_acids': u'A/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 131}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1237, u'impact': u'MODERATE', u'protein_start': 113, u'codons': u'gCc/gTc', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.06, u'polyphen_score': 0.155, u'variant_allele': u'A', u'cds_end': 338, u'cds_start': 338, u'polyphen_prediction': u'benign', u'cdna_end': 1237, u'amino_acids': u'A/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 113}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1800, u'cds_start': 338, u'protein_start': 113, u'codons': u'gCc/gTc', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.06, u'polyphen_score': 0.155, u'variant_allele': u'A', u'cds_end': 338, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1800, u'amino_acids': u'A/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 113}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 831, u'impact': u'MODERATE', u'protein_start': 113, u'codons': u'gCc/gTc', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.06, u'polyphen_score': 0.218, u'variant_allele': u'A', u'cds_end': 338, u'cds_start': 338, u'polyphen_prediction': u'benign', u'cdna_end': 831, u'amino_acids': u'A/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 113}, {u'impact': u'MODIFIER', u'distance': 4247, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 484, u'cds_start': 338, u'protein_start': 113, u'consequence_terms': [u'missense_variant'], u'codons': u'gCc/gTc', u'gene_symbol': u'BCL2', u'sift_score': 0.06, u'polyphen_score': 0.218, u'variant_allele': u'A', u'cds_end': 338, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 484, u'amino_acids': u'A/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 113}]\n",
      "[{u'polyphen_score': 0, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1006, u'impact': u'MODERATE', u'protein_start': 36, u'codons': u'gGc/gAc', u'gene_symbol': u'BCL2', u'sift_score': 0.75, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 107, u'cds_start': 107, u'polyphen_prediction': u'benign', u'cdna_end': 1006, u'amino_acids': u'G/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 36}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1569, u'cds_start': 107, u'protein_start': 36, u'codons': u'gGc/gAc', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.75, u'polyphen_score': 0, u'variant_allele': u'T', u'cds_end': 107, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1569, u'amino_acids': u'G/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 36}, {u'polyphen_score': 0.001, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 600, u'impact': u'MODERATE', u'protein_start': 36, u'codons': u'gGc/gAc', u'gene_symbol': u'BCL2', u'sift_score': 0.65, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 107, u'cds_start': 107, u'polyphen_prediction': u'benign', u'cdna_end': 600, u'amino_acids': u'G/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 36}, {u'impact': u'MODIFIER', u'distance': 4478, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'polyphen_score': 0.001, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 253, u'impact': u'MODERATE', u'protein_start': 36, u'codons': u'gGc/gAc', u'gene_symbol': u'BCL2', u'sift_score': 0.65, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 107, u'cds_start': 107, u'polyphen_prediction': u'benign', u'cdna_end': 253, u'amino_acids': u'G/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 36}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1074, u'cds_start': 175, u'protein_start': 59, u'consequence_terms': [u'missense_variant'], u'codons': u'Cca/Tca', u'gene_symbol': u'BCL2', u'sift_score': 0.37, u'polyphen_score': 0, u'variant_allele': u'A', u'cds_end': 175, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1074, u'amino_acids': u'P/S', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 59}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1637, u'impact': u'MODERATE', u'protein_start': 59, u'codons': u'Cca/Tca', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.37, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'A', u'cds_end': 175, u'cds_start': 175, u'polyphen_prediction': u'benign', u'cdna_end': 1637, u'amino_acids': u'P/S', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'polyphen_score': 0, u'strand': -1, u'protein_end': 59}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 668, u'impact': u'MODERATE', u'protein_start': 59, u'codons': u'Cca/Tca', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.42, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'A', u'cds_end': 175, u'cds_start': 175, u'polyphen_prediction': u'benign', u'cdna_end': 668, u'amino_acids': u'P/S', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'polyphen_score': 0.003, u'strand': -1, u'protein_end': 59}, {u'impact': u'MODIFIER', u'distance': 4410, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'polyphen_score': 0.003, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 321, u'cds_start': 175, u'protein_start': 59, u'codons': u'Cca/Tca', u'gene_symbol': u'BCL2', u'sift_score': 0.42, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'A', u'cds_end': 175, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 321, u'amino_acids': u'P/S', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 59}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1078, u'impact': u'MODERATE', u'protein_start': 60, u'codons': u'gCc/gTc', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.23, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'A', u'cds_end': 179, u'cds_start': 179, u'polyphen_prediction': u'benign', u'cdna_end': 1078, u'amino_acids': u'A/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'polyphen_score': 0.001, u'strand': -1, u'protein_end': 60}, {u'polyphen_score': 0.001, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1641, u'impact': u'MODERATE', u'protein_start': 60, u'codons': u'gCc/gTc', u'gene_symbol': u'BCL2', u'sift_score': 0.23, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'A', u'cds_end': 179, u'cds_start': 179, u'polyphen_prediction': u'benign', u'cdna_end': 1641, u'amino_acids': u'A/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 60}, {u'polyphen_score': 0.083, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 672, u'cds_start': 179, u'protein_start': 60, u'codons': u'gCc/gTc', u'gene_symbol': u'BCL2', u'sift_score': 0.33, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'A', u'cds_end': 179, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 672, u'amino_acids': u'A/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 60}, {u'impact': u'MODIFIER', u'distance': 4406, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 325, u'cds_start': 179, u'protein_start': 60, u'consequence_terms': [u'missense_variant'], u'codons': u'gCc/gTc', u'gene_symbol': u'BCL2', u'sift_score': 0.33, u'polyphen_score': 0.083, u'variant_allele': u'A', u'cds_end': 179, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 325, u'amino_acids': u'A/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 60}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1017, u'impact': u'MODERATE', u'protein_start': 40, u'codons': u'Ccg/Gcg', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.71, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 118, u'cds_start': 118, u'polyphen_prediction': u'benign', u'cdna_end': 1017, u'amino_acids': u'P/A', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'polyphen_score': 0, u'strand': -1, u'protein_end': 40}, {u'polyphen_score': 0, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1580, u'impact': u'MODERATE', u'protein_start': 40, u'codons': u'Ccg/Gcg', u'gene_symbol': u'BCL2', u'sift_score': 0.71, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 118, u'cds_start': 118, u'polyphen_prediction': u'benign', u'cdna_end': 1580, u'amino_acids': u'P/A', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 40}, {u'polyphen_score': 0.001, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 611, u'cds_start': 118, u'protein_start': 40, u'codons': u'Ccg/Gcg', u'gene_symbol': u'BCL2', u'sift_score': 0.68, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 118, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 611, u'amino_acids': u'P/A', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 40}, {u'impact': u'MODIFIER', u'distance': 4467, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'C', u'strand': -1}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 264, u'impact': u'MODERATE', u'protein_start': 40, u'codons': u'Ccg/Gcg', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.68, u'polyphen_score': 0.001, u'variant_allele': u'C', u'cds_end': 118, u'cds_start': 118, u'polyphen_prediction': u'benign', u'cdna_end': 264, u'amino_acids': u'P/A', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 40}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 959, u'impact': u'MODERATE', u'protein_start': 20, u'consequence_terms': [u'missense_variant'], u'codons': u'caT/caG', u'gene_symbol': u'BCL2', u'sift_score': 0.26, u'polyphen_score': 0.919, u'variant_allele': u'C', u'cds_end': 60, u'cds_start': 60, u'polyphen_prediction': u'probably_damaging', u'cdna_end': 959, u'amino_acids': u'H/Q', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 20}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1522, u'cds_start': 60, u'protein_start': 20, u'codons': u'caT/caG', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.26, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 60, u'impact': u'MODERATE', u'polyphen_prediction': u'probably_damaging', u'cdna_end': 1522, u'amino_acids': u'H/Q', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'polyphen_score': 0.919, u'strand': -1, u'protein_end': 20}, {u'polyphen_score': 0.94, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 553, u'cds_start': 60, u'protein_start': 20, u'codons': u'caT/caG', u'gene_symbol': u'BCL2', u'sift_score': 0.31, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 60, u'impact': u'MODERATE', u'polyphen_prediction': u'probably_damaging', u'cdna_end': 553, u'amino_acids': u'H/Q', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 20}, {u'impact': u'MODIFIER', u'distance': 4525, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'C', u'strand': -1}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 206, u'impact': u'MODERATE', u'protein_start': 20, u'codons': u'caT/caG', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.31, u'polyphen_score': 0.94, u'variant_allele': u'C', u'cds_end': 60, u'cds_start': 60, u'polyphen_prediction': u'probably_damaging', u'cdna_end': 206, u'amino_acids': u'H/Q', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 20}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1415, u'cds_start': 516, u'protein_start': 172, u'consequence_terms': [u'missense_variant'], u'codons': u'aaC/aaA', u'gene_symbol': u'BCL2', u'sift_score': 0.04, u'polyphen_score': 0.743, u'variant_allele': u'T', u'cds_end': 516, u'impact': u'MODERATE', u'polyphen_prediction': u'possibly_damaging', u'cdna_end': 1415, u'amino_acids': u'N/K', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'deleterious', u'strand': -1, u'protein_end': 172}, {u'polyphen_score': 0.743, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1978, u'cds_start': 516, u'protein_start': 172, u'codons': u'aaC/aaA', u'gene_symbol': u'BCL2', u'sift_score': 0.04, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 516, u'impact': u'MODERATE', u'polyphen_prediction': u'possibly_damaging', u'cdna_end': 1978, u'amino_acids': u'N/K', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'deleterious', u'strand': -1, u'protein_end': 172}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1009, u'cds_start': 516, u'protein_start': 172, u'consequence_terms': [u'missense_variant'], u'codons': u'aaC/aaA', u'gene_symbol': u'BCL2', u'sift_score': 0.31, u'polyphen_score': 0.827, u'variant_allele': u'T', u'cds_end': 516, u'impact': u'MODERATE', u'polyphen_prediction': u'possibly_damaging', u'cdna_end': 1009, u'amino_acids': u'N/K', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 172}, {u'impact': u'MODIFIER', u'distance': 4069, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 662, u'cds_start': 516, u'protein_start': 172, u'consequence_terms': [u'missense_variant'], u'codons': u'aaC/aaA', u'gene_symbol': u'BCL2', u'sift_score': 0.31, u'polyphen_score': 0.827, u'variant_allele': u'T', u'cds_end': 516, u'impact': u'MODERATE', u'polyphen_prediction': u'possibly_damaging', u'cdna_end': 662, u'amino_acids': u'N/K', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 172}]\n",
      "[{u'polyphen_score': 0.685, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1206, u'cds_start': 307, u'protein_start': 103, u'codons': u'Gac/Aac', u'gene_symbol': u'BCL2', u'sift_score': 0, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 307, u'impact': u'MODERATE', u'polyphen_prediction': u'possibly_damaging', u'cdna_end': 1206, u'amino_acids': u'D/N', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'deleterious', u'strand': -1, u'protein_end': 103}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1769, u'impact': u'MODERATE', u'protein_start': 103, u'codons': u'Gac/Aac', u'sift_prediction': u'deleterious', u'gene_symbol': u'BCL2', u'sift_score': 0, u'polyphen_score': 0.685, u'variant_allele': u'T', u'cds_end': 307, u'cds_start': 307, u'polyphen_prediction': u'possibly_damaging', u'cdna_end': 1769, u'amino_acids': u'D/N', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 103}, {u'polyphen_score': 0.792, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 800, u'impact': u'MODERATE', u'protein_start': 103, u'codons': u'Gac/Aac', u'gene_symbol': u'BCL2', u'sift_score': 0, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 307, u'cds_start': 307, u'polyphen_prediction': u'possibly_damaging', u'cdna_end': 800, u'amino_acids': u'D/N', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'sift_prediction': u'deleterious', u'strand': -1, u'protein_end': 103}, {u'impact': u'MODIFIER', u'distance': 4278, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'polyphen_score': 0.792, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 453, u'cds_start': 307, u'protein_start': 103, u'codons': u'Gac/Aac', u'gene_symbol': u'BCL2', u'sift_score': 0, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 307, u'impact': u'MODERATE', u'polyphen_prediction': u'possibly_damaging', u'cdna_end': 453, u'amino_acids': u'D/N', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'sift_prediction': u'deleterious', u'strand': -1, u'protein_end': 103}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1001, u'cds_start': 102, u'protein_start': 34, u'codons': u'gaT/gaA', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.64, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 102, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1001, u'amino_acids': u'D/E', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'polyphen_score': 0, u'strand': -1, u'protein_end': 34}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1564, u'cds_start': 102, u'protein_start': 34, u'consequence_terms': [u'missense_variant'], u'codons': u'gaT/gaA', u'gene_symbol': u'BCL2', u'sift_score': 0.64, u'polyphen_score': 0, u'variant_allele': u'T', u'cds_end': 102, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1564, u'amino_acids': u'D/E', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 34}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 595, u'cds_start': 102, u'protein_start': 34, u'codons': u'gaT/gaA', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.62, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 102, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 595, u'amino_acids': u'D/E', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'polyphen_score': 0, u'strand': -1, u'protein_end': 34}, {u'impact': u'MODIFIER', u'distance': 4483, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 248, u'cds_start': 102, u'protein_start': 34, u'codons': u'gaT/gaA', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.62, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 102, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 248, u'amino_acids': u'D/E', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'polyphen_score': 0, u'strand': -1, u'protein_end': 34}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1237, u'cds_start': 338, u'protein_start': 113, u'consequence_terms': [u'missense_variant'], u'codons': u'gCc/gGc', u'gene_symbol': u'BCL2', u'sift_score': 0.06, u'polyphen_score': 0.048, u'variant_allele': u'C', u'cds_end': 338, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1237, u'amino_acids': u'A/G', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 113}, {u'polyphen_score': 0.048, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1800, u'cds_start': 338, u'protein_start': 113, u'codons': u'gCc/gGc', u'gene_symbol': u'BCL2', u'sift_score': 0.06, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 338, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1800, u'amino_acids': u'A/G', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 113}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 831, u'cds_start': 338, u'protein_start': 113, u'consequence_terms': [u'missense_variant'], u'codons': u'gCc/gGc', u'gene_symbol': u'BCL2', u'sift_score': 0.06, u'polyphen_score': 0.072, u'variant_allele': u'C', u'cds_end': 338, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 831, u'amino_acids': u'A/G', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 113}, {u'impact': u'MODIFIER', u'distance': 4247, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'C', u'strand': -1}, {u'polyphen_score': 0.072, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 484, u'cds_start': 338, u'protein_start': 113, u'codons': u'gCc/gGc', u'gene_symbol': u'BCL2', u'sift_score': 0.06, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 338, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 484, u'amino_acids': u'A/G', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 113}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 910, u'cds_start': 11, u'protein_start': 4, u'codons': u'gCt/gTt', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.06, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'A', u'cds_end': 11, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 910, u'amino_acids': u'A/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'polyphen_score': 0.321, u'strand': -1, u'protein_end': 4}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1473, u'impact': u'MODERATE', u'protein_start': 4, u'consequence_terms': [u'missense_variant'], u'codons': u'gCt/gTt', u'gene_symbol': u'BCL2', u'sift_score': 0.06, u'polyphen_score': 0.321, u'variant_allele': u'A', u'cds_end': 11, u'cds_start': 11, u'polyphen_prediction': u'benign', u'cdna_end': 1473, u'amino_acids': u'A/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 4}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 504, u'impact': u'MODERATE', u'protein_start': 4, u'consequence_terms': [u'missense_variant'], u'codons': u'gCt/gTt', u'gene_symbol': u'BCL2', u'sift_score': 0.08, u'polyphen_score': 0.202, u'variant_allele': u'A', u'cds_end': 11, u'cds_start': 11, u'polyphen_prediction': u'benign', u'cdna_end': 504, u'amino_acids': u'A/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 4}, {u'impact': u'MODIFIER', u'distance': 4574, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 157, u'cds_start': 11, u'protein_start': 4, u'codons': u'gCt/gTt', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.08, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'A', u'cds_end': 11, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 157, u'amino_acids': u'A/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'polyphen_score': 0.202, u'strand': -1, u'protein_end': 4}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 997, u'cds_start': 98, u'protein_start': 33, u'consequence_terms': [u'missense_variant'], u'codons': u'gGa/gAa', u'gene_symbol': u'BCL2', u'sift_score': 0.59, u'polyphen_score': 0.591, u'variant_allele': u'T', u'cds_end': 98, u'impact': u'MODERATE', u'polyphen_prediction': u'possibly_damaging', u'cdna_end': 997, u'amino_acids': u'G/E', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 33}, {u'polyphen_score': 0.591, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1560, u'cds_start': 98, u'protein_start': 33, u'codons': u'gGa/gAa', u'gene_symbol': u'BCL2', u'sift_score': 0.59, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 98, u'impact': u'MODERATE', u'polyphen_prediction': u'possibly_damaging', u'cdna_end': 1560, u'amino_acids': u'G/E', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 33}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 591, u'impact': u'MODERATE', u'protein_start': 33, u'codons': u'gGa/gAa', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.77, u'polyphen_score': 0.114, u'variant_allele': u'T', u'cds_end': 98, u'cds_start': 98, u'polyphen_prediction': u'benign', u'cdna_end': 591, u'amino_acids': u'G/E', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 33}, {u'impact': u'MODIFIER', u'distance': 4487, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'polyphen_score': 0.114, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 244, u'cds_start': 98, u'protein_start': 33, u'codons': u'gGa/gAa', u'gene_symbol': u'BCL2', u'sift_score': 0.77, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 98, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 244, u'amino_acids': u'G/E', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 33}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1291, u'cds_start': 392, u'protein_start': 131, u'codons': u'gCc/gGc', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.16, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 392, u'impact': u'MODERATE', u'polyphen_prediction': u'possibly_damaging', u'cdna_end': 1291, u'amino_acids': u'A/G', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'polyphen_score': 0.567, u'strand': -1, u'protein_end': 131}, {u'polyphen_score': 0.567, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1854, u'cds_start': 392, u'protein_start': 131, u'codons': u'gCc/gGc', u'gene_symbol': u'BCL2', u'sift_score': 0.16, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 392, u'impact': u'MODERATE', u'polyphen_prediction': u'possibly_damaging', u'cdna_end': 1854, u'amino_acids': u'A/G', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 131}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 885, u'impact': u'MODERATE', u'protein_start': 131, u'codons': u'gCc/gGc', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.32, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 392, u'cds_start': 392, u'polyphen_prediction': u'possibly_damaging', u'cdna_end': 885, u'amino_acids': u'A/G', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'polyphen_score': 0.761, u'strand': -1, u'protein_end': 131}, {u'impact': u'MODIFIER', u'distance': 4193, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'C', u'strand': -1}, {u'polyphen_score': 0.761, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 538, u'cds_start': 392, u'protein_start': 131, u'codons': u'gCc/gGc', u'gene_symbol': u'BCL2', u'sift_score': 0.32, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 392, u'impact': u'MODERATE', u'polyphen_prediction': u'possibly_damaging', u'cdna_end': 538, u'amino_acids': u'A/G', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 131}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1291, u'cds_start': 392, u'protein_start': 131, u'codons': u'gCc/gTc', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.4, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'A', u'cds_end': 392, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1291, u'amino_acids': u'A/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'polyphen_score': 0.055, u'strand': -1, u'protein_end': 131}, {u'polyphen_score': 0.055, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1854, u'impact': u'MODERATE', u'protein_start': 131, u'codons': u'gCc/gTc', u'gene_symbol': u'BCL2', u'sift_score': 0.4, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'A', u'cds_end': 392, u'cds_start': 392, u'polyphen_prediction': u'benign', u'cdna_end': 1854, u'amino_acids': u'A/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 131}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 885, u'cds_start': 392, u'protein_start': 131, u'codons': u'gCc/gTc', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.39, u'polyphen_score': 0.123, u'variant_allele': u'A', u'cds_end': 392, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 885, u'amino_acids': u'A/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 131}, {u'impact': u'MODIFIER', u'distance': 4193, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'polyphen_score': 0.123, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 538, u'impact': u'MODERATE', u'protein_start': 131, u'codons': u'gCc/gTc', u'gene_symbol': u'BCL2', u'sift_score': 0.39, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'A', u'cds_end': 392, u'cds_start': 392, u'polyphen_prediction': u'benign', u'cdna_end': 538, u'amino_acids': u'A/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 131}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1029, u'impact': u'MODERATE', u'protein_start': 44, u'codons': u'Ccc/Gcc', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.68, u'polyphen_score': 0.007, u'variant_allele': u'C', u'cds_end': 130, u'cds_start': 130, u'polyphen_prediction': u'benign', u'cdna_end': 1029, u'amino_acids': u'P/A', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 44}, {u'polyphen_score': 0.007, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1592, u'cds_start': 130, u'protein_start': 44, u'codons': u'Ccc/Gcc', u'gene_symbol': u'BCL2', u'sift_score': 0.68, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 130, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1592, u'amino_acids': u'P/A', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 44}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 623, u'cds_start': 130, u'protein_start': 44, u'codons': u'Ccc/Gcc', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.68, u'polyphen_score': 0.003, u'variant_allele': u'C', u'cds_end': 130, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 623, u'amino_acids': u'P/A', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 44}, {u'impact': u'MODIFIER', u'distance': 4455, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'C', u'strand': -1}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 276, u'impact': u'MODERATE', u'protein_start': 44, u'codons': u'Ccc/Gcc', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.68, u'polyphen_score': 0.003, u'variant_allele': u'C', u'cds_end': 130, u'cds_start': 130, u'polyphen_prediction': u'benign', u'cdna_end': 276, u'amino_acids': u'P/A', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 44}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 961, u'impact': u'MODERATE', u'protein_start': 21, u'consequence_terms': [u'missense_variant'], u'codons': u'tAt/tCt', u'gene_symbol': u'BCL2', u'sift_score': 0.01, u'polyphen_score': 0.974, u'variant_allele': u'G', u'cds_end': 62, u'cds_start': 62, u'polyphen_prediction': u'probably_damaging', u'cdna_end': 961, u'amino_acids': u'Y/S', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'deleterious', u'strand': -1, u'protein_end': 21}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1524, u'cds_start': 62, u'protein_start': 21, u'codons': u'tAt/tCt', u'sift_prediction': u'deleterious', u'gene_symbol': u'BCL2', u'sift_score': 0.01, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'G', u'cds_end': 62, u'impact': u'MODERATE', u'polyphen_prediction': u'probably_damaging', u'cdna_end': 1524, u'amino_acids': u'Y/S', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'polyphen_score': 0.974, u'strand': -1, u'protein_end': 21}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 555, u'impact': u'MODERATE', u'protein_start': 21, u'consequence_terms': [u'missense_variant'], u'codons': u'tAt/tCt', u'gene_symbol': u'BCL2', u'sift_score': 0.01, u'polyphen_score': 0.976, u'variant_allele': u'G', u'cds_end': 62, u'cds_start': 62, u'polyphen_prediction': u'probably_damaging', u'cdna_end': 555, u'amino_acids': u'Y/S', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'sift_prediction': u'deleterious', u'strand': -1, u'protein_end': 21}, {u'impact': u'MODIFIER', u'distance': 4523, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'G', u'strand': -1}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 208, u'cds_start': 62, u'protein_start': 21, u'codons': u'tAt/tCt', u'sift_prediction': u'deleterious', u'gene_symbol': u'BCL2', u'sift_score': 0.01, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'G', u'cds_end': 62, u'impact': u'MODERATE', u'polyphen_prediction': u'probably_damaging', u'cdna_end': 208, u'amino_acids': u'Y/S', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'polyphen_score': 0.976, u'strand': -1, u'protein_end': 21}]\n",
      "[{u'polyphen_score': 0.99, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 922, u'impact': u'MODERATE', u'protein_start': 8, u'codons': u'gGg/gAg', u'gene_symbol': u'BCL2', u'sift_score': 0.04, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 23, u'cds_start': 23, u'polyphen_prediction': u'probably_damaging', u'cdna_end': 922, u'amino_acids': u'G/E', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'deleterious', u'strand': -1, u'protein_end': 8}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1485, u'impact': u'MODERATE', u'protein_start': 8, u'codons': u'gGg/gAg', u'sift_prediction': u'deleterious', u'gene_symbol': u'BCL2', u'sift_score': 0.04, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 23, u'cds_start': 23, u'polyphen_prediction': u'probably_damaging', u'cdna_end': 1485, u'amino_acids': u'G/E', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'polyphen_score': 0.99, u'strand': -1, u'protein_end': 8}, {u'polyphen_score': 0.994, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 516, u'impact': u'MODERATE', u'protein_start': 8, u'codons': u'gGg/gAg', u'gene_symbol': u'BCL2', u'sift_score': 0.02, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 23, u'cds_start': 23, u'polyphen_prediction': u'probably_damaging', u'cdna_end': 516, u'amino_acids': u'G/E', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'sift_prediction': u'deleterious', u'strand': -1, u'protein_end': 8}, {u'impact': u'MODIFIER', u'distance': 4562, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 169, u'impact': u'MODERATE', u'protein_start': 8, u'codons': u'gGg/gAg', u'sift_prediction': u'deleterious', u'gene_symbol': u'BCL2', u'sift_score': 0.02, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 23, u'cds_start': 23, u'polyphen_prediction': u'probably_damaging', u'cdna_end': 169, u'amino_acids': u'G/E', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'polyphen_score': 0.994, u'strand': -1, u'protein_end': 8}]\n",
      "[{u'polyphen_score': 0.781, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1182, u'cds_start': 283, u'protein_start': 95, u'codons': u'Ctg/Atg', u'gene_symbol': u'BCL2', u'sift_score': 0.16, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 283, u'impact': u'MODERATE', u'polyphen_prediction': u'possibly_damaging', u'cdna_end': 1182, u'amino_acids': u'L/M', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 95}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1745, u'impact': u'MODERATE', u'protein_start': 95, u'codons': u'Ctg/Atg', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.16, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 283, u'cds_start': 283, u'polyphen_prediction': u'possibly_damaging', u'cdna_end': 1745, u'amino_acids': u'L/M', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'polyphen_score': 0.781, u'strand': -1, u'protein_end': 95}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 776, u'impact': u'MODERATE', u'protein_start': 95, u'codons': u'Ctg/Atg', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.07, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 283, u'cds_start': 283, u'polyphen_prediction': u'possibly_damaging', u'cdna_end': 776, u'amino_acids': u'L/M', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'polyphen_score': 0.656, u'strand': -1, u'protein_end': 95}, {u'impact': u'MODIFIER', u'distance': 4302, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 429, u'impact': u'MODERATE', u'protein_start': 95, u'codons': u'Ctg/Atg', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.07, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 283, u'cds_start': 283, u'polyphen_prediction': u'possibly_damaging', u'cdna_end': 429, u'amino_acids': u'L/M', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'polyphen_score': 0.656, u'strand': -1, u'protein_end': 95}]\n",
      "[{u'polyphen_score': 0.154, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1365, u'cds_start': 466, u'protein_start': 156, u'codons': u'Gtc/Ctc', u'gene_symbol': u'BCL2', u'sift_score': 0.08, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'G', u'cds_end': 466, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1365, u'amino_acids': u'V/L', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 156}, {u'polyphen_score': 0.154, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1928, u'cds_start': 466, u'protein_start': 156, u'codons': u'Gtc/Ctc', u'gene_symbol': u'BCL2', u'sift_score': 0.08, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'G', u'cds_end': 466, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1928, u'amino_acids': u'V/L', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 156}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 959, u'cds_start': 466, u'protein_start': 156, u'codons': u'Gtc/Ctc', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.06, u'polyphen_score': 0.195, u'variant_allele': u'G', u'cds_end': 466, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 959, u'amino_acids': u'V/L', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 156}, {u'impact': u'MODIFIER', u'distance': 4119, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'G', u'strand': -1}, {u'polyphen_score': 0.195, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 612, u'cds_start': 466, u'protein_start': 156, u'codons': u'Gtc/Ctc', u'gene_symbol': u'BCL2', u'sift_score': 0.06, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'G', u'cds_end': 466, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 612, u'amino_acids': u'V/L', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 156}]\n",
      "[{u'polyphen_score': 0.999, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1349, u'impact': u'MODERATE', u'protein_start': 150, u'codons': u'ttC/ttG', u'gene_symbol': u'BCL2', u'sift_score': 0, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 450, u'cds_start': 450, u'polyphen_prediction': u'probably_damaging', u'cdna_end': 1349, u'amino_acids': u'F/L', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'deleterious', u'strand': -1, u'protein_end': 150}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1912, u'cds_start': 450, u'protein_start': 150, u'codons': u'ttC/ttG', u'sift_prediction': u'deleterious', u'gene_symbol': u'BCL2', u'sift_score': 0, u'polyphen_score': 0.999, u'variant_allele': u'C', u'cds_end': 450, u'impact': u'MODERATE', u'polyphen_prediction': u'probably_damaging', u'cdna_end': 1912, u'amino_acids': u'F/L', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 150}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 943, u'cds_start': 450, u'protein_start': 150, u'codons': u'ttC/ttG', u'sift_prediction': u'deleterious', u'gene_symbol': u'BCL2', u'sift_score': 0, u'polyphen_score': 0.999, u'variant_allele': u'C', u'cds_end': 450, u'impact': u'MODERATE', u'polyphen_prediction': u'probably_damaging', u'cdna_end': 943, u'amino_acids': u'F/L', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 150}, {u'impact': u'MODIFIER', u'distance': 4135, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'C', u'strand': -1}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 596, u'cds_start': 450, u'protein_start': 150, u'codons': u'ttC/ttG', u'sift_prediction': u'deleterious', u'gene_symbol': u'BCL2', u'sift_score': 0, u'polyphen_score': 0.999, u'variant_allele': u'C', u'cds_end': 450, u'impact': u'MODERATE', u'polyphen_prediction': u'probably_damaging', u'cdna_end': 596, u'amino_acids': u'F/L', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 150}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1066, u'cds_start': 167, u'protein_start': 56, u'codons': u'aCg/aTg', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.15, u'polyphen_score': 0.093, u'variant_allele': u'A', u'cds_end': 167, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1066, u'amino_acids': u'T/M', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 56}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1629, u'cds_start': 167, u'protein_start': 56, u'codons': u'aCg/aTg', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.15, u'polyphen_score': 0.093, u'variant_allele': u'A', u'cds_end': 167, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1629, u'amino_acids': u'T/M', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 56}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 660, u'impact': u'MODERATE', u'protein_start': 56, u'consequence_terms': [u'missense_variant'], u'codons': u'aCg/aTg', u'gene_symbol': u'BCL2', u'sift_score': 0.09, u'polyphen_score': 0.007, u'variant_allele': u'A', u'cds_end': 167, u'cds_start': 167, u'polyphen_prediction': u'benign', u'cdna_end': 660, u'amino_acids': u'T/M', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 56}, {u'impact': u'MODIFIER', u'distance': 4418, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 313, u'impact': u'MODERATE', u'protein_start': 56, u'consequence_terms': [u'missense_variant'], u'codons': u'aCg/aTg', u'gene_symbol': u'BCL2', u'sift_score': 0.09, u'polyphen_score': 0.007, u'variant_allele': u'A', u'cds_end': 167, u'cds_start': 167, u'polyphen_prediction': u'benign', u'cdna_end': 313, u'amino_acids': u'T/M', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 56}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1039, u'cds_start': 140, u'protein_start': 47, u'consequence_terms': [u'missense_variant'], u'codons': u'gGc/gAc', u'gene_symbol': u'BCL2', u'sift_score': 0.11, u'polyphen_score': 0.218, u'variant_allele': u'T', u'cds_end': 140, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1039, u'amino_acids': u'G/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 47}, {u'polyphen_score': 0.218, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1602, u'impact': u'MODERATE', u'protein_start': 47, u'codons': u'gGc/gAc', u'gene_symbol': u'BCL2', u'sift_score': 0.11, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 140, u'cds_start': 140, u'polyphen_prediction': u'benign', u'cdna_end': 1602, u'amino_acids': u'G/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 47}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 633, u'impact': u'MODERATE', u'protein_start': 47, u'codons': u'gGc/gAc', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.15, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 140, u'cds_start': 140, u'polyphen_prediction': u'benign', u'cdna_end': 633, u'amino_acids': u'G/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'polyphen_score': 0.379, u'strand': -1, u'protein_end': 47}, {u'impact': u'MODIFIER', u'distance': 4445, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'polyphen_score': 0.379, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 286, u'impact': u'MODERATE', u'protein_start': 47, u'codons': u'gGc/gAc', u'gene_symbol': u'BCL2', u'sift_score': 0.15, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 140, u'cds_start': 140, u'polyphen_prediction': u'benign', u'cdna_end': 286, u'amino_acids': u'G/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 47}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 907, u'cds_start': 8, u'protein_start': 3, u'codons': u'cAc/cTc', u'sift_prediction': u'deleterious_low_confidence', u'gene_symbol': u'BCL2', u'sift_score': 0, u'polyphen_score': 0.1, u'variant_allele': u'A', u'cds_end': 8, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 907, u'amino_acids': u'H/L', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 3}, {u'polyphen_score': 0.1, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1470, u'impact': u'MODERATE', u'protein_start': 3, u'codons': u'cAc/cTc', u'gene_symbol': u'BCL2', u'sift_score': 0, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'A', u'cds_end': 8, u'cds_start': 8, u'polyphen_prediction': u'benign', u'cdna_end': 1470, u'amino_acids': u'H/L', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'deleterious_low_confidence', u'strand': -1, u'protein_end': 3}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 501, u'cds_start': 8, u'protein_start': 3, u'codons': u'cAc/cTc', u'sift_prediction': u'deleterious_low_confidence', u'gene_symbol': u'BCL2', u'sift_score': 0, u'polyphen_score': 0.154, u'variant_allele': u'A', u'cds_end': 8, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 501, u'amino_acids': u'H/L', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 3}, {u'impact': u'MODIFIER', u'distance': 4577, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'polyphen_score': 0.154, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 154, u'impact': u'MODERATE', u'protein_start': 3, u'codons': u'cAc/cTc', u'gene_symbol': u'BCL2', u'sift_score': 0, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'A', u'cds_end': 8, u'cds_start': 8, u'polyphen_prediction': u'benign', u'cdna_end': 154, u'amino_acids': u'H/L', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'sift_prediction': u'deleterious_low_confidence', u'strand': -1, u'protein_end': 3}]\n",
      "[{u'polyphen_score': 0.007, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1009, u'cds_start': 110, u'protein_start': 37, u'codons': u'gCc/gTc', u'gene_symbol': u'BCL2', u'sift_score': 0.23, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'A', u'cds_end': 110, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1009, u'amino_acids': u'A/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 37}, {u'polyphen_score': 0.007, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1572, u'cds_start': 110, u'protein_start': 37, u'codons': u'gCc/gTc', u'gene_symbol': u'BCL2', u'sift_score': 0.23, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'A', u'cds_end': 110, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1572, u'amino_acids': u'A/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 37}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 603, u'impact': u'MODERATE', u'protein_start': 37, u'codons': u'gCc/gTc', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.21, u'polyphen_score': 0.001, u'variant_allele': u'A', u'cds_end': 110, u'cds_start': 110, u'polyphen_prediction': u'benign', u'cdna_end': 603, u'amino_acids': u'A/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 37}, {u'impact': u'MODIFIER', u'distance': 4475, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'polyphen_score': 0.001, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 256, u'cds_start': 110, u'protein_start': 37, u'codons': u'gCc/gTc', u'gene_symbol': u'BCL2', u'sift_score': 0.21, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'A', u'cds_end': 110, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 256, u'amino_acids': u'A/V', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 37}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 922, u'impact': u'MODERATE', u'protein_start': 8, u'consequence_terms': [u'missense_variant'], u'codons': u'gGg/gAg', u'gene_symbol': u'BCL2', u'sift_score': 0.04, u'polyphen_score': 0.99, u'variant_allele': u'T', u'cds_end': 23, u'cds_start': 23, u'polyphen_prediction': u'probably_damaging', u'cdna_end': 922, u'amino_acids': u'G/E', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'deleterious', u'strand': -1, u'protein_end': 8}, {u'polyphen_score': 0.99, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1485, u'impact': u'MODERATE', u'protein_start': 8, u'codons': u'gGg/gAg', u'gene_symbol': u'BCL2', u'sift_score': 0.04, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 23, u'cds_start': 23, u'polyphen_prediction': u'probably_damaging', u'cdna_end': 1485, u'amino_acids': u'G/E', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'deleterious', u'strand': -1, u'protein_end': 8}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 516, u'impact': u'MODERATE', u'protein_start': 8, u'codons': u'gGg/gAg', u'sift_prediction': u'deleterious', u'gene_symbol': u'BCL2', u'sift_score': 0.02, u'polyphen_score': 0.994, u'variant_allele': u'T', u'cds_end': 23, u'cds_start': 23, u'polyphen_prediction': u'probably_damaging', u'cdna_end': 516, u'amino_acids': u'G/E', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 8}, {u'impact': u'MODIFIER', u'distance': 4562, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 169, u'cds_start': 23, u'protein_start': 8, u'codons': u'gGg/gAg', u'sift_prediction': u'deleterious', u'gene_symbol': u'BCL2', u'sift_score': 0.02, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 23, u'impact': u'MODERATE', u'polyphen_prediction': u'probably_damaging', u'cdna_end': 169, u'amino_acids': u'G/E', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'polyphen_score': 0.994, u'strand': -1, u'protein_end': 8}]\n",
      "[{u'polyphen_score': 0.994, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1285, u'impact': u'MODERATE', u'protein_start': 129, u'codons': u'cGc/cTc', u'gene_symbol': u'BCL2', u'sift_score': 0.01, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'A', u'cds_end': 386, u'cds_start': 386, u'polyphen_prediction': u'probably_damaging', u'cdna_end': 1285, u'amino_acids': u'R/L', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'deleterious', u'strand': -1, u'protein_end': 129}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1848, u'cds_start': 386, u'protein_start': 129, u'consequence_terms': [u'missense_variant'], u'codons': u'cGc/cTc', u'gene_symbol': u'BCL2', u'sift_score': 0.01, u'polyphen_score': 0.994, u'variant_allele': u'A', u'cds_end': 386, u'impact': u'MODERATE', u'polyphen_prediction': u'probably_damaging', u'cdna_end': 1848, u'amino_acids': u'R/L', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'deleterious', u'strand': -1, u'protein_end': 129}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 879, u'impact': u'MODERATE', u'protein_start': 129, u'consequence_terms': [u'missense_variant'], u'codons': u'cGc/cTc', u'gene_symbol': u'BCL2', u'sift_score': 0.02, u'polyphen_score': 0.994, u'variant_allele': u'A', u'cds_end': 386, u'cds_start': 386, u'polyphen_prediction': u'probably_damaging', u'cdna_end': 879, u'amino_acids': u'R/L', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'sift_prediction': u'deleterious', u'strand': -1, u'protein_end': 129}, {u'impact': u'MODIFIER', u'distance': 4199, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'polyphen_score': 0.994, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 532, u'impact': u'MODERATE', u'protein_start': 129, u'codons': u'cGc/cTc', u'gene_symbol': u'BCL2', u'sift_score': 0.02, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'A', u'cds_end': 386, u'cds_start': 386, u'polyphen_prediction': u'probably_damaging', u'cdna_end': 532, u'amino_acids': u'R/L', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'sift_prediction': u'deleterious', u'strand': -1, u'protein_end': 129}]\n",
      "[{u'polyphen_score': 0.536, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1300, u'impact': u'MODERATE', u'protein_start': 134, u'codons': u'gTg/gCg', u'gene_symbol': u'BCL2', u'sift_score': 0.26, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'G', u'cds_end': 401, u'cds_start': 401, u'polyphen_prediction': u'possibly_damaging', u'cdna_end': 1300, u'amino_acids': u'V/A', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 134}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1863, u'cds_start': 401, u'protein_start': 134, u'codons': u'gTg/gCg', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.26, u'polyphen_score': 0.536, u'variant_allele': u'G', u'cds_end': 401, u'impact': u'MODERATE', u'polyphen_prediction': u'possibly_damaging', u'cdna_end': 1863, u'amino_acids': u'V/A', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 134}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 894, u'cds_start': 401, u'protein_start': 134, u'codons': u'gTg/gCg', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.1, u'polyphen_score': 0.536, u'variant_allele': u'G', u'cds_end': 401, u'impact': u'MODERATE', u'polyphen_prediction': u'possibly_damaging', u'cdna_end': 894, u'amino_acids': u'V/A', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 134}, {u'impact': u'MODIFIER', u'distance': 4184, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'G', u'strand': -1}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 547, u'cds_start': 401, u'protein_start': 134, u'codons': u'gTg/gCg', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.1, u'polyphen_score': 0.536, u'variant_allele': u'G', u'cds_end': 401, u'impact': u'MODERATE', u'polyphen_prediction': u'possibly_damaging', u'cdna_end': 547, u'amino_acids': u'V/A', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 134}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1291, u'cds_start': 392, u'protein_start': 131, u'codons': u'gCc/gAc', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.25, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 392, u'impact': u'MODERATE', u'polyphen_prediction': u'possibly_damaging', u'cdna_end': 1291, u'amino_acids': u'A/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'polyphen_score': 0.786, u'strand': -1, u'protein_end': 131}, {u'polyphen_score': 0.786, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1854, u'impact': u'MODERATE', u'protein_start': 131, u'codons': u'gCc/gAc', u'gene_symbol': u'BCL2', u'sift_score': 0.25, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 392, u'cds_start': 392, u'polyphen_prediction': u'possibly_damaging', u'cdna_end': 1854, u'amino_acids': u'A/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 131}, {u'polyphen_score': 0.936, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 885, u'impact': u'MODERATE', u'protein_start': 131, u'codons': u'gCc/gAc', u'gene_symbol': u'BCL2', u'sift_score': 0.52, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 392, u'cds_start': 392, u'polyphen_prediction': u'probably_damaging', u'cdna_end': 885, u'amino_acids': u'A/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 131}, {u'impact': u'MODIFIER', u'distance': 4193, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 538, u'cds_start': 392, u'protein_start': 131, u'codons': u'gCc/gAc', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.52, u'polyphen_score': 0.936, u'variant_allele': u'T', u'cds_end': 392, u'impact': u'MODERATE', u'polyphen_prediction': u'probably_damaging', u'cdna_end': 538, u'amino_acids': u'A/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 131}]\n",
      "[{u'polyphen_score': 0.047, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1035, u'impact': u'MODERATE', u'protein_start': 46, u'codons': u'Ccg/Gcg', u'gene_symbol': u'BCL2', u'sift_score': 0.84, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 136, u'cds_start': 136, u'polyphen_prediction': u'benign', u'cdna_end': 1035, u'amino_acids': u'P/A', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 46}, {u'polyphen_score': 0.047, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1598, u'cds_start': 136, u'protein_start': 46, u'codons': u'Ccg/Gcg', u'gene_symbol': u'BCL2', u'sift_score': 0.84, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 136, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1598, u'amino_acids': u'P/A', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 46}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 629, u'cds_start': 136, u'protein_start': 46, u'codons': u'Ccg/Gcg', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.62, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 136, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 629, u'amino_acids': u'P/A', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'polyphen_score': 0.001, u'strand': -1, u'protein_end': 46}, {u'impact': u'MODIFIER', u'distance': 4449, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'C', u'strand': -1}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 282, u'cds_start': 136, u'protein_start': 46, u'codons': u'Ccg/Gcg', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.62, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 136, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 282, u'amino_acids': u'P/A', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'polyphen_score': 0.001, u'strand': -1, u'protein_end': 46}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 931, u'cds_start': 32, u'protein_start': 11, u'consequence_terms': [u'missense_variant'], u'codons': u'aAc/aCc', u'gene_symbol': u'BCL2', u'sift_score': 0.03, u'polyphen_score': 0.99, u'variant_allele': u'G', u'cds_end': 32, u'impact': u'MODERATE', u'polyphen_prediction': u'probably_damaging', u'cdna_end': 931, u'amino_acids': u'N/T', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'deleterious', u'strand': -1, u'protein_end': 11}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1494, u'impact': u'MODERATE', u'protein_start': 11, u'consequence_terms': [u'missense_variant'], u'codons': u'aAc/aCc', u'gene_symbol': u'BCL2', u'sift_score': 0.03, u'polyphen_score': 0.99, u'variant_allele': u'G', u'cds_end': 32, u'cds_start': 32, u'polyphen_prediction': u'probably_damaging', u'cdna_end': 1494, u'amino_acids': u'N/T', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'deleterious', u'strand': -1, u'protein_end': 11}, {u'polyphen_score': 0.985, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 525, u'cds_start': 32, u'protein_start': 11, u'codons': u'aAc/aCc', u'gene_symbol': u'BCL2', u'sift_score': 0.02, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'G', u'cds_end': 32, u'impact': u'MODERATE', u'polyphen_prediction': u'probably_damaging', u'cdna_end': 525, u'amino_acids': u'N/T', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'sift_prediction': u'deleterious', u'strand': -1, u'protein_end': 11}, {u'impact': u'MODIFIER', u'distance': 4553, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'G', u'strand': -1}, {u'polyphen_score': 0.985, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 178, u'cds_start': 32, u'protein_start': 11, u'codons': u'aAc/aCc', u'gene_symbol': u'BCL2', u'sift_score': 0.02, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'G', u'cds_end': 32, u'impact': u'MODERATE', u'polyphen_prediction': u'probably_damaging', u'cdna_end': 178, u'amino_acids': u'N/T', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'sift_prediction': u'deleterious', u'strand': -1, u'protein_end': 11}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1072, u'cds_start': 173, u'protein_start': 58, u'consequence_terms': [u'missense_variant'], u'codons': u'cAt/cCt', u'gene_symbol': u'BCL2', u'sift_score': 0.3, u'polyphen_score': 0, u'variant_allele': u'G', u'cds_end': 173, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1072, u'amino_acids': u'H/P', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 58}, {u'polyphen_score': 0, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1635, u'cds_start': 173, u'protein_start': 58, u'codons': u'cAt/cCt', u'gene_symbol': u'BCL2', u'sift_score': 0.3, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'G', u'cds_end': 173, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1635, u'amino_acids': u'H/P', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 58}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 666, u'impact': u'MODERATE', u'protein_start': 58, u'codons': u'cAt/cCt', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.38, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'G', u'cds_end': 173, u'cds_start': 173, u'polyphen_prediction': u'benign', u'cdna_end': 666, u'amino_acids': u'H/P', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'polyphen_score': 0, u'strand': -1, u'protein_end': 58}, {u'impact': u'MODIFIER', u'distance': 4412, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'G', u'strand': -1}, {u'polyphen_score': 0, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 319, u'cds_start': 173, u'protein_start': 58, u'codons': u'cAt/cCt', u'gene_symbol': u'BCL2', u'sift_score': 0.38, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'G', u'cds_end': 173, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 319, u'amino_acids': u'H/P', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 58}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1039, u'impact': u'MODERATE', u'protein_start': 47, u'codons': u'gGc/gAc', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.11, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 140, u'cds_start': 140, u'polyphen_prediction': u'benign', u'cdna_end': 1039, u'amino_acids': u'G/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'polyphen_score': 0.218, u'strand': -1, u'protein_end': 47}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1602, u'impact': u'MODERATE', u'protein_start': 47, u'codons': u'gGc/gAc', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.11, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 140, u'cds_start': 140, u'polyphen_prediction': u'benign', u'cdna_end': 1602, u'amino_acids': u'G/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'polyphen_score': 0.218, u'strand': -1, u'protein_end': 47}, {u'polyphen_score': 0.379, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 633, u'impact': u'MODERATE', u'protein_start': 47, u'codons': u'gGc/gAc', u'gene_symbol': u'BCL2', u'sift_score': 0.15, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 140, u'cds_start': 140, u'polyphen_prediction': u'benign', u'cdna_end': 633, u'amino_acids': u'G/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 47}, {u'impact': u'MODIFIER', u'distance': 4445, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'polyphen_score': 0.379, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 286, u'impact': u'MODERATE', u'protein_start': 47, u'codons': u'gGc/gAc', u'gene_symbol': u'BCL2', u'sift_score': 0.15, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 140, u'cds_start': 140, u'polyphen_prediction': u'benign', u'cdna_end': 286, u'amino_acids': u'G/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 47}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1006, u'cds_start': 107, u'protein_start': 36, u'codons': u'gGc/gAc', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.75, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 107, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1006, u'amino_acids': u'G/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'polyphen_score': 0, u'strand': -1, u'protein_end': 36}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1569, u'cds_start': 107, u'protein_start': 36, u'consequence_terms': [u'missense_variant'], u'codons': u'gGc/gAc', u'gene_symbol': u'BCL2', u'sift_score': 0.75, u'polyphen_score': 0, u'variant_allele': u'T', u'cds_end': 107, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1569, u'amino_acids': u'G/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 36}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 600, u'impact': u'MODERATE', u'protein_start': 36, u'codons': u'gGc/gAc', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.65, u'polyphen_score': 0.001, u'variant_allele': u'T', u'cds_end': 107, u'cds_start': 107, u'polyphen_prediction': u'benign', u'cdna_end': 600, u'amino_acids': u'G/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 36}, {u'impact': u'MODIFIER', u'distance': 4478, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 253, u'cds_start': 107, u'protein_start': 36, u'codons': u'gGc/gAc', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.65, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'T', u'cds_end': 107, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 253, u'amino_acids': u'G/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'polyphen_score': 0.001, u'strand': -1, u'protein_end': 36}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 986, u'cds_start': 87, u'protein_start': 29, u'codons': u'gaG/gaT', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.73, u'polyphen_score': 0.022, u'variant_allele': u'A', u'cds_end': 87, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 986, u'amino_acids': u'E/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 29}, {u'polyphen_score': 0.022, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1549, u'impact': u'MODERATE', u'protein_start': 29, u'codons': u'gaG/gaT', u'gene_symbol': u'BCL2', u'sift_score': 0.73, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'A', u'cds_end': 87, u'cds_start': 87, u'polyphen_prediction': u'benign', u'cdna_end': 1549, u'amino_acids': u'E/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 29}, {u'polyphen_score': 0.039, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 580, u'impact': u'MODERATE', u'protein_start': 29, u'codons': u'gaG/gaT', u'gene_symbol': u'BCL2', u'sift_score': 0.66, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'A', u'cds_end': 87, u'cds_start': 87, u'polyphen_prediction': u'benign', u'cdna_end': 580, u'amino_acids': u'E/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 29}, {u'impact': u'MODIFIER', u'distance': 4498, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'polyphen_score': 0.039, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 233, u'impact': u'MODERATE', u'protein_start': 29, u'codons': u'gaG/gaT', u'gene_symbol': u'BCL2', u'sift_score': 0.66, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'A', u'cds_end': 87, u'cds_start': 87, u'polyphen_prediction': u'benign', u'cdna_end': 233, u'amino_acids': u'E/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 29}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1073, u'impact': u'MODERATE', u'protein_start': 58, u'consequence_terms': [u'missense_variant'], u'codons': u'caT/caG', u'gene_symbol': u'BCL2', u'sift_score': 0.61, u'polyphen_score': 0, u'variant_allele': u'C', u'cds_end': 174, u'cds_start': 174, u'polyphen_prediction': u'benign', u'cdna_end': 1073, u'amino_acids': u'H/Q', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 58}, {u'polyphen_score': 0, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1636, u'impact': u'MODERATE', u'protein_start': 58, u'codons': u'caT/caG', u'gene_symbol': u'BCL2', u'sift_score': 0.61, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 174, u'cds_start': 174, u'polyphen_prediction': u'benign', u'cdna_end': 1636, u'amino_acids': u'H/Q', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 58}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 667, u'cds_start': 174, u'protein_start': 58, u'codons': u'caT/caG', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.66, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 174, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 667, u'amino_acids': u'H/Q', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'polyphen_score': 0, u'strand': -1, u'protein_end': 58}, {u'impact': u'MODIFIER', u'distance': 4411, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'C', u'strand': -1}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 320, u'cds_start': 174, u'protein_start': 58, u'codons': u'caT/caG', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.66, u'polyphen_score': 0, u'variant_allele': u'C', u'cds_end': 174, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 320, u'amino_acids': u'H/Q', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 58}]\n",
      "[{u'polyphen_score': 0.976, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1388, u'impact': u'MODERATE', u'protein_start': 163, u'codons': u'aaC/aaG', u'gene_symbol': u'BCL2', u'sift_score': 0.11, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 489, u'cds_start': 489, u'polyphen_prediction': u'probably_damaging', u'cdna_end': 1388, u'amino_acids': u'N/K', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 163}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1951, u'impact': u'MODERATE', u'protein_start': 163, u'consequence_terms': [u'missense_variant'], u'codons': u'aaC/aaG', u'gene_symbol': u'BCL2', u'sift_score': 0.11, u'polyphen_score': 0.976, u'variant_allele': u'C', u'cds_end': 489, u'cds_start': 489, u'polyphen_prediction': u'probably_damaging', u'cdna_end': 1951, u'amino_acids': u'N/K', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 163}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 982, u'cds_start': 489, u'protein_start': 163, u'consequence_terms': [u'missense_variant'], u'codons': u'aaC/aaG', u'gene_symbol': u'BCL2', u'sift_score': 0.1, u'polyphen_score': 0.941, u'variant_allele': u'C', u'cds_end': 489, u'impact': u'MODERATE', u'polyphen_prediction': u'probably_damaging', u'cdna_end': 982, u'amino_acids': u'N/K', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 163}, {u'impact': u'MODIFIER', u'distance': 4096, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'C', u'strand': -1}, {u'polyphen_score': 0.941, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 635, u'cds_start': 489, u'protein_start': 163, u'codons': u'aaC/aaG', u'gene_symbol': u'BCL2', u'sift_score': 0.1, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 489, u'impact': u'MODERATE', u'polyphen_prediction': u'probably_damaging', u'cdna_end': 635, u'amino_acids': u'N/K', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 163}]\n",
      "[{u'polyphen_score': 0.006, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 938, u'impact': u'MODERATE', u'protein_start': 13, u'codons': u'gaG/gaC', u'gene_symbol': u'BCL2', u'sift_score': 0.23, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'G', u'cds_end': 39, u'cds_start': 39, u'polyphen_prediction': u'benign', u'cdna_end': 938, u'amino_acids': u'E/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 13}, {u'polyphen_score': 0.006, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1501, u'impact': u'MODERATE', u'protein_start': 13, u'codons': u'gaG/gaC', u'gene_symbol': u'BCL2', u'sift_score': 0.23, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'G', u'cds_end': 39, u'cds_start': 39, u'polyphen_prediction': u'benign', u'cdna_end': 1501, u'amino_acids': u'E/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 13}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 532, u'cds_start': 39, u'protein_start': 13, u'codons': u'gaG/gaC', u'sift_prediction': u'tolerated', u'gene_symbol': u'BCL2', u'sift_score': 0.21, u'polyphen_score': 0.023, u'variant_allele': u'G', u'cds_end': 39, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 532, u'amino_acids': u'E/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'missense_variant'], u'strand': -1, u'protein_end': 13}, {u'impact': u'MODIFIER', u'distance': 4546, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'G', u'strand': -1}, {u'polyphen_score': 0.023, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 185, u'impact': u'MODERATE', u'protein_start': 13, u'codons': u'gaG/gaC', u'gene_symbol': u'BCL2', u'sift_score': 0.21, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'G', u'cds_end': 39, u'cds_start': 39, u'polyphen_prediction': u'benign', u'cdna_end': 185, u'amino_acids': u'E/D', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 13}]\n",
      "[{u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1074, u'cds_start': 175, u'protein_start': 59, u'consequence_terms': [u'missense_variant'], u'codons': u'Cca/Gca', u'gene_symbol': u'BCL2', u'sift_score': 0.28, u'polyphen_score': 0, u'variant_allele': u'C', u'cds_end': 175, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1074, u'amino_acids': u'P/A', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 59}, {u'polyphen_score': 0, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1637, u'cds_start': 175, u'protein_start': 59, u'codons': u'Cca/Gca', u'gene_symbol': u'BCL2', u'sift_score': 0.28, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 175, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 1637, u'amino_acids': u'P/A', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 59}, {u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 668, u'cds_start': 175, u'protein_start': 59, u'consequence_terms': [u'missense_variant'], u'codons': u'Cca/Gca', u'gene_symbol': u'BCL2', u'sift_score': 0.32, u'polyphen_score': 0.001, u'variant_allele': u'C', u'cds_end': 175, u'impact': u'MODERATE', u'polyphen_prediction': u'benign', u'cdna_end': 668, u'amino_acids': u'P/A', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 59}, {u'impact': u'MODIFIER', u'distance': 4410, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'C', u'strand': -1}, {u'polyphen_score': 0.001, u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 321, u'impact': u'MODERATE', u'protein_start': 59, u'codons': u'Cca/Gca', u'gene_symbol': u'BCL2', u'sift_score': 0.32, u'consequence_terms': [u'missense_variant'], u'variant_allele': u'C', u'cds_end': 175, u'cds_start': 175, u'polyphen_prediction': u'benign', u'cdna_end': 321, u'amino_acids': u'P/A', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'sift_prediction': u'tolerated', u'strand': -1, u'protein_end': 59}]\n",
      "g.chr18:60788979A>G IGR\n",
      "g.chr18:60787589T>C IGR\n",
      "g.chr18:60790273G>A IGR\n",
      "g.chr18:60987915T>C 5'Flank\n",
      "g.chr18:60988035C>G 5'Flank\n",
      "g.chr18:60988051G>C 5'Flank\n",
      "g.chr18:60988074C>T 5'Flank\n",
      "g.chr18:60988287T>C 5'Flank\n",
      "g.chr18:60988303T>A 5'Flank\n",
      "g.chr18:60988411C>T 5'Flank\n",
      "g.chr18:60988646A>G 5'Flank\n",
      "g.chr18:60988364T>C 5'Flank\n",
      "g.chr18:60987807C>T 5'Flank\n",
      "g.chr18:60988038T>A 5'Flank\n",
      "g.chr18:60988051G>T 5'Flank\n",
      "g.chr18:60987989C>G 5'Flank\n",
      "g.chr18:60988054C>T 5'Flank\n",
      "g.chr18:60988203C>G 5'Flank\n",
      "g.chr18:60988352T>C 5'Flank\n",
      "g.chr18:60988372T>C 5'Flank\n",
      "g.chr18:60988451A>G 5'Flank\n",
      "g.chr18:60988547A>C 5'Flank\n",
      "g.chr18:60988554A>T 5'Flank\n",
      "g.chr18:60987879C>G 5'Flank\n",
      "g.chr18:60987916T>A 5'Flank\n",
      "g.chr18:60988051G>A 5'Flank\n",
      "g.chr18:60988199A>G 5'Flank\n",
      "g.chr18:60988240G>C 5'Flank\n",
      "g.chr18:60988291C>T 5'Flank\n",
      "g.chr18:60988293C>T 5'Flank\n",
      "g.chr18:60987893G>A 5'Flank\n",
      "g.chr18:60987916T>C 5'Flank\n",
      "g.chr18:60987999A>G 5'Flank\n",
      "g.chr18:60988080G>T 5'Flank\n",
      "g.chr18:60988139A>C 5'Flank\n",
      "g.chr18:60988173A>G 5'Flank\n",
      "g.chr18:60988228T>C 5'Flank\n",
      "g.chr18:60988230C>G 5'Flank\n",
      "g.chr18:60988240G>C 5'Flank\n",
      "g.chr18:60988250T>A 5'Flank\n",
      "g.chr18:60988292A>T 5'Flank\n",
      "g.chr18:60988376A>C 5'Flank\n",
      "g.chr18:60988486G>A 5'Flank\n",
      "g.chr18:60988652A>G 5'Flank\n",
      "g.chr18:60988066C>A 5'Flank\n",
      "g.chr18:60988075G>A 5'Flank\n",
      "g.chr18:60988087G>C 5'Flank\n",
      "g.chr18:60988089G>C 5'Flank\n",
      "g.chr18:60988106A>G 5'Flank\n",
      "g.chr18:60988108A>T 5'Flank\n",
      "g.chr18:60988140G>A 5'Flank\n",
      "g.chr18:60988155C>T 5'Flank\n",
      "g.chr18:60988157C>G 5'Flank\n",
      "g.chr18:60988160C>T 5'Flank\n",
      "g.chr18:60988189G>A 5'Flank\n",
      "g.chr18:60988225G>A 5'Flank\n",
      "g.chr18:60988240G>C 5'Flank\n",
      "g.chr18:60988448G>C 5'Flank\n",
      "g.chr18:60988488G>A 5'Flank\n",
      "g.chr18:60988552G>A 5'Flank\n",
      "g.chr18:60987894C>T 5'Flank\n",
      "g.chr18:60987900G>A 5'Flank\n",
      "g.chr18:60988051G>A 5'Flank\n",
      "g.chr18:60988066C>T 5'Flank\n",
      "g.chr18:60988072C>A 5'Flank\n",
      "g.chr18:60988084C>T 5'Flank\n",
      "g.chr18:60988197C>T 5'Flank\n",
      "g.chr18:60988228T>A 5'Flank\n",
      "g.chr18:60988270A>G 5'Flank\n",
      "g.chr18:60988279C>T 5'Flank\n",
      "g.chr18:60987894C>G 5'Flank\n",
      "g.chr18:60987956T>A 5'Flank\n",
      "g.chr18:60988078G>C 5'Flank\n",
      "g.chr18:60988087G>C 5'Flank\n",
      "g.chr18:60988105A>G 5'Flank\n",
      "g.chr18:60988229A>G 5'Flank\n",
      "g.chr18:60988235A>T 5'Flank\n",
      "g.chr18:60987884G>A 5'Flank\n",
      "g.chr18:60987999A>G 5'Flank\n",
      "g.chr18:60988041A>T 5'Flank\n",
      "g.chr18:60988084C>T 5'Flank\n",
      "g.chr18:60988210C>T 5'Flank\n",
      "g.chr18:60988278C>G 5'Flank\n",
      "g.chr18:60988291C>T 5'Flank\n",
      "g.chr18:60988381A>T 5'Flank\n",
      "g.chr18:60988383G>T 5'Flank\n",
      "g.chr18:60988084C>T 5'Flank\n",
      "g.chr18:60988119T>G 5'Flank\n",
      "g.chr18:60988198A>T 5'Flank\n",
      "g.chr18:60988230C>G 5'Flank\n",
      "g.chr18:60988237G>C 5'Flank\n",
      "g.chr18:60988246T>C 5'Flank\n",
      "g.chr18:60987884G>A 5'Flank\n",
      "g.chr18:60987963G>T 5'Flank\n",
      "g.chr18:60987997G>A 5'Flank\n",
      "g.chr18:60988075G>A 5'Flank\n",
      "g.chr18:60988084C>T 5'Flank\n",
      "g.chr18:60988171A>G 5'Flank\n",
      "g.chr18:60988381A>C 5'Flank\n",
      "g.chr18:60987913C>T 5'Flank\n",
      "g.chr18:60987993C>T 5'Flank\n",
      "g.chr18:60988239A>G 5'Flank\n",
      "g.chr18:60988299C>A 5'Flank\n",
      "g.chr18:60988649A>G 5'Flank\n",
      "g.chr18:60987913C>T 5'Flank\n",
      "g.chr18:60987979T>G 5'Flank\n",
      "g.chr18:60988065A>G 5'Flank\n",
      "g.chr18:60988083A>G 5'Flank\n",
      "g.chr18:60988165C>G 5'Flank\n",
      "g.chr18:60988225G>A 5'Flank\n",
      "g.chr18:60988228T>A 5'Flank\n",
      "g.chr18:60988240G>A 5'Flank\n",
      "g.chr18:60988242T>G 5'Flank\n",
      "g.chr18:60988321A>C 5'Flank\n",
      "g.chr18:60988340A>G 5'Flank\n",
      "g.chr18:60988353A>T 5'Flank\n",
      "g.chr18:60987841C>A 5'Flank\n",
      "g.chr18:60987903C>G 5'Flank\n",
      "g.chr18:60988028T>C 5'Flank\n",
      "g.chr18:60988066C>A 5'Flank\n",
      "g.chr18:60988068C>A 5'Flank\n",
      "g.chr18:60988070C>G 5'Flank\n",
      "g.chr18:60988074C>A 5'Flank\n",
      "g.chr18:60988084C>T 5'Flank\n",
      "g.chr18:60988189G>T 5'Flank\n",
      "g.chr18:60988241C>G 5'Flank\n",
      "g.chr18:60988519C>T 5'Flank\n",
      "g.chr18:60987993C>T 5'Flank\n",
      "g.chr18:60987997G>C 5'Flank\n",
      "g.chr18:60988005G>A 5'Flank\n",
      "g.chr18:60988134C>G 5'Flank\n",
      "g.chr18:60988150C>G 5'Flank\n",
      "g.chr18:60988157C>G 5'Flank\n",
      "g.chr18:60988230C>G 5'Flank\n",
      "g.chr18:60988535G>A 5'Flank\n",
      "g.chr18:60987913C>A 5'Flank\n",
      "g.chr18:60987918C>G 5'Flank\n",
      "g.chr18:60987984C>A 5'Flank\n",
      "g.chr18:60987999A>T 5'Flank\n",
      "g.chr18:60988011G>T 5'Flank\n",
      "g.chr18:60988043G>C 5'Flank\n",
      "g.chr18:60988045G>A 5'Flank\n",
      "g.chr18:60988156G>A 5'Flank\n",
      "g.chr18:60988201G>C 5'Flank\n",
      "g.chr18:60988230C>G 5'Flank\n",
      "g.chr18:60988119T>C 5'Flank\n",
      "g.chr18:60988189G>A 5'Flank\n",
      "g.chr18:60987999A>T 5'Flank\n",
      "g.chr18:60988022G>A 5'Flank\n",
      "g.chr18:60988035C>G 5'Flank\n",
      "g.chr18:60988051G>A 5'Flank\n",
      "g.chr18:60988059G>A 5'Flank\n",
      "g.chr18:60988082T>C 5'Flank\n",
      "g.chr18:60988139A>C 5'Flank\n",
      "g.chr18:60988230C>G 5'Flank\n",
      "g.chr18:60988298A>G 5'Flank\n",
      "g.chr18:60988353A>T 5'Flank\n",
      "g.chr18:60988426A>C 5'Flank\n",
      "g.chr18:60988130A>C 5'Flank\n",
      "g.chr18:60988278C>T 5'Flank\n",
      "g.chr18:60987924G>C 5'Flank\n",
      "g.chr18:60987993C>G 5'Flank\n",
      "g.chr18:60988035C>A 5'Flank\n",
      "g.chr18:60988052T>A 5'Flank\n",
      "g.chr18:60988194A>G 5'Flank\n",
      "g.chr18:60987836G>A 5'Flank\n",
      "g.chr18:60987838G>A 5'Flank\n",
      "g.chr18:60987879C>T 5'Flank\n",
      "g.chr18:60987894C>T 5'Flank\n",
      "g.chr18:60987918C>T 5'Flank\n",
      "g.chr18:60987930G>A 5'Flank\n",
      "g.chr18:60987980G>A 5'Flank\n",
      "g.chr18:60987993C>T 5'Flank\n",
      "g.chr18:60988013G>T 5'Flank\n",
      "g.chr18:60988020C>T 5'Flank\n",
      "g.chr18:60988026C>T 5'Flank\n",
      "g.chr18:60988045G>A 5'Flank\n",
      "g.chr18:60988051G>C 5'Flank\n",
      "g.chr18:60988066C>T 5'Flank\n",
      "g.chr18:60988080G>A 5'Flank\n",
      "g.chr18:60988084C>G 5'Flank\n",
      "g.chr18:60988170A>G 5'Flank\n",
      "g.chr18:60988186C>T 5'Flank\n",
      "g.chr18:60988189G>A 5'Flank\n",
      "g.chr18:60988195A>G 5'Flank\n",
      "g.chr18:60988201G>A 5'Flank\n",
      "g.chr18:60988222G>T 5'Flank\n",
      "g.chr18:60988225G>A 5'Flank\n",
      "g.chr18:60988302C>T 5'Flank\n",
      "g.chr18:60988368A>G 5'Flank\n",
      "g.chr18:60988376A>G 5'Flank\n",
      "g.chr18:60988416G>A 5'Flank\n",
      "g.chr18:60988450A>C 5'Flank\n",
      "g.chr18:60988756A>G 5'Flank\n",
      "g.chr18:60988772A>G 5'Flank\n",
      "g.chr18:60987978C>G 5'Flank\n",
      "g.chr18:60987993C>T 5'Flank\n",
      "g.chr18:60988009G>C 5'Flank\n",
      "g.chr18:60988031C>G 5'Flank\n",
      "g.chr18:60988041A>T 5'Flank\n",
      "g.chr18:60988062T>C 5'Flank\n",
      "g.chr18:60988072C>G 5'Flank\n",
      "g.chr18:60988118A>G 5'Flank\n",
      "g.chr18:60988225G>C 5'Flank\n",
      "g.chr18:60988240G>C 5'Flank\n",
      "g.chr18:60988349G>A 5'Flank\n",
      "g.chr18:60987961G>T 5'Flank\n",
      "g.chr18:60987997G>C 5'Flank\n",
      "g.chr18:60988041A>C 5'Flank\n",
      "g.chr18:60988119T>C 5'Flank\n",
      "g.chr18:60988240G>A 5'Flank\n",
      "g.chr18:60988279C>T 5'Flank\n",
      "g.chr18:60988688T>C 5'Flank\n",
      "g.chr18:60988739G>C 5'Flank\n",
      "g.chr18:60988815A>C 5'Flank\n",
      "g.chr18:60988821T>C 5'Flank\n",
      "g.chr18:60988834A>T 5'Flank\n",
      "g.chr18:60988858T>G 5'Flank\n",
      "g.chr18:60987903C>G 5'Flank\n",
      "g.chr18:60988119T>C 5'Flank\n",
      "g.chr18:60988193T>C 5'Flank\n",
      "g.chr18:60988228T>C 5'Flank\n",
      "g.chr18:60988230C>G 5'Flank\n",
      "g.chr18:60988241C>T 5'Flank\n",
      "g.chr18:60988245C>T 5'Flank\n",
      "g.chr18:60988279C>A 5'Flank\n",
      "g.chr18:60988299C>G 5'Flank\n",
      "g.chr18:60988400A>G 5'Flank\n",
      "g.chr18:60988423G>A 5'Flank\n",
      "g.chr18:60988537A>G 5'Flank\n",
      "g.chr18:60987853G>A 5'Flank\n",
      "g.chr18:60988028T>C 5'Flank\n",
      "g.chr18:60988035C>T 5'Flank\n",
      "g.chr18:60988040T>A 5'Flank\n",
      "g.chr18:60988080G>A 5'Flank\n",
      "g.chr18:60988082T>G 5'Flank\n",
      "g.chr18:60988230C>T 5'Flank\n",
      "g.chr18:60988401A>G 5'Flank\n",
      "g.chr18:60988012T>A 5'Flank\n",
      "g.chr18:60988062T>C 5'Flank\n",
      "g.chr18:60988084C>T 5'Flank\n",
      "g.chr18:60988087G>A 5'Flank\n",
      "g.chr18:60988035C>G 5'Flank\n",
      "g.chr18:60988051G>A 5'Flank\n",
      "g.chr18:60988089G>C 5'Flank\n",
      "g.chr18:60988131C>G 5'Flank\n",
      "g.chr18:60988141C>T 5'Flank\n",
      "g.chr18:60988228T>A 5'Flank\n",
      "g.chr18:60988258C>T 5'Flank\n",
      "g.chr18:60988299C>T 5'Flank\n",
      "g.chr18:60988302C>T 5'Flank\n",
      "g.chr18:60988357A>G 5'Flank\n",
      "g.chr18:60988134C>G 5'Flank\n",
      "g.chr18:60988165C>T 5'Flank\n",
      "g.chr18:60988205A>G 5'Flank\n",
      "g.chr18:60988424C>G 5'Flank\n",
      "g.chr18:60988444T>A 5'Flank\n",
      "g.chr18:60987879C>T 5'Flank\n",
      "g.chr18:60988085C>T 5'Flank\n",
      "g.chr18:60988164G>A 5'Flank\n",
      "g.chr18:60988180A>G 5'Flank\n",
      "g.chr18:60988189G>A 5'Flank\n",
      "g.chr18:60988349G>A 5'Flank\n",
      "g.chr18:60988726A>C 5'Flank\n",
      "g.chr18:60988066C>G 5'Flank\n",
      "g.chr18:60988229A>T 5'Flank\n",
      "g.chr18:60988053A>T 5'Flank\n",
      "g.chr18:60988291C>T 5'Flank\n",
      "g.chr18:60987916T>C 5'Flank\n",
      "g.chr18:60988032T>C 5'Flank\n",
      "g.chr18:60988062T>G 5'Flank\n",
      "g.chr18:60988066C>T 5'Flank\n",
      "g.chr18:60988068C>A 5'Flank\n",
      "g.chr18:60988070C>G 5'Flank\n",
      "g.chr18:60988088C>G 5'Flank\n",
      "g.chr18:60988091T>G 5'Flank\n",
      "g.chr18:60988179G>A 5'Flank\n",
      "g.chr18:60988197C>T 5'Flank\n",
      "g.chr18:60987989C>G 5'Flank\n",
      "g.chr18:60987993C>T 5'Flank\n",
      "g.chr18:60988022G>A 5'Flank\n",
      "g.chr18:60988105A>C 5'Flank\n",
      "g.chr18:60988779A>T 5'Flank\n",
      "g.chr18:60987900G>A 5'Flank\n",
      "g.chr18:60988035C>T 5'Flank\n",
      "g.chr18:60988076T>G 5'Flank\n",
      "g.chr18:60988085C>T 5'Flank\n",
      "g.chr18:60988241C>A 5'Flank\n",
      "g.chr18:60988268G>A 5'Flank\n",
      "g.chr18:60988416G>A 5'Flank\n",
      "g.chr18:60988540T>G 5'Flank\n",
      "g.chr18:60988752C>T 5'Flank\n",
      "g.chr18:60988825A>C 5'Flank\n",
      "g.chr18:60987926G>A 5'Flank\n",
      "g.chr18:60988031C>T 5'Flank\n",
      "g.chr18:60988070C>G 5'Flank\n",
      "g.chr18:60988082T>A 5'Flank\n",
      "g.chr18:60988087G>A 5'Flank\n",
      "g.chr18:60988475C>G 5'Flank\n",
      "g.chr18:60988121C>T 5'Flank\n",
      "g.chr18:60987880A>C 5'Flank\n",
      "g.chr18:60987986T>G 5'Flank\n",
      "g.chr18:60987991A>G 5'Flank\n",
      "g.chr18:60988035C>T 5'Flank\n",
      "g.chr18:60988074C>T 5'Flank\n",
      "g.chr18:60988160C>T 5'Flank\n",
      "g.chr18:60988240G>C 5'Flank\n",
      "g.chr18:60988259A>G 5'Flank\n",
      "g.chr18:60988291C>G 5'Flank\n",
      "g.chr18:60987893G>C 5'Flank\n",
      "g.chr18:60987975C>G 5'Flank\n",
      "g.chr18:60987993C>T 5'Flank\n",
      "g.chr18:60988031C>T 5'Flank\n",
      "g.chr18:60988084C>A 5'Flank\n",
      "g.chr18:60988119T>A 5'Flank\n",
      "g.chr18:60988219C>T 5'Flank\n",
      "g.chr18:60988261T>A 5'Flank\n",
      "g.chr18:60989272G>C 5'Flank\n",
      "g.chr18:60987993C>T 5'Flank\n",
      "g.chr18:60987996T>C 5'Flank\n",
      "g.chr18:60988023G>A 5'Flank\n",
      "g.chr18:60988031C>G 5'Flank\n",
      "g.chr18:60988035C>T 5'Flank\n",
      "g.chr18:60988051G>A 5'Flank\n",
      "g.chr18:60988074C>G 5'Flank\n",
      "g.chr18:60988080G>A 5'Flank\n",
      "g.chr18:60988084C>T 5'Flank\n",
      "g.chr18:60988191T>C 5'Flank\n",
      "g.chr18:60988259A>T 5'Flank\n",
      "g.chr18:60988265C>G 5'Flank\n",
      "g.chr18:60988288G>T 5'Flank\n",
      "g.chr18:60988444T>G 5'Flank\n",
      "g.chr18:60988451A>T 5'Flank\n",
      "g.chr18:60986212A>C De_novo_Start_InFrame\n",
      "g.chr18:60986138G>A De_novo_Start_InFrame\n",
      "g.chr18:60986140T>C De_novo_Start_InFrame\n",
      "g.chr18:60986138G>A De_novo_Start_InFrame\n",
      "g.chr18:60986154A>G 5'UTR\n",
      "g.chr18:60986235T>C 5'UTR\n",
      "g.chr18:60986822G>A 5'UTR\n",
      "g.chr18:60986651G>C 5'UTR\n",
      "g.chr18:60986567A>G 5'UTR\n",
      "g.chr18:60986895C>T 5'UTR\n",
      "g.chr18:60986906T>G 5'UTR\n",
      "g.chr18:60987048T>C 5'UTR\n",
      "g.chr18:60987060T>A 5'UTR\n",
      "g.chr18:60987284G>A 5'UTR\n",
      "g.chr18:60987040C>T 5'UTR\n",
      "g.chr18:60987042C>T 5'UTR\n",
      "g.chr18:60986948G>A 5'UTR\n",
      "g.chr18:60985917C>T 5'UTR\n",
      "g.chr18:60986069T>C 5'UTR\n",
      "g.chr18:60986195T>A 5'UTR\n",
      "g.chr18:60986318T>C 5'UTR\n",
      "g.chr18:60986339T>A 5'UTR\n",
      "g.chr18:60986377T>G 5'UTR\n",
      "g.chr18:60986389C>A 5'UTR\n",
      "g.chr18:60986410T>G 5'UTR\n",
      "g.chr18:60986430A>G 5'UTR\n",
      "g.chr18:60986433C>T 5'UTR\n",
      "g.chr18:60986449T>C 5'UTR\n",
      "g.chr18:60986452T>C 5'UTR\n",
      "g.chr18:60986454T>C 5'UTR\n",
      "g.chr18:60986469T>A 5'UTR\n",
      "g.chr18:60986509G>A 5'UTR\n",
      "g.chr18:60986525A>G 5'UTR\n",
      "g.chr18:60986805T>C 5'UTR\n",
      "g.chr18:60986915A>C 5'UTR\n",
      "g.chr18:60986960C>G 5'UTR\n",
      "g.chr18:60986987T>A 5'UTR\n",
      "g.chr18:60987036T>C 5'UTR\n",
      "g.chr18:60985974G>A 5'UTR\n",
      "g.chr18:60986192G>C 5'UTR\n",
      "g.chr18:60986386T>C 5'UTR\n",
      "g.chr18:60987047T>C 5'UTR\n",
      "g.chr18:60985905C>T 5'UTR\n",
      "g.chr18:60986124A>G 5'UTR\n",
      "g.chr18:60986138G>C 5'UTR\n",
      "g.chr18:60986192G>C 5'UTR\n",
      "g.chr18:60986204A>T 5'UTR\n",
      "g.chr18:60986255A>G 5'UTR\n",
      "g.chr18:60986382C>T 5'UTR\n",
      "g.chr18:60986420G>C 5'UTR\n",
      "g.chr18:60986423C>T 5'UTR\n",
      "g.chr18:60986709A>C 5'UTR\n",
      "g.chr18:60986906T>G 5'UTR\n",
      "g.chr18:60986908T>A 5'UTR\n",
      "g.chr18:60986968T>C 5'UTR\n",
      "g.chr18:60986420G>A 5'UTR\n",
      "g.chr18:60986652C>T 5'UTR\n",
      "g.chr18:60985981A>G 5'UTR\n",
      "g.chr18:60985993G>C 5'UTR\n",
      "g.chr18:60986141A>T 5'UTR\n",
      "g.chr18:60986215A>G 5'UTR\n",
      "g.chr18:60986304A>G 5'UTR\n",
      "g.chr18:60986331A>C 5'UTR\n",
      "g.chr18:60986339T>G 5'UTR\n",
      "g.chr18:60986477G>C 5'UTR\n",
      "g.chr18:60986516C>T 5'UTR\n",
      "g.chr18:60986706G>A 5'UTR\n",
      "g.chr18:60986740C>G 5'UTR\n",
      "g.chr18:60986750T>G 5'UTR\n",
      "g.chr18:60986916A>G 5'UTR\n",
      "g.chr18:60986935G>C 5'UTR\n",
      "g.chr18:60987030T>C 5'UTR\n",
      "g.chr18:60985916G>A 5'UTR\n",
      "g.chr18:60986002T>G 5'UTR\n",
      "g.chr18:60986142T>C 5'UTR\n",
      "g.chr18:60986144A>T 5'UTR\n",
      "g.chr18:60986154A>G 5'UTR\n",
      "g.chr18:60986163A>G 5'UTR\n",
      "g.chr18:60986167G>A 5'UTR\n",
      "g.chr18:60986188G>C 5'UTR\n",
      "g.chr18:60986209A>G 5'UTR\n",
      "g.chr18:60986266T>C 5'UTR\n",
      "g.chr18:60986425G>A 5'UTR\n",
      "g.chr18:60986514A>T 5'UTR\n",
      "g.chr18:60986516C>G 5'UTR\n",
      "g.chr18:60986523G>C 5'UTR\n",
      "g.chr18:60986581C>G 5'UTR\n",
      "g.chr18:60986637C>T 5'UTR\n",
      "g.chr18:60986720T>G 5'UTR\n",
      "g.chr18:60986744G>A 5'UTR\n",
      "g.chr18:60986803G>A 5'UTR\n",
      "g.chr18:60986836A>C 5'UTR\n",
      "g.chr18:60986142T>A 5'UTR\n",
      "g.chr18:60986300A>C 5'UTR\n",
      "g.chr18:60986321A>G 5'UTR\n",
      "g.chr18:60986374A>T 5'UTR\n",
      "g.chr18:60986426C>A 5'UTR\n",
      "g.chr18:60986515G>A 5'UTR\n",
      "g.chr18:60986652C>T 5'UTR\n",
      "g.chr18:60986794C>T 5'UTR\n",
      "g.chr18:60986918C>G 5'UTR\n",
      "g.chr18:60986979G>A 5'UTR\n",
      "g.chr18:60987041C>G 5'UTR\n",
      "g.chr18:60986252G>A 5'UTR\n",
      "g.chr18:60986286G>T 5'UTR\n",
      "g.chr18:60986296A>G 5'UTR\n",
      "g.chr18:60986300A>T 5'UTR\n",
      "g.chr18:60986731T>C 5'UTR\n",
      "g.chr18:60985948C>T 5'UTR\n",
      "g.chr18:60986420G>C 5'UTR\n",
      "g.chr18:60986699T>C 5'UTR\n",
      "g.chr18:60986709A>T 5'UTR\n",
      "g.chr18:60986923G>A 5'UTR\n",
      "g.chr18:60985916G>T 5'UTR\n",
      "g.chr18:60986189A>G 5'UTR\n",
      "g.chr18:60986205A>G 5'UTR\n",
      "g.chr18:60986243A>C 5'UTR\n",
      "g.chr18:60986386T>A 5'UTR\n",
      "g.chr18:60986444T>C 5'UTR\n",
      "g.chr18:60986604T>C 5'UTR\n",
      "g.chr18:60986888C>T 5'UTR\n",
      "g.chr18:60985914A>T 5'UTR\n",
      "g.chr18:60985916G>C 5'UTR\n",
      "g.chr18:60986015G>C 5'UTR\n",
      "g.chr18:60986107A>C 5'UTR\n",
      "g.chr18:60986124A>T 5'UTR\n",
      "g.chr18:60986129A>T 5'UTR\n",
      "g.chr18:60986144A>T 5'UTR\n",
      "g.chr18:60986153T>G 5'UTR\n",
      "g.chr18:60986171A>T 5'UTR\n",
      "g.chr18:60986202A>G 5'UTR\n",
      "g.chr18:60986226C>A 5'UTR\n",
      "g.chr18:60986257A>G 5'UTR\n",
      "g.chr18:60986277A>G 5'UTR\n",
      "g.chr18:60986299T>A 5'UTR\n",
      "g.chr18:60986337T>C 5'UTR\n",
      "g.chr18:60986421T>A 5'UTR\n",
      "g.chr18:60986425G>A 5'UTR\n",
      "g.chr18:60986652C>T 5'UTR\n",
      "g.chr18:60986729G>A 5'UTR\n",
      "g.chr18:60986746A>T 5'UTR\n",
      "g.chr18:60986852G>T 5'UTR\n",
      "g.chr18:60986874C>G 5'UTR\n",
      "g.chr18:60986878T>C 5'UTR\n",
      "g.chr18:60986888C>T 5'UTR\n",
      "g.chr18:60986924G>T 5'UTR\n",
      "g.chr18:60987032T>A 5'UTR\n",
      "g.chr18:60987060T>G 5'UTR\n",
      "g.chr18:60987150C>G 5'UTR\n",
      "g.chr18:60987177T>G 5'UTR\n",
      "g.chr18:60985916G>A 5'UTR\n",
      "g.chr18:60986218T>C 5'UTR\n",
      "g.chr18:60986266T>A 5'UTR\n",
      "g.chr18:60986279T>G 5'UTR\n",
      "g.chr18:60986299T>A 5'UTR\n",
      "g.chr18:60986319A>C 5'UTR\n",
      "g.chr18:60986322T>G 5'UTR\n",
      "g.chr18:60986342G>A 5'UTR\n",
      "g.chr18:60986446C>A 5'UTR\n",
      "g.chr18:60986527C>G 5'UTR\n",
      "g.chr18:60986793C>T 5'UTR\n",
      "g.chr18:60986935G>C 5'UTR\n",
      "g.chr18:60986980C>A 5'UTR\n",
      "g.chr18:60987047T>G 5'UTR\n",
      "g.chr18:60987060T>A 5'UTR\n",
      "g.chr18:60987195C>T 5'UTR\n",
      "g.chr18:60986516C>G 5'UTR\n",
      "g.chr18:60986744G>A 5'UTR\n",
      "g.chr18:60986892G>A 5'UTR\n",
      "g.chr18:60985917C>A 5'UTR\n",
      "g.chr18:60986053C>G 5'UTR\n",
      "g.chr18:60986152G>A 5'UTR\n",
      "g.chr18:60986170T>A 5'UTR\n",
      "g.chr18:60986195T>A 5'UTR\n",
      "g.chr18:60986279T>G 5'UTR\n",
      "g.chr18:60986526A>G 5'UTR\n",
      "g.chr18:60986917G>A 5'UTR\n",
      "g.chr18:60985916G>A 5'UTR\n",
      "g.chr18:60986039T>G 5'UTR\n",
      "g.chr18:60986173T>C 5'UTR\n",
      "g.chr18:60986228G>A 5'UTR\n",
      "g.chr18:60986278G>C 5'UTR\n",
      "g.chr18:60986296A>C 5'UTR\n",
      "g.chr18:60986311T>A 5'UTR\n",
      "g.chr18:60986331A>C 5'UTR\n",
      "g.chr18:60986334C>G 5'UTR\n",
      "g.chr18:60986382C>T 5'UTR\n",
      "g.chr18:60986387T>A 5'UTR\n",
      "g.chr18:60986516C>G 5'UTR\n",
      "g.chr18:60986573A>G 5'UTR\n",
      "g.chr18:60986614T>G 5'UTR\n",
      "g.chr18:60986697T>C 5'UTR\n",
      "g.chr18:60986708T>G 5'UTR\n",
      "g.chr18:60986805T>C 5'UTR\n",
      "g.chr18:60986874C>T 5'UTR\n",
      "g.chr18:60986877T>C 5'UTR\n",
      "g.chr18:60986888C>G 5'UTR\n",
      "g.chr18:60986910C>G 5'UTR\n",
      "g.chr18:60986926T>C 5'UTR\n",
      "g.chr18:60986935G>A 5'UTR\n",
      "g.chr18:60986968T>C 5'UTR\n",
      "g.chr18:60987049C>T 5'UTR\n",
      "g.chr18:60985916G>A 5'UTR\n",
      "g.chr18:60985997T>C 5'UTR\n",
      "g.chr18:60986113T>C 5'UTR\n",
      "g.chr18:60986163A>G 5'UTR\n",
      "g.chr18:60986561G>A 5'UTR\n",
      "g.chr18:60986630T>C 5'UTR\n",
      "g.chr18:60986982G>A 5'UTR\n",
      "g.chr18:60985982G>A 5'UTR\n",
      "g.chr18:60986069T>C 5'UTR\n",
      "g.chr18:60986141A>C 5'UTR\n",
      "g.chr18:60986152G>C 5'UTR\n",
      "g.chr18:60986159T>G 5'UTR\n",
      "g.chr18:60986205A>C 5'UTR\n",
      "g.chr18:60986252G>A 5'UTR\n",
      "g.chr18:60986280A>G 5'UTR\n",
      "g.chr18:60986351C>G 5'UTR\n",
      "g.chr18:60986396A>G 5'UTR\n",
      "g.chr18:60986423C>T 5'UTR\n",
      "g.chr18:60986471C>A 5'UTR\n",
      "g.chr18:60986651G>C 5'UTR\n",
      "g.chr18:60986710T>A 5'UTR\n",
      "g.chr18:60986715A>G 5'UTR\n",
      "g.chr18:60986853C>T 5'UTR\n",
      "g.chr18:60986935G>C 5'UTR\n",
      "g.chr18:60986960C>G 5'UTR\n",
      "g.chr18:60986987T>A 5'UTR\n",
      "g.chr18:60987005G>A 5'UTR\n",
      "g.chr18:60987192G>C 5'UTR\n",
      "g.chr18:60986097T>C 5'UTR\n",
      "g.chr18:60986150A>C 5'UTR\n",
      "g.chr18:60986170T>C 5'UTR\n",
      "g.chr18:60986206C>G 5'UTR\n",
      "g.chr18:60986268C>G 5'UTR\n",
      "g.chr18:60986387T>C 5'UTR\n",
      "g.chr18:60986443T>A 5'UTR\n",
      "g.chr18:60986460C>G 5'UTR\n",
      "g.chr18:60986578T>A 5'UTR\n",
      "g.chr18:60986651G>A 5'UTR\n",
      "g.chr18:60986691T>G 5'UTR\n",
      "g.chr18:60986888C>G 5'UTR\n",
      "g.chr18:60987031T>C 5'UTR\n",
      "g.chr18:60987040C>T 5'UTR\n",
      "g.chr18:60985976A>G 5'UTR\n",
      "g.chr18:60986104C>T 5'UTR\n",
      "g.chr18:60986114T>C 5'UTR\n",
      "g.chr18:60986226C>T 5'UTR\n",
      "g.chr18:60987017G>A 5'UTR\n",
      "g.chr18:60985916G>A 5'UTR\n",
      "g.chr18:60985982G>A 5'UTR\n",
      "g.chr18:60986001A>C 5'UTR\n",
      "g.chr18:60986011C>T 5'UTR\n",
      "g.chr18:60986053C>T 5'UTR\n",
      "g.chr18:60986082T>C 5'UTR\n",
      "g.chr18:60986098T>C 5'UTR\n",
      "g.chr18:60986102T>C 5'UTR\n",
      "g.chr18:60986146T>C 5'UTR\n",
      "g.chr18:60986153T>A 5'UTR\n",
      "g.chr18:60986155T>A 5'UTR\n",
      "g.chr18:60986162T>A 5'UTR\n",
      "g.chr18:60986224A>T 5'UTR\n",
      "g.chr18:60986278G>A 5'UTR\n",
      "g.chr18:60986321A>G 5'UTR\n",
      "g.chr18:60986332T>C 5'UTR\n",
      "g.chr18:60986450T>C 5'UTR\n",
      "g.chr18:60986611A>G 5'UTR\n",
      "g.chr18:60986639G>C 5'UTR\n",
      "g.chr18:60986652C>T 5'UTR\n",
      "g.chr18:60986701A>C 5'UTR\n",
      "g.chr18:60986744G>A 5'UTR\n",
      "g.chr18:60986791A>G 5'UTR\n",
      "g.chr18:60986833G>T 5'UTR\n",
      "g.chr18:60986835G>T 5'UTR\n",
      "g.chr18:60987054A>T 5'UTR\n",
      "g.chr18:60987110C>G 5'UTR\n",
      "g.chr18:60987135G>A 5'UTR\n",
      "g.chr18:60985916G>A 5'UTR\n",
      "g.chr18:60985953A>T 5'UTR\n",
      "g.chr18:60985974G>C 5'UTR\n",
      "g.chr18:60986167G>C 5'UTR\n",
      "g.chr18:60986442G>C 5'UTR\n",
      "g.chr18:60985979G>C 5'UTR\n",
      "g.chr18:60985900C>A 5'UTR\n",
      "g.chr18:60985917C>G 5'UTR\n",
      "g.chr18:60986082T>G 5'UTR\n",
      "g.chr18:60986204A>G 5'UTR\n",
      "g.chr18:60986297T>C 5'UTR\n",
      "g.chr18:60986986G>A 5'UTR\n",
      "g.chr18:60985994C>T 5'UTR\n",
      "g.chr18:60986286G>A 5'UTR\n",
      "g.chr18:60986300A>T 5'UTR\n",
      "g.chr18:60986391T>C 5'UTR\n",
      "g.chr18:60986409G>A 5'UTR\n",
      "g.chr18:60986423C>T 5'UTR\n",
      "g.chr18:60986441A>G 5'UTR\n",
      "g.chr18:60986476G>A 5'UTR\n",
      "g.chr18:60986598T>C 5'UTR\n",
      "g.chr18:60986621C>T 5'UTR\n",
      "g.chr18:60986642A>G 5'UTR\n",
      "g.chr18:60986793C>G 5'UTR\n",
      "g.chr18:60986878T>C 5'UTR\n",
      "g.chr18:60986883T>C 5'UTR\n",
      "g.chr18:60986919T>A 5'UTR\n",
      "g.chr18:60986924G>A 5'UTR\n",
      "g.chr18:60986980C>T 5'UTR\n",
      "g.chr18:60987029T>C 5'UTR\n",
      "g.chr18:60985900C>T 5'UTR\n",
      "g.chr18:60985982G>A 5'UTR\n",
      "g.chr18:60985994C>T 5'UTR\n",
      "g.chr18:60986055C>G 5'UTR\n",
      "g.chr18:60986140T>A 5'UTR\n",
      "g.chr18:60986189A>G 5'UTR\n",
      "g.chr18:60986191A>G 5'UTR\n",
      "g.chr18:60986205A>G 5'UTR\n",
      "g.chr18:60986243A>G 5'UTR\n",
      "g.chr18:60986339T>C 5'UTR\n",
      "g.chr18:60986341A>G 5'UTR\n",
      "g.chr18:60986351C>T 5'UTR\n",
      "g.chr18:60986422A>C 5'UTR\n",
      "g.chr18:60986697T>C 5'UTR\n",
      "g.chr18:60986821A>G 5'UTR\n",
      "g.chr18:60986867A>G 5'UTR\n",
      "g.chr18:60986875T>G 5'UTR\n",
      "g.chr18:60987043T>C 5'UTR\n",
      "g.chr18:60986002T>C 5'UTR\n",
      "g.chr18:60986014T>C 5'UTR\n",
      "g.chr18:60986077T>G 5'UTR\n",
      "g.chr18:60986097T>G 5'UTR\n",
      "g.chr18:60986101A>C 5'UTR\n",
      "g.chr18:60986119T>G 5'UTR\n",
      "g.chr18:60986152G>A 5'UTR\n",
      "g.chr18:60986211T>A 5'UTR\n",
      "g.chr18:60986409G>A 5'UTR\n",
      "g.chr18:60986461A>G 5'UTR\n",
      "g.chr18:60986464A>G 5'UTR\n",
      "g.chr18:60986468A>G 5'UTR\n",
      "g.chr18:60986471C>T 5'UTR\n",
      "g.chr18:60986598T>G 5'UTR\n",
      "g.chr18:60986666A>G 5'UTR\n",
      "g.chr18:60986773G>A 5'UTR\n",
      "g.chr18:60985955C>T 5'UTR\n",
      "g.chr18:60985990C>G 5'UTR\n",
      "g.chr18:60986042A>G 5'UTR\n",
      "g.chr18:60986163A>G 5'UTR\n",
      "g.chr18:60986210C>T 5'UTR\n",
      "g.chr18:60986242T>A 5'UTR\n",
      "g.chr18:60986288T>A 5'UTR\n",
      "g.chr18:60986300A>T 5'UTR\n",
      "g.chr18:60986304A>T 5'UTR\n",
      "g.chr18:60986321A>G 5'UTR\n",
      "g.chr18:60986359T>G 5'UTR\n",
      "g.chr18:60986381A>G 5'UTR\n",
      "g.chr18:60986409G>C 5'UTR\n",
      "g.chr18:60986411A>T 5'UTR\n",
      "g.chr18:60986420G>A 5'UTR\n",
      "g.chr18:60986427A>G 5'UTR\n",
      "g.chr18:60986442G>C 5'UTR\n",
      "g.chr18:60986449T>G 5'UTR\n",
      "g.chr18:60986459C>G 5'UTR\n",
      "g.chr18:60986467A>G 5'UTR\n",
      "g.chr18:60986478C>A 5'UTR\n",
      "g.chr18:60986509G>C 5'UTR\n",
      "g.chr18:60986511T>G 5'UTR\n",
      "g.chr18:60986526A>G 5'UTR\n",
      "g.chr18:60986541A>G 5'UTR\n",
      "g.chr18:60986602G>A 5'UTR\n",
      "g.chr18:60986610G>A 5'UTR\n",
      "g.chr18:60986629A>G 5'UTR\n",
      "g.chr18:60986652C>T 5'UTR\n",
      "g.chr18:60986669G>A 5'UTR\n",
      "g.chr18:60986766C>G 5'UTR\n",
      "g.chr18:60986773G>A 5'UTR\n",
      "g.chr18:60986793C>T 5'UTR\n",
      "g.chr18:60986888C>G 5'UTR\n",
      "g.chr18:60986897G>C 5'UTR\n",
      "g.chr18:60986903G>A 5'UTR\n",
      "g.chr18:60986967C>G 5'UTR\n",
      "g.chr18:60986974C>T 5'UTR\n",
      "g.chr18:60987052A>T 5'UTR\n",
      "g.chr18:60987066C>T 5'UTR\n",
      "g.chr18:60987100C>A 5'UTR\n",
      "g.chr18:60987144C>G 5'UTR\n",
      "g.chr18:60987175G>A 5'UTR\n",
      "g.chr18:60987182C>T 5'UTR\n",
      "g.chr18:60987209C>G 5'UTR\n",
      "g.chr18:60985900C>T 5'UTR\n",
      "g.chr18:60986016C>T 5'UTR\n",
      "g.chr18:60986069T>C 5'UTR\n",
      "g.chr18:60986150A>G 5'UTR\n",
      "g.chr18:60986153T>G 5'UTR\n",
      "g.chr18:60986172C>T 5'UTR\n",
      "g.chr18:60986243A>C 5'UTR\n",
      "g.chr18:60986407T>C 5'UTR\n",
      "g.chr18:60986420G>C 5'UTR\n",
      "g.chr18:60986515G>A 5'UTR\n",
      "g.chr18:60986917G>A 5'UTR\n",
      "g.chr18:60986960C>T 5'UTR\n",
      "g.chr18:60986965C>T 5'UTR\n",
      "g.chr18:60986989G>A 5'UTR\n",
      "g.chr18:60986994G>A 5'UTR\n",
      "g.chr18:60985900C>T 5'UTR\n",
      "g.chr18:60986005G>T 5'UTR\n",
      "g.chr18:60986018T>G 5'UTR\n",
      "g.chr18:60986092G>C 5'UTR\n",
      "g.chr18:60986117G>A 5'UTR\n",
      "g.chr18:60986163A>G 5'UTR\n",
      "g.chr18:60986244C>T 5'UTR\n",
      "g.chr18:60986305G>A 5'UTR\n",
      "g.chr18:60986325A>T 5'UTR\n",
      "g.chr18:60986409G>A 5'UTR\n",
      "g.chr18:60986444T>C 5'UTR\n",
      "g.chr18:60986509G>A 5'UTR\n",
      "g.chr18:60986534A>G 5'UTR\n",
      "g.chr18:60986568T>C 5'UTR\n",
      "g.chr18:60985974G>A 5'UTR\n",
      "g.chr18:60986053C>T 5'UTR\n",
      "g.chr18:60986067G>A 5'UTR\n",
      "g.chr18:60986083G>A 5'UTR\n",
      "g.chr18:60986146T>G 5'UTR\n",
      "g.chr18:60986148G>A 5'UTR\n",
      "g.chr18:60986333G>A 5'UTR\n",
      "g.chr18:60986375C>T 5'UTR\n",
      "g.chr18:60986377T>G 5'UTR\n",
      "g.chr18:60986386T>C 5'UTR\n",
      "g.chr18:60986397G>A 5'UTR\n",
      "g.chr18:60986399G>A 5'UTR\n",
      "g.chr18:60986516C>G 5'UTR\n",
      "g.chr18:60986641G>A 5'UTR\n",
      "g.chr18:60986740C>T 5'UTR\n",
      "g.chr18:60986744G>A 5'UTR\n",
      "g.chr18:60986773G>C 5'UTR\n",
      "g.chr18:60986847A>T 5'UTR\n",
      "g.chr18:60986918C>T 5'UTR\n",
      "g.chr18:60986980C>T 5'UTR\n",
      "g.chr18:60985900C>G 5'UTR\n",
      "g.chr18:60985910G>C 5'UTR\n",
      "g.chr18:60985950C>T 5'UTR\n",
      "g.chr18:60986343T>A 5'UTR\n",
      "g.chr18:60986402C>G 5'UTR\n",
      "g.chr18:60986411A>C 5'UTR\n",
      "g.chr18:60986421T>G 5'UTR\n",
      "g.chr18:60986744G>A 5'UTR\n",
      "g.chr18:60986750T>A 5'UTR\n",
      "g.chr18:60986829A>C 5'UTR\n",
      "g.chr18:60986921C>G 5'UTR\n",
      "g.chr18:60986926T>C 5'UTR\n",
      "g.chr18:60987039T>C 5'UTR\n",
      "g.chr18:60987048T>A 5'UTR\n",
      "g.chr18:60985932G>A 5'UTR\n",
      "g.chr18:60986057T>G 5'UTR\n",
      "g.chr18:60986069T>C 5'UTR\n",
      "g.chr18:60986134G>C 5'UTR\n",
      "g.chr18:60986144A>T 5'UTR\n",
      "g.chr18:60986148G>A 5'UTR\n",
      "g.chr18:60986210C>T 5'UTR\n",
      "g.chr18:60986222A>G 5'UTR\n",
      "g.chr18:60986280A>G 5'UTR\n",
      "g.chr18:60986669G>T 5'UTR\n",
      "g.chr18:60987028G>C 5'UTR\n",
      "g.chr18:60986148G>C 5'UTR\n",
      "g.chr18:60986652C>T 5'UTR\n",
      "g.chr18:60986254G>A 5'UTR\n",
      "g.chr18:60986378T>C 5'UTR\n",
      "g.chr18:60986469T>C 5'UTR\n",
      "g.chr18:60986517T>C 5'UTR\n",
      "g.chr18:60986992C>T 5'UTR\n",
      "g.chr18:60986022T>A 5'UTR\n",
      "g.chr18:60986046C>G 5'UTR\n",
      "g.chr18:60986305G>A 5'UTR\n",
      "g.chr18:60986372T>G 5'UTR\n",
      "g.chr18:60986473T>A 5'UTR\n",
      "g.chr18:60986935G>T 5'UTR\n",
      "g.chr18:60986994G>C 5'UTR\n",
      "g.chr18:60985947G>A 5'UTR\n",
      "g.chr18:60986070T>G 5'UTR\n",
      "g.chr18:60986138G>C 5'UTR\n",
      "g.chr18:60986293A>G 5'UTR\n",
      "g.chr18:60986306C>G 5'UTR\n",
      "g.chr18:60985917C>T 5'UTR\n",
      "g.chr18:60986018T>G 5'UTR\n",
      "g.chr18:60986082T>G 5'UTR\n",
      "g.chr18:60986234T>A 5'UTR\n",
      "g.chr18:60986322T>A 5'UTR\n",
      "g.chr18:60986326C>G 5'UTR\n",
      "g.chr18:60986470G>C 5'UTR\n",
      "g.chr18:60986727A>G 5'UTR\n",
      "g.chr18:60986137C>G 5'UTR\n",
      "g.chr18:60986154A>T 5'UTR\n",
      "g.chr18:60986232C>T 5'UTR\n",
      "g.chr18:60986241G>A 5'UTR\n",
      "g.chr18:60986263G>A 5'UTR\n",
      "g.chr18:60986278G>C 5'UTR\n",
      "g.chr18:60986306C>T 5'UTR\n",
      "g.chr18:60986417T>A 5'UTR\n",
      "g.chr18:60986420G>A 5'UTR\n",
      "g.chr18:60986515G>A 5'UTR\n",
      "g.chr18:60986518G>A 5'UTR\n",
      "g.chr18:60986651G>A 5'UTR\n",
      "g.chr18:60986959G>A 5'UTR\n",
      "g.chr18:60986169C>G 5'UTR\n",
      "g.chr18:60985905C>T 5'UTR\n",
      "g.chr18:60986138G>T 5'UTR\n",
      "g.chr18:60986151A>T 5'UTR\n",
      "g.chr18:60985916G>C 5'UTR\n",
      "g.chr18:60985997T>C 5'UTR\n",
      "g.chr18:60986076T>C 5'UTR\n",
      "g.chr18:60986145T>G 5'UTR\n",
      "g.chr18:60986252G>T 5'UTR\n",
      "g.chr18:60986280A>G 5'UTR\n",
      "g.chr18:60986301T>A 5'UTR\n",
      "g.chr18:60986420G>A 5'UTR\n",
      "g.chr18:60986669G>T 5'UTR\n",
      "g.chr18:60986857G>C 5'UTR\n",
      "g.chr18:60986888C>T 5'UTR\n",
      "g.chr18:60985993G>C 5'UTR\n",
      "g.chr18:60986159T>A 5'UTR\n",
      "g.chr18:60986228G>C 5'UTR\n",
      "g.chr18:60986234T>A 5'UTR\n",
      "g.chr18:60986278G>A 5'UTR\n",
      "g.chr18:60986312G>A 5'UTR\n",
      "g.chr18:60986322T>G 5'UTR\n",
      "g.chr18:60986347A>G 5'UTR\n",
      "g.chr18:60986378T>G 5'UTR\n",
      "g.chr18:60986407T>C 5'UTR\n",
      "g.chr18:60986417T>G 5'UTR\n",
      "g.chr18:60986500C>T 5'UTR\n",
      "g.chr18:60986502C>A 5'UTR\n",
      "g.chr18:60986535T>C 5'UTR\n",
      "g.chr18:60986565A>G 5'UTR\n",
      "g.chr18:60986614T>A 5'UTR\n",
      "g.chr18:60986630T>C 5'UTR\n",
      "g.chr18:60986686T>C 5'UTR\n",
      "g.chr18:60986748T>G 5'UTR\n",
      "g.chr18:60986750T>G 5'UTR\n",
      "g.chr18:60986752C>T 5'UTR\n",
      "g.chr18:60986793C>T 5'UTR\n",
      "[{u'impact': u'LOW', u'protein_start': 8, u'gene_id': u'ENSG00000171791', u'protein_end': 8, u'gene_symbol': u'BCL2', u'cdna_end': 923, u'amino_acids': u'G', u'hgnc_id': 990, u'codons': u'ggG/ggA', u'gene_symbol_source': u'HGNC', u'cdna_start': 923, u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 24, u'strand': -1, u'cds_start': 24}, {u'cds_start': 24, u'impact': u'LOW', u'protein_end': 8, u'cdna_start': 1486, u'protein_start': 8, u'cdna_end': 1486, u'amino_acids': u'G', u'hgnc_id': 990, u'codons': u'ggG/ggA', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000398117', u'gene_id': u'ENSG00000171791', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 24, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'LOW', u'protein_start': 8, u'protein_end': 8, u'gene_symbol': u'BCL2', u'gene_id': u'ENSG00000171791', u'amino_acids': u'G', u'hgnc_id': 990, u'codons': u'ggG/ggA', u'gene_symbol_source': u'HGNC', u'cdna_start': 517, u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'cdna_end': 517, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 24, u'strand': -1, u'cds_start': 24}, {u'impact': u'MODIFIER', u'distance': 4561, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'cds_start': 24, u'protein_start': 8, u'protein_end': 8, u'cdna_start': 170, u'cdna_end': 170, u'impact': u'LOW', u'amino_acids': u'G', u'hgnc_id': 990, u'codons': u'ggG/ggA', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000589955', u'gene_id': u'ENSG00000171791', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 24, u'strand': -1, u'biotype': u'protein_coding'}]\n",
      "[{u'impact': u'LOW', u'protein_start': 152, u'protein_end': 152, u'cdna_start': 1355, u'cdna_end': 1355, u'amino_acids': u'E', u'hgnc_id': 990, u'codons': u'gaG/gaA', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000333681', u'gene_id': u'ENSG00000171791', u'cds_start': 456, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 456, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'LOW', u'protein_start': 152, u'protein_end': 152, u'cdna_start': 1918, u'cdna_end': 1918, u'amino_acids': u'E', u'hgnc_id': 990, u'codons': u'gaG/gaA', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000398117', u'gene_id': u'ENSG00000171791', u'cds_start': 456, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 456, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'LOW', u'protein_start': 152, u'protein_end': 152, u'cdna_start': 949, u'cdna_end': 949, u'amino_acids': u'E', u'hgnc_id': 990, u'codons': u'gaG/gaA', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000444484', u'gene_id': u'ENSG00000171791', u'cds_start': 456, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 456, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'MODIFIER', u'distance': 4129, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'cds_start': 456, u'impact': u'LOW', u'gene_id': u'ENSG00000171791', u'protein_end': 152, u'gene_symbol': u'BCL2', u'cdna_end': 602, u'amino_acids': u'E', u'hgnc_id': 990, u'codons': u'gaG/gaA', u'gene_symbol_source': u'HGNC', u'cdna_start': 602, u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'protein_start': 152, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 456, u'strand': -1}]\n",
      "[{u'impact': u'LOW', u'protein_start': 143, u'protein_end': 143, u'gene_symbol': u'BCL2', u'gene_id': u'ENSG00000171791', u'amino_acids': u'N', u'hgnc_id': 990, u'codons': u'aaC/aaT', u'gene_symbol_source': u'HGNC', u'cdna_start': 1328, u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'cdna_end': 1328, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 429, u'strand': -1, u'cds_start': 429}, {u'impact': u'LOW', u'protein_start': 143, u'gene_id': u'ENSG00000171791', u'protein_end': 143, u'gene_symbol': u'BCL2', u'cdna_end': 1891, u'amino_acids': u'N', u'hgnc_id': 990, u'codons': u'aaC/aaT', u'gene_symbol_source': u'HGNC', u'cdna_start': 1891, u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 429, u'strand': -1, u'cds_start': 429}, {u'impact': u'LOW', u'protein_start': 143, u'protein_end': 143, u'gene_symbol': u'BCL2', u'gene_id': u'ENSG00000171791', u'amino_acids': u'N', u'hgnc_id': 990, u'codons': u'aaC/aaT', u'gene_symbol_source': u'HGNC', u'cdna_start': 922, u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'cdna_end': 922, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 429, u'strand': -1, u'cds_start': 429}, {u'impact': u'MODIFIER', u'distance': 4156, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'cds_start': 429, u'impact': u'LOW', u'protein_end': 143, u'cdna_start': 575, u'protein_start': 143, u'cdna_end': 575, u'amino_acids': u'N', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000589955', u'codons': u'aaC/aaT', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 429, u'strand': -1, u'biotype': u'protein_coding'}]\n",
      "[{u'impact': u'LOW', u'cds_start': 408, u'codons': u'gaG/gaA', u'protein_end': 136, u'gene_symbol': u'BCL2', u'cdna_end': 1307, u'amino_acids': u'E', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 1307, u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'protein_start': 136, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 408, u'strand': -1}, {u'impact': u'LOW', u'cds_start': 408, u'codons': u'gaG/gaA', u'protein_end': 136, u'gene_symbol': u'BCL2', u'cdna_end': 1870, u'amino_acids': u'E', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 1870, u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'protein_start': 136, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 408, u'strand': -1}, {u'cds_start': 408, u'protein_start': 136, u'protein_end': 136, u'cdna_start': 901, u'cdna_end': 901, u'impact': u'LOW', u'amino_acids': u'E', u'hgnc_id': 990, u'codons': u'gaG/gaA', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000444484', u'gene_id': u'ENSG00000171791', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 408, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'MODIFIER', u'distance': 4177, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'impact': u'LOW', u'protein_start': 136, u'codons': u'gaG/gaA', u'protein_end': 136, u'cdna_start': 554, u'cdna_end': 554, u'amino_acids': u'E', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 408, u'strand': -1, u'cds_start': 408}]\n",
      "[{u'cds_start': 327, u'impact': u'LOW', u'codons': u'cgC/cgT', u'protein_end': 109, u'protein_start': 109, u'cdna_end': 1226, u'amino_acids': u'R', u'hgnc_id': 990, u'cdna_start': 1226, u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 327, u'strand': -1}, {u'impact': u'LOW', u'protein_start': 109, u'protein_end': 109, u'gene_symbol': u'BCL2', u'cdna_end': 1789, u'amino_acids': u'R', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 1789, u'transcript_id': u'ENST00000398117', u'codons': u'cgC/cgT', u'cds_start': 327, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 327, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'LOW', u'cds_start': 327, u'codons': u'cgC/cgT', u'protein_end': 109, u'protein_start': 109, u'cdna_end': 820, u'amino_acids': u'R', u'hgnc_id': 990, u'cdna_start': 820, u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 327, u'strand': -1}, {u'impact': u'MODIFIER', u'distance': 4258, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'impact': u'LOW', u'protein_start': 109, u'protein_end': 109, u'gene_symbol': u'BCL2', u'cdna_end': 473, u'amino_acids': u'R', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 473, u'transcript_id': u'ENST00000589955', u'codons': u'cgC/cgT', u'cds_start': 327, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 327, u'strand': -1, u'biotype': u'protein_coding'}]\n",
      "[{u'impact': u'LOW', u'protein_start': 178, u'codons': u'acT/acC', u'protein_end': 178, u'gene_symbol': u'BCL2', u'cdna_end': 1433, u'amino_acids': u'T', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 1433, u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'G', u'cds_end': 534, u'strand': -1, u'cds_start': 534}, {u'impact': u'LOW', u'cds_start': 534, u'codons': u'acT/acC', u'protein_end': 178, u'gene_symbol': u'BCL2', u'cdna_end': 1996, u'amino_acids': u'T', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 1996, u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'protein_start': 178, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'G', u'cds_end': 534, u'strand': -1}, {u'cds_start': 534, u'protein_start': 178, u'codons': u'acT/acC', u'protein_end': 178, u'gene_symbol': u'BCL2', u'cdna_end': 1027, u'impact': u'LOW', u'amino_acids': u'T', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 1027, u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'G', u'cds_end': 534, u'strand': -1}, {u'impact': u'MODIFIER', u'distance': 4051, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'G', u'strand': -1}, {u'cds_start': 534, u'protein_start': 178, u'protein_end': 178, u'impact': u'LOW', u'cdna_end': 680, u'amino_acids': u'T', u'hgnc_id': 990, u'codons': u'acT/acC', u'cdna_start': 680, u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000589955', u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'G', u'cds_end': 534, u'strand': -1, u'biotype': u'protein_coding'}]\n",
      "[{u'impact': u'LOW', u'protein_start': 155, u'codons': u'ggG/ggA', u'protein_end': 155, u'cdna_start': 1364, u'cdna_end': 1364, u'amino_acids': u'G', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 465, u'strand': -1, u'cds_start': 465}, {u'impact': u'LOW', u'protein_start': 155, u'codons': u'ggG/ggA', u'protein_end': 155, u'gene_symbol': u'BCL2', u'cdna_end': 1927, u'amino_acids': u'G', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 1927, u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 465, u'strand': -1, u'cds_start': 465}, {u'cds_start': 465, u'protein_start': 155, u'protein_end': 155, u'gene_symbol': u'BCL2', u'cdna_end': 958, u'impact': u'LOW', u'amino_acids': u'G', u'hgnc_id': 990, u'codons': u'ggG/ggA', u'gene_symbol_source': u'HGNC', u'cdna_start': 958, u'transcript_id': u'ENST00000444484', u'gene_id': u'ENSG00000171791', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 465, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'MODIFIER', u'distance': 4120, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'cds_start': 465, u'protein_start': 155, u'protein_end': 155, u'gene_symbol': u'BCL2', u'cdna_end': 611, u'impact': u'LOW', u'amino_acids': u'G', u'hgnc_id': 990, u'codons': u'ggG/ggA', u'gene_symbol_source': u'HGNC', u'cdna_start': 611, u'transcript_id': u'ENST00000589955', u'gene_id': u'ENSG00000171791', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 465, u'strand': -1, u'biotype': u'protein_coding'}]\n",
      "[{u'cds_start': 66, u'protein_start': 22, u'gene_id': u'ENSG00000171791', u'protein_end': 22, u'impact': u'LOW', u'cdna_end': 965, u'gene_symbol': u'BCL2', u'amino_acids': u'K', u'hgnc_id': 990, u'codons': u'aaG/aaA', u'gene_symbol_source': u'HGNC', u'cdna_start': 965, u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 66, u'strand': -1}, {u'impact': u'LOW', u'cds_start': 66, u'codons': u'aaG/aaA', u'protein_end': 22, u'gene_symbol': u'BCL2', u'cdna_end': 1528, u'amino_acids': u'K', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 1528, u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'protein_start': 22, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 66, u'strand': -1}, {u'impact': u'LOW', u'protein_start': 22, u'codons': u'aaG/aaA', u'protein_end': 22, u'cdna_start': 559, u'cdna_end': 559, u'amino_acids': u'K', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 66, u'strand': -1, u'cds_start': 66}, {u'impact': u'MODIFIER', u'distance': 4519, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'impact': u'LOW', u'protein_start': 22, u'codons': u'aaG/aaA', u'protein_end': 22, u'cdna_start': 212, u'cdna_end': 212, u'amino_acids': u'K', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 66, u'strand': -1, u'cds_start': 66}]\n",
      "[{u'cds_start': 24, u'impact': u'LOW', u'protein_end': 8, u'cdna_start': 923, u'protein_start': 8, u'cdna_end': 923, u'amino_acids': u'G', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000333681', u'codons': u'ggG/ggT', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 24, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'LOW', u'protein_start': 8, u'codons': u'ggG/ggT', u'protein_end': 8, u'gene_symbol': u'BCL2', u'cdna_end': 1486, u'amino_acids': u'G', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 1486, u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 24, u'strand': -1, u'cds_start': 24}, {u'impact': u'LOW', u'protein_start': 8, u'codons': u'ggG/ggT', u'protein_end': 8, u'gene_symbol': u'BCL2', u'cdna_end': 517, u'amino_acids': u'G', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 517, u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 24, u'strand': -1, u'cds_start': 24}, {u'impact': u'MODIFIER', u'distance': 4561, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'cds_start': 24, u'impact': u'LOW', u'protein_end': 8, u'cdna_start': 170, u'protein_start': 8, u'cdna_end': 170, u'amino_acids': u'G', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000589955', u'codons': u'ggG/ggT', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 24, u'strand': -1, u'biotype': u'protein_coding'}]\n",
      "[{u'cds_start': 66, u'protein_start': 22, u'gene_id': u'ENSG00000171791', u'protein_end': 22, u'cdna_start': 965, u'impact': u'LOW', u'cdna_end': 965, u'amino_acids': u'K', u'hgnc_id': 990, u'codons': u'aaG/aaA', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 66, u'strand': -1}, {u'impact': u'LOW', u'protein_start': 22, u'codons': u'aaG/aaA', u'protein_end': 22, u'cdna_start': 1528, u'cdna_end': 1528, u'amino_acids': u'K', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 66, u'strand': -1, u'cds_start': 66}, {u'cds_start': 66, u'protein_start': 22, u'gene_id': u'ENSG00000171791', u'protein_end': 22, u'impact': u'LOW', u'cdna_end': 559, u'gene_symbol': u'BCL2', u'amino_acids': u'K', u'hgnc_id': 990, u'codons': u'aaG/aaA', u'gene_symbol_source': u'HGNC', u'cdna_start': 559, u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 66, u'strand': -1}, {u'impact': u'MODIFIER', u'distance': 4519, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'cds_start': 66, u'protein_start': 22, u'gene_id': u'ENSG00000171791', u'protein_end': 22, u'cdna_start': 212, u'impact': u'LOW', u'cdna_end': 212, u'amino_acids': u'K', u'hgnc_id': 990, u'codons': u'aaG/aaA', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 66, u'strand': -1}]\n",
      "[{u'cds_start': 67, u'protein_start': 23, u'protein_end': 23, u'impact': u'LOW', u'cdna_end': 966, u'gene_symbol': u'BCL2', u'amino_acids': u'L', u'hgnc_id': 990, u'codons': u'Ctg/Ttg', u'gene_symbol_source': u'HGNC', u'cdna_start': 966, u'transcript_id': u'ENST00000333681', u'gene_id': u'ENSG00000171791', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 67, u'strand': -1, u'biotype': u'protein_coding'}, {u'cds_start': 67, u'impact': u'LOW', u'codons': u'Ctg/Ttg', u'protein_end': 23, u'cdna_start': 1529, u'cdna_end': 1529, u'amino_acids': u'L', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'protein_start': 23, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 67, u'strand': -1}, {u'impact': u'LOW', u'protein_start': 23, u'codons': u'Ctg/Ttg', u'protein_end': 23, u'cdna_start': 560, u'cdna_end': 560, u'amino_acids': u'L', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 67, u'strand': -1, u'cds_start': 67}, {u'impact': u'MODIFIER', u'distance': 4518, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'cds_start': 67, u'protein_start': 23, u'codons': u'Ctg/Ttg', u'protein_end': 23, u'cdna_start': 213, u'impact': u'LOW', u'cdna_end': 213, u'amino_acids': u'L', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 67, u'strand': -1}]\n",
      "[{u'impact': u'LOW', u'protein_start': 181, u'protein_end': 181, u'gene_symbol': u'BCL2', u'cdna_end': 1440, u'amino_acids': u'L', u'hgnc_id': 990, u'codons': u'Ctg/Ttg', u'biotype': u'protein_coding', u'gene_symbol_source': u'HGNC', u'cdna_start': 1440, u'transcript_id': u'ENST00000333681', u'gene_id': u'ENSG00000171791', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 541, u'strand': -1, u'cds_start': 541}, {u'cds_start': 541, u'impact': u'LOW', u'codons': u'Ctg/Ttg', u'protein_end': 181, u'cdna_start': 2003, u'protein_start': 181, u'cdna_end': 2003, u'amino_acids': u'L', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 541, u'strand': -1}, {u'impact': u'LOW', u'cds_start': 541, u'codons': u'Ctg/Ttg', u'protein_end': 181, u'protein_start': 181, u'cdna_end': 1034, u'amino_acids': u'L', u'hgnc_id': 990, u'cdna_start': 1034, u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 541, u'strand': -1}, {u'impact': u'MODIFIER', u'distance': 4044, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'cds_start': 541, u'protein_start': 181, u'protein_end': 181, u'gene_symbol': u'BCL2', u'cdna_end': 687, u'impact': u'LOW', u'amino_acids': u'L', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 687, u'transcript_id': u'ENST00000589955', u'codons': u'Ctg/Ttg', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 541, u'strand': -1, u'biotype': u'protein_coding'}]\n",
      "[{u'impact': u'LOW', u'protein_start': 178, u'gene_id': u'ENSG00000171791', u'protein_end': 178, u'cdna_start': 1433, u'cdna_end': 1433, u'amino_acids': u'T', u'hgnc_id': 990, u'codons': u'acT/acA', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 534, u'strand': -1, u'cds_start': 534}, {u'impact': u'LOW', u'protein_start': 178, u'gene_id': u'ENSG00000171791', u'protein_end': 178, u'cdna_start': 1996, u'cdna_end': 1996, u'amino_acids': u'T', u'hgnc_id': 990, u'codons': u'acT/acA', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 534, u'strand': -1, u'cds_start': 534}, {u'impact': u'LOW', u'protein_start': 178, u'codons': u'acT/acA', u'protein_end': 178, u'gene_symbol': u'BCL2', u'cdna_end': 1027, u'amino_acids': u'T', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 1027, u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 534, u'strand': -1, u'cds_start': 534}, {u'impact': u'MODIFIER', u'distance': 4051, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'impact': u'LOW', u'cds_start': 534, u'gene_id': u'ENSG00000171791', u'protein_end': 178, u'cdna_start': 680, u'protein_start': 178, u'cdna_end': 680, u'amino_acids': u'T', u'hgnc_id': 990, u'codons': u'acT/acA', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 534, u'strand': -1}]\n",
      "[{u'impact': u'LOW', u'cds_start': 39, u'codons': u'gaG/gaA', u'protein_end': 13, u'gene_symbol': u'BCL2', u'cdna_end': 938, u'amino_acids': u'E', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 938, u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'protein_start': 13, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 39, u'strand': -1}, {u'cds_start': 39, u'protein_start': 13, u'protein_end': 13, u'impact': u'LOW', u'gene_id': u'ENSG00000171791', u'amino_acids': u'E', u'hgnc_id': 990, u'codons': u'gaG/gaA', u'cdna_start': 1501, u'gene_symbol': u'BCL2', u'cdna_end': 1501, u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'gene_symbol_source': u'HGNC', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 39, u'strand': -1}, {u'cds_start': 39, u'protein_start': 13, u'protein_end': 13, u'impact': u'LOW', u'gene_id': u'ENSG00000171791', u'amino_acids': u'E', u'hgnc_id': 990, u'codons': u'gaG/gaA', u'cdna_start': 532, u'gene_symbol': u'BCL2', u'cdna_end': 532, u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'gene_symbol_source': u'HGNC', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 39, u'strand': -1}, {u'impact': u'MODIFIER', u'distance': 4546, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'cds_start': 39, u'protein_start': 13, u'gene_id': u'ENSG00000171791', u'protein_end': 13, u'impact': u'LOW', u'cdna_end': 185, u'gene_symbol': u'BCL2', u'amino_acids': u'E', u'hgnc_id': 990, u'codons': u'gaG/gaA', u'gene_symbol_source': u'HGNC', u'cdna_start': 185, u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 39, u'strand': -1}]\n",
      "[{u'cds_start': 67, u'impact': u'LOW', u'protein_end': 23, u'cdna_start': 966, u'cdna_end': 966, u'amino_acids': u'L', u'hgnc_id': 990, u'codons': u'Ctg/Ttg', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000333681', u'gene_id': u'ENSG00000171791', u'protein_start': 23, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 67, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'LOW', u'protein_start': 23, u'gene_id': u'ENSG00000171791', u'protein_end': 23, u'gene_symbol': u'BCL2', u'cdna_end': 1529, u'amino_acids': u'L', u'hgnc_id': 990, u'codons': u'Ctg/Ttg', u'gene_symbol_source': u'HGNC', u'cdna_start': 1529, u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 67, u'strand': -1, u'cds_start': 67}, {u'cds_start': 67, u'impact': u'LOW', u'protein_end': 23, u'cdna_start': 560, u'cdna_end': 560, u'amino_acids': u'L', u'hgnc_id': 990, u'codons': u'Ctg/Ttg', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000444484', u'gene_id': u'ENSG00000171791', u'protein_start': 23, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 67, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'MODIFIER', u'distance': 4518, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'impact': u'LOW', u'protein_start': 23, u'codons': u'Ctg/Ttg', u'protein_end': 23, u'gene_symbol': u'BCL2', u'cdna_end': 213, u'amino_acids': u'L', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 213, u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 67, u'strand': -1, u'cds_start': 67}]\n",
      "[{u'impact': u'LOW', u'cds_start': 66, u'gene_id': u'ENSG00000171791', u'protein_end': 22, u'cdna_start': 965, u'protein_start': 22, u'cdna_end': 965, u'amino_acids': u'K', u'hgnc_id': 990, u'codons': u'aaG/aaA', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 66, u'strand': -1}, {u'impact': u'LOW', u'protein_start': 22, u'gene_id': u'ENSG00000171791', u'protein_end': 22, u'cdna_start': 1528, u'cdna_end': 1528, u'amino_acids': u'K', u'hgnc_id': 990, u'codons': u'aaG/aaA', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 66, u'strand': -1, u'cds_start': 66}, {u'impact': u'LOW', u'cds_start': 66, u'gene_id': u'ENSG00000171791', u'protein_end': 22, u'cdna_start': 559, u'protein_start': 22, u'cdna_end': 559, u'amino_acids': u'K', u'hgnc_id': 990, u'codons': u'aaG/aaA', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 66, u'strand': -1}, {u'impact': u'MODIFIER', u'distance': 4519, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'cds_start': 66, u'protein_start': 22, u'codons': u'aaG/aaA', u'protein_end': 22, u'gene_symbol': u'BCL2', u'cdna_end': 212, u'impact': u'LOW', u'amino_acids': u'K', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 212, u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 66, u'strand': -1}]\n",
      "[{u'cds_start': 351, u'protein_start': 117, u'codons': u'agC/agT', u'protein_end': 117, u'impact': u'LOW', u'cdna_end': 1250, u'amino_acids': u'S', u'hgnc_id': 990, u'cdna_start': 1250, u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'gene_id': u'ENSG00000171791', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 351, u'strand': -1}, {u'impact': u'LOW', u'protein_start': 117, u'protein_end': 117, u'gene_symbol': u'BCL2', u'cdna_end': 1813, u'amino_acids': u'S', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'biotype': u'protein_coding', u'gene_symbol_source': u'HGNC', u'cdna_start': 1813, u'transcript_id': u'ENST00000398117', u'codons': u'agC/agT', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 351, u'strand': -1, u'cds_start': 351}, {u'cds_start': 351, u'protein_start': 117, u'protein_end': 117, u'impact': u'LOW', u'cdna_end': 844, u'amino_acids': u'S', u'hgnc_id': 990, u'codons': u'agC/agT', u'cdna_start': 844, u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000444484', u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 351, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'MODIFIER', u'distance': 4234, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'impact': u'LOW', u'cds_start': 351, u'gene_id': u'ENSG00000171791', u'protein_end': 117, u'gene_symbol': u'BCL2', u'cdna_end': 497, u'amino_acids': u'S', u'protein_start': 117, u'codons': u'agC/agT', u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 497, u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 351, u'strand': -1}]\n",
      "[{u'cds_start': 189, u'protein_start': 63, u'gene_id': u'ENSG00000171791', u'protein_end': 63, u'impact': u'LOW', u'cdna_end': 1088, u'gene_symbol': u'BCL2', u'amino_acids': u'R', u'hgnc_id': 990, u'codons': u'cgG/cgT', u'gene_symbol_source': u'HGNC', u'cdna_start': 1088, u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 189, u'strand': -1}, {u'impact': u'LOW', u'cds_start': 189, u'codons': u'cgG/cgT', u'protein_end': 63, u'gene_symbol': u'BCL2', u'cdna_end': 1651, u'amino_acids': u'R', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 1651, u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'protein_start': 63, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 189, u'strand': -1}, {u'cds_start': 189, u'protein_start': 63, u'protein_end': 63, u'impact': u'LOW', u'gene_id': u'ENSG00000171791', u'amino_acids': u'R', u'hgnc_id': 990, u'codons': u'cgG/cgT', u'cdna_start': 682, u'gene_symbol': u'BCL2', u'cdna_end': 682, u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'gene_symbol_source': u'HGNC', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 189, u'strand': -1}, {u'impact': u'MODIFIER', u'distance': 4396, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'cds_start': 189, u'protein_start': 63, u'gene_id': u'ENSG00000171791', u'protein_end': 63, u'impact': u'LOW', u'cdna_end': 335, u'gene_symbol': u'BCL2', u'amino_acids': u'R', u'hgnc_id': 990, u'codons': u'cgG/cgT', u'gene_symbol_source': u'HGNC', u'cdna_start': 335, u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 189, u'strand': -1}]\n",
      "[{u'impact': u'LOW', u'protein_start': 117, u'codons': u'agC/agT', u'protein_end': 117, u'gene_symbol': u'BCL2', u'cdna_end': 1250, u'amino_acids': u'S', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 1250, u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 351, u'strand': -1, u'cds_start': 351}, {u'cds_start': 351, u'protein_start': 117, u'protein_end': 117, u'cdna_start': 1813, u'cdna_end': 1813, u'impact': u'LOW', u'amino_acids': u'S', u'hgnc_id': 990, u'codons': u'agC/agT', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000398117', u'gene_id': u'ENSG00000171791', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 351, u'strand': -1, u'biotype': u'protein_coding'}, {u'cds_start': 351, u'protein_start': 117, u'protein_end': 117, u'cdna_start': 844, u'cdna_end': 844, u'impact': u'LOW', u'amino_acids': u'S', u'hgnc_id': 990, u'codons': u'agC/agT', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000444484', u'gene_id': u'ENSG00000171791', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 351, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'MODIFIER', u'distance': 4234, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'impact': u'LOW', u'protein_start': 117, u'codons': u'agC/agT', u'protein_end': 117, u'gene_symbol': u'BCL2', u'cdna_end': 497, u'amino_acids': u'S', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 497, u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 351, u'strand': -1, u'cds_start': 351}]\n",
      "[{u'cds_start': 24, u'protein_start': 8, u'protein_end': 8, u'impact': u'LOW', u'cdna_end': 923, u'amino_acids': u'G', u'hgnc_id': 990, u'codons': u'ggG/ggA', u'cdna_start': 923, u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000333681', u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 24, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'LOW', u'cds_start': 24, u'gene_id': u'ENSG00000171791', u'protein_end': 8, u'cdna_start': 1486, u'protein_start': 8, u'cdna_end': 1486, u'amino_acids': u'G', u'hgnc_id': 990, u'codons': u'ggG/ggA', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 24, u'strand': -1}, {u'impact': u'LOW', u'protein_start': 8, u'gene_id': u'ENSG00000171791', u'protein_end': 8, u'cdna_start': 517, u'cdna_end': 517, u'amino_acids': u'G', u'hgnc_id': 990, u'codons': u'ggG/ggA', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 24, u'strand': -1, u'cds_start': 24}, {u'impact': u'MODIFIER', u'distance': 4561, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'impact': u'LOW', u'protein_start': 8, u'codons': u'ggG/ggA', u'protein_end': 8, u'gene_symbol': u'BCL2', u'cdna_end': 170, u'amino_acids': u'G', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 170, u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 24, u'strand': -1, u'cds_start': 24}]\n",
      "[{u'impact': u'LOW', u'protein_start': 124, u'gene_id': u'ENSG00000171791', u'protein_end': 124, u'gene_symbol': u'BCL2', u'cdna_end': 1271, u'amino_acids': u'F', u'hgnc_id': 990, u'codons': u'ttC/ttT', u'gene_symbol_source': u'HGNC', u'cdna_start': 1271, u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 372, u'strand': -1, u'cds_start': 372}, {u'cds_start': 372, u'protein_start': 124, u'codons': u'ttC/ttT', u'protein_end': 124, u'cdna_start': 1834, u'cdna_end': 1834, u'impact': u'LOW', u'amino_acids': u'F', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 372, u'strand': -1}, {u'cds_start': 372, u'protein_start': 124, u'protein_end': 124, u'impact': u'LOW', u'cdna_end': 865, u'gene_symbol': u'BCL2', u'amino_acids': u'F', u'hgnc_id': 990, u'codons': u'ttC/ttT', u'gene_symbol_source': u'HGNC', u'cdna_start': 865, u'transcript_id': u'ENST00000444484', u'gene_id': u'ENSG00000171791', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 372, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'MODIFIER', u'distance': 4213, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'impact': u'LOW', u'protein_start': 124, u'codons': u'ttC/ttT', u'protein_end': 124, u'cdna_start': 518, u'cdna_end': 518, u'amino_acids': u'F', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 372, u'strand': -1, u'cds_start': 372}]\n",
      "[{u'cds_start': 66, u'impact': u'LOW', u'gene_id': u'ENSG00000171791', u'protein_end': 22, u'cdna_start': 965, u'cdna_end': 965, u'amino_acids': u'K', u'protein_start': 22, u'codons': u'aaG/aaA', u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 66, u'strand': -1}, {u'cds_start': 66, u'impact': u'LOW', u'gene_id': u'ENSG00000171791', u'protein_end': 22, u'cdna_start': 1528, u'protein_start': 22, u'cdna_end': 1528, u'amino_acids': u'K', u'hgnc_id': 990, u'codons': u'aaG/aaA', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 66, u'strand': -1}, {u'cds_start': 66, u'protein_start': 22, u'protein_end': 22, u'gene_symbol': u'BCL2', u'cdna_end': 559, u'impact': u'LOW', u'amino_acids': u'K', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 559, u'transcript_id': u'ENST00000444484', u'codons': u'aaG/aaA', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 66, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'MODIFIER', u'distance': 4519, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'cds_start': 66, u'impact': u'LOW', u'gene_id': u'ENSG00000171791', u'protein_end': 22, u'cdna_start': 212, u'cdna_end': 212, u'amino_acids': u'K', u'protein_start': 22, u'codons': u'aaG/aaA', u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 66, u'strand': -1}]\n",
      "[{u'impact': u'LOW', u'protein_start': 120, u'codons': u'caC/caT', u'protein_end': 120, u'cdna_start': 1259, u'cdna_end': 1259, u'amino_acids': u'H', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 360, u'strand': -1, u'cds_start': 360}, {u'impact': u'LOW', u'protein_start': 120, u'codons': u'caC/caT', u'protein_end': 120, u'cdna_start': 1822, u'cdna_end': 1822, u'amino_acids': u'H', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 360, u'strand': -1, u'cds_start': 360}, {u'cds_start': 360, u'impact': u'LOW', u'protein_end': 120, u'cdna_start': 853, u'gene_id': u'ENSG00000171791', u'amino_acids': u'H', u'hgnc_id': 990, u'codons': u'caC/caT', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'cdna_end': 853, u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'protein_start': 120, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 360, u'strand': -1}, {u'impact': u'MODIFIER', u'distance': 4225, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'impact': u'LOW', u'protein_start': 120, u'protein_end': 120, u'gene_symbol': u'BCL2', u'cdna_end': 506, u'amino_acids': u'H', u'hgnc_id': 990, u'codons': u'caC/caT', u'gene_symbol_source': u'HGNC', u'cdna_start': 506, u'transcript_id': u'ENST00000589955', u'consequence_terms': [u'synonymous_variant'], u'biotype': u'protein_coding', u'gene_id': u'ENSG00000171791', u'variant_allele': u'A', u'cds_end': 360, u'strand': -1, u'cds_start': 360}]\n",
      "[{u'cds_start': 187, u'protein_start': 63, u'protein_end': 63, u'cdna_start': 1086, u'cdna_end': 1086, u'impact': u'LOW', u'amino_acids': u'R', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000333681', u'codons': u'Cgg/Agg', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 187, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'LOW', u'protein_start': 63, u'codons': u'Cgg/Agg', u'protein_end': 63, u'gene_symbol': u'BCL2', u'cdna_end': 1649, u'amino_acids': u'R', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 1649, u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 187, u'strand': -1, u'cds_start': 187}, {u'cds_start': 187, u'protein_start': 63, u'protein_end': 63, u'cdna_start': 680, u'cdna_end': 680, u'impact': u'LOW', u'amino_acids': u'R', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000444484', u'codons': u'Cgg/Agg', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 187, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'MODIFIER', u'distance': 4398, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'impact': u'LOW', u'protein_start': 63, u'protein_end': 63, u'gene_symbol': u'BCL2', u'gene_id': u'ENSG00000171791', u'amino_acids': u'R', u'hgnc_id': 990, u'codons': u'Cgg/Agg', u'gene_symbol_source': u'HGNC', u'cdna_start': 333, u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'cdna_end': 333, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 187, u'strand': -1, u'cds_start': 187}]\n",
      "[{u'cds_start': 537, u'impact': u'LOW', u'codons': u'gaG/gaA', u'protein_end': 179, u'cdna_start': 1436, u'protein_start': 179, u'cdna_end': 1436, u'amino_acids': u'E', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 537, u'strand': -1}, {u'impact': u'LOW', u'protein_start': 179, u'protein_end': 179, u'gene_symbol': u'BCL2', u'cdna_end': 1999, u'amino_acids': u'E', u'hgnc_id': 990, u'codons': u'gaG/gaA', u'biotype': u'protein_coding', u'gene_symbol_source': u'HGNC', u'cdna_start': 1999, u'transcript_id': u'ENST00000398117', u'gene_id': u'ENSG00000171791', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 537, u'strand': -1, u'cds_start': 537}, {u'cds_start': 537, u'protein_start': 179, u'protein_end': 179, u'impact': u'LOW', u'cdna_end': 1030, u'gene_symbol': u'BCL2', u'amino_acids': u'E', u'hgnc_id': 990, u'codons': u'gaG/gaA', u'gene_symbol_source': u'HGNC', u'cdna_start': 1030, u'transcript_id': u'ENST00000444484', u'gene_id': u'ENSG00000171791', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 537, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'MODIFIER', u'distance': 4048, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'cds_start': 537, u'protein_start': 179, u'protein_end': 179, u'gene_symbol': u'BCL2', u'cdna_end': 683, u'impact': u'LOW', u'amino_acids': u'E', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 683, u'transcript_id': u'ENST00000589955', u'codons': u'gaG/gaA', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 537, u'strand': -1, u'biotype': u'protein_coding'}]\n",
      "[{u'cds_start': 483, u'protein_start': 161, u'protein_end': 161, u'impact': u'LOW', u'cdna_end': 1382, u'amino_acids': u'S', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'cdna_start': 1382, u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000333681', u'codons': u'agC/agT', u'gene_symbol_source': u'HGNC', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 483, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'LOW', u'cds_start': 483, u'protein_end': 161, u'gene_symbol': u'BCL2', u'cdna_end': 1945, u'amino_acids': u'S', u'protein_start': 161, u'gene_id': u'ENSG00000171791', u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 1945, u'transcript_id': u'ENST00000398117', u'codons': u'agC/agT', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 483, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'LOW', u'protein_start': 161, u'protein_end': 161, u'gene_symbol': u'BCL2', u'cdna_end': 976, u'amino_acids': u'S', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'biotype': u'protein_coding', u'gene_symbol_source': u'HGNC', u'cdna_start': 976, u'transcript_id': u'ENST00000444484', u'codons': u'agC/agT', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 483, u'strand': -1, u'cds_start': 483}, {u'impact': u'MODIFIER', u'distance': 4102, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'cds_start': 483, u'protein_start': 161, u'protein_end': 161, u'cdna_start': 629, u'cdna_end': 629, u'impact': u'LOW', u'amino_acids': u'S', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000589955', u'codons': u'agC/agT', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 483, u'strand': -1, u'biotype': u'protein_coding'}]\n",
      "[{u'cds_start': 183, u'impact': u'LOW', u'protein_end': 61, u'cdna_start': 1082, u'cdna_end': 1082, u'amino_acids': u'A', u'hgnc_id': 990, u'codons': u'gcA/gcG', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000333681', u'gene_id': u'ENSG00000171791', u'protein_start': 61, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'C', u'cds_end': 183, u'strand': -1, u'biotype': u'protein_coding'}, {u'cds_start': 183, u'impact': u'LOW', u'protein_end': 61, u'cdna_start': 1645, u'cdna_end': 1645, u'amino_acids': u'A', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000398117', u'codons': u'gcA/gcG', u'protein_start': 61, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'C', u'cds_end': 183, u'strand': -1, u'biotype': u'protein_coding'}, {u'cds_start': 183, u'protein_start': 61, u'protein_end': 61, u'cdna_start': 676, u'cdna_end': 676, u'impact': u'LOW', u'amino_acids': u'A', u'hgnc_id': 990, u'codons': u'gcA/gcG', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000444484', u'gene_id': u'ENSG00000171791', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'C', u'cds_end': 183, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'MODIFIER', u'distance': 4402, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'C', u'strand': -1}, {u'cds_start': 183, u'impact': u'LOW', u'protein_end': 61, u'cdna_start': 329, u'protein_start': 61, u'cdna_end': 329, u'amino_acids': u'A', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000589955', u'codons': u'gcA/gcG', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'C', u'cds_end': 183, u'strand': -1, u'biotype': u'protein_coding'}]\n",
      "[{u'cds_start': 105, u'impact': u'LOW', u'protein_end': 35, u'protein_start': 35, u'cdna_end': 1004, u'gene_symbol': u'BCL2', u'amino_acids': u'V', u'hgnc_id': 990, u'codons': u'gtG/gtC', u'gene_symbol_source': u'HGNC', u'cdna_start': 1004, u'transcript_id': u'ENST00000333681', u'gene_id': u'ENSG00000171791', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'G', u'cds_end': 105, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'LOW', u'protein_start': 35, u'codons': u'gtG/gtC', u'protein_end': 35, u'cdna_start': 1567, u'cdna_end': 1567, u'amino_acids': u'V', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'G', u'cds_end': 105, u'strand': -1, u'cds_start': 105}, {u'impact': u'LOW', u'protein_start': 35, u'gene_id': u'ENSG00000171791', u'protein_end': 35, u'gene_symbol': u'BCL2', u'cdna_end': 598, u'amino_acids': u'V', u'hgnc_id': 990, u'codons': u'gtG/gtC', u'gene_symbol_source': u'HGNC', u'cdna_start': 598, u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'G', u'cds_end': 105, u'strand': -1, u'cds_start': 105}, {u'impact': u'MODIFIER', u'distance': 4480, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'G', u'strand': -1}, {u'impact': u'LOW', u'protein_start': 35, u'protein_end': 35, u'gene_symbol': u'BCL2', u'gene_id': u'ENSG00000171791', u'amino_acids': u'V', u'hgnc_id': 990, u'codons': u'gtG/gtC', u'gene_symbol_source': u'HGNC', u'cdna_start': 251, u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'cdna_end': 251, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'G', u'cds_end': 105, u'strand': -1, u'cds_start': 105}]\n",
      "[{u'impact': u'LOW', u'protein_start': 8, u'protein_end': 8, u'cdna_start': 923, u'cdna_end': 923, u'amino_acids': u'G', u'hgnc_id': 990, u'codons': u'ggG/ggC', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000333681', u'gene_id': u'ENSG00000171791', u'cds_start': 24, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'G', u'cds_end': 24, u'strand': -1, u'biotype': u'protein_coding'}, {u'cds_start': 24, u'protein_start': 8, u'protein_end': 8, u'gene_symbol': u'BCL2', u'cdna_end': 1486, u'impact': u'LOW', u'amino_acids': u'G', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 1486, u'transcript_id': u'ENST00000398117', u'codons': u'ggG/ggC', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'G', u'cds_end': 24, u'strand': -1, u'biotype': u'protein_coding'}, {u'cds_start': 24, u'protein_start': 8, u'protein_end': 8, u'gene_symbol': u'BCL2', u'cdna_end': 517, u'impact': u'LOW', u'amino_acids': u'G', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 517, u'transcript_id': u'ENST00000444484', u'codons': u'ggG/ggC', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'G', u'cds_end': 24, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'MODIFIER', u'distance': 4561, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'G', u'strand': -1}, {u'impact': u'LOW', u'protein_start': 8, u'protein_end': 8, u'gene_symbol': u'BCL2', u'cdna_end': 170, u'amino_acids': u'G', u'hgnc_id': 990, u'codons': u'ggG/ggC', u'gene_symbol_source': u'HGNC', u'cdna_start': 170, u'transcript_id': u'ENST00000589955', u'consequence_terms': [u'synonymous_variant'], u'biotype': u'protein_coding', u'gene_id': u'ENSG00000171791', u'variant_allele': u'G', u'cds_end': 24, u'strand': -1, u'cds_start': 24}]\n",
      "[{u'impact': u'LOW', u'cds_start': 408, u'codons': u'gaG/gaA', u'protein_end': 136, u'gene_symbol': u'BCL2', u'cdna_end': 1307, u'amino_acids': u'E', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 1307, u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'protein_start': 136, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 408, u'strand': -1}, {u'cds_start': 408, u'protein_start': 136, u'protein_end': 136, u'cdna_start': 1870, u'cdna_end': 1870, u'impact': u'LOW', u'amino_acids': u'E', u'hgnc_id': 990, u'codons': u'gaG/gaA', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000398117', u'gene_id': u'ENSG00000171791', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 408, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'LOW', u'protein_start': 136, u'codons': u'gaG/gaA', u'protein_end': 136, u'cdna_start': 901, u'cdna_end': 901, u'amino_acids': u'E', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 408, u'strand': -1, u'cds_start': 408}, {u'impact': u'MODIFIER', u'distance': 4177, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'cds_start': 408, u'protein_start': 136, u'gene_id': u'ENSG00000171791', u'protein_end': 136, u'impact': u'LOW', u'cdna_end': 554, u'gene_symbol': u'BCL2', u'amino_acids': u'E', u'hgnc_id': 990, u'codons': u'gaG/gaA', u'gene_symbol_source': u'HGNC', u'cdna_start': 554, u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 408, u'strand': -1}]\n",
      "[{u'impact': u'LOW', u'cds_start': 318, u'codons': u'cgC/cgT', u'protein_end': 106, u'gene_symbol': u'BCL2', u'cdna_end': 1217, u'amino_acids': u'R', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 1217, u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'protein_start': 106, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 318, u'strand': -1}, {u'impact': u'LOW', u'protein_start': 106, u'codons': u'cgC/cgT', u'protein_end': 106, u'cdna_start': 1780, u'cdna_end': 1780, u'amino_acids': u'R', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 318, u'strand': -1, u'cds_start': 318}, {u'cds_start': 318, u'protein_start': 106, u'protein_end': 106, u'impact': u'LOW', u'cdna_end': 811, u'gene_symbol': u'BCL2', u'amino_acids': u'R', u'hgnc_id': 990, u'codons': u'cgC/cgT', u'gene_symbol_source': u'HGNC', u'cdna_start': 811, u'transcript_id': u'ENST00000444484', u'gene_id': u'ENSG00000171791', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 318, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'MODIFIER', u'distance': 4267, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'cds_start': 318, u'protein_start': 106, u'protein_end': 106, u'impact': u'LOW', u'cdna_end': 464, u'gene_symbol': u'BCL2', u'amino_acids': u'R', u'hgnc_id': 990, u'codons': u'cgC/cgT', u'gene_symbol_source': u'HGNC', u'cdna_start': 464, u'transcript_id': u'ENST00000589955', u'gene_id': u'ENSG00000171791', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 318, u'strand': -1, u'biotype': u'protein_coding'}]\n",
      "[{u'impact': u'LOW', u'protein_start': 49, u'protein_end': 49, u'gene_symbol': u'BCL2', u'cdna_end': 1046, u'amino_acids': u'F', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'biotype': u'protein_coding', u'gene_symbol_source': u'HGNC', u'cdna_start': 1046, u'transcript_id': u'ENST00000333681', u'codons': u'ttC/ttT', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 147, u'strand': -1, u'cds_start': 147}, {u'impact': u'LOW', u'protein_start': 49, u'protein_end': 49, u'gene_symbol': u'BCL2', u'cdna_end': 1609, u'amino_acids': u'F', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'biotype': u'protein_coding', u'gene_symbol_source': u'HGNC', u'cdna_start': 1609, u'transcript_id': u'ENST00000398117', u'codons': u'ttC/ttT', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 147, u'strand': -1, u'cds_start': 147}, {u'impact': u'LOW', u'protein_start': 49, u'protein_end': 49, u'gene_symbol': u'BCL2', u'cdna_end': 640, u'amino_acids': u'F', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'biotype': u'protein_coding', u'gene_symbol_source': u'HGNC', u'cdna_start': 640, u'transcript_id': u'ENST00000444484', u'codons': u'ttC/ttT', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 147, u'strand': -1, u'cds_start': 147}, {u'impact': u'MODIFIER', u'distance': 4438, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'cds_start': 147, u'impact': u'LOW', u'protein_end': 49, u'protein_start': 49, u'cdna_end': 293, u'amino_acids': u'F', u'hgnc_id': 990, u'codons': u'ttC/ttT', u'cdna_start': 293, u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000589955', u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 147, u'strand': -1, u'biotype': u'protein_coding'}]\n",
      "[{u'cds_start': 516, u'protein_start': 172, u'protein_end': 172, u'impact': u'LOW', u'cdna_end': 1415, u'gene_symbol': u'BCL2', u'amino_acids': u'N', u'hgnc_id': 990, u'codons': u'aaC/aaT', u'gene_symbol_source': u'HGNC', u'cdna_start': 1415, u'transcript_id': u'ENST00000333681', u'gene_id': u'ENSG00000171791', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 516, u'strand': -1, u'biotype': u'protein_coding'}, {u'cds_start': 516, u'protein_start': 172, u'protein_end': 172, u'impact': u'LOW', u'cdna_end': 1978, u'gene_symbol': u'BCL2', u'amino_acids': u'N', u'hgnc_id': 990, u'codons': u'aaC/aaT', u'gene_symbol_source': u'HGNC', u'cdna_start': 1978, u'transcript_id': u'ENST00000398117', u'gene_id': u'ENSG00000171791', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 516, u'strand': -1, u'biotype': u'protein_coding'}, {u'cds_start': 516, u'protein_start': 172, u'protein_end': 172, u'impact': u'LOW', u'cdna_end': 1009, u'gene_symbol': u'BCL2', u'amino_acids': u'N', u'hgnc_id': 990, u'codons': u'aaC/aaT', u'gene_symbol_source': u'HGNC', u'cdna_start': 1009, u'transcript_id': u'ENST00000444484', u'gene_id': u'ENSG00000171791', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 516, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'MODIFIER', u'distance': 4069, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'cds_start': 516, u'impact': u'LOW', u'codons': u'aaC/aaT', u'protein_end': 172, u'cdna_start': 662, u'protein_start': 172, u'cdna_end': 662, u'amino_acids': u'N', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 516, u'strand': -1}]\n",
      "[{u'impact': u'LOW', u'protein_start': 119, u'codons': u'Ctg/Ttg', u'protein_end': 119, u'gene_symbol': u'BCL2', u'cdna_end': 1254, u'amino_acids': u'L', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 1254, u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 355, u'strand': -1, u'cds_start': 355}, {u'cds_start': 355, u'protein_start': 119, u'protein_end': 119, u'cdna_start': 1817, u'cdna_end': 1817, u'impact': u'LOW', u'amino_acids': u'L', u'hgnc_id': 990, u'codons': u'Ctg/Ttg', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000398117', u'gene_id': u'ENSG00000171791', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 355, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'LOW', u'protein_start': 119, u'codons': u'Ctg/Ttg', u'protein_end': 119, u'gene_symbol': u'BCL2', u'cdna_end': 848, u'amino_acids': u'L', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 848, u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 355, u'strand': -1, u'cds_start': 355}, {u'impact': u'MODIFIER', u'distance': 4230, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'impact': u'LOW', u'protein_start': 119, u'gene_id': u'ENSG00000171791', u'protein_end': 119, u'gene_symbol': u'BCL2', u'cdna_end': 501, u'amino_acids': u'L', u'hgnc_id': 990, u'codons': u'Ctg/Ttg', u'gene_symbol_source': u'HGNC', u'cdna_start': 501, u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 355, u'strand': -1, u'cds_start': 355}]\n",
      "[{u'impact': u'LOW', u'protein_start': 54, u'codons': u'ggG/ggA', u'protein_end': 54, u'cdna_start': 1061, u'cdna_end': 1061, u'amino_acids': u'G', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 162, u'strand': -1, u'cds_start': 162}, {u'cds_start': 162, u'protein_start': 54, u'codons': u'ggG/ggA', u'protein_end': 54, u'cdna_start': 1624, u'impact': u'LOW', u'cdna_end': 1624, u'amino_acids': u'G', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 162, u'strand': -1}, {u'impact': u'LOW', u'protein_start': 54, u'codons': u'ggG/ggA', u'protein_end': 54, u'cdna_start': 655, u'cdna_end': 655, u'amino_acids': u'G', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 162, u'strand': -1, u'cds_start': 162}, {u'impact': u'MODIFIER', u'distance': 4423, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'impact': u'LOW', u'protein_start': 54, u'protein_end': 54, u'gene_symbol': u'BCL2', u'gene_id': u'ENSG00000171791', u'amino_acids': u'G', u'hgnc_id': 990, u'codons': u'ggG/ggA', u'gene_symbol_source': u'HGNC', u'cdna_start': 308, u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'cdna_end': 308, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 162, u'strand': -1, u'cds_start': 162}]\n",
      "[{u'impact': u'LOW', u'protein_start': 95, u'codons': u'ctG/ctA', u'protein_end': 95, u'cdna_start': 1184, u'cdna_end': 1184, u'amino_acids': u'L', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 285, u'strand': -1, u'cds_start': 285}, {u'impact': u'LOW', u'protein_start': 95, u'protein_end': 95, u'cdna_start': 1747, u'cdna_end': 1747, u'amino_acids': u'L', u'hgnc_id': 990, u'codons': u'ctG/ctA', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000398117', u'consequence_terms': [u'synonymous_variant'], u'biotype': u'protein_coding', u'gene_id': u'ENSG00000171791', u'variant_allele': u'T', u'cds_end': 285, u'strand': -1, u'cds_start': 285}, {u'impact': u'LOW', u'protein_start': 95, u'protein_end': 95, u'gene_symbol': u'BCL2', u'cdna_end': 778, u'amino_acids': u'L', u'hgnc_id': 990, u'codons': u'ctG/ctA', u'gene_symbol_source': u'HGNC', u'cdna_start': 778, u'transcript_id': u'ENST00000444484', u'consequence_terms': [u'synonymous_variant'], u'biotype': u'protein_coding', u'gene_id': u'ENSG00000171791', u'variant_allele': u'T', u'cds_end': 285, u'strand': -1, u'cds_start': 285}, {u'impact': u'MODIFIER', u'distance': 4300, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'impact': u'LOW', u'protein_start': 95, u'codons': u'ctG/ctA', u'protein_end': 95, u'gene_symbol': u'BCL2', u'cdna_end': 431, u'amino_acids': u'L', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 431, u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 285, u'strand': -1, u'cds_start': 285}]\n",
      "[{u'impact': u'LOW', u'protein_start': 22, u'codons': u'aaG/aaA', u'protein_end': 22, u'gene_symbol': u'BCL2', u'cdna_end': 965, u'amino_acids': u'K', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 965, u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 66, u'strand': -1, u'cds_start': 66}, {u'cds_start': 66, u'impact': u'LOW', u'protein_end': 22, u'cdna_start': 1528, u'cdna_end': 1528, u'amino_acids': u'K', u'hgnc_id': 990, u'codons': u'aaG/aaA', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000398117', u'gene_id': u'ENSG00000171791', u'protein_start': 22, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 66, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'LOW', u'protein_start': 22, u'codons': u'aaG/aaA', u'protein_end': 22, u'gene_symbol': u'BCL2', u'cdna_end': 559, u'amino_acids': u'K', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 559, u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 66, u'strand': -1, u'cds_start': 66}, {u'impact': u'MODIFIER', u'distance': 4519, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'impact': u'LOW', u'protein_start': 22, u'codons': u'aaG/aaA', u'protein_end': 22, u'gene_symbol': u'BCL2', u'cdna_end': 212, u'amino_acids': u'K', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 212, u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 66, u'strand': -1, u'cds_start': 66}]\n",
      "[{u'impact': u'LOW', u'protein_start': 155, u'codons': u'ggG/ggA', u'protein_end': 155, u'cdna_start': 1364, u'cdna_end': 1364, u'amino_acids': u'G', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 465, u'strand': -1, u'cds_start': 465}, {u'impact': u'LOW', u'protein_start': 155, u'gene_id': u'ENSG00000171791', u'protein_end': 155, u'gene_symbol': u'BCL2', u'cdna_end': 1927, u'amino_acids': u'G', u'hgnc_id': 990, u'codons': u'ggG/ggA', u'gene_symbol_source': u'HGNC', u'cdna_start': 1927, u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 465, u'strand': -1, u'cds_start': 465}, {u'cds_start': 465, u'protein_start': 155, u'protein_end': 155, u'impact': u'LOW', u'cdna_end': 958, u'gene_symbol': u'BCL2', u'amino_acids': u'G', u'hgnc_id': 990, u'codons': u'ggG/ggA', u'gene_symbol_source': u'HGNC', u'cdna_start': 958, u'transcript_id': u'ENST00000444484', u'gene_id': u'ENSG00000171791', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 465, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'MODIFIER', u'distance': 4120, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'impact': u'LOW', u'protein_start': 155, u'protein_end': 155, u'gene_symbol': u'BCL2', u'gene_id': u'ENSG00000171791', u'amino_acids': u'G', u'hgnc_id': 990, u'codons': u'ggG/ggA', u'gene_symbol_source': u'HGNC', u'cdna_start': 611, u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'cdna_end': 611, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 465, u'strand': -1, u'cds_start': 465}]\n",
      "[{u'cds_start': 456, u'protein_start': 152, u'protein_end': 152, u'cdna_start': 1355, u'impact': u'LOW', u'gene_id': u'ENSG00000171791', u'amino_acids': u'E', u'hgnc_id': 990, u'codons': u'gaG/gaA', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'cdna_end': 1355, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 456, u'strand': -1}, {u'cds_start': 456, u'protein_start': 152, u'protein_end': 152, u'cdna_start': 1918, u'cdna_end': 1918, u'impact': u'LOW', u'amino_acids': u'E', u'hgnc_id': 990, u'codons': u'gaG/gaA', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000398117', u'gene_id': u'ENSG00000171791', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 456, u'strand': -1, u'biotype': u'protein_coding'}, {u'impact': u'LOW', u'cds_start': 456, u'codons': u'gaG/gaA', u'protein_end': 152, u'gene_symbol': u'BCL2', u'cdna_end': 949, u'amino_acids': u'E', u'protein_start': 152, u'gene_id': u'ENSG00000171791', u'hgnc_id': 990, u'gene_symbol_source': u'HGNC', u'cdna_start': 949, u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 456, u'strand': -1}, {u'impact': u'MODIFIER', u'distance': 4129, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'impact': u'LOW', u'cds_start': 456, u'codons': u'gaG/gaA', u'protein_end': 152, u'gene_symbol': u'BCL2', u'cdna_end': 602, u'amino_acids': u'E', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 602, u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'protein_start': 152, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 456, u'strand': -1}]\n",
      "[{u'impact': u'LOW', u'protein_start': 11, u'codons': u'aaC/aaT', u'protein_end': 11, u'cdna_start': 932, u'cdna_end': 932, u'amino_acids': u'N', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 33, u'strand': -1, u'cds_start': 33}, {u'impact': u'LOW', u'cds_start': 33, u'codons': u'aaC/aaT', u'protein_end': 11, u'gene_symbol': u'BCL2', u'cdna_end': 1495, u'amino_acids': u'N', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 1495, u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'protein_start': 11, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 33, u'strand': -1}, {u'impact': u'LOW', u'cds_start': 33, u'codons': u'aaC/aaT', u'protein_end': 11, u'gene_symbol': u'BCL2', u'cdna_end': 526, u'amino_acids': u'N', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 526, u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'protein_start': 11, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 33, u'strand': -1}, {u'impact': u'MODIFIER', u'distance': 4552, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'cds_start': 33, u'protein_start': 11, u'protein_end': 11, u'impact': u'LOW', u'cdna_end': 179, u'amino_acids': u'N', u'hgnc_id': 990, u'codons': u'aaC/aaT', u'cdna_start': 179, u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000589955', u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 33, u'strand': -1, u'biotype': u'protein_coding'}]\n",
      "[{u'impact': u'LOW', u'protein_start': 55, u'codons': u'caC/caT', u'protein_end': 55, u'cdna_start': 1064, u'cdna_end': 1064, u'amino_acids': u'H', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 165, u'strand': -1, u'cds_start': 165}, {u'impact': u'LOW', u'protein_start': 55, u'codons': u'caC/caT', u'protein_end': 55, u'cdna_start': 1627, u'cdna_end': 1627, u'amino_acids': u'H', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 165, u'strand': -1, u'cds_start': 165}, {u'cds_start': 165, u'impact': u'LOW', u'gene_id': u'ENSG00000171791', u'protein_end': 55, u'cdna_start': 658, u'cdna_end': 658, u'amino_acids': u'H', u'hgnc_id': 990, u'codons': u'caC/caT', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'protein_start': 55, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 165, u'strand': -1}, {u'impact': u'MODIFIER', u'distance': 4420, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'cds_start': 165, u'protein_start': 55, u'protein_end': 55, u'gene_symbol': u'BCL2', u'cdna_end': 311, u'impact': u'LOW', u'amino_acids': u'H', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 311, u'transcript_id': u'ENST00000589955', u'codons': u'caC/caT', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 165, u'strand': -1, u'biotype': u'protein_coding'}]\n",
      "[{u'cds_start': 456, u'protein_start': 152, u'protein_end': 152, u'impact': u'LOW', u'cdna_end': 1355, u'amino_acids': u'E', u'hgnc_id': 990, u'codons': u'gaG/gaA', u'cdna_start': 1355, u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000333681', u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 456, u'strand': -1, u'biotype': u'protein_coding'}, {u'cds_start': 456, u'protein_start': 152, u'protein_end': 152, u'impact': u'LOW', u'cdna_end': 1918, u'amino_acids': u'E', u'hgnc_id': 990, u'codons': u'gaG/gaA', u'cdna_start': 1918, u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000398117', u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 456, u'strand': -1, u'biotype': u'protein_coding'}, {u'cds_start': 456, u'protein_start': 152, u'codons': u'gaG/gaA', u'protein_end': 152, u'gene_symbol': u'BCL2', u'cdna_end': 949, u'impact': u'LOW', u'amino_acids': u'E', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 949, u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 456, u'strand': -1}, {u'impact': u'MODIFIER', u'distance': 4129, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'T', u'strand': -1}, {u'cds_start': 456, u'protein_start': 152, u'codons': u'gaG/gaA', u'protein_end': 152, u'gene_symbol': u'BCL2', u'cdna_end': 602, u'impact': u'LOW', u'amino_acids': u'E', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 602, u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'T', u'cds_end': 456, u'strand': -1}]\n",
      "[{u'impact': u'LOW', u'protein_start': 62, u'codons': u'tcC/tcT', u'protein_end': 62, u'cdna_start': 1085, u'cdna_end': 1085, u'amino_acids': u'S', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000333681', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 186, u'strand': -1, u'cds_start': 186}, {u'impact': u'LOW', u'cds_start': 186, u'codons': u'tcC/tcT', u'protein_end': 62, u'gene_symbol': u'BCL2', u'cdna_end': 1648, u'amino_acids': u'S', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'cdna_start': 1648, u'transcript_id': u'ENST00000398117', u'biotype': u'protein_coding', u'protein_start': 62, u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 186, u'strand': -1}, {u'impact': u'LOW', u'protein_start': 62, u'codons': u'tcC/tcT', u'protein_end': 62, u'cdna_start': 679, u'cdna_end': 679, u'amino_acids': u'S', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000444484', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 186, u'strand': -1, u'cds_start': 186}, {u'impact': u'MODIFIER', u'distance': 4399, u'gene_id': u'ENSG00000267701', u'gene_symbol_source': u'Clone_based_vega_gene', u'gene_symbol': u'RP11-28F1.2', u'transcript_id': u'ENST00000588307', u'biotype': u'sense_intronic', u'consequence_terms': [u'upstream_gene_variant'], u'variant_allele': u'A', u'strand': -1}, {u'impact': u'LOW', u'protein_start': 62, u'codons': u'tcC/tcT', u'protein_end': 62, u'cdna_start': 332, u'cdna_end': 332, u'amino_acids': u'S', u'hgnc_id': 990, u'gene_id': u'ENSG00000171791', u'gene_symbol_source': u'HGNC', u'gene_symbol': u'BCL2', u'transcript_id': u'ENST00000589955', u'biotype': u'protein_coding', u'consequence_terms': [u'synonymous_variant'], u'variant_allele': u'A', u'cds_end': 186, u'strand': -1, u'cds_start': 186}]\n"
     ]
    }
   ],
   "source": [
    "input_col = ['Hugo_Symbol','Sample_ID','Protein_Change','Mutation_Type']\n",
    "df_input_plot = pd.DataFrame(columns = input_col)\n",
    "l1 = []\n",
    "l2 = []\n",
    "l3 = []\n",
    "l4 = []\n",
    "\n",
    "for c in df_mut.iloc[:,2:10].columns:\n",
    "    for s,v in df_mut[df_mut[c].notnull()][c].iteritems():\n",
    "        for mut in [v]:\n",
    "            mutation = re.findall('g.chr\\d+:\\d+[A-Z]+>[A-Z]+', v)\n",
    "            for muts in mutation:\n",
    "                chrom = re.findall(r'\\d+', muts.split(':')[0])[0]\n",
    "                startp = re.findall(r'\\d+', muts.split(':')[1])[0]\n",
    "                var = muts[-1]\n",
    "                mut = str(chrom)+':'+str(startp)+'-'+str(startp)+'/'+str(var)\n",
    "                res = get_overlapping_cons(mut)\n",
    "                try:\n",
    "                    aa = res[0]['transcript_consequences'][0]['amino_acids']\n",
    "                    print res[0]['transcript_consequences']\n",
    "                    proteinpos = res[0]['transcript_consequences'][0]['protein_end']\n",
    "                    if '/' in aa:\n",
    "                        rp = aa.split('/')[0]\n",
    "                        mp = aa.split('/')[1]\n",
    "                        pchange = str(rp)+str(proteinpos)+str(mp)\n",
    "                    else:\n",
    "                        pchange = str(aa)+str(proteinpos)+str(aa)\n",
    "                except KeyError:\n",
    "                    print muts, c\n",
    "                    continue\n",
    "\n",
    "                l1.append('BCL2')\n",
    "                l2.append(s)\n",
    "                l3.append(pchange)\n",
    "                l4.append(c)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "input_col = ['Hugo_Symbol','Sample_ID','Protein_Change','Mutation_Type']\n",
    "df_input_plot = pd.DataFrame(columns = input_col)\n",
    "l1 = []\n",
    "l2 = []\n",
    "l3 = []\n",
    "l4 = []\n",
    "\n",
    "for s,v in df_mut[df_mut['Silent'].notnull()]['Silent'].iteritems():\n",
    "    for mut in [v]:\n",
    "        mutation = re.findall('g.chr\\d+:\\d+[A-Z]+>[A-Z]+', v)\n",
    "        for muts in mutation:\n",
    "            chrom = re.findall(r'\\d+', muts.split(':')[0])[0]\n",
    "            startp = re.findall(r'\\d+', muts.split(':')[1])[0]\n",
    "            var = muts[-1]\n",
    "            mut = str(chrom)+':'+str(startp)+'-'+str(startp)+'/'+str(var)\n",
    "            res = get_overlapping_cons(mut)\n",
    "\n",
    "            aa = res[0]['transcript_consequences'][0]['amino_acids']\n",
    "            proteinpos = res[0]['transcript_consequences'][0]['protein_end']\n",
    "            pchange = str(aa)+str(proteinpos)+str(aa)\n",
    "\n",
    "            l1.append('BCL2')\n",
    "            l2.append(s)\n",
    "            l3.append(pchange)\n",
    "            l4.append('Silent')\n",
    "            \n",
    "df_input_plot['Hugo_Symbol'] = l1\n",
    "df_input_plot['Sample_ID'] = l2\n",
    "df_input_plot['Protein_Change'] = l3\n",
    "df_input_plot['Mutation_Type'] = l4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "df_input_plot.to_csv('track_input.txt', sep = '\\t', index = None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "df_input_plot['Hugo_Symbol'] = l1\n",
    "df_input_plot['Sample_ID'] = l2\n",
    "df_input_plot['Protein_Change'] = l3\n",
    "df_input_plot['Mutation_Type'] = l4"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get fusion break point"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'pd' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-1-c88bb2716111>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mffus\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'../data/anno_refs/pcawg_fusion.tsv'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mdffus\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mffus\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msep\u001b[0m\u001b[0;34m=\u001b[0m \u001b[0;34m'\\t'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0mdffus\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdffus\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mset_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'wgs_aliquot_id'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;31m# select BCL2 fusion\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0;31m# dfbcl2 = dffus[(dffus['gene_id1']=='ENSG00000171791') | (dffus['gene_id2'] == 'ENSG00000171791')]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'pd' is not defined"
     ]
    }
   ],
   "source": [
    "ffus = '../data/anno_refs/pcawg_fusion.tsv'\n",
    "dffus = pd.read_csv(ffus, sep= '\\t')\n",
    "dffus = dffus.set_index('wgs_aliquot_id')\n",
    "# select BCL2 fusion\n",
    "# dfbcl2 = dffus[(dffus['gene_id1']=='ENSG00000171791') | (dffus['gene_id2'] == 'ENSG00000171791')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "# Select dataframe b overlapping patients\n",
    "idx1 = dffus.index\n",
    "idx2 = df_mut.index\n",
    "interidx = idx1.intersection(idx2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "dffus_lymph = dffus.loc[interidx]\n",
    "dfbcl2 = dffus_lymph[(dffus_lymph['gene_id1']=='ENSG00000171791') | (dffus_lymph['gene_id2'] == 'ENSG00000171791')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "for p in dfbcl2.index:\n",
    "    if not isinstance(dfbcl2.loc[p,'class'],str):\n",
    "        df_mut.loc[p,'fusion'] = \"Y\"\n",
    "        df_mut.loc[p,'fusion_class'] = dfbcl2.loc[p,'class'].values[0]\n",
    "        df_mut.loc[p,'fusion_point'] = int(dfbcl2.loc[p,'break1'][0])\n",
    "    else:\n",
    "        df_mut.loc[p,'fusion'] = \"Y\"\n",
    "        df_mut.loc[p,'fusion_class'] = dfbcl2.loc[p,'class']\n",
    "        df_mut.loc[p,'fusion_point'] = int(dfbcl2.loc[p,'break1'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>type</th>\n",
       "      <th>Silent</th>\n",
       "      <th>fusion</th>\n",
       "      <th>fusion_class</th>\n",
       "      <th>fusion_point</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2c3ada8b-9650-4859-9857-43209c0f7cea</th>\n",
       "      <td>Not Follicular</td>\n",
       "      <td>g.chr18:60985833G&gt;A</td>\n",
       "      <td>Y</td>\n",
       "      <td>SV independent</td>\n",
       "      <td>60843599.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3e94aa64-7dd3-4d63-a5de-b4050d3dfafa</th>\n",
       "      <td>Follicular</td>\n",
       "      <td>g.chr18:60985833G&gt;A</td>\n",
       "      <td>Y</td>\n",
       "      <td>composite</td>\n",
       "      <td>60812505.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ebe0ed67-2d3f-45cd-8f9b-4912595b16a0</th>\n",
       "      <td>Follicular</td>\n",
       "      <td>g.chr18:60985834C&gt;T</td>\n",
       "      <td>Y</td>\n",
       "      <td>composite</td>\n",
       "      <td>60794689.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>f2c08ac1-4755-4c36-a11d-338d8df9fa0d</th>\n",
       "      <td>Follicular</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Y</td>\n",
       "      <td>composite</td>\n",
       "      <td>60794689.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>f60007b3-0603-4ad1-8b47-d2d4c4c621e4</th>\n",
       "      <td>Follicular</td>\n",
       "      <td>g.chr18:60985444C&gt;T</td>\n",
       "      <td>Y</td>\n",
       "      <td>composite</td>\n",
       "      <td>60847806.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                type               Silent  \\\n",
       "2c3ada8b-9650-4859-9857-43209c0f7cea  Not Follicular  g.chr18:60985833G>A   \n",
       "3e94aa64-7dd3-4d63-a5de-b4050d3dfafa      Follicular  g.chr18:60985833G>A   \n",
       "ebe0ed67-2d3f-45cd-8f9b-4912595b16a0      Follicular  g.chr18:60985834C>T   \n",
       "f2c08ac1-4755-4c36-a11d-338d8df9fa0d      Follicular                  NaN   \n",
       "f60007b3-0603-4ad1-8b47-d2d4c4c621e4      Follicular  g.chr18:60985444C>T   \n",
       "\n",
       "                                     fusion    fusion_class  fusion_point  \n",
       "2c3ada8b-9650-4859-9857-43209c0f7cea      Y  SV independent    60843599.0  \n",
       "3e94aa64-7dd3-4d63-a5de-b4050d3dfafa      Y       composite    60812505.0  \n",
       "ebe0ed67-2d3f-45cd-8f9b-4912595b16a0      Y       composite    60794689.0  \n",
       "f2c08ac1-4755-4c36-a11d-338d8df9fa0d      Y       composite    60794689.0  \n",
       "f60007b3-0603-4ad1-8b47-d2d4c4c621e4      Y       composite    60847806.0  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_mut[df_mut['fusion'] == 'Y'][['type','Silent','fusion','fusion_class','fusion_point']]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get structure variant"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "sv_dir = '/gpfs/scratch/yur97/yur97/gitlab/pcawg-to-mutsigcv/anno_ref/icgc/open'\n",
    "lsv = listdir(sv_dir)\n",
    "\n",
    "for f in lsv:\n",
    "    if f.endswith('bedpe'):\n",
    "        patient = f.split('.')[0]\n",
    "        dfsv = pd.read_csv(join(sv_dir,f), sep = '\\t')\n",
    "        df_bcl = dfsv[((dfsv['chrom1']==18) & (dfsv['start1'] >60790579) & (dfsv['end1'] <60987361)) | \\\n",
    "                  ((dfsv['chrom2']==18) & (dfsv['start2'] >60790579) & (dfsv['end2'] <60987361))]\n",
    "        \n",
    "        \n",
    "        if not df_bcl.empty:\n",
    "            if (df_bcl['chrom2'] == 18).all():\n",
    "                df_mut.loc[patient,'sv'] = 'Y'\n",
    "                df_mut.loc[patient,'sv_class'] = df_bcl['svclass'].tolist()[0]\n",
    "                df_mut.loc[patient,'sv_start'] = df_bcl['start2'].tolist()[0]\n",
    "                df_mut.loc[patient,'sv_end'] = df_bcl['end2'].tolist()[0]\n",
    "            elif (df_bcl['chrom1'] == 18).all():\n",
    "                df_mut.loc[patient,'sv'] = 'Y'\n",
    "                df_mut.loc[patient,'sv_class'] = df_bcl['svclass'].tolist()[0]\n",
    "                df_mut.loc[patient,'sv_start'] = df_bcl['start1'].tolist()[0]\n",
    "                df_mut.loc[patient,'sv_end'] = df_bcl['end1'].tolist()[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "df_fusion = df_mut[df_mut['fusion'] == 'Y'][['type','Silent','fusion_class','fusion_point']]\n",
    "df_fusion['fusion_point'] = df_fusion['fusion_point'].astype('int64')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df_structure = df_mut[df_mut['sv'] == 'Y'][['type','Silent','sv_class','sv_start']]\n",
    "df_structure['sv_start'] = df_structure['sv_start'].astype('int64')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "df_structure = df_structure.iloc[:-1,:]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check the expression of BCL2 and AID/ sno gene and RPL gene  \n",
    "RP11-28F1.2 ENSG00000267701 ENST00000588307.1  \n",
    "RP11-299P1.2 ENSG00000267766 \tENST00000587475.1  \n",
    "snoU13 ENSG00000238988 ENST00000458827.1\t\n",
    "Expect: same expression?\n",
    "If not same expression then why?\n",
    "I hope to find which one express, so that it could be related to function  \n",
    "['ENSG00000267701','ENSG00000267766','ENSG00000238988']\n",
    "['ENST00000588307.1','ENST00000587475.1','ENST00000458827.1']\n",
    "AID: ENSG00000111732\n",
    "BCL2: ENSG00000171791"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "df_mut = df_mut.iloc[1:,:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index([u'2c3ada8b-9650-4859-9857-43209c0f7cea',\n",
       "       u'3e94aa64-7dd3-4d63-a5de-b4050d3dfafa',\n",
       "       u'ebe0ed67-2d3f-45cd-8f9b-4912595b16a0',\n",
       "       u'f2c08ac1-4755-4c36-a11d-338d8df9fa0d',\n",
       "       u'f60007b3-0603-4ad1-8b47-d2d4c4c621e4'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_fusion.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "df_exp = pd.read_csv(join(exp_dir,gene_tophat), sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "df_exp_bcl2 = df_exp[df_exp['feature'].str.contains('ENSG00000171791')]\n",
    "df_exp_aid = df_exp[df_exp['feature'].str.contains('ENSG00000111732')]"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "exp_dir = '../anno_ref/icgc/pcawg_rnaseq/'\n",
    "gene_gtex = 'GTEx.tophat2.gene.fpkm.tsv'\n",
    "gene_tophat = 'tophat_star_fpkm.v2.aliquot_gl.tsv'\n",
    "gene_tophatuq = 'tophat_star_fpkm_uq.v2_aliquot_gl.tsv'\n",
    "transcript = 'pcawg.rnaseq.transcript.expr.tpm.tsv'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "df_exp_info = pd.read_csv(join(exp_dir,'rnaseq.metadata.tsv'), sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "df_mut_bcls = df_mut[df_mut['Silent'].notnull()]['Silent']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "col = ['q1','q2','q3','q4']\n",
    "df_stat = pd.DataFrame(0,index = ['bcl2', 'aid'],columns = col)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "col = ['q1','q2','q3','q4']\n",
    "df_stat_bclq4 = pd.DataFrame(0,index = ['aid'],columns = col)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "list_bcl2 = []\n",
    "list_aid = []\n",
    "for mut in set(ls):\n",
    "    tit = mut\n",
    "    # get patients with mutation:\n",
    "    for rows in df_mut_bcls:\n",
    "        if mut in rows:\n",
    "            list_patients = df_mut_bcls[df_mut_bcls == rows].index.tolist()\n",
    "            \n",
    "            list_exp_syn = []\n",
    "            list_exp_other = []\n",
    "            list_exp_syn_aid = []\n",
    "            list_exp_other_aid = []\n",
    "            for pat in list_patients:\n",
    "                if pat in df_fusion.index.tolist():\n",
    "                    tit += ' Fusion with IgG'\n",
    "                if pat in df_structure.index.tolist():\n",
    "                    tit += ' SV'\n",
    "                exp_id = df_exp_info[df_exp_info['wgs_aliquot_id'] == pat]['aliquot_id'].values\n",
    "                exp_val = df_exp_bcl2[exp_id].values\n",
    "                exp_val_aid = df_exp_aid[exp_id].values\n",
    "                if exp_val:\n",
    "                    list_exp_syn.append(exp_val[0][0])\n",
    "                    list_exp_syn_aid.append(exp_val_aid[0][0])\n",
    "                    if pat in df_structure.index.tolist():\n",
    "                        list_bcl2.append(exp_val[0][0])\n",
    "                        list_aid.append(exp_val_aid[0][0])\n",
    "            \n",
    "            for p in df_mut.index.tolist():\n",
    "                if p not in list_patients:\n",
    "                    exp_id = df_exp_info[df_exp_info['wgs_aliquot_id'] == p]['aliquot_id'].values\n",
    "                    exp_val = df_exp_bcl2[exp_id].values\n",
    "                    exp_val_aid = df_exp_aid[exp_id].values\n",
    "                    if exp_val:\n",
    "                        list_exp_other.append(exp_val[0][0])\n",
    "                        list_exp_other_aid.append(exp_val_aid[0][0])\n",
    "    \n",
    "    dict_syn = {'syn_bcl2': list_exp_syn, 'syn_aid': list_exp_syn_aid}\n",
    "    dict_other = {'other_bcl2': list_exp_other, 'other_aid': list_exp_other_aid}\n",
    "    df_syn = pd.DataFrame(dict_syn) \n",
    "    df_other = pd.DataFrame(dict_other) \n",
    "    \n",
    "    col = ['q1','q2','q3','q4']\n",
    "    df_stat = pd.DataFrame(0,index = ['bcl2', 'aid'],columns = col)\n",
    "    other25 = df_other.quantile(.25)['other_bcl2']\n",
    "    other50 = df_other.quantile(.50)['other_bcl2']\n",
    "    other75 = df_other.quantile(.75)['other_bcl2']\n",
    "    other25aid = df_other.quantile(.25)['other_aid']\n",
    "    other50aid = df_other.quantile(.50)['other_aid']\n",
    "    other75aid = df_other.quantile(.75)['other_aid']\n",
    "    for x in df_syn['syn_bcl2']:\n",
    "        if x < other25:\n",
    "            df_stat.loc['bcl2','q1'] +=1\n",
    "        elif x >= other25 and x < other50:\n",
    "            df_stat.loc['bcl2','q2'] +=1\n",
    "        elif x >= other50 and x < other75:\n",
    "            df_stat.loc['bcl2','q3'] +=1\n",
    "        elif x >= other75:\n",
    "            df_stat.loc['bcl2','q4'] +=1\n",
    "            for x in df_syn['syn_aid']:\n",
    "                if x < other25aid:\n",
    "                    df_stat_bclq4.loc['aid','q1'] +=1\n",
    "                elif x >= other25aid and x < other50aid:\n",
    "                    df_stat_bclq4.loc['aid','q2'] +=1\n",
    "                elif x >= other50aid and x < other75aid:\n",
    "                    df_stat_bclq4.loc['aid','q3'] +=1\n",
    "                elif x >= other75aid:\n",
    "                    df_stat_bclq4.loc['aid','q4'] +=1\n",
    "    for x in df_syn['syn_aid']:\n",
    "        if x < other25aid:\n",
    "            df_stat.loc['aid','q1'] +=1\n",
    "        elif x >= other25aid and x < other50aid:\n",
    "            df_stat.loc['aid','q2'] +=1\n",
    "        elif x >= other50aid and x < other75aid:\n",
    "            df_stat.loc['aid','q3'] +=1\n",
    "        elif x >= other75aid:\n",
    "            df_stat.loc['aid','q4'] +=1\n",
    "\n",
    "\n",
    "    \n",
    "#     ax1 = df_other.plot(kind='scatter', x='other_bcl2', y='other_aid', color='r', alpha = 0.5, title = tit )    \n",
    "#     ax2 = df_syn.plot(kind='scatter', x='syn_bcl2', y='syn_aid', color='g', ax=ax1)\n",
    "#     plt.plot([df_other.quantile(.25)['other_bcl2'], df_other.quantile(.25)['other_bcl2']], [0, 150], '--')\n",
    "#     plt.plot([df_other.quantile(.50)['other_bcl2'], df_other.quantile(.50)['other_bcl2']], [0, 150], '--')\n",
    "#     plt.plot([df_other.quantile(.75)['other_bcl2'], df_other.quantile(.75)['other_bcl2']], [0, 150], '--')\n",
    "#     plt.plot([0, 150],[df_other.quantile(.25)['other_aid'], df_other.quantile(.25)['other_aid']], '--')\n",
    "#     plt.plot([0, 150],[df_other.quantile(.50)['other_aid'], df_other.quantile(.50)['other_aid']], '--')\n",
    "#     plt.plot([0, 150],[df_other.quantile(.75)['other_aid'], df_other.quantile(.75)['other_aid']], '--')\n",
    "#     plt.show()\n",
    "            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "list_bcl2\n",
    "list_aid\n",
    "dict_lexp = {'bcl2': list_bcl2, 'aid': list_aid}\n",
    "df_lexp = pd.DataFrame(dict_lexp) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[0.66610017733000004,\n",
       " 0.44642940620800003,\n",
       " 4.14261165987,\n",
       " 4.14261165987,\n",
       " 19.049046671099997,\n",
       " 0.84703191540000011,\n",
       " 4.1966716946199991,\n",
       " 5.7113879087999999,\n",
       " 0.0,\n",
       " 4.1966716946199991,\n",
       " 3.7197345452800006,\n",
       " 0.44642940620800003,\n",
       " 2.9579352208599996,\n",
       " 2.9579352208599996,\n",
       " 2.9579352208599996,\n",
       " 0.44642940620800003,\n",
       " 3.7111029606199999,\n",
       " 3.7111029606199999,\n",
       " 5.7113879087999999]"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list_aid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAX4AAAD8CAYAAABw1c+bAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDIuMS4wLCBo\ndHRwOi8vbWF0cGxvdGxpYi5vcmcvpW3flQAAFMxJREFUeJzt3X+sX/V93/Hna8ZsHmEzhBsKxoSU\nIS9pKTb7ih/yhkhSMFAUvChbQaQjXRS3FZESqWOFLmo2mimZUNNWpQoyPxa6UpIuAQetJAYlmZIg\nQrjGUEMI44dIsc3wTR0DWawmdt/743suuVy+X9/r+/1ef33veT6kr77nfM7nfM/nHvi+zvHnfM73\npKqQJLXHPxh1AyRJh5bBL0ktY/BLUssY/JLUMga/JLWMwS9JLTNj8CdZmeTrSZ5M8kSSjzTlxya5\nP8nTzfsxfda/qqnzdJKrhv0HSJIOTmYax5/kBOCEqnokydHAFmA98AFgd1V9Ksm1wDFV9TvT1j0W\nGAc6QDXr/ouq+uHQ/xJJ0qzMeMZfVS9W1SPN9KvAk8AK4DLg9qba7XQPBtOtA+6vqt1N2N8PXDSM\nhkuS5uaIg6mc5BRgDfAQcHxVvQjdg0OSt/RYZQXwwpT57U3ZAR133HF1yimnHEzTJKnVtmzZ8oOq\nGptN3VkHf5I3AV8EPlpVrySZ1Wo9ynr2LSXZAGwAOPnkkxkfH59t0ySp9ZJ8f7Z1ZzWqJ8lSuqF/\nR1Xd1RS/1PT/T14H2NVj1e3AyinzJwE7e22jqjZWVaeqOmNjszpoSZLmYDajegLcCjxZVZ+esuge\nYHKUzlXAl3qsvhm4MMkxzaifC5sySdKIzOaMfy3wa8C7kjzavC4BPgVckORp4IJmniSdJLcAVNVu\n4PeBh5vX9U2ZJGlEZhzOOQqdTqfs45ek2Uuypao6s6nrnbuS1DIHNZxTWog2bd3BDZufYueevZy4\nfBnXrFvF+jUzjiqWFi2DX4vapq07uO6ubez96X4AduzZy3V3bQMw/NVadvVoUbth81Ovhf6kvT/d\nzw2bnxpRi6TRM/i1qO3cs/egyqU2MPi1qJ24fNlBlUttYPBrUbtm3SqWLV3yurJlS5dwzbpVI2qR\nNHpe3NWiNnkB11E90s8Y/Fr01q9ZYdBLU9jVI0ktY/BLUssY/JLUMga/JLWMwS9JLWPwS1LLGPyS\n1DIGvyS1zIw3cCW5DbgU2FVVv9iUfR6YvOd9ObCnqlb3WPd54FVgP7Bvtk+HkSTNn9ncuftZ4Ebg\nzyYLqupXJ6eT/AHw8gHWf2dV/WCuDZQkDdeMwV9V30hySq9lSQL8W+Bdw22WJGm+DNrH/6+Al6rq\n6T7LC7gvyZYkGw70QUk2JBlPMj4xMTFgsyRJ/Qwa/FcAdx5g+dqqOhO4GLg6yXn9KlbVxqrqVFVn\nbGxswGZJkvqZc/AnOQJ4L/D5fnWqamfzvgu4GzhrrtuTJA3HIGf8vwx8r6q291qY5KgkR09OAxcC\njw+wPUnSEMwY/EnuBB4EViXZnuSDzaLLmdbNk+TEJPc2s8cD30ryGPAd4K+q6ivDa7okaS5mM6rn\nij7lH+hRthO4pJl+DjhjwPZJkobMO3clqWUMfklqGYNfklrG4JekljH4JallDH5JahmDX5JaxuCX\npJYx+CWpZQx+SWoZg1+SWsbgl6SWMfglqWUMfklqGYNfklrG4JekljH4JallZvPoxduS7Ery+JSy\n/5xkR5JHm9clfda9KMlTSZ5Jcu0wGy5JmpvZnPF/FrioR/kfVtXq5nXv9IVJlgB/ClwMvAO4Isk7\nBmmsJGlwMwZ/VX0D2D2Hzz4LeKaqnquqnwCfAy6bw+dIkoZokD7+Dyf566Yr6Jgey1cAL0yZ396U\n9ZRkQ5LxJOMTExMDNEuSdCBzDf7PAKcCq4EXgT/oUSc9yqrfB1bVxqrqVFVnbGxsjs2SJM1kTsFf\nVS9V1f6q+nvgZrrdOtNtB1ZOmT8J2DmX7UmShmdOwZ/khCmz/xp4vEe1h4HTkrwtyZHA5cA9c9me\nJGl4jpipQpI7gfOB45JsBz4OnJ9kNd2um+eB32jqngjcUlWXVNW+JB8GNgNLgNuq6ol5+SskSbOW\nqr7d7iPT6XRqfHx81M2QpAUjyZaq6symrnfuSlLLGPyS1DIGvyS1jMEvSS1j8EtSyxj8ktQyBr8k\ntYzBL0ktY/BLUssY/JLUMga/JLWMwS9JLWPwS1LLGPyS1DIGvyS1jMEvSS1j8EtSy8wY/EluS7Ir\nyeNTym5I8r0kf53k7iTL+6z7fJJtSR5N4iO1JOkwMJsz/s8CF00rux/4xar6JeD/ANcdYP13VtXq\n2T4STJI0v2YM/qr6BrB7Wtl9VbWvmf02cNI8tE2SNA+G0cf/74Ev91lWwH1JtiTZcKAPSbIhyXiS\n8YmJiSE0S5LUy0DBn+Q/AfuAO/pUWVtVZwIXA1cnOa/fZ1XVxqrqVFVnbGxskGZJkg5gzsGf5Crg\nUuDKqqpedapqZ/O+C7gbOGuu25MkDcecgj/JRcDvAO+pqh/3qXNUkqMnp4ELgcd71ZUkHTqzGc55\nJ/AgsCrJ9iQfBG4Ejgbub4Zq3tTUPTHJvc2qxwPfSvIY8B3gr6rqK/PyV0iSZu2ImSpU1RU9im/t\nU3cncEkz/RxwxkCtkyQNnXfuSlLLGPyS1DIGvyS1jMEvSS1j8EtSyxj8ktQyBr8ktYzBL0ktY/BL\nUssY/JLUMga/JLWMwS9JLWPwS1LLGPyS1DIGvyS1jMEvSS1j8EtSy8wq+JPclmRXksenlB2b5P4k\nTzfvx/RZ96qmztPNA9olSSM02zP+zwIXTSu7FvhqVZ0GfLWZf50kxwIfB84GzgI+3u8AIUk6NGYV\n/FX1DWD3tOLLgNub6duB9T1WXQfcX1W7q+qHwP288QAiSTqEBunjP76qXgRo3t/So84K4IUp89ub\nsjdIsiHJeJLxiYmJAZolSTqQ+b64mx5l1atiVW2sqk5VdcbGxua5WZLUXoME/0tJTgBo3nf1qLMd\nWDll/iRg5wDblCQNaJDgvweYHKVzFfClHnU2AxcmOaa5qHthUyZJGpHZDue8E3gQWJVke5IPAp8C\nLkjyNHBBM0+STpJbAKpqN/D7wMPN6/qmTJI0Iqnq2eU+Up1Op8bHx0fdDElaMJJsqarObOp6564k\ntYzBL0ktY/BLUssY/JLUMga/JLWMwS9JLWPwS1LLGPyS1DIGvyS1jMEvSS1j8EtSyxj8ktQyBr8k\ntYzBL0ktY/BLUssY/JLUMga/JLXMnIM/yaokj055vZLko9PqnJ/k5Sl1fm/wJkuSBnHEXFesqqeA\n1QBJlgA7gLt7VP1mVV061+1IkoZrWF097waerarvD+nzJEnzZFjBfzlwZ59l5yZ5LMmXk/xCvw9I\nsiHJeJLxiYmJITVLkjTdwMGf5EjgPcD/7LH4EeCtVXUG8CfApn6fU1Ubq6pTVZ2xsbFBmyVJ6mMY\nZ/wXA49U1UvTF1TVK1X1o2b6XmBpkuOGsE1J0hwNI/ivoE83T5KfS5Jm+qxme387hG1KkuZozqN6\nAJL8Y+AC4DemlP0mQFXdBLwP+K0k+4C9wOVVVYNsU5I0mIGCv6p+DLx5WtlNU6ZvBG4cZBuSpOHy\nzl1JahmDX5JaxuCXpJYx+CWpZQx+SWoZg1+SWsbgl6SWMfglqWUMfklqGYNfklpmoJ9skGbjY5u2\ncedDL7C/iiUJV5y9kk+sP33UzZJay+DXvPrYpm38+bf/5rX5/VWvzRv+0mjY1aN5dedDLxxUuaT5\nZ/BrXu3v8yvc/colzT+DX/NqSfc5PLMulzT/DH7NqyvOXnlQ5ZLmnxd3Na8mL+A6qkc6fGTQJyEm\neR54FdgP7KuqzrTlAf4YuAT4MfCBqnrkQJ/Z6XRqfHx8oHZJUpsk2TI9f/sZ1hn/O6vqB32WXQyc\n1rzOBj7TvEuSRuBQ9PFfBvxZdX0bWJ7khEOwXUlSD8MI/gLuS7IlyYYey1cAUwdtb2/KXifJhiTj\nScYnJiaG0CxJUi/DCP61VXUm3S6dq5OcN215r3F7b7iwUFUbq6pTVZ2xsbEhNEuS1MvAwV9VO5v3\nXcDdwFnTqmwHpo7dOwnYOeh2JUlzM1DwJzkqydGT08CFwOPTqt0D/Lt0nQO8XFUvDrJdSdLcDTqq\n53jg7u6ITY4A/qKqvpLkNwGq6ibgXrpDOZ+hO5zz1wfcpiRpAAMFf1U9B5zRo/ymKdMFXD3IdnRw\n/BlkSQfinbuLjD+DPDweQLVY+Vs9i4w/gzwckwfQyV8RnTyAfmzTthG3TBqcwb/I+DPIw+EBVIuZ\nwb/I+DPIw+EBVIuZffyLzBVnr3xdH//U8sVg09Yd3LD5KXbu2cuJy5dxzbpVrF/zhhvBB7Yk6Rny\nHkAXBq/PHJhn/IvMJ9afzvvPOfm1gFqS8P5zTl4U/9Nv2rqD6+7axo49eylgx569XHfXNjZt3TH0\nbfkcgYXL6zMzG/hnmeeDP8usXtZ+6mvs2LP3DeUrli/jgWvfNfTteda4MJ163b19/7X27CcvGUGL\nDo1R/CyzNO929gj9A5UP6hPrTzfoFyCvz8zMrh4tGCcuX3ZQ5WonBzjMzODXgnHNulUsW7rkdWXL\nli7hmnWrRtQiHY68PjMzu3q0YEyO3jkUo3q0cPmc55l5cVeSFgEv7o7IlTc/yAPP7n5tfu2px3LH\nh84dYYsk6Y3s4x+S6aEP8MCzu7ny5gdH1CJJ6s3gH5LpoT9TuSSNisEvSS1j8EtSy8w5+JOsTPL1\nJE8meSLJR3rUOT/Jy0kebV6/N1hzD19rTz32oMolaVQGOePfB/x2Vb0dOAe4Osk7etT7ZlWtbl7X\nD7C9w9odHzr3DSHvqB5Jh6M5D+esqheBF5vpV5M8CawAvjukti04hrykhWAoffxJTgHWAA/1WHxu\nkseSfDnJLxzgMzYkGU8yPjExMYxmSZJ6GDj4k7wJ+CLw0ap6ZdriR4C3VtUZwJ8Am/p9TlVtrKpO\nVXXGxsYGbZYkqY+Bgj/JUrqhf0dV3TV9eVW9UlU/aqbvBZYmOW6QbUqSBjPIqJ4AtwJPVtWn+9T5\nuaYeSc5qtve3c92mJGlwg/xWz1rg14BtSR5tyn4XOBmgqm4C3gf8VpJ9wF7g8jocfxVOklpkkFE9\n3wIO+GSDqroRuHGu25AkDZ937kpSyxj8ktQyBr8ktYzBL0ktY/BLUssY/JLUMga/JLWMwS9JLTPI\nnbsakk1bd3DD5qfYuWcvJy5fxjXrVrF+zYpRN0vSImXwj9imrTu47q5t7P3pfgB27NnLdXdtAzD8\nJc2LVgb/4XSGfcPmp14L/Ul7f7qfGzY/ZfBLmhetC/7D7Qx75569B1UuSYNaNMF/5c0P8sCzu1+b\n7/e828PtDPvE5cvY0SPkT1y+7JC3RVI7LIpRPdNDH+CBZ3dz5c0PvqHu4XaGfc26VSxbuuR1ZcuW\nLuGadatG0h5Ji9+iCP7poX+g8n5n0qM6w16/ZgWffO/prFi+jAArli/jk+893f59SfNm0XT1zNY1\n61a9ro8fRn+GvX7NCoNe0iEz6DN3L0ryVJJnklzbY/k/TPL5ZvlDSU4ZZHvD4Bm2pLab8xl/kiXA\nnwIXANuBh5PcU1XfnVLtg8APq+qfJbkc+G/Arw7S4F7Wnnpsz26dtace27O+Z9iSDieHeoj5IGf8\nZwHPVNVzVfUT4HPAZdPqXAbc3kx/AXj35MPXh+mOD537hpDvN6pHkg4nk0PMd+zZS/GzIeabtu6Y\nt20O0se/Anhhyvx24Ox+dapqX5KXgTcDPxhguz0Z8pIWolEMMR/kjL/XmXvNoU63YrIhyXiS8YmJ\niQGaJUkLxyiGmA8S/NuBlVPmTwJ29quT5AjgnwI9x15W1caq6lRVZ2xsbIBmSdLCMYoh5oME/8PA\naUneluRI4HLgnml17gGuaqbfB3ytqnqe8UtSG43iJs459/E3ffYfBjYDS4DbquqJJNcD41V1D3Ar\n8D+SPEP3TP/yYTRakhaLyX78QzmqJ4fjCXin06nx8fFRN0OSFowkW6qqM5u6i+InGyRJs2fwS1LL\nGPyS1DIGvyS1jMEvSS1zWI7qSTIBfH8WVY9jHn7+YZFw3/TnvunPfdPf4b5v3lpVs7r79bAM/tlK\nMj7b4Utt477pz33Tn/umv8W0b+zqkaSWMfglqWUWevBvHHUDDmPum/7cN/25b/pbNPtmQffxS5IO\n3kI/45ckHaQFEfxJVib5epInkzyR5CNN+bFJ7k/ydPN+zKjbeqgl+UdJvpPksWbf/Jem/G3NA+6f\nbh54f+So2zoqSZYk2ZrkfzXz7ptGkueTbEvyaJLxpqz13yuAJMuTfCHJ95rsOXex7JsFEfzAPuC3\nq+rtwDnA1UneAVwLfLWqTgO+2sy3zd8B76qqM4DVwEVJzqH7YPs/bPbND+k++L6tPgI8OWXeffN6\n76yq1VOGKvq96vpj4CtV9c+BM+j+P7Qo9s2CCP6qerGqHmmmX6X7H2AFr3+Y++3A+tG0cHSq60fN\n7NLmVcC76D7gHlq6bwCSnAT8CnBLMx/cNzNp/fcqyT8BzqP7TBGq6idVtYdFsm8WRPBPleQUYA3w\nEHB8Vb0I3YMD8JbRtWx0mq6MR4FdwP3As8CeqtrXVNlO90DZRn8E/Efg75v5N+O+maqA+5JsSbKh\nKfN7BT8PTAD/vekmvCXJUSySfbOggj/Jm4AvAh+tqldG3Z7DRVXtr6rVdJ97fBbw9l7VDm2rRi/J\npcCuqtoytbhH1dbtmynWVtWZwMV0u1DPG3WDDhNHAGcCn6mqNcD/Y4F26/SyYII/yVK6oX9HVd3V\nFL+U5IRm+Ql0z3hbq/mn6P+mex1kefOAe+geEHaOql0jtBZ4T5Lngc/R7eL5I9w3r6mqnc37LuBu\nuicOfq+6/xLcXlUPNfNfoHsgWBT7ZkEEf9MveyvwZFV9esqiqQ9zvwr40qFu26glGUuyvJleBvwy\n3WsgX6f7gHto6b6pquuq6qSqOoXu856/VlVX4r4BIMlRSY6enAYuBB7H7xVV9X+BF5JMPvH83cB3\nWST7ZkHcwJXkXwLfBLbxs77a36Xbz/+XwMnA3wD/pqp2j6SRI5Lkl+heZFpC90D+l1V1fZKfp3uW\neyywFXh/Vf3d6Fo6WknOB/5DVV3qvulq9sPdzewRwF9U1X9N8mZa/r0CSLKa7qCAI4HngF+n+Y6x\nwPfNggh+SdLwLIiuHknS8Bj8ktQyBr8ktYzBL0ktY/BLUssY/JLUMga/JLWMwS9JLfP/AYu0E6rY\njlJ2AAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x2adad8211ed0>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# scatter plot\n",
    "plt.scatter(df_lexp['bcl2'], df_lexp['aid'])\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>q1</th>\n",
       "      <th>q2</th>\n",
       "      <th>q3</th>\n",
       "      <th>q4</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>aid</th>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     q1  q2  q3  q4\n",
       "aid   0   2   6   7"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_stat_bclq4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>feature</th>\n",
       "      <th>00055bad-54b7-4898-a9ae-3dd8036e62d3</th>\n",
       "      <th>0031d433-d703-4b2c-9fdf-2920008eb457</th>\n",
       "      <th>0046fef7-052b-4d4e-b264-b3bf2e2699b1</th>\n",
       "      <th>008292c8-81f5-4930-b179-a198686fc79b</th>\n",
       "      <th>00b6a02c-1b13-4b9a-b63f-400230f83973</th>\n",
       "      <th>010d78d6-f2fa-4112-8ebc-c41082e6ddcc</th>\n",
       "      <th>01253f54-c6b4-48c0-83da-84b73e12b6af</th>\n",
       "      <th>01370d42-f75c-4532-9b9c-24ff7302b033</th>\n",
       "      <th>0148a96e-972e-4762-9a75-ee2791dc3d95</th>\n",
       "      <th>...</th>\n",
       "      <th>fd4eb5ce-47b3-4c31-b9c5-ca08b3f580db</th>\n",
       "      <th>fe39a1a6-f141-4501-849b-19929c7c25de</th>\n",
       "      <th>feb1dc41-7013-4d2a-8b69-45830c456f7d</th>\n",
       "      <th>febead73-3b12-44e6-8bab-a7cfa7bf1129</th>\n",
       "      <th>fecacab2-ccf7-4c94-8dcd-fd319a3e630a</th>\n",
       "      <th>fef90a5d-49de-4277-a927-e6cfe7b3936e</th>\n",
       "      <th>ff0461fe-4c5f-476c-aade-de30a5c12a15</th>\n",
       "      <th>ff555cad-2ab4-425b-83b5-c56ac20905a8</th>\n",
       "      <th>ff781a4f-7281-4195-880f-3df099a401cc</th>\n",
       "      <th>ffac8b7c-64ac-429c-800b-fb2a5b6deefe</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>13279</th>\n",
       "      <td>ENSG00000171791.10</td>\n",
       "      <td>1.907678</td>\n",
       "      <td>0.58942</td>\n",
       "      <td>1.155672</td>\n",
       "      <td>5.457458</td>\n",
       "      <td>10.820279</td>\n",
       "      <td>2.416373</td>\n",
       "      <td>0.259505</td>\n",
       "      <td>9.90359</td>\n",
       "      <td>1.280249</td>\n",
       "      <td>...</td>\n",
       "      <td>14.56363</td>\n",
       "      <td>26.291428</td>\n",
       "      <td>13.216233</td>\n",
       "      <td>0.607463</td>\n",
       "      <td>62.128938</td>\n",
       "      <td>2.103286</td>\n",
       "      <td>7.782993</td>\n",
       "      <td>24.388436</td>\n",
       "      <td>1.140305</td>\n",
       "      <td>0.454289</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows Ã— 1522 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  feature  00055bad-54b7-4898-a9ae-3dd8036e62d3  \\\n",
       "13279  ENSG00000171791.10                              1.907678   \n",
       "\n",
       "       0031d433-d703-4b2c-9fdf-2920008eb457  \\\n",
       "13279                               0.58942   \n",
       "\n",
       "       0046fef7-052b-4d4e-b264-b3bf2e2699b1  \\\n",
       "13279                              1.155672   \n",
       "\n",
       "       008292c8-81f5-4930-b179-a198686fc79b  \\\n",
       "13279                              5.457458   \n",
       "\n",
       "       00b6a02c-1b13-4b9a-b63f-400230f83973  \\\n",
       "13279                             10.820279   \n",
       "\n",
       "       010d78d6-f2fa-4112-8ebc-c41082e6ddcc  \\\n",
       "13279                              2.416373   \n",
       "\n",
       "       01253f54-c6b4-48c0-83da-84b73e12b6af  \\\n",
       "13279                              0.259505   \n",
       "\n",
       "       01370d42-f75c-4532-9b9c-24ff7302b033  \\\n",
       "13279                               9.90359   \n",
       "\n",
       "       0148a96e-972e-4762-9a75-ee2791dc3d95  \\\n",
       "13279                              1.280249   \n",
       "\n",
       "                       ...                   \\\n",
       "13279                  ...                    \n",
       "\n",
       "       fd4eb5ce-47b3-4c31-b9c5-ca08b3f580db  \\\n",
       "13279                              14.56363   \n",
       "\n",
       "       fe39a1a6-f141-4501-849b-19929c7c25de  \\\n",
       "13279                             26.291428   \n",
       "\n",
       "       feb1dc41-7013-4d2a-8b69-45830c456f7d  \\\n",
       "13279                             13.216233   \n",
       "\n",
       "       febead73-3b12-44e6-8bab-a7cfa7bf1129  \\\n",
       "13279                              0.607463   \n",
       "\n",
       "       fecacab2-ccf7-4c94-8dcd-fd319a3e630a  \\\n",
       "13279                             62.128938   \n",
       "\n",
       "       fef90a5d-49de-4277-a927-e6cfe7b3936e  \\\n",
       "13279                              2.103286   \n",
       "\n",
       "       ff0461fe-4c5f-476c-aade-de30a5c12a15  \\\n",
       "13279                              7.782993   \n",
       "\n",
       "       ff555cad-2ab4-425b-83b5-c56ac20905a8  \\\n",
       "13279                             24.388436   \n",
       "\n",
       "       ff781a4f-7281-4195-880f-3df099a401cc  \\\n",
       "13279                              1.140305   \n",
       "\n",
       "       ffac8b7c-64ac-429c-800b-fb2a5b6deefe  \n",
       "13279                              0.454289  \n",
       "\n",
       "[1 rows x 1522 columns]"
      ]
     },
     "execution_count": 162,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_exp_bcl2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "def get_expression(gene, trp):\n",
    "    df_exp = pd.DataFrame()\n",
    "    for filename in [gene_gtex, gene_tophat, gene_tophatuq,transcript]:\n",
    "        flag = 0 \n",
    "        with open(join(exp_dir,filename),'r+b') as fexp:\n",
    "            mexp = mmap.mmap(fexp.fileno(),0)\n",
    "            for line in iter(mexp.readline, \"\"):\n",
    "                name = line.split('\\t')[0].split('.')[0]\n",
    "                if name == gene or name == trp:\n",
    "                    if flag == 0:\n",
    "                        exp = line.split('\\t')[1:]\n",
    "                        exp = np.array([float(v) for v in exp])\n",
    "                        df_tmp = pd.DataFrame()\n",
    "                        df_tmp[filename] = exp\n",
    "\n",
    "                        colname = df_exp.columns.values.tolist()\n",
    "                        if not colname:\n",
    "                            colname = []\n",
    "                        colname.append(filename)\n",
    "                        df_exp = pd.concat([df_exp,df_tmp], axis = 1, ignore_index = True)\n",
    "                        df_exp.columns = colname\n",
    "                    else:\n",
    "                        print \"ERROR FIND DUPLICATE GENE/TRANSCRIPT\"\n",
    "            mexp.close()\n",
    "            \n",
    "    return df_exp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "def get_expression_CCLE(gene):\n",
    "    fn = '../anno_ref/CCLE/CCLE_RNAseq_rsem_genes_tpm_20180929.txt'\n",
    "    df_exp = pd.DataFrame()\n",
    "    flag = 0 \n",
    "    with open(fn,'r+b') as fexp:\n",
    "        mexp = mmap.mmap(fexp.fileno(),0)\n",
    "        for line in iter(mexp.readline, \"\"):\n",
    "            name = line.split('\\t')[0].split('.')[0]\n",
    "            if name == gene:\n",
    "                if flag == 0:\n",
    "                    exp = line.split('\\t')[2:]\n",
    "                    exp = np.array([float(v) for v in exp])\n",
    "                    df_tmp = pd.DataFrame()\n",
    "                    df_tmp[fn] = exp\n",
    "\n",
    "                    colname = df_exp.columns.values.tolist()\n",
    "                    if not colname:\n",
    "                        colname = []\n",
    "                    colname.append(fn)\n",
    "                    df_exp = pd.concat([df_exp,df_tmp], axis = 1, ignore_index = True)\n",
    "                    df_exp.columns = colname\n",
    "                else:\n",
    "                    print \"ERROR FIND DUPLICATE GENE/TRANSCRIPT\"\n",
    "        mexp.close()\n",
    "\n",
    "    return df_exp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "exp_RP28_ccle = get_expression_CCLE('ENSG00000267701')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "exp_RP28_ccle.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "exp_RP29_ccle = get_expression_CCLE('ENSG00000267766')\n",
    "exp_RP29_ccle.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "exp_sno_ccle = get_expression_CCLE('ENSG00000238988')\n",
    "exp_sno_ccle.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "exp_bcl2_ccle = get_expression_CCLE('ENSG00000171791')\n",
    "exp_bcl2_ccle.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "exp_RP28 = get_expression('ENSG00000267701','ENST00000588307')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "exp_RP29 = get_expression('ENSG00000267766','ENST00000587475')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "exp_sno = get_expression('ENSG00000238988','ENST00000458827')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "exp_bcl2 = get_expression('ENSG00000171791', 'ENST00000398117')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GTEx.tophat2.gene.fpkm.tsv                4.166680\n",
       "tophat_star_fpkm.v2.aliquot_gl.tsv        8.855226\n",
       "tophat_star_fpkm_uq.v2_aliquot_gl.tsv     9.652404\n",
       "pcawg.rnaseq.transcript.expr.tpm.tsv     13.553974\n",
       "dtype: float64"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exp_bcl2.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GTEx.tophat2.gene.fpkm.tsv               0.435999\n",
       "tophat_star_fpkm.v2.aliquot_gl.tsv       1.485859\n",
       "tophat_star_fpkm_uq.v2_aliquot_gl.tsv    1.639796\n",
       "dtype: float64"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exp_RP28.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GTEx.tophat2.gene.fpkm.tsv               0.062703\n",
       "tophat_star_fpkm.v2.aliquot_gl.tsv       0.360591\n",
       "tophat_star_fpkm_uq.v2_aliquot_gl.tsv    0.383730\n",
       "dtype: float64"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exp_RP29.mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GTEx.tophat2.gene.fpkm.tsv               0.108119\n",
       "tophat_star_fpkm.v2.aliquot_gl.tsv       0.650623\n",
       "tophat_star_fpkm_uq.v2_aliquot_gl.tsv    0.689528\n",
       "dtype: float64"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exp_sno.mean()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check if the mutations in CCLE cell lines  \n",
    "only SC1 is follicular lymphoma cell line and one syn 60985327 one missense 60985366!  \n",
    "May be should check if those mutations exists in cell line at all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "cols = ['cell_type']\n",
    "df_ccle = pd.DataFrame(columns = cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "lsp = [re.findall(r'\\d+',v)[1] for v in ls]\n",
    "fcmut = '../anno_ref/CCLE/CCLE_DepMap_18q3_maf_20180718.txt'\n",
    "with open(fcmut,'r+b') as fmut:\n",
    "    mmut = mmap.mmap(fmut.fileno(), 0)\n",
    "    for line in iter(mmut.readline, \"\"):\n",
    "        name,code, build, chrom, sp,ep,strand,vc,vt = line.split('\\t')[0:9]\n",
    "        cell = line.split('\\t')[15]\n",
    "        muts = line.split('\\t')[17]\n",
    "        if chrom == '18' and (sp in lsp) and vc == 'Silent':\n",
    "            if sp not in df_ccle.index:\n",
    "                df_ccle.loc[sp,'cell_type'] = cell\n",
    "                df_ccle.loc[sp,'silent_mutation'] = muts\n",
    "            else\n",
    "        if chrom == '18' and (sp in lsp) and vc != 'Silent':\n",
    "            if sp not in df_ccle.index:\n",
    "                df_ccle.loc[sp,'cell_type'] = cell\n",
    "                df_ccle.loc[sp,'other_mutation'] = muts\n",
    "\n",
    "#                 if not pd.isnull(df_ccle.loc[sp,['cell_type']].values):\n",
    "#                     df_ccle.loc[sp,'cell_type'] += ','+cell\n",
    "#                 else:\n",
    "#                     df_ccle.loc[sp,'cell_type'] = cell\n",
    "            \n",
    "    mmut.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cell_type</th>\n",
       "      <th>silent_mutation</th>\n",
       "      <th>other_mutation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>60985528</th>\n",
       "      <td>NCIH1048_LUNG</td>\n",
       "      <td>c.(370-372)ttC&gt;ttT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60985492</th>\n",
       "      <td>BICR18_UPPER_AERODIGESTIVE_TRACT</td>\n",
       "      <td>c.(406-408)gaG&gt;gaA</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60985833</th>\n",
       "      <td>OCILY19_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</td>\n",
       "      <td>c.(67-69)Ctg&gt;Ttg</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60985861</th>\n",
       "      <td>OCILY19_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.(37-39)gaG&gt;gaC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60985417</th>\n",
       "      <td>OCILY3_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</td>\n",
       "      <td>c.(481-483)agC&gt;agT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60985876</th>\n",
       "      <td>PFEIFFER_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</td>\n",
       "      <td>c.(22-24)ggG&gt;ggT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60985549</th>\n",
       "      <td>SUDHL4_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.(349-351)agC&gt;agG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60985384</th>\n",
       "      <td>DB_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</td>\n",
       "      <td>c.(514-516)aaC&gt;aaT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60985545</th>\n",
       "      <td>DB_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</td>\n",
       "      <td>c.(355-357)Ctg&gt;Ttg</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60985582</th>\n",
       "      <td>ROS50_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</td>\n",
       "      <td>c.(316-318)cgC&gt;cgA</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60985435</th>\n",
       "      <td>VAL_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.(463-465)gGG&gt;gCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60985711</th>\n",
       "      <td>NUDHL1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE</td>\n",
       "      <td>c.(187-189)cgG&gt;cgA</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            cell_type     silent_mutation  \\\n",
       "60985528                                NCIH1048_LUNG  c.(370-372)ttC>ttT   \n",
       "60985492             BICR18_UPPER_AERODIGESTIVE_TRACT  c.(406-408)gaG>gaA   \n",
       "60985833   OCILY19_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE    c.(67-69)Ctg>Ttg   \n",
       "60985861   OCILY19_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE                 NaN   \n",
       "60985417    OCILY3_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  c.(481-483)agC>agT   \n",
       "60985876  PFEIFFER_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE    c.(22-24)ggG>ggT   \n",
       "60985549    SUDHL4_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE                 NaN   \n",
       "60985384        DB_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  c.(514-516)aaC>aaT   \n",
       "60985545        DB_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  c.(355-357)Ctg>Ttg   \n",
       "60985582     ROS50_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  c.(316-318)cgC>cgA   \n",
       "60985435       VAL_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE                 NaN   \n",
       "60985711    NUDHL1_HAEMATOPOIETIC_AND_LYMPHOID_TISSUE  c.(187-189)cgG>cgA   \n",
       "\n",
       "              other_mutation  \n",
       "60985528                 NaN  \n",
       "60985492                 NaN  \n",
       "60985833                 NaN  \n",
       "60985861    c.(37-39)gaG>gaC  \n",
       "60985417                 NaN  \n",
       "60985876                 NaN  \n",
       "60985549  c.(349-351)agC>agG  \n",
       "60985384                 NaN  \n",
       "60985545                 NaN  \n",
       "60985582                 NaN  \n",
       "60985435  c.(463-465)gGG>gCA  \n",
       "60985711                 NaN  "
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ccle"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# does COSMIC patients have those mutations, just mutation, don't care about follicular lymphoma"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "# Check the COSMIC patients, better from TCGA\n",
    "cosmicf = '../anno_ref/WES/CosmicMutantExport.tsv'\n",
    "df_cosmic = pd.DataFrame()\n",
    "lcosmic = []\n",
    "with open(cosmicf,'r+b') as cf:\n",
    "    next(cf)\n",
    "    mcosmic = mmap.mmap(cf.fileno(), 0)\n",
    "    for line in iter(mcosmic.readline, \"\"):\n",
    "        name = line.split('\\t')[0]\n",
    "        if '_' not in name:\n",
    "            mutgpos = line.split('\\t')[25]\n",
    "            val = re.findall(r'\\d+',mutgpos)\n",
    "            if val:\n",
    "                chrom,sp,ep = val[0:3]\n",
    "            if sp in lsp or ep in lsp:\n",
    "                lcosmic.append(line.split('\\t'))\n",
    "df_cosmic = pd.DataFrame(lcosmic)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0      NS\n",
       "1      NS\n",
       "2      NS\n",
       "3      NS\n",
       "4      NS\n",
       "5      NS\n",
       "6      NS\n",
       "7      NS\n",
       "8      NS\n",
       "9      NS\n",
       "10     NS\n",
       "11     NS\n",
       "12     NS\n",
       "13     NS\n",
       "14     NS\n",
       "15     NS\n",
       "16     NS\n",
       "17     NS\n",
       "18     NS\n",
       "19     NS\n",
       "20     NS\n",
       "21     NS\n",
       "22     NS\n",
       "23     NS\n",
       "24     NS\n",
       "25     NS\n",
       "26     NS\n",
       "27     NS\n",
       "28     NS\n",
       "29     NS\n",
       "       ..\n",
       "72     NS\n",
       "73     NS\n",
       "74     NS\n",
       "75     NS\n",
       "76     NS\n",
       "77     NS\n",
       "78     NS\n",
       "79     NS\n",
       "80     NS\n",
       "81     NS\n",
       "82     NS\n",
       "83     NS\n",
       "84     NS\n",
       "85     NS\n",
       "86     NS\n",
       "87     NS\n",
       "88     NS\n",
       "89     NS\n",
       "90     NS\n",
       "91     NS\n",
       "92     NS\n",
       "93     NS\n",
       "94     NS\n",
       "95     NS\n",
       "96     NS\n",
       "97     NS\n",
       "98     NS\n",
       "99     NS\n",
       "100    NS\n",
       "101    NS\n",
       "Name: 10, Length: 102, dtype: object"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cosmic[10]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# What about the previous gene list?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": [
    "# # Check the coverage at mutation position\n",
    "# dict_patient = {}\n",
    "# for p in lfp:\n",
    "#     print p\n",
    "#     dict_patient[p] = {}\n",
    "#     fcov = '/gpfs/scratch/yur97/compressed_coverage_tracks/'+p+'.coverage'\n",
    "#     with open(fcov,'r+b') as covf:\n",
    "#         mcov = mmap.mmap(covf.fileno(), 0)\n",
    "#         flag = 0\n",
    "#         for line in iter(mcov.readline, \"\"): \n",
    "#             if 'chrom=18 start=60248270 step=1' in line:\n",
    "#                 line_list = re.findall(r'\\d+', line)\n",
    "#                 chr_n = line_list[0]\n",
    "#                 start_pos = int(line_list[1])\n",
    "#                 position = start_pos - 1\n",
    "#                 flag = 1\n",
    "#             else:\n",
    "#                 if flag ==1:\n",
    "#                     position +=1\n",
    "#             if flag ==1:\n",
    "# #                 if chr_n not in dict_patient:\n",
    "# #                     dict_patient[p][chr_n] = []\n",
    "#                 if line == '0\\n' and position< 60985900 and position > 60985300:\n",
    "#                     print p\n",
    "#                     print position\n",
    "# #                     dict_patient[p][chr_n].append(position)\n",
    "#             if 'fixedStep chrom=18 start=63580815 step=1' in line:\n",
    "#                 break\n",
    "\n",
    "#         mcov.close()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
