{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# This script read and parse the annotation file and find out if all genes has principle transcripts, \\\n",
    "# than compare to PCAWG genes\n",
    "# Author: Yiyun\n",
    "\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "f_anno = ('../anno_ref/gencode_v19/gencode.v19.annotation.gtf')\n",
    "f_gene = ('../anno_ref/gene-PCAWG-all.tsv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "      <th>12</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "      <th>15</th>\n",
       "      <th>16</th>\n",
       "      <th>17</th>\n",
       "      <th>18</th>\n",
       "      <th>19</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>14363</td>\n",
       "      <td>29806</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000227232.4\"</td>\n",
       "      <td>transcript_id \"ENSG00000227232.4\"</td>\n",
       "      <td>gene_type \"pseudogene\"</td>\n",
       "      <td>gene_status \"KNOWN\"</td>\n",
       "      <td>gene_name \"WASH7P\"</td>\n",
       "      <td>transcript_type \"pseudogene\"</td>\n",
       "      <td>transcript_status \"KNOWN\"</td>\n",
       "      <td>transcript_name \"WASH7P\"</td>\n",
       "      <td>level 2</td>\n",
       "      <td>havana_gene \"OTTHUMG00000000958.1\"</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>29554</td>\n",
       "      <td>31109</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000243485.2\"</td>\n",
       "      <td>transcript_id \"ENSG00000243485.2\"</td>\n",
       "      <td>gene_type \"lincRNA\"</td>\n",
       "      <td>gene_status \"NOVEL\"</td>\n",
       "      <td>gene_name \"MIR1302-11\"</td>\n",
       "      <td>transcript_type \"lincRNA\"</td>\n",
       "      <td>transcript_status \"NOVEL\"</td>\n",
       "      <td>transcript_name \"MIR1302-11\"</td>\n",
       "      <td>level 2</td>\n",
       "      <td>havana_gene \"OTTHUMG00000000959.2\"</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>34554</td>\n",
       "      <td>36081</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000237613.2\"</td>\n",
       "      <td>transcript_id \"ENSG00000237613.2\"</td>\n",
       "      <td>gene_type \"lincRNA\"</td>\n",
       "      <td>gene_status \"KNOWN\"</td>\n",
       "      <td>gene_name \"FAM138A\"</td>\n",
       "      <td>transcript_type \"lincRNA\"</td>\n",
       "      <td>transcript_status \"KNOWN\"</td>\n",
       "      <td>transcript_name \"FAM138A\"</td>\n",
       "      <td>level 2</td>\n",
       "      <td>havana_gene \"OTTHUMG00000000960.1\"</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>52473</td>\n",
       "      <td>54936</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000268020.2\"</td>\n",
       "      <td>transcript_id \"ENSG00000268020.2\"</td>\n",
       "      <td>gene_type \"pseudogene\"</td>\n",
       "      <td>gene_status \"KNOWN\"</td>\n",
       "      <td>gene_name \"OR4G4P\"</td>\n",
       "      <td>transcript_type \"pseudogene\"</td>\n",
       "      <td>transcript_status \"KNOWN\"</td>\n",
       "      <td>transcript_name \"OR4G4P\"</td>\n",
       "      <td>level 2</td>\n",
       "      <td>havana_gene \"OTTHUMG00000185779.1\"</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>62948</td>\n",
       "      <td>63887</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000240361.1\"</td>\n",
       "      <td>transcript_id \"ENSG00000240361.1\"</td>\n",
       "      <td>gene_type \"pseudogene\"</td>\n",
       "      <td>gene_status \"KNOWN\"</td>\n",
       "      <td>gene_name \"OR4G11P\"</td>\n",
       "      <td>transcript_type \"pseudogene\"</td>\n",
       "      <td>transcript_status \"KNOWN\"</td>\n",
       "      <td>transcript_name \"OR4G11P\"</td>\n",
       "      <td>level 2</td>\n",
       "      <td>havana_gene \"OTTHUMG00000001095.2\"</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>69091</td>\n",
       "      <td>70008</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000186092.4\"</td>\n",
       "      <td>transcript_id \"ENSG00000186092.4\"</td>\n",
       "      <td>gene_type \"protein_coding\"</td>\n",
       "      <td>gene_status \"KNOWN\"</td>\n",
       "      <td>gene_name \"OR4F5\"</td>\n",
       "      <td>transcript_type \"protein_coding\"</td>\n",
       "      <td>transcript_status \"KNOWN\"</td>\n",
       "      <td>transcript_name \"OR4F5\"</td>\n",
       "      <td>level 2</td>\n",
       "      <td>havana_gene \"OTTHUMG00000001094.1\"</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>89295</td>\n",
       "      <td>133566</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000238009.2\"</td>\n",
       "      <td>transcript_id \"ENSG00000238009.2\"</td>\n",
       "      <td>gene_type \"lincRNA\"</td>\n",
       "      <td>gene_status \"NOVEL\"</td>\n",
       "      <td>gene_name \"RP11-34P13.7\"</td>\n",
       "      <td>transcript_type \"lincRNA\"</td>\n",
       "      <td>transcript_status \"NOVEL\"</td>\n",
       "      <td>transcript_name \"RP11-34P13.7\"</td>\n",
       "      <td>level 2</td>\n",
       "      <td>havana_gene \"OTTHUMG00000001096.2\"</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>89551</td>\n",
       "      <td>91105</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000239945.1\"</td>\n",
       "      <td>transcript_id \"ENSG00000239945.1\"</td>\n",
       "      <td>gene_type \"lincRNA\"</td>\n",
       "      <td>gene_status \"NOVEL\"</td>\n",
       "      <td>gene_name \"RP11-34P13.8\"</td>\n",
       "      <td>transcript_type \"lincRNA\"</td>\n",
       "      <td>transcript_status \"NOVEL\"</td>\n",
       "      <td>transcript_name \"RP11-34P13.8\"</td>\n",
       "      <td>level 2</td>\n",
       "      <td>havana_gene \"OTTHUMG00000001097.2\"</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>131025</td>\n",
       "      <td>134836</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000233750.3\"</td>\n",
       "      <td>transcript_id \"ENSG00000233750.3\"</td>\n",
       "      <td>gene_type \"pseudogene\"</td>\n",
       "      <td>gene_status \"KNOWN\"</td>\n",
       "      <td>gene_name \"CICP27\"</td>\n",
       "      <td>transcript_type \"pseudogene\"</td>\n",
       "      <td>transcript_status \"KNOWN\"</td>\n",
       "      <td>transcript_name \"CICP27\"</td>\n",
       "      <td>level 1</td>\n",
       "      <td>tag \"pseudo_consens\"</td>\n",
       "      <td>havana_gene \"OTTHUMG00000001257.3\"</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>gene</td>\n",
       "      <td>134901</td>\n",
       "      <td>139379</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000237683.5\"</td>\n",
       "      <td>transcript_id \"ENSG00000237683.5\"</td>\n",
       "      <td>gene_type \"protein_coding\"</td>\n",
       "      <td>gene_status \"KNOWN\"</td>\n",
       "      <td>gene_name \"AL627309.1\"</td>\n",
       "      <td>transcript_type \"protein_coding\"</td>\n",
       "      <td>transcript_status \"KNOWN\"</td>\n",
       "      <td>transcript_name \"AL627309.1\"</td>\n",
       "      <td>level 3</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>135141</td>\n",
       "      <td>135895</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000268903.1\"</td>\n",
       "      <td>transcript_id \"ENSG00000268903.1\"</td>\n",
       "      <td>gene_type \"pseudogene\"</td>\n",
       "      <td>gene_status \"KNOWN\"</td>\n",
       "      <td>gene_name \"RP11-34P13.15\"</td>\n",
       "      <td>transcript_type \"pseudogene\"</td>\n",
       "      <td>transcript_status \"KNOWN\"</td>\n",
       "      <td>transcript_name \"RP11-34P13.15\"</td>\n",
       "      <td>level 2</td>\n",
       "      <td>havana_gene \"OTTHUMG00000182518.2\"</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>137682</td>\n",
       "      <td>137965</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000269981.1\"</td>\n",
       "      <td>transcript_id \"ENSG00000269981.1\"</td>\n",
       "      <td>gene_type \"pseudogene\"</td>\n",
       "      <td>gene_status \"KNOWN\"</td>\n",
       "      <td>gene_name \"RP11-34P13.16\"</td>\n",
       "      <td>transcript_type \"pseudogene\"</td>\n",
       "      <td>transcript_status \"KNOWN\"</td>\n",
       "      <td>transcript_name \"RP11-34P13.16\"</td>\n",
       "      <td>level 2</td>\n",
       "      <td>havana_gene \"OTTHUMG00000182738.2\"</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156</th>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>139790</td>\n",
       "      <td>140339</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000239906.1\"</td>\n",
       "      <td>transcript_id \"ENSG00000239906.1\"</td>\n",
       "      <td>gene_type \"antisense\"</td>\n",
       "      <td>gene_status \"NOVEL\"</td>\n",
       "      <td>gene_name \"RP11-34P13.14\"</td>\n",
       "      <td>transcript_type \"antisense\"</td>\n",
       "      <td>transcript_status \"NOVEL\"</td>\n",
       "      <td>transcript_name \"RP11-34P13.14\"</td>\n",
       "      <td>level 2</td>\n",
       "      <td>havana_gene \"OTTHUMG00000002481.1\"</td>\n",
       "      <td></td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160</th>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>141474</td>\n",
       "      <td>173862</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000241860.2\"</td>\n",
       "      <td>transcript_id \"ENSG00000241860.2\"</td>\n",
       "      <td>gene_type \"processed_transcript\"</td>\n",
       "      <td>gene_status \"NOVEL\"</td>\n",
       "      <td>gene_name \"RP11-34P13.13\"</td>\n",
       "      <td>transcript_type \"processed_transcript\"</td>\n",
       "      <td>transcript_status \"NOVEL\"</td>\n",
       "      <td>transcript_name \"RP11-34P13.13\"</td>\n",
       "      <td>level 2</td>\n",
       "      <td>tag \"ncRNA_host\"</td>\n",
       "      <td>havana_gene \"OTTHUMG00000002480.3\"</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          1     2       3       4  5  6  7                            8   \\\n",
       "21    HAVANA  gene   14363   29806  .  -  .  gene_id \"ENSG00000227232.4\"   \n",
       "82    HAVANA  gene   29554   31109  .  +  .  gene_id \"ENSG00000243485.2\"   \n",
       "92    HAVANA  gene   34554   36081  .  -  .  gene_id \"ENSG00000237613.2\"   \n",
       "100   HAVANA  gene   52473   54936  .  +  .  gene_id \"ENSG00000268020.2\"   \n",
       "106   HAVANA  gene   62948   63887  .  +  .  gene_id \"ENSG00000240361.1\"   \n",
       "109   HAVANA  gene   69091   70008  .  +  .  gene_id \"ENSG00000186092.4\"   \n",
       "116   HAVANA  gene   89295  133566  .  -  .  gene_id \"ENSG00000238009.2\"   \n",
       "134   HAVANA  gene   89551   91105  .  -  .  gene_id \"ENSG00000239945.1\"   \n",
       "138   HAVANA  gene  131025  134836  .  +  .  gene_id \"ENSG00000233750.3\"   \n",
       "141  ENSEMBL  gene  134901  139379  .  -  .  gene_id \"ENSG00000237683.5\"   \n",
       "150   HAVANA  gene  135141  135895  .  -  .  gene_id \"ENSG00000268903.1\"   \n",
       "153   HAVANA  gene  137682  137965  .  -  .  gene_id \"ENSG00000269981.1\"   \n",
       "156   HAVANA  gene  139790  140339  .  -  .  gene_id \"ENSG00000239906.1\"   \n",
       "160   HAVANA  gene  141474  173862  .  -  .  gene_id \"ENSG00000241860.2\"   \n",
       "\n",
       "                                     9                                  10  \\\n",
       "21    transcript_id \"ENSG00000227232.4\"             gene_type \"pseudogene\"   \n",
       "82    transcript_id \"ENSG00000243485.2\"                gene_type \"lincRNA\"   \n",
       "92    transcript_id \"ENSG00000237613.2\"                gene_type \"lincRNA\"   \n",
       "100   transcript_id \"ENSG00000268020.2\"             gene_type \"pseudogene\"   \n",
       "106   transcript_id \"ENSG00000240361.1\"             gene_type \"pseudogene\"   \n",
       "109   transcript_id \"ENSG00000186092.4\"         gene_type \"protein_coding\"   \n",
       "116   transcript_id \"ENSG00000238009.2\"                gene_type \"lincRNA\"   \n",
       "134   transcript_id \"ENSG00000239945.1\"                gene_type \"lincRNA\"   \n",
       "138   transcript_id \"ENSG00000233750.3\"             gene_type \"pseudogene\"   \n",
       "141   transcript_id \"ENSG00000237683.5\"         gene_type \"protein_coding\"   \n",
       "150   transcript_id \"ENSG00000268903.1\"             gene_type \"pseudogene\"   \n",
       "153   transcript_id \"ENSG00000269981.1\"             gene_type \"pseudogene\"   \n",
       "156   transcript_id \"ENSG00000239906.1\"              gene_type \"antisense\"   \n",
       "160   transcript_id \"ENSG00000241860.2\"   gene_type \"processed_transcript\"   \n",
       "\n",
       "                       11                          12  \\\n",
       "21    gene_status \"KNOWN\"          gene_name \"WASH7P\"   \n",
       "82    gene_status \"NOVEL\"      gene_name \"MIR1302-11\"   \n",
       "92    gene_status \"KNOWN\"         gene_name \"FAM138A\"   \n",
       "100   gene_status \"KNOWN\"          gene_name \"OR4G4P\"   \n",
       "106   gene_status \"KNOWN\"         gene_name \"OR4G11P\"   \n",
       "109   gene_status \"KNOWN\"           gene_name \"OR4F5\"   \n",
       "116   gene_status \"NOVEL\"    gene_name \"RP11-34P13.7\"   \n",
       "134   gene_status \"NOVEL\"    gene_name \"RP11-34P13.8\"   \n",
       "138   gene_status \"KNOWN\"          gene_name \"CICP27\"   \n",
       "141   gene_status \"KNOWN\"      gene_name \"AL627309.1\"   \n",
       "150   gene_status \"KNOWN\"   gene_name \"RP11-34P13.15\"   \n",
       "153   gene_status \"KNOWN\"   gene_name \"RP11-34P13.16\"   \n",
       "156   gene_status \"NOVEL\"   gene_name \"RP11-34P13.14\"   \n",
       "160   gene_status \"NOVEL\"   gene_name \"RP11-34P13.13\"   \n",
       "\n",
       "                                          13                          14  \\\n",
       "21              transcript_type \"pseudogene\"   transcript_status \"KNOWN\"   \n",
       "82                 transcript_type \"lincRNA\"   transcript_status \"NOVEL\"   \n",
       "92                 transcript_type \"lincRNA\"   transcript_status \"KNOWN\"   \n",
       "100             transcript_type \"pseudogene\"   transcript_status \"KNOWN\"   \n",
       "106             transcript_type \"pseudogene\"   transcript_status \"KNOWN\"   \n",
       "109         transcript_type \"protein_coding\"   transcript_status \"KNOWN\"   \n",
       "116                transcript_type \"lincRNA\"   transcript_status \"NOVEL\"   \n",
       "134                transcript_type \"lincRNA\"   transcript_status \"NOVEL\"   \n",
       "138             transcript_type \"pseudogene\"   transcript_status \"KNOWN\"   \n",
       "141         transcript_type \"protein_coding\"   transcript_status \"KNOWN\"   \n",
       "150             transcript_type \"pseudogene\"   transcript_status \"KNOWN\"   \n",
       "153             transcript_type \"pseudogene\"   transcript_status \"KNOWN\"   \n",
       "156              transcript_type \"antisense\"   transcript_status \"NOVEL\"   \n",
       "160   transcript_type \"processed_transcript\"   transcript_status \"NOVEL\"   \n",
       "\n",
       "                                   15        16  \\\n",
       "21           transcript_name \"WASH7P\"   level 2   \n",
       "82       transcript_name \"MIR1302-11\"   level 2   \n",
       "92          transcript_name \"FAM138A\"   level 2   \n",
       "100          transcript_name \"OR4G4P\"   level 2   \n",
       "106         transcript_name \"OR4G11P\"   level 2   \n",
       "109           transcript_name \"OR4F5\"   level 2   \n",
       "116    transcript_name \"RP11-34P13.7\"   level 2   \n",
       "134    transcript_name \"RP11-34P13.8\"   level 2   \n",
       "138          transcript_name \"CICP27\"   level 1   \n",
       "141      transcript_name \"AL627309.1\"   level 3   \n",
       "150   transcript_name \"RP11-34P13.15\"   level 2   \n",
       "153   transcript_name \"RP11-34P13.16\"   level 2   \n",
       "156   transcript_name \"RP11-34P13.14\"   level 2   \n",
       "160   transcript_name \"RP11-34P13.13\"   level 2   \n",
       "\n",
       "                                      17                                   18  \\\n",
       "21    havana_gene \"OTTHUMG00000000958.1\"                                        \n",
       "82    havana_gene \"OTTHUMG00000000959.2\"                                        \n",
       "92    havana_gene \"OTTHUMG00000000960.1\"                                        \n",
       "100   havana_gene \"OTTHUMG00000185779.1\"                                        \n",
       "106   havana_gene \"OTTHUMG00000001095.2\"                                        \n",
       "109   havana_gene \"OTTHUMG00000001094.1\"                                        \n",
       "116   havana_gene \"OTTHUMG00000001096.2\"                                        \n",
       "134   havana_gene \"OTTHUMG00000001097.2\"                                        \n",
       "138                 tag \"pseudo_consens\"   havana_gene \"OTTHUMG00000001257.3\"   \n",
       "141                                                                      None   \n",
       "150   havana_gene \"OTTHUMG00000182518.2\"                                        \n",
       "153   havana_gene \"OTTHUMG00000182738.2\"                                        \n",
       "156   havana_gene \"OTTHUMG00000002481.1\"                                        \n",
       "160                     tag \"ncRNA_host\"   havana_gene \"OTTHUMG00000002480.3\"   \n",
       "\n",
       "       19  \n",
       "21   None  \n",
       "82   None  \n",
       "92   None  \n",
       "100  None  \n",
       "106  None  \n",
       "109  None  \n",
       "116  None  \n",
       "134  None  \n",
       "138        \n",
       "141  None  \n",
       "150  None  \n",
       "153  None  \n",
       "156  None  \n",
       "160        "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gtf = pd.read_csv(f_anno, skiprows =[0,1,2,3,4], header = None, sep='\\t')\n",
    "df_info_expand = df_gtf[8].str.split(';', expand = True)\n",
    "df_gtf.drop(8, axis=1, inplace=True)\n",
    "df_gtf = pd.concat([df_gtf,df_info_expand], axis = 1)\n",
    "df_gtf.columns =range(0,len(df_gtf.columns) )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of all KNOWN PROTEIN CODING GENES in GENCODE GTF FILES: 19456\n"
     ]
    }
   ],
   "source": [
    "df_allgene = df_gtf[df_gtf[13].str.contains('protein_coding') & df_gtf[14].str.contains('KNOWN')& df_gtf[2].str.contains('gene')]\n",
    "print 'Number of all KNOWN PROTEIN CODING GENES in GENCODE GTF FILES: '+ str(len(df_allgene))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read in all genes in PCAWG in a list\n",
    "list_pcawg = []\n",
    "with open(f_gene,'r') as gf:\n",
    "    for lines in gf:\n",
    "        list_pcawg.append(lines.strip('\\n'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes in GTF file but not in PCAWG: 75\n"
     ]
    }
   ],
   "source": [
    "print 'Number of genes in GTF file but not in PCAWG: '+\\\n",
    "str(len(df_allgene[~df_allgene[12].str.split(\"\\\"\",expand = True)[1].isin(list_pcawg)].iloc[:,1:15]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "list_png = []\n",
    "df_gene = df_allgene[12].str.split(\"\\\"\",expand = True)[1]\n",
    "for genes in list_pcawg:\n",
    "    if not any(df_gene==genes):\n",
    "        if genes not in list_png:\n",
    "            list_png.append(genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes in PCAWG file but not in GTF: 850\n"
     ]
    }
   ],
   "source": [
    "print 'Number of genes in PCAWG file but not in GTF: '+\\\n",
    "str(len(list_png))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "45"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print 'Number of duplicated genes in GTF file: '+ str(len(df_duplicate[df_duplicate[12].\\\n",
    "                                                str.split(\"\\\"\",expand = True)[1].isin(list_pcawg)].iloc[:,1:15]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_agnd = df_allgene[~df_allgene[12].str.split(\"\\\"\",expand = True)[1].duplicated()]\n",
    "df_agndinp = df_agnd[df_agnd[12].str.split(\"\\\"\",expand = True)[1].isin(list_pcawg)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_finalg = df_agndinp[12].str.split(\"\\\"\",expand = True)[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "df_finalg.to_csv('../anno_ref/genes-final.csv', index=False, header=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
