{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "import operator\n",
    "import pickle\n",
    "import pandas as pd\n",
    "import mmap\n",
    "import tqdm as tqdm, notebook"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "final_genelist = []\n",
    "with open('../anno_ref/genes-final.csv','r') as f:\n",
    "    for lines in f:\n",
    "        final_genelist.append(lines.strip('\\n'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_annotation(annotation_file_path):\n",
    "    name_dict = {}  # {transcript:[gene_name]}\n",
    "    transcript_dict = {}  # {'gene_name':['ENST00000']}\n",
    "    transcript_info_dict = {}  # {transcriptID:'exon'[]; 'CDS'[];'gene'[];'UTR\n",
    "    \n",
    "    with open(annotation_file_path, 'r') as annotation_f:\n",
    "        for lines in annotation_f:\n",
    "            if lines.startswith(\"chr\"):\n",
    "                line_split = lines.split('\\t')\n",
    "                chr_n, source, feature, start_pos, end_pos = line_split[0:5]\n",
    "                strand, frame, attribute = line_split[6:9]\n",
    "                position_list = [int(start_pos), int(end_pos)]\n",
    "                    \n",
    "                # feature = line_split[2] # 'exon' / 'CDS'/ 'UTR'/ 'start_codon'/ 'stop_codon'/ 'transcript'/ 'gene'\n",
    "                if feature !='gene':\n",
    "                    # if this is a protein coding gene and is the primary transcript:\n",
    "                    if 'protein_coding' in attribute:\n",
    "                        if 'appris_principal' in attribute or 'appris_candidate_longest' in attribute:\n",
    "                            # parse the attributes\n",
    "                            attribute_split = attribute.split(';')\n",
    "                            gene_id_col, transcript_id_col, gene_type_col, gene_name_col, transcript_type_col, \\\n",
    "                            transcript_name_col = attribute_split[0:6]\n",
    "                            # not include the version number\n",
    "                            gene_id = re.findall(r'(ENSG\\d+|ENSGR\\d+)', gene_id_col)[0]\n",
    "                            transcript_id = re.findall(r'(ENST\\d+|ENSTR\\d+)', transcript_id_col)[0]\n",
    "                            gene_name = re.findall(r'\\\"(.*?)\\\"', gene_name_col)[0]\n",
    "#                             print(gene_id, gene_name)\n",
    "\n",
    "                            # ignore the genes not in final gene_list\n",
    "                            if gene_name not in final_genelist:\n",
    "                                continue\n",
    "\n",
    "                            if gene_name not in transcript_dict:\n",
    "                                transcript_dict[gene_name] = []\n",
    "                            if transcript_id not in transcript_dict[gene_name]:\n",
    "                                transcript_dict[gene_name].append(transcript_id)\n",
    "\n",
    "                            if transcript_id not in transcript_info_dict:\n",
    "                                transcript_info_dict[transcript_id] = {}\n",
    "                                transcript_info_dict[transcript_id]['strand'] = strand\n",
    "                                transcript_info_dict[transcript_id]['chr'] = chr_n\n",
    "                                transcript_info_dict[transcript_id]['exon'] = []\n",
    "                                transcript_info_dict[transcript_id]['CDS'] = []\n",
    "                                transcript_info_dict[transcript_id]['UTR'] = []\n",
    "                                transcript_info_dict[transcript_id]['transcript'] = []\n",
    "\n",
    "                                # if the transcript and genes are known, parse according to feature type\n",
    "                            if feature == 'exon':\n",
    "                                # exon_n = re.findall(r'\\d+', attribute_split[8])\n",
    "                                transcript_info_dict[transcript_id]['exon'].append(position_list)\n",
    "                            elif feature == 'CDS':\n",
    "                                transcript_info_dict[transcript_id]['CDS'].append(position_list)\n",
    "                            elif feature == 'UTR':\n",
    "                                transcript_info_dict[transcript_id]['UTR'].append(position_list)\n",
    "                            elif feature == 'transcript':\n",
    "                                transcript_info_dict[transcript_id]['transcript'] = position_list\n",
    "\n",
    "                            if transcript_id not in name_dict:\n",
    "                                name_dict[transcript_id] = gene_name\n",
    "                                \n",
    "\n",
    "    # delete the transcript record that have 2 or more transcripts, only keep the longest transcript\n",
    "    for names in transcript_dict:\n",
    "        if len(transcript_dict[names]) > 1:\n",
    "            dict_t={}\n",
    "            for t in transcript_dict[names]:\n",
    "                len_t = abs(transcript_info_dict[t]['transcript'][1] - transcript_info_dict[t]['transcript'][0])\n",
    "                dict_t[t]=len_t\n",
    "            dt = sorted(dict_t.items(), key=operator.itemgetter(1), reverse=True)[1:]\n",
    "            for t in dt:\n",
    "                del transcript_info_dict[t[0]]\n",
    "        \n",
    "\n",
    "    return name_dict, transcript_info_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "f_anno = ('../anno_ref/gencode_v38/gencode.v38.annotation.gtf')\n",
    "dict_name, dict_transcript_info = get_annotation(f_anno)\n",
    "# pickle.dump(dict_name, open('../proc_09152020/dict_name_v38.pkl','wb'))\n",
    "# pickle.dump(dict_transcript_info, open('../proc_09152020/dict_transcript_info_v38.pkl','wb'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "f_anno = ('../anno_ref/gencode_v38/gencode.v38.annotation.gtf')\n",
    "df_anno = pd.read_csv(f_anno, skiprows = 5, sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr1</th>\n",
       "      <th>HAVANA</th>\n",
       "      <th>gene</th>\n",
       "      <th>11869</th>\n",
       "      <th>14412</th>\n",
       "      <th>.</th>\n",
       "      <th>+</th>\n",
       "      <th>..1</th>\n",
       "      <th>gene_id \"ENSG00000223972.4\"; transcript_id \"ENSG00000223972.4\"; gene_type \"pseudogene\"; gene_status \"KNOWN\"; gene_name \"DDX11L1\"; transcript_type \"pseudogene\"; transcript_status \"KNOWN\"; transcript_name \"DDX11L1\"; level 2; havana_gene \"OTTHUMG00000000961.2\";</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>11869</td>\n",
       "      <td>14409</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000223972.4\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>11869</td>\n",
       "      <td>12227</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000223972.4\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>12613</td>\n",
       "      <td>12721</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000223972.4\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>13221</td>\n",
       "      <td>14409</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000223972.4\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>transcript</td>\n",
       "      <td>11872</td>\n",
       "      <td>14412</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000223972.4\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2619438</th>\n",
       "      <td>chrM</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>transcript</td>\n",
       "      <td>15888</td>\n",
       "      <td>15953</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000210195.2\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2619439</th>\n",
       "      <td>chrM</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>exon</td>\n",
       "      <td>15888</td>\n",
       "      <td>15953</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000210195.2\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2619440</th>\n",
       "      <td>chrM</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>gene</td>\n",
       "      <td>15956</td>\n",
       "      <td>16023</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000210196.2\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2619441</th>\n",
       "      <td>chrM</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>transcript</td>\n",
       "      <td>15956</td>\n",
       "      <td>16023</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000210196.2\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2619442</th>\n",
       "      <td>chrM</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>exon</td>\n",
       "      <td>15956</td>\n",
       "      <td>16023</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000210196.2\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2619443 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr1   HAVANA        gene  11869  14412  .  + ..1  \\\n",
       "0        chr1   HAVANA  transcript  11869  14409  .  +   .   \n",
       "1        chr1   HAVANA        exon  11869  12227  .  +   .   \n",
       "2        chr1   HAVANA        exon  12613  12721  .  +   .   \n",
       "3        chr1   HAVANA        exon  13221  14409  .  +   .   \n",
       "4        chr1  ENSEMBL  transcript  11872  14412  .  +   .   \n",
       "...       ...      ...         ...    ...    ... .. ..  ..   \n",
       "2619438  chrM  ENSEMBL  transcript  15888  15953  .  +   .   \n",
       "2619439  chrM  ENSEMBL        exon  15888  15953  .  +   .   \n",
       "2619440  chrM  ENSEMBL        gene  15956  16023  .  -   .   \n",
       "2619441  chrM  ENSEMBL  transcript  15956  16023  .  -   .   \n",
       "2619442  chrM  ENSEMBL        exon  15956  16023  .  -   .   \n",
       "\n",
       "        gene_id \"ENSG00000223972.4\"; transcript_id \"ENSG00000223972.4\"; gene_type \"pseudogene\"; gene_status \"KNOWN\"; gene_name \"DDX11L1\"; transcript_type \"pseudogene\"; transcript_status \"KNOWN\"; transcript_name \"DDX11L1\"; level 2; havana_gene \"OTTHUMG00000000961.2\";  \n",
       "0        gene_id \"ENSG00000223972.4\"; transcript_id \"EN...                                                                                                                                                                                                                  \n",
       "1        gene_id \"ENSG00000223972.4\"; transcript_id \"EN...                                                                                                                                                                                                                  \n",
       "2        gene_id \"ENSG00000223972.4\"; transcript_id \"EN...                                                                                                                                                                                                                  \n",
       "3        gene_id \"ENSG00000223972.4\"; transcript_id \"EN...                                                                                                                                                                                                                  \n",
       "4        gene_id \"ENSG00000223972.4\"; transcript_id \"EN...                                                                                                                                                                                                                  \n",
       "...                                                    ...                                                                                                                                                                                                                  \n",
       "2619438  gene_id \"ENSG00000210195.2\"; transcript_id \"EN...                                                                                                                                                                                                                  \n",
       "2619439  gene_id \"ENSG00000210195.2\"; transcript_id \"EN...                                                                                                                                                                                                                  \n",
       "2619440  gene_id \"ENSG00000210196.2\"; transcript_id \"EN...                                                                                                                                                                                                                  \n",
       "2619441  gene_id \"ENSG00000210196.2\"; transcript_id \"EN...                                                                                                                                                                                                                  \n",
       "2619442  gene_id \"ENSG00000210196.2\"; transcript_id \"EN...                                                                                                                                                                                                                  \n",
       "\n",
       "[2619443 rows x 9 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_anno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "dict_transcript_info = pickle.load(open('../proc_09152020/dict_transcript_info_v38.pkl','rb'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'strand': '-',\n",
       " 'chr': 'chr13',\n",
       " 'exon': [[102758702, 102759072],\n",
       "  [102758543, 102758576],\n",
       "  [102758410, 102758452],\n",
       "  [102729367, 102750388]],\n",
       " 'CDS': [[102758702, 102758932],\n",
       "  [102758543, 102758576],\n",
       "  [102758410, 102758452],\n",
       "  [102729454, 102750388]],\n",
       " 'UTR': [[102758933, 102759072], [102729367, 102729453]],\n",
       " 'transcript': [102729367, 102759072]}"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dict_transcript_info['ENST00000322527']"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
