{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '../figure4/histology.syn.df_all_forheatmap.cohort_072221.1.csv'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-1-958049bb170c>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;31m### This is a gene specific analysis notebook\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0;32mfrom\u001b[0m \u001b[0mcandidate_utils\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/gpfs/group/epo2/default/yur97/gitlab/pcawg-to-mutsigcv/notebook/figure5/candidate_utils.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     14\u001b[0m \u001b[0mdir_out\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'../figure4/'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     15\u001b[0m \u001b[0mfeature_type\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'histology'\u001b[0m\u001b[0;34m;\u001b[0m\u001b[0msyn_nsyn\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'syn'\u001b[0m\u001b[0;34m;\u001b[0m\u001b[0mrun\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'cohort_072221'\u001b[0m\u001b[0;34m;\u001b[0m\u001b[0mthreshold\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m;\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 16\u001b[0;31m \u001b[0mdf_syn\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdir_out\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mfeature_type\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m'.syn.df_all_forheatmap.'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mrun\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m'.'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mthreshold\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m'.csv'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mindex_col\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     17\u001b[0m \u001b[0mdf_syn\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdf_syn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mset_index\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'gene'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     18\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.6/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36mread_csv\u001b[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, dialect, error_bad_lines, warn_bad_lines, delim_whitespace, low_memory, memory_map, float_precision)\u001b[0m\n\u001b[1;32m    686\u001b[0m     )\n\u001b[1;32m    687\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 688\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0m_read\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    689\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    690\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.6/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m_read\u001b[0;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[1;32m    452\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    453\u001b[0m     \u001b[0;31m# Create the parser.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 454\u001b[0;31m     \u001b[0mparser\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mTextFileReader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfp_or_buf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    455\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    456\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mchunksize\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0miterator\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.6/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[1;32m    946\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"has_index_names\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"has_index_names\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    947\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 948\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_make_engine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mengine\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    949\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    950\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mclose\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.6/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m_make_engine\u001b[0;34m(self, engine)\u001b[0m\n\u001b[1;32m   1178\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_make_engine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mengine\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"c\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1179\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mengine\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"c\"\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1180\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mCParserWrapper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1181\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1182\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mengine\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"python\"\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.6/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, src, **kwds)\u001b[0m\n\u001b[1;32m   2008\u001b[0m         \u001b[0mkwds\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"usecols\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0musecols\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2009\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2010\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_reader\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mparsers\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mTextReader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msrc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2011\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munnamed_cols\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_reader\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munnamed_cols\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2012\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader.__cinit__\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader._setup_parser_source\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '../figure4/histology.syn.df_all_forheatmap.cohort_072221.1.csv'"
     ]
    }
   ],
   "source": [
    "### This is a gene specific analysis notebook\n",
    "from candidate_utils import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Gene-transcript dict\n",
    "dict_name = pickle.load(open(os.path.join(dir_refs,'dict_name_forcov_062121.pkl'), 'rb'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_name = 'SIGLEC15'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Find transcript for the gene\n",
    "for transcript,gene in dict_name.items():\n",
    "    if gene == gene_name:\n",
    "        print(transcript)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<span style=\"color:blue\">**The histology subtypes of patients with the mutation**</span>.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/aci/sw/anaconda3/2020.07_gcc-4.8.5-bzb/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3263: DtypeWarning: Columns (1,17,24,33,34,35,36) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  if (await self.run_code(code, result,  async_=asy)):\n"
     ]
    }
   ],
   "source": [
    "df_str, df_num = get_num_str(gene_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dcc_specimen_type</th>\n",
       "      <th>histology_tier3</th>\n",
       "      <th>histology_tier4</th>\n",
       "      <th>tumour_histological_type</th>\n",
       "      <th>tumour_stage</th>\n",
       "      <th>tumour_histological_comment</th>\n",
       "      <th>specimen_donor_treatment_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>syn</th>\n",
       "      <td>{'Primary tumour - other': 1}</td>\n",
       "      <td>{'Adenocarcinoma': 1}</td>\n",
       "      <td>{'Adenocarcinoma': 1}</td>\n",
       "      <td>{nan: 1}</td>\n",
       "      <td>{nan: 1}</td>\n",
       "      <td>{'WARNING: Tumour histology inferred from proj...</td>\n",
       "      <td>{'no treatment': 1}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>all</th>\n",
       "      <td>{'Primary tumour - other': 129, 'Primary tumou...</td>\n",
       "      <td>{'Adenocarcinoma': 232}</td>\n",
       "      <td>{'Adenocarcinoma': 60, 'Pancreatic ductal carc...</td>\n",
       "      <td>{nan: 60, 'Infiltrating duct carcinoma, NOS': ...</td>\n",
       "      <td>{nan: 57, 'IIB': 56, 'T2N0MX': 5, 'T1N1bMX': 1...</td>\n",
       "      <td>{'WARNING: Tumour histology inferred from proj...</td>\n",
       "      <td>{'no treatment': 171, 'surgery': 38, 'chemothe...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     dcc_specimen_type  \\\n",
       "syn                      {'Primary tumour - other': 1}   \n",
       "all  {'Primary tumour - other': 129, 'Primary tumou...   \n",
       "\n",
       "             histology_tier3  \\\n",
       "syn    {'Adenocarcinoma': 1}   \n",
       "all  {'Adenocarcinoma': 232}   \n",
       "\n",
       "                                       histology_tier4  \\\n",
       "syn                              {'Adenocarcinoma': 1}   \n",
       "all  {'Adenocarcinoma': 60, 'Pancreatic ductal carc...   \n",
       "\n",
       "                              tumour_histological_type  \\\n",
       "syn                                           {nan: 1}   \n",
       "all  {nan: 60, 'Infiltrating duct carcinoma, NOS': ...   \n",
       "\n",
       "                                          tumour_stage  \\\n",
       "syn                                           {nan: 1}   \n",
       "all  {nan: 57, 'IIB': 56, 'T2N0MX': 5, 'T1N1bMX': 1...   \n",
       "\n",
       "                           tumour_histological_comment  \\\n",
       "syn  {'WARNING: Tumour histology inferred from proj...   \n",
       "all  {'WARNING: Tumour histology inferred from proj...   \n",
       "\n",
       "                         specimen_donor_treatment_type  \n",
       "syn                                {'no treatment': 1}  \n",
       "all  {'no treatment': 171, 'surgery': 38, 'chemothe...  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_str"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>SV.events</th>\n",
       "      <th>Coding.SNVs</th>\n",
       "      <th>Non.coding.SNVs</th>\n",
       "      <th>CNA.events..do_not_use.</th>\n",
       "      <th>Retrotransposon.insertions</th>\n",
       "      <th>Mitochondrial.mutations</th>\n",
       "      <th>all.Indels</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mut_status</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"8\" valign=\"top\">syn</th>\n",
       "      <th>count</th>\n",
       "      <td>6.000000</td>\n",
       "      <td>6.000000</td>\n",
       "      <td>6.000000</td>\n",
       "      <td>6.000000</td>\n",
       "      <td>6.000000</td>\n",
       "      <td>6.000000</td>\n",
       "      <td>6.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>84.333333</td>\n",
       "      <td>1615.500000</td>\n",
       "      <td>212489.666667</td>\n",
       "      <td>275.333333</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>2.666667</td>\n",
       "      <td>1283.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>88.991385</td>\n",
       "      <td>1293.003751</td>\n",
       "      <td>158158.019245</td>\n",
       "      <td>206.418668</td>\n",
       "      <td>1.211060</td>\n",
       "      <td>2.422120</td>\n",
       "      <td>936.930520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>30.000000</td>\n",
       "      <td>542.000000</td>\n",
       "      <td>70331.000000</td>\n",
       "      <td>157.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>431.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>40.250000</td>\n",
       "      <td>600.500000</td>\n",
       "      <td>79879.000000</td>\n",
       "      <td>179.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>615.500000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>56.500000</td>\n",
       "      <td>1070.000000</td>\n",
       "      <td>166630.000000</td>\n",
       "      <td>206.500000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>909.500000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>62.250000</td>\n",
       "      <td>2626.250000</td>\n",
       "      <td>332539.000000</td>\n",
       "      <td>215.250000</td>\n",
       "      <td>0.750000</td>\n",
       "      <td>4.500000</td>\n",
       "      <td>1875.500000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>264.000000</td>\n",
       "      <td>3422.000000</td>\n",
       "      <td>432543.000000</td>\n",
       "      <td>694.000000</td>\n",
       "      <td>3.000000</td>\n",
       "      <td>6.000000</td>\n",
       "      <td>2733.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"8\" valign=\"top\">all</th>\n",
       "      <th>count</th>\n",
       "      <td>106.000000</td>\n",
       "      <td>107.000000</td>\n",
       "      <td>107.000000</td>\n",
       "      <td>107.000000</td>\n",
       "      <td>107.000000</td>\n",
       "      <td>107.000000</td>\n",
       "      <td>107.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>182.490566</td>\n",
       "      <td>803.971963</td>\n",
       "      <td>112468.271028</td>\n",
       "      <td>487.065421</td>\n",
       "      <td>4.990654</td>\n",
       "      <td>2.168224</td>\n",
       "      <td>1000.140187</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>213.635080</td>\n",
       "      <td>1008.382232</td>\n",
       "      <td>144315.255781</td>\n",
       "      <td>461.390500</td>\n",
       "      <td>34.356756</td>\n",
       "      <td>1.840252</td>\n",
       "      <td>2274.979637</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>12.000000</td>\n",
       "      <td>20.000000</td>\n",
       "      <td>2126.000000</td>\n",
       "      <td>118.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>79.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>49.250000</td>\n",
       "      <td>121.500000</td>\n",
       "      <td>17297.000000</td>\n",
       "      <td>202.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>401.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>98.000000</td>\n",
       "      <td>496.000000</td>\n",
       "      <td>70331.000000</td>\n",
       "      <td>317.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>556.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>242.750000</td>\n",
       "      <td>932.500000</td>\n",
       "      <td>138316.500000</td>\n",
       "      <td>636.000000</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>3.000000</td>\n",
       "      <td>900.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>1234.000000</td>\n",
       "      <td>5164.000000</td>\n",
       "      <td>854014.000000</td>\n",
       "      <td>3128.000000</td>\n",
       "      <td>355.000000</td>\n",
       "      <td>8.000000</td>\n",
       "      <td>22963.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    SV.events  Coding.SNVs  Non.coding.SNVs  \\\n",
       "mut_status                                                    \n",
       "syn        count     6.000000     6.000000         6.000000   \n",
       "           mean     84.333333  1615.500000    212489.666667   \n",
       "           std      88.991385  1293.003751    158158.019245   \n",
       "           min      30.000000   542.000000     70331.000000   \n",
       "           25%      40.250000   600.500000     79879.000000   \n",
       "           50%      56.500000  1070.000000    166630.000000   \n",
       "           75%      62.250000  2626.250000    332539.000000   \n",
       "           max     264.000000  3422.000000    432543.000000   \n",
       "all        count   106.000000   107.000000       107.000000   \n",
       "           mean    182.490566   803.971963    112468.271028   \n",
       "           std     213.635080  1008.382232    144315.255781   \n",
       "           min      12.000000    20.000000      2126.000000   \n",
       "           25%      49.250000   121.500000     17297.000000   \n",
       "           50%      98.000000   496.000000     70331.000000   \n",
       "           75%     242.750000   932.500000    138316.500000   \n",
       "           max    1234.000000  5164.000000    854014.000000   \n",
       "\n",
       "                  CNA.events..do_not_use.  Retrotransposon.insertions  \\\n",
       "mut_status                                                              \n",
       "syn        count                 6.000000                    6.000000   \n",
       "           mean                275.333333                    0.666667   \n",
       "           std                 206.418668                    1.211060   \n",
       "           min                 157.000000                    0.000000   \n",
       "           25%                 179.000000                    0.000000   \n",
       "           50%                 206.500000                    0.000000   \n",
       "           75%                 215.250000                    0.750000   \n",
       "           max                 694.000000                    3.000000   \n",
       "all        count               107.000000                  107.000000   \n",
       "           mean                487.065421                    4.990654   \n",
       "           std                 461.390500                   34.356756   \n",
       "           min                 118.000000                    0.000000   \n",
       "           25%                 202.000000                    0.000000   \n",
       "           50%                 317.000000                    0.000000   \n",
       "           75%                 636.000000                    2.000000   \n",
       "           max                3128.000000                  355.000000   \n",
       "\n",
       "                  Mitochondrial.mutations    all.Indels  \n",
       "mut_status                                               \n",
       "syn        count                 6.000000      6.000000  \n",
       "           mean                  2.666667   1283.000000  \n",
       "           std                   2.422120    936.930520  \n",
       "           min                   0.000000    431.000000  \n",
       "           25%                   1.000000    615.500000  \n",
       "           50%                   2.000000    909.500000  \n",
       "           75%                   4.500000   1875.500000  \n",
       "           max                   6.000000   2733.000000  \n",
       "all        count               107.000000    107.000000  \n",
       "           mean                  2.168224   1000.140187  \n",
       "           std                   1.840252   2274.979637  \n",
       "           min                   0.000000     79.000000  \n",
       "           25%                   1.000000    401.000000  \n",
       "           50%                   2.000000    556.000000  \n",
       "           75%                   3.000000    900.000000  \n",
       "           max                   8.000000  22963.000000  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_num"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<span style=\"color:blue\">**Mutation**</span>.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/aci/sw/anaconda3/2020.07_gcc-4.8.5-bzb/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3263: DtypeWarning: Columns (1,17,24,33,34,35,36) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  if (await self.run_code(code, result,  async_=asy)):\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total 230 patients\n",
      "1 patients have synonymous mutations, they are: ['DO51546']\n",
      "18:g.43418687G>C\n",
      "18:g.43418708G>T\n"
     ]
    }
   ],
   "source": [
    "df_all, df_mut, donor_list = get_mut(gene_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Hugo_Symbol  Variant_Classification  Donor_ID\n",
       "SIGLEC15     Silent                  DO51546     g.chr18:43418687G>C\n",
       "                                     DO51546     g.chr18:43418708G>T\n",
       "Name: Genome_Change, dtype: object"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_all.loc[pd.IndexSlice[gene_name, :,donor_list],'Genome_Change']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<span style=\"color:blue\">**CNV**</span>.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                               754d26af-f184-497c-856a-cd1ea7a5115c\n",
      "Gene Symbol Locus ID Cytoband                                      \n",
      "SIGLEC15    284266   18q12.3                                    1.0\n"
     ]
    }
   ],
   "source": [
    "cnv = get_cnv(donor_list, gene_name)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<span style=\"color:blue\">**Expression PCAWG**</span>.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/aci/sw/anaconda3/2020.07_gcc-4.8.5-bzb/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3343: DtypeWarning: Columns (1,17,24,33,34,35,36) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of normal patient: 0\n",
      "Number of synonymous patients: 0\n",
      "Number of other tumor patients:78\n",
      "test statistic:0.0,p-value:0.0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/aci/sw/anaconda3/2020.07_gcc-4.8.5-bzb/lib/python3.8/site-packages/scipy/stats/stats.py:7039: RuntimeWarning: divide by zero encountered in double_scalars\n",
      "  z = (bigu - meanrank) / sd\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAV4AAADhCAYAAACAwN0OAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAAaWUlEQVR4nO3debhcVZnv8e8vA0kgKOQkRqYQJSjNGDEEuBAMAnZwArsRCPRNsLsZuoXgcEWu3a1pbWj62n31gtAt2kiQSVRAQKYoMkSZEokkEVCG0CGQEA5TAomS8N4/1iqyUzmnTlX61K6TOr/P89Rzak9rv2sPb629dtU+igjMzKw8A1odgJlZf+PEa2ZWMideM7OSOfGamZXMidfMrGROvGZmJeu3iVfSZEnPtDoOM0lfkvTdTVx2laR393ZMfZ2ksZJC0qBWx7Ipei3xSlosaXU+EJZL+p6k4b1VfgNxnJR3yLFlr7s7kiZKulnSy5JelPSApE9VzfMuSW9KuqiBco+SNF/Sq5JekPRzSWPztJmSLi/MK0mnS3pY0uuSlkm6U9LxhXnulPTXXayncpCvqnodV08dJW0h6Uf5GAlJk6vKnynpjaqye0wmuaxx9W6vGuVssK2auWxXH/gRcW5EbLTdu1h2o/0TEcMj4sn6I36rrPGS5uVjYZ6k8TXmHSLpknycLZP0uUbXtzlptL6STpD0tKTXJF0vaURP6+jtFu/HImI4sC+wH/D3vVx+PaYDL+a/LSfpQOAO4C5gHNAB/A1wZNWs04CXgOMlDamj3HHAZcDngbcD7wIuAt7sZpHzgc/k+TuAHUj7Z0oD1dkmn+iV1w8aqOMc4C+AZd2U/YOqshtOJlYfSVsAPwEuB7YFZgE/yeO7MhPYFdgZOBQ4S1Ijx83mZiZ11lfSHsC3gf8JjAZeJ52HtUVEr7yAxcDhheGvAzeRduxNwApSYrkJ2LEw353A14BfAiuB24GRhekHA78CXgaWACfViGFnUuL5c2AtMLowbRhwaY7ht8AXgGcK07cHfpzjfAqYUZg2E7iGlOhWAouACYXpf5Lr8XKe9vHCtDnAhXVsvydIyWo5cEwd8x8DzK8xfSZweX7/HmBdMeZulrkT+Osuxo8FAhjUzXJ11THP+wwwubtYGzje7s4xvQasAo7L4z8KzM/74lfA3oVlvggszfvwMeAw0gfPH4E3cjm/6WZ9dS8LfAp4JM/7JHBqHr8VsDofo6vya/uqfTWUlBA7cx0eJJ3Q5+R9uCYv9608fwDjCsf4vwFPA6/k/TKsi7p8KNdFhXH/BUzppu5LgQ8Vhr8GXN3NvCeRzuULcgyPAod1M+/xwNyqcZ8FbsjvPwI8BLxKOvdndndMsnH+2eCYAg5gfR75TfUx+N+o77nAlYXhXfIxsXXN47eRg72HE+GtigM7kRLQ10itnz8HtgS2Bn4IXF91sj9BSg7D8vB5edqYfPBOBQbnssbXiOEfgAfy+wXA5wrTzgPuAUbk+BaSEy+p5T8P+DKwBfBu0gnzp4WduAb4MDAQ+GfgvjxtMPA48KW87AdzzO/NdV4HHNrDtpsE/IH0IXVB5cDrYZl355i+QfpUHl41/a0DDzgNWFxHmXfSYOKtt46F+btLvK+QrlQWAX9TZ1lvJZ08vC/wPLB/3k/T83E5JO+PJcD2hTrt0tVJ2sV6GlqWlDB2AQR8gNQK2jdPm0zhA7+LfXUqcGPergOB9wNv627/sGHivTDPs0Ne9n8AQ/K0h4ET8vvPArdUlXMT8Pku6r5tXkexEXMMsKCbbXUSqdHzWdK5cVzetyO6OXZWArsWxj0IHF/YVnuRzs+9SY2So7s6JqmRePP26CSdvwOAI/LwqDz9bOCmTazvT4AvVo1bBby/1rHb210N10t6mfRJexdwbkR0RsSPI+L1iFhJ+uT+QNVy34uI30XEalLLstLfdCLws4i4KiLeyGXNr7H+acCV+f2VbNjdcCxwTkS8GBFLSJfeFfuRdsJXI+KPkS5zv0P6RK6YExE3R8Q64PvAPnn8AcBw0ofFHyPiDtJBPJW0EwcAz9WImRznLRHxUo77SEnvqLVAjnEy6aC6BnhB0qXd9KuPpOoSX9IzuT92jaSde4iv4oW8TOX1J9Rfx1quIV01jAJOBr4saeomlHMy8O2IuD8i1kXELNIH2gGkD4chwO6SBkfE4oh4os5yG1o2In4aEU9EchfpKm5Snet6g9TAGJfrMC8iXu1pIUkDgL8EzoyIpXnZX0XEH3JMe0dE5dwYTkqGRa+QGkbVhhem9zRvxfPAN/M5+wPSFcJHqmeKiNdJiWtqrsOuwG7ADXn6nRGxICLejIiHgavYOHfU4y+Am/P5+2ZEzAbmkhIxEXFeRHx0E+vbyLZ8S28n3qMjYpuI2Dki/jYiVkvaUtK3c+fzq6RLxG0kDSwsV0wKr7O+8juRWsMbkDSmeCMmjzuI1M95dZ7tSmCvwk2D7UmtloqnC+93BrYvJhVSC3Z0jRiHKt1R3R5YEhHFvtWnSQnxJdJl5XYbb6q36jIM+CRwBUBE3Eu67Duhu2UqIuK+iDg2IkaRTuxDgL/rYtbO6hgiYkdSQh5CapnVY2Tev5XXI9RRxzrq8duIeLaSLID/R2plNGpn4PNV+3EnUkv1cVIf90zgeUlXS9q+zvgaWlbSkZLuyzcZXyad4CPrrMP3gduAqyU9K+n/SBpcx3IjSd0U9XyYrALeVjXubaTWZ1fzVqb3NG/F0shNv+xp0vk1qXDeLsrTriQnXtIxf31OyEjaX9IvJK2Q9Arpyq3e7Vi0M/DJquPiYLo+ZhutbyPb8i1lfJ3s86RLtf0j4m2k5AD1nexLSJdsG4iI/4rCjZg8enouc76kZcD9efy0/Pc50klYMaZqPU9VJZWtI+LDdcT4LLBTbnEUy16aD6B7SV0t3fkEaUddlO+gLiMl7Wk1ltlIRDwIXAvs2cXkO4AdJU1opMw611tPHRsulvo/DIqWkK5qivtxy4i4Ksd6ZUQcTDoRA/iXwvpqB1TnsvnG6I+BfyVdrm4D3FyoT8115VbiP0bE7qSugo+y/liotewLpK6njc6XLiwC9pZU3MZ75/HV8bxEOnf2KYzep6t5C3aoKnsM8GxE3FM4b/fI024HRuYG0lTWX7GS398A7BQRbwf+g+6Pi9dIXRcV7yy8XwJ8v+q42CoizuuF+i4qzpu/jTME+F038wPlJN6tSTcUXs5fs/hKA8teARwu6VhJgyR1dPW1F0lDSV0Jp5C6KSqvM4ATc8v0GuB/S9pW0o55WsUDwKuSvihpmKSBkvaUtF8dMd5P2ulnSRqs9FWpj7G+5X0WcJKkL0jqyPHuI6kyfTpwCakvqxL3QcB4SXt1t1JJB0s6udIlIWk34OPAfdXzRsRjpDuvV0s6olJH0oldbZCkoYVXPa2tnupY+YrO0Dy4RS5bedpReb9I0kRgBukStCfLSX3dFd8BTsstJUnaStJHJG0t6b2SPpgT4xrSMbmuUM7Yqg/PtzS47BakE28FsFbSkaSbWcWYOyS9vZt1HSppr7x/XiV1PRTX1eXX7PIV1yXA/5W0fT6GD1TX35C5M5c5I++X0/P4O7oqm3RT+e/zPtqN1KVzaSHmxZJOKsz/jlz2YEmfJHUj3dxN3GuBH5Fuxo8AZhcmbw28GBFr8nFR6ypwPukbQYNzA6N4xXQ58DFJf5q3y1Clr/XtuCn1rXJFLnuSpK2ArwLX5m7V7kWNDuBGXlR1bhfGb0/a0atInwKnsmGn+J0UbhiQOufnFIYnkZJb5c7m9C7WcTzpU2pw1fihpJbAR0mfhpeR7mp2962Gq0hdCi+REljlZuFMNrxDOraqDnuQ+rRfyWV/oiqOicAtrL+BdD+pFbMD6UbEXl3U6WbgX2ts7z1JN2GW5227mNQKG9xNzCIltAWkxPFcjvlYYEBhX0TV6/JCfVdVvT7XUx2rjo/qssfmaVeRukNWke6Cz+iu3lXb4LRcj5eBY/O4KaQbNC/naT8kncB7kz5gV+b4bmL9zbIO0n2Jl4Bfd7GehpYFPp33y8ukroOrgX8qlHcJ67+1UP2thqmkPtHXchnns/44O5B0Dr0EnJ/HFW+uDQO+Sbor/wqpW29YnrYIOLEQw/tIN5RXA78G3leYdiKwqDA8JMf8ao6puN+3yNtlt8L5+0vgWzmG31H4hkA3+3FSrseFVeOPIXVTrMzb/FuF7TSWDc/Bd5OOuVXAT/N2Kx7/+5OO9xdJH4o/BcbkaV+icLOxVn1j/c2zSYXhE0jdg6+RGgwb3Uisfik26IoxM6ufpIOBT0dE5QbZSaSG1MEtDayP2yx/bmdmfUNEzCG1+K0BTrx9mKRJpMv3jcT6m4ptq7/X39qXuxrMzErWb59OZmbWKk68ZmYl2yz6eEeOHBljx45tdRhm1mbmzZv3QqRffpZqs0i8Y8eOZe7cua0Ow8zajKSne56r97mrwcysZE681jY6OzuZMWMGnZ2drQ7FrCYnXmsbs2bNYsGCBVx22WWtDsWsJideawudnZ3ceuutRAS33nqrW73WpznxWluYNWsWb76ZHom8bt06t3qtT3Pitbbws5/9jLVr1wKwdu1aZs+e3cMSZq3jxGtt4fDDD2fQoPTtyEGDBnHEEUe0OCKz7jnxWluYPn06Awakw3ngwIFMm9bQP/AwK5UTr7WFjo4OpkyZgiSmTJlCR0dHq0My69Zm8cs1s3pMnz6dxYsXu7VrfZ4Tr7WNjo4Ozj///FaHYdYjdzWYmZXMidfMrGROvGZmJXPiNTMrmROvmVnJnHjNzErmxGtmVjInXjOzkjnxmpmVzInXzKxkTrxmZiVz4jUzK5kTr5lZyZx4zcxK5sRrZlYyJ14zs5I58ZqZlcyJ18ysZE1LvJIukfS8pIWFcTMlLZU0P78+3Kz1m5n1Vc1s8V4KTOli/DciYnx+3dzE9ZuZ9UlNS7wRcTfwYrPKNzPbXLWij/d0SQ/nrohtW7B+M7OWKjvx/juwCzAeeA74t+5mlHSKpLmS5q5YsaKs+MzMmq7UxBsRyyNiXUS8CXwHmFhj3osjYkJETBg1alR5QZqZNVmpiVfSdoXBTwALu5vXzKxdDWpWwZKuAiYDIyU9A3wFmCxpPBDAYuDUZq3fzKyvalrijYipXYz+z2atz8xsc+FfrpmZlcyJ18ysZE68ZmYlc+I1MyuZE6+ZWcmceM3MSubEa2ZWMideM7OSOfGamZXMidfMrGROvGZmJXPiNTMrmROvmVnJnHjNzErmxGtmVjInXjOzkjnxmpmVzInXzKxkTrxmZiVz4jUzK5kTr5lZyZx4zcxK5sRrZlYyJ14zs5I58ZqZlcyJ18ysZE68ZmYlc+I1MyuZE6+ZWcmceM3MSubEa2ZWsqYlXkmXSHpe0sLCuBGSZkv6ff67bbPWb2bWVzWzxXspMKVq3NnAzyNiV+DnedjMrF9pWuKNiLuBF6tGHwXMyu9nAUc3a/1mZn1V2X28oyPiOYD89x0lr9/MrOUG1TujpD8DDgYCmBMR1zUtqrS+U4BTAMaMGdPMVZmZlaquFq+ki4DTgAXAQuBUSRduwvqWS9oul7kd8Hx3M0bExRExISImjBo1ahNWZWbWN9Xb4v0AsGdEBICkWaQk3KgbgOnAefnvTzahDDOzzVq9fbyPAcXr/Z2Ah2stIOkq4F7gvZKekfRXpIR7hKTfA0fkYTOzfqXeFm8H8IikB/LwfsB9km4AiIiPVy8QEVO7KeuwhqM0M2sj9SbeLzc1CjOzfqTexLsiIn5bHCFpckTc2fshmZm1t3r7eK+RdJaSYZIuAP65mYGZmbWrehPv/qSba78CHgSeBQ5qVlBmZu2s3sT7BrAaGAYMBZ6KiDebFpWZWRurN/E+SEq8E0i/Xpsq6UdNi8rMrI3Vm3hPBn4PfCkilgFnAPObFpWZWRurN/F+CjgAqHw3dyXpSWNmZtager9Otn9E7CvpIYCIeEnS4CbGZWbWtuq+uSZpIOnJZEgaVXlvZmaNqTfxng9cB7xD0jnAHODcpkVlZtbG6upqiIgrJM0jPWdBwNER8UhTIzMza1N1Pwg9Ih4FHm1iLGZm/YL/vbuZWcmceM3MSubEa2ZWMideM7OSOfGamZXMidfMrGROvGZmJXPiNTMrmROvmVnJnHjNzErmxGtmVjInXjOzkjnxmpmVzInXzKxkTrxmZiVz4jUzK5kTr5lZyZx4zcxK5sRrZlayuv/nWm+StBhYCawD1kbEhFbEYWbWCq1s8R4aEeOddK23dHZ2MmPGDDo7O1sdillN7mqwtjFr1iwWLFjAZZdd1upQzGpqVeIN4HZJ8ySd0tUMkk6RNFfS3BUrVpQcnm1uOjs7ueWWW4gIbrnlFrd6rU9rVeI9KCL2BY4EPi3pkOoZIuLiiJgQERNGjRpVfoS2WZk1axZr164F4I033nCr1/q0liTeiHg2/30euA6Y2Io4rH3Mnj2biAAgIrj99ttbHJFZ90pPvJK2krR15T3wIWBh2XFYexk9enTNYbO+pBVfJxsNXCepsv4rI+LWFsRhbWT58uU1h836ktITb0Q8CexT9nqtvR1yyCHcdtttGwyb9VX+Opm1hUr/rtnmwInX2sKcOXM2GL7nnntaFIlZz5x4rS1MmjSp5rBZX+LEa23BXQ22OXHitbZQ3bVw9913tygSs5458Vpb8Pd4bXPixGttYdmyZTWHzfoSJ15rC+985ztrDpv1JU681hb8yzXbnDjxWluYOHHD5yztv//+LYrErGdOvNYWnnzyyQ2Gn3jiiRZFYtYzJ15rC0uWLKk5bNaXOPFaWxg7dmzNYbO+xInX2sLpp5++wfAZZ5zRokjMeubEa22h+pdq/uWa9WWteBC6tdAFF1zA448/3uowet2CBQs2GL7xxhtZvHhxa4JponHjxrk13wbc4rW2sO2229YcNutL3OLtZ9q1tdTZ2ckxxxxDRDBkyBAuvvhiOjo6Wh2WWZfc4rW20NHRwYgRIwCYMmWKk671aW7xWtsYPXo0a9asYdq0aa0Oxawmt3itbQwePJhx48a5tWt9nhOvmVnJnHjNzErmxGtmVjInXjOzkvlbDVXa9Zdd/UFlv5155pktjsQ2VX/5ZZ4Tb5XHH3+c+QsfYd2WI1odijVowB/Tv3if96T/+8TmaODrL7Y6hNI48XZh3ZYjWL3bh1sdhlm/MuzRm1sdQmncx2tmVjInXjOzkrmrocrSpUsZ+Por/eqyx6wvGPh6J0uXrm11GKVoSYtX0hRJj0l6XNLZrYjBzKxVSk+8kgYCFwJHArsDUyXtXnYc3dlhhx0AtToM2wQD1rzKgDWvtjoM22TK51/7a0VXw0Tg8Yh4EkDS1cBRwG9bEMtGxo0b1+oQmmrp0qWsXr261WE0xep1awAY9mb7fnAOGzasjZPT6LY//ypakXh3AIr/e/sZYP8WxNGldv/ydjv/QGTp0qUAbZyY+s8PDNpdKxJvV82R2Ggm6RTgFIAxY8Y0O6Z+wyetWeu14ubaM8BOheEdgWerZ4qIiyNiQkRMGDVqVGnBmZk1WysS74PArpLeJWkL4HjghhbEYWbWEqV3NUTEWkmnA7cBA4FLImJR2XGYmbVKS35AERE3A/6Fgpn1S/7JsJlZyZx4zcxKpoiNvsnV50haATzd6jhsszASeKHVQdhmY+eIKP1rU5tF4jWrl6S5ETGh1XGY1eKuBjOzkjnxmpmVzInX2s3FrQ7ArCfu4zUzK5lbvGZmJXPiNTMrmROvNYWkbST9bavj6E51fJImS7qplTFZ/+HEa82yDVBa4s3/UqoRvRqfJP/jWKubE681y3nALpLmS3qw2JqU9C1JJ+X3iyWdK+leSXMl7SvpNklPSDotzyNJX5e0UNICScfl8ZMl/ULSlcCC7gKR9Lm87EJJn+kivq/nccMl/UjSo5KukKS8/Psl3SVpXo5tuzz+zhz7XcCZvbr1rK35U9qa5Wxgz4gYL2ky8L9qzLskIg6U9A3gUuAgYCiwCPgP4M+A8cA+pJ8EPyjp7rzsxLyep7oqWNL7gU+R/r2UgPtzonwrvjzfZOB9wB6kB/P/EjhI0v3ABcBREbEiJ/1zgL/Mq9gmIj7QyIYxc+K1vqDyIPwFwPCIWAmslLRG0jbAwcBVEbEOWJ4T537Aq8AD3SXd7GDguoh4DUDStcAkun74/gMR8Uyebz4wFngZ2BOYnRvAA4HnCsv8YBPqa/2cE6+VYS0bdmsNrZr+h/z3zcL7yvAguv4/fRWv9bDuRv7lcHHd6wrrXhQRB27i+s024j5ea5aVwNb5/dPA7pKGSHo7cFiDZd0NHCdpoKRRwCHAAw0se7SkLSVtBXwCuKcqvloeA0ZJOhBA0mBJezQYv9kG3OK1poiITkm/lLQQuAW4BngY+D3wUIPFXQccCPyG9B+pz4qIZZJ2qyOOX0u6lPWJ+rsR8RBAVXw/7Wb5P0o6Bjg/f2gMAr5J6n822yT+ybCZWcnc1WBmVjJ3NVhbkNQB/LyLSYdFRGfZ8ZjV4q4GM7OSuavBzKxkTrxmZiVz4rVe09efSNZskmZKqvXTaDPAidd6V19/Ilmtsnyj2UrjxGu9qaVPJMtPC/sXSQ9I+p2kSXn8UEnfy+U8JOnQPP4kST+UdCNwex6+XtKNkp6SdHp+stlDku6TNCIvd3Ku328k/VjSls3ftNZOnHitN50NPJGf+PWFHuZdkp9/cA/piWTHAAcAX83Ti08kOxz4euVxjKQnkv1dROzeRbmDImIi8BngK3ncpwEiYi9gKjBLUuV5EQcC0yPig3l4T+CEvI5zgNcj4n3AvcC0PM+1EbFfROwDPAL8VQ91NduAL6+sVZr1RLJr8995pKeLkcu6ACAiHpX0NPCePG12RLxYWP4XhVheAW4sxLl3fr+npH8ida0MB25rvPrWn7nFa83SqieSVcqqPF2MBsuqjqUYZ6W8S4HTcwv6H9m4bmY1OfFab+orTyTrqqwTASS9BxhDeurYptoaeE7S4Eq5Zo1w4rVek3+aW3ni1wzWP5HsCjbtiWQPk55Idgf5iWTVM0n6rqQJPZR1ETBQ0gLSg8tPiog/9LBMLf8A3A/MBh79b5Rj/ZR/MmxmVjK3eM3MSubEa2ZWMideM7OSOfGamZXMidfMrGROvGZmJXPiNTMrmROvmVnJ/j89kodCtg4rdwAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 360x216 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "get_expression(gene_name)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<span style=\"color:blue\">**CRISPR**</span>.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "### This is a gene specific analysis notebook\n",
    "import pickle, os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from scipy.stats import mannwhitneyu"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_name = 'SIGLEC15'\n",
    "dir_depmap = '../figure5/depmap'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_lineage(gene, histology):\n",
    "    ### Read file\n",
    "    df_depmap = pd.read_csv(os.path.join(dir_depmap, gene+'_21Q2.csv'))\n",
    "    print(df_depmap[df_depmap['Lineage'] == histology]['Lineage Subtype'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_CERES(gene, lineage):\n",
    "    df_depmap = pd.read_csv(os.path.join(dir_depmap, gene+'_21Q2.csv'))\n",
    "    df_depmap.columns = ['ID', 'CERES', 'Name', 'Primary Disease', 'Lineage', 'Lineage Subtype', 'Expression', 'Mutation']\n",
    "\n",
    "    df_gene = df_depmap[df_depmap['Lineage Subtype'] == lineage]\n",
    "    print(f'{len(df_gene)} in histology subtype')\n",
    "    df_other = df_depmap[df_depmap['Lineage Subtype'] != lineage]\n",
    "    print(f'{len(df_other)} not in histology subtype')\n",
    "    df_all = pd.DataFrame({'inlineage':df_gene['Expression'],\n",
    "                          'other':df_other['Expression']})\n",
    "    df_all = df_all.melt()\n",
    "    \n",
    "    return df_all, df_gene, df_other"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Exocrine']\n"
     ]
    }
   ],
   "source": [
    "get_lineage(gene_name, 'Pancreas')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "44 in histology subtype\n",
      "945 not in histology subtype\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(       variable     value\n",
       " 0     inlineage       NaN\n",
       " 1     inlineage       NaN\n",
       " 2     inlineage       NaN\n",
       " 3     inlineage       NaN\n",
       " 4     inlineage       NaN\n",
       " ...         ...       ...\n",
       " 1973      other  0.150560\n",
       " 1974      other  0.084064\n",
       " 1975      other  3.606442\n",
       " 1976      other  0.000000\n",
       " 1977      other  2.920293\n",
       " \n",
       " [1978 rows x 2 columns],\n",
       "              ID     CERES       Name    Primary Disease   Lineage  \\\n",
       " 8    ACH-001107  0.033505       KP1N  Pancreatic Cancer  Pancreas   \n",
       " 66   ACH-001353  0.134641    JOPACA1  Pancreatic Cancer  Pancreas   \n",
       " 73   ACH-001375  0.073135  PACADD119  Pancreatic Cancer  Pancreas   \n",
       " 75   ACH-001377  0.141798  PACADD137  Pancreatic Cancer  Pancreas   \n",
       " 76   ACH-001380  0.100504  PACADD165  Pancreatic Cancer  Pancreas   \n",
       " 77   ACH-001382 -0.054845  PACADD188  Pancreatic Cancer  Pancreas   \n",
       " 88   ACH-001379  0.123033  PACADD161  Pancreatic Cancer  Pancreas   \n",
       " 247  ACH-001999  0.216406    9505BIK  Pancreatic Cancer  Pancreas   \n",
       " 349  ACH-002186 -0.023219        PL4  Pancreatic Cancer  Pancreas   \n",
       " 497  ACH-000281  0.087753        KP2  Pancreatic Cancer  Pancreas   \n",
       " 512  ACH-000307  0.023396        PK1  Pancreatic Cancer  Pancreas   \n",
       " 524  ACH-000235  0.114546   PANC0403  Pancreatic Cancer  Pancreas   \n",
       " 527  ACH-000243  0.125141       DANG  Pancreatic Cancer  Pancreas   \n",
       " 547  ACH-000265  0.055981        KP4  Pancreatic Cancer  Pancreas   \n",
       " 548  ACH-000266  0.079473     SNU213  Pancreatic Cancer  Pancreas   \n",
       " 584  ACH-000320  0.096088       PSN1  Pancreatic Cancer  Pancreas   \n",
       " 592  ACH-000332  0.044893       YAPC  Pancreatic Cancer  Pancreas   \n",
       " 604  ACH-000354  0.104152     CAPAN1  Pancreatic Cancer  Pancreas   \n",
       " 618  ACH-000060  0.030446   PANC1005  Pancreatic Cancer  Pancreas   \n",
       " 626  ACH-000085  0.141224       T3M4  Pancreatic Cancer  Pancreas   \n",
       " 630  ACH-000093 -0.028396   PANC0504  Pancreatic Cancer  Pancreas   \n",
       " 631  ACH-000094  0.153254     HPAFII  Pancreatic Cancer  Pancreas   \n",
       " 638  ACH-000108  0.008919        KP3  Pancreatic Cancer  Pancreas   \n",
       " 641  ACH-000114  0.068236     SU8686  Pancreatic Cancer  Pancreas   \n",
       " 659  ACH-000022  0.161471  PATU8988S  Pancreatic Cancer  Pancreas   \n",
       " 660  ACH-000023 -0.009701  PATU8988T  Pancreatic Cancer  Pancreas   \n",
       " 672  ACH-000042  0.131781   PANC0203  Pancreatic Cancer  Pancreas   \n",
       " 679  ACH-000178  0.051676     HS766T  Pancreatic Cancer  Pancreas   \n",
       " 697  ACH-000213  0.062732      HUPT4  Pancreatic Cancer  Pancreas   \n",
       " 701  ACH-000222 -0.107244      ASPC1  Pancreatic Cancer  Pancreas   \n",
       " 705  ACH-000118  0.094202      HUPT3  Pancreatic Cancer  Pancreas   \n",
       " 715  ACH-000138  0.087482     CFPAC1  Pancreatic Cancer  Pancreas   \n",
       " 716  ACH-000139  0.075497   PANC0327  Pancreatic Cancer  Pancreas   \n",
       " 729  ACH-000155 -0.110762     SW1990  Pancreatic Cancer  Pancreas   \n",
       " 734  ACH-000164 -0.003319      PANC1  Pancreatic Cancer  Pancreas   \n",
       " 738  ACH-000417  0.103661   PANC0813  Pancreatic Cancer  Pancreas   \n",
       " 742  ACH-000652  0.003762      SUIT2  Pancreatic Cancer  Pancreas   \n",
       " 772  ACH-000601  0.201335   MIAPACA2  Pancreatic Cancer  Pancreas   \n",
       " 796  ACH-000685 -0.129659        L33  Pancreatic Cancer  Pancreas   \n",
       " 859  ACH-000599 -0.065548   PATU8902  Pancreatic Cancer  Pancreas   \n",
       " 862  ACH-000468 -0.000625      PK45H  Pancreatic Cancer  Pancreas   \n",
       " 920  ACH-000502  0.120645    TCCPAN2  Pancreatic Cancer  Pancreas   \n",
       " 961  ACH-000517 -0.020521     SNU410  Pancreatic Cancer  Pancreas   \n",
       " 973  ACH-000535  0.127784      BXPC3  Pancreatic Cancer  Pancreas   \n",
       " \n",
       "     Lineage Subtype  Expression Mutation  \n",
       " 8          Exocrine         NaN    Other  \n",
       " 66         Exocrine    0.847997    Other  \n",
       " 73         Exocrine    0.056584    Other  \n",
       " 75         Exocrine    0.831877    Other  \n",
       " 76         Exocrine    2.643856    Other  \n",
       " 77         Exocrine    1.344828    Other  \n",
       " 88         Exocrine    1.090853    Other  \n",
       " 247        Exocrine    0.454176    Other  \n",
       " 349        Exocrine         NaN    Other  \n",
       " 497        Exocrine    0.124328    Other  \n",
       " 512        Exocrine    0.731183    Other  \n",
       " 524        Exocrine    0.356144    Other  \n",
       " 527        Exocrine    6.030998    Other  \n",
       " 547        Exocrine    0.238787    Other  \n",
       " 548        Exocrine    0.111031    Other  \n",
       " 584        Exocrine    0.201634    Other  \n",
       " 592        Exocrine    1.448901    Other  \n",
       " 604        Exocrine    0.250962    Other  \n",
       " 618        Exocrine    0.475085    Other  \n",
       " 626        Exocrine    0.150560    Other  \n",
       " 630        Exocrine    1.599318    Other  \n",
       " 631        Exocrine    0.941106    Other  \n",
       " 638        Exocrine    0.097611    Other  \n",
       " 641        Exocrine    0.124328    Other  \n",
       " 659        Exocrine    0.713696    Other  \n",
       " 660        Exocrine    0.042644    Other  \n",
       " 672        Exocrine    0.594549    Other  \n",
       " 679        Exocrine    0.831877    Other  \n",
       " 697        Exocrine    2.998196    Other  \n",
       " 701        Exocrine    0.910733    Other  \n",
       " 705        Exocrine    0.189034    Other  \n",
       " 715        Exocrine    1.327687    Other  \n",
       " 716        Exocrine    0.475085    Other  \n",
       " 729        Exocrine    0.344828    Other  \n",
       " 734        Exocrine    0.226509    Other  \n",
       " 738        Exocrine    3.058316    Other  \n",
       " 742        Exocrine    0.910733    Other  \n",
       " 772        Exocrine    0.000000    Other  \n",
       " 796        Exocrine    0.014355    Other  \n",
       " 859        Exocrine    1.459432    Other  \n",
       " 862        Exocrine    0.084064    Other  \n",
       " 920        Exocrine    3.984589    Other  \n",
       " 961        Exocrine    0.056584    Other  \n",
       " 973        Exocrine    1.182692    Other  ,\n",
       "              ID     CERES       Name             Primary Disease  \\\n",
       " 0    ACH-000001 -0.074509  NIHOVCAR3              Ovarian Cancer   \n",
       " 1    ACH-001063 -0.003193      DOV13              Ovarian Cancer   \n",
       " 2    ACH-001065  0.103948      EVSAT               Breast Cancer   \n",
       " 3    ACH-001067  0.104490         F5                Brain Cancer   \n",
       " 4    ACH-001075 -0.066433    NCIH292                 Lung Cancer   \n",
       " ..          ...       ...        ...                         ...   \n",
       " 984  ACH-000937 -0.001135      PF382                    Leukemia   \n",
       " 985  ACH-001416 -0.005437      UMUC9              Bladder Cancer   \n",
       " 986  ACH-000272  0.185129      SLR24               Kidney Cancer   \n",
       " 987  ACH-000879  0.107438     MFE296  Endometrial/Uterine Cancer   \n",
       " 988  ACH-000791  0.039455  RERFLCAD1                 Lung Cancer   \n",
       " \n",
       "                     Lineage             Lineage Subtype  Expression  \\\n",
       " 0                     Ovary        Ovary Adenocarcinoma    0.042644   \n",
       " 1                     Ovary        Ovary Adenocarcinoma         NaN   \n",
       " 2                    Breast            Breast Carcinoma         NaN   \n",
       " 3    Central Nervous System                  Meningioma    0.263034   \n",
       " 4                      Lung                       NSCLC    0.070389   \n",
       " ..                      ...                         ...         ...   \n",
       " 984                   Blood                         ALL    0.150560   \n",
       " 985           Urinary Tract           Bladder Carcinoma    0.084064   \n",
       " 986                  Kidney        Renal Cell Carcinoma    3.606442   \n",
       " 987                  Uterus  Endometrial Adenocarcinoma    0.000000   \n",
       " 988                    Lung                       NSCLC    2.920293   \n",
       " \n",
       "                  Mutation  \n",
       " 0                   Other  \n",
       " 1                   Other  \n",
       " 2                   Other  \n",
       " 3                   Other  \n",
       " 4                   Other  \n",
       " ..                    ...  \n",
       " 984  Other non-conserving  \n",
       " 985  Other non-conserving  \n",
       " 986  Other non-conserving  \n",
       " 987              Damaging  \n",
       " 988              Damaging  \n",
       " \n",
       " [945 rows x 8 columns])"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_CERES(gene_name, 'Exocrine')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<span style=\"color:blue\">**Alternative Splicing**</span>  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "dir_splice = '../../anno_ref/ICGC/alternative_splicing'\n",
    "exp_dir = '../../anno_ref/ICGC/pcawg_rnaseq/'\n",
    "# Read aliquot id information\n",
    "df_exp_info = pd.read_csv(os.path.join(exp_dir,'rnaseq.metadata.tsv'), sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DO51546 is not in PCAWG transcriptome analysis\n"
     ]
    }
   ],
   "source": [
    "### If patient is in the splicing file(transcriptome analysis info)\n",
    "for donor in donor_list:\n",
    "    tumor_id = df_exp_info[df_exp_info['icgc_donor_id'] == donor]['aliquot_id']\n",
    "    if len(tumor_id) ==0:\n",
    "        print(f'{donor} is not in PCAWG transcriptome analysis')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
